---
title: "BAM_reads_filtering"
author: "Dani"
date: "14 de enero de 2019"
output: html_document
---

#0: Define paths.

```{r Define paths, eval=FALSE, engine='bash'}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

```

#1: Subset BAM files to gene reads.
##Obtain bed file with gene coordinates +- 100 bp.

```{r Subset BAM files to gene reads, eval=FALSE, engine='bash'}

#Keep only genes annotation from Maria's file and then convert to bed file and to rf file.
cd /GRUPOS/grupolince/Lyp_annotation_Apr14_final
awk '$3 == "gene" {print $0;}' LYPA23C.GENE.mRNA.CDS.GENE_promoters.GENE_introns.UTRs.ncRNA.lncRNA.lncRNA_introns.lncRNA_promoters.UCNE.intergenic.nr.gff3 | cut -d$'\t' -f1,3,4,5 | awk '{printf ("%s\t%s\t%s\t%s\n", $1, $3, $4, $2)}' > LYPA23C.GENE.nr.bed
awk '{printf ("%s:%s-%s\n", $1, $2, $3)}' LYPA23C.GENE.nr.bed > LYPA23C.GENE.nr.rf

```

##Subset the BAM files.
###DON:
```{r Subset BAM files to gene reads, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
screen -S c_lp_do_genes_BAMs.log
script c_lp_do_genes_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
declare SAMPLES=$(ls c_lp_do*.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  samtools view -b -h -L /GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.GENE.nr.bed c_lp_do_"${i}"_recal_round-1.bam > BAM_genes_5x/c_lp_do_"${i}"_recal_round-1.genes.bam
  done

```

###SMO:
```{r Subset BAM files to gene reads, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
screen -S c_lp_sm_genes_BAMs.log
script c_lp_sm_genes_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
declare SAMPLES=$(ls c_lp_sm*.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  samtools view -b -h -L /GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.GENE.nr.bed c_lp_sm_"${i}"_recal_round-1.bam > BAM_genes_5x/c_lp_sm_"${i}"_recal_round-1.genes.bam
  done

```

#2: Obtain distribution of edit distance.
##W-g BAMs.
###Extract edit distance from BAMs.

```{r Obtain distribution of edit distance, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/edit_distance_distr
screen -S c_lp_do_edit_distance_distr.log
script c_lp_do_edit_distance_distr.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
declare SAMPLES=$(ls c_lp_do*.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  #samtools view c_lp_do_"${i}"_recal_round-1.bam | grep -o '\bNM:i:\w*' | awk -v var="${i}" -F ":" '{printf ("%s\t%s\n", $3, var)}' > edit_distance_distr/c_lp_do_"${i}"_NM_distr.txt
  samtools view c_lp_do_"${i}"_recal_round-1.bam | grep -o '\bNM:i:\w*' | cut -d':' -f3 > edit_distance_distr/c_lp_do_"${i}"_NM_distr.txt
  done
  
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/edit_distance_distr/c_lp_do_*_NM_distr.txt /Users/Dani/ownCloud/backup/contamination/edit_distance_tests

```

###Draw edit distance distributions.

```{r Other stuff}

library(readr)
library(dplyr)
library(ggplot2)

#First draw the NM distribution for each individual:
sample_files <- list.files("/Users/Dani/ownCloud/backup/contamination/edit_distance_tests/", pattern="*distr.txt")
for (file in sample_files) {
  edit_distance_distribution <- read_tsv(paste0("/Users/Dani/ownCloud/backup/contamination/edit_distance_tests/",file),col_names=c("NM"))
  edit_distance_distribution
  sample <- strsplit(file,"_")[[1]][4]
  print(sample)
  plot_data <- edit_distance_distribution %>% group_by(NM) %>% tally()
  plot_data
  #edit_distance_distribution$pop <- as.factor(edit_distance_distribution$pop)
  #plot_data <- edit_distance_distribution %>% filter(edit_distance_distribution$pop == !!pop) #the two !! allow R to evaluate the text and distinguish looping variables from col_names
  #plot_data
  NM_distr_ggplot <- ggplot(data=plot_data, aes(NM,n)) +
  #geom_histogram(aes(NM),binwidth=1) +
  geom_col() +
  ggtitle(paste0("NM distribution for ",sample)) +
  ylab("count") +
  #xlab("heritability") +
  scale_x_continuous(breaks=seq(0,nrow(plot_data),2)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.92,0.86),
        legend.title=element_blank()
  )
  NM_distr_ggplot
  ggsave(paste0("c_lp_do_",sample,"_NM_distribution.lr_ann.pdf"), width=20, height=20, units="cm", device="pdf", path="/Users/Dani/ownCloud/backup/contamination/edit_distance_tests")
}
rm(edit_distance_distribution)

```

##Genes-only BAMs.
###Extract edit distance and quality qfrom BAMs.

```{r Obtain distribution of edit distance, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/edit_distance_distr
screen -S c_lp_do_edit_distance_distr.genes.log
script c_lp_do_edit_distance_distr.genes.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
declare SAMPLES=$(ls c_lp_do*_recal_round-1.genes.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  #samtools view c_lp_do_"${i}"_recal_round-1.genes.bam | grep -o '\bNM:i:\w*' | awk -v var="${i}" -F ":" '{printf ("%s\t%s\n", $3, var)}' > ./../edit_distance_distr/c_lp_do_"${i}"_NM_distr.genes.txt
  samtools view c_lp_do_"${i}"_recal_round-1.genes.bam | grep -o '\bNM:i:\w*' | cut -d':' -f3 > ./../edit_distance_distr/c_lp_do_"${i}"_NM_distr.genes.txt
  done
  
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/edit_distance_distr/c_lp_do_*_NM_distr.genes.txt /Users/Dani/ownCloud/backup/contamination/edit_distance_tests

```


###Draw edit distance distributions.

```{r Other stuff}

library(readr)
library(dplyr)
library(ggplot2)

#First draw the NM distribution for each individual:
sample_files <- list.files("/Users/Dani/ownCloud/backup/contamination/edit_distance_tests/", pattern="*distr.genes.txt")
for (file in sample_files) {
  edit_distance_distribution <- read_tsv(paste0("/Users/Dani/ownCloud/backup/contamination/edit_distance_tests/",file),col_names=c("NM"))
  edit_distance_distribution
  sample <- strsplit(file,"_")[[1]][4]
  print(sample)
  plot_data <- edit_distance_distribution %>% group_by(NM) %>% tally()
  plot_data
  #edit_distance_distribution$pop <- as.factor(edit_distance_distribution$pop)
  #plot_data <- edit_distance_distribution %>% filter(edit_distance_distribution$pop == !!pop) #the two !! allow R to evaluate the text and distinguish looping variables from col_names
  #plot_data
  NM_distr_ggplot <- ggplot(data=plot_data, aes(NM,n)) +
  #geom_histogram(aes(NM),binwidth=1) +
  geom_col() +
  ggtitle(paste0("NM distribution for ",sample)) +
  ylab("count") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.92,0.86),
        legend.title=element_blank()
  )
  NM_distr_ggplot
  ggsave(paste0("c_lp_do_",sample,"_NM_distribution.genes.pdf"), width=30, height=20, units="cm", device="pdf", path="/Users/Dani/ownCloud/backup/contamination/edit_distance_tests")
}
rm(edit_distance_distribution)

reads_totales <- sum(plot_data$n)
plot_bis <- mutate(plot_data,prop=as.numeric(100*n/reads_totales))
plot_bis$cum_prop <- cumsum(plot_bis$prop)
plot_bis

```

#3: Perform NM-based BAM filterings.

##Filter in reads with high NM.

```{r Perform NM-based BAM filterings, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
screen -S c_lp_do_genes_hm_BAMs.log
script c_lp_do_genes_hm_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
declare SAMPLES=$(ls c_lp_do*.genes.bam | cut -c9-12 | sort | uniq)
cd /opt/bamtools/lib
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  bamtools filter -tag "NM:>12" -in /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x/c_lp_do_"${i}"_recal_round-1.genes.bam -out /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x/c_lp_do_"${i}"_recal_round-1.genes-hm.bam
  samtools index c_lp_do_"${i}"_recal_round-1.genes-hm.bam
  done

```

##Filter out reads with high NM. Based on the previously obtained distribution of NM, we decide to filter out reads with NM>4.
###Genes-only BAMs.
####DON:
```{r Perform NM-based BAM filterings, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
screen -S c_lp_do_genes_nm_BAMs.log
script c_lp_do_genes_nm_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
declare SAMPLES=$(ls c_lp_do_*_recal_round-1.genes.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  bamtools filter -tag "NM:<=4" -in c_lp_do_"${i}"_recal_round-1.genes.bam -out c_lp_do_"${i}"_recal_round-1.genes-nm.bam
  samtools index c_lp_do_"${i}"_recal_round-1.genes-nm.bam
  done

```

####SMO:
```{r Perform NM-based BAM filterings, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
screen -S c_lp_sm_genes_nm_BAMs.log
script c_lp_sm_genes_nm_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_genes_5x
declare SAMPLES=$(ls c_lp_sm_*_recal_round-1.genes.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  bamtools filter -tag "NM:<=4" -in c_lp_sm_"${i}"_recal_round-1.genes.bam -out c_lp_sm_"${i}"_recal_round-1.genes-nm.bam
  samtools index c_lp_sm_"${i}"_recal_round-1.genes-nm.bam
  done

```

###W-g BAMs.
####DON:
```{r Perform NM-based BAM filterings, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
screen -S c_lp_do_nm_BAMs.log
script c_lp_do_nm_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
declare SAMPLES=$(ls c_lp_do_*_recal_round-1.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  bamtools filter -tag "NM:<=4" -in c_lp_do_"${i}"_recal_round-1.bam -out c_lp_do_"${i}"_recal_round-1.nm.bam
  samtools index c_lp_do_"${i}"_recal_round-1.nm.bam
  done

```

####SMO:
```{r Perform NM-based BAM filterings, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
screen -S c_lp_sm_nm_BAMs.log
script c_lp_sm_nm_BAMs.log

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
declare SAMPLES=$(ls c_lp_sm_*_recal_round-1.bam | cut -c9-12 | sort | uniq)
for i in ${SAMPLES[@]}
  do
  echo "${i}"
  bamtools filter -tag "NM:<=4" -in c_lp_sm_"${i}"_recal_round-1.bam -out c_lp_sm_"${i}"_recal_round-1.nm.bam
  samtools index c_lp_sm_"${i}"_recal_round-1.nm.bam
  done

```


#4: Perform variant calling for each species. Combine all BAMs of interest into the per species VCF. These won't include substitutions between species.
##A: Genes-subset only. 
###For Lynx pardinus.

```{r Perform variant calling for each species, eval=FALSE, engine='bash'}

#Perform direct variant calling (without gVCFs) on the NMâ‰¤4 filtered Lynx pardinus BAMs.
cd $V_PATH
screen -S c_lp_sm_c_lp_do_nm_genes.log
script c_lp_sm_c_lp_do_nm_genes.log

cd $B_PATH/BAM_genes_5x
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -jar $GATK \
-T HaplotypeCaller \
-R $REF \
$(for var in c_lp*recal_round-1.genes-nm.bam; do echo -I ${var}" ";done) \
-o $V_PATH/c_lp_sm_c_lp_do_nm_genes.vcf

grep -v '#' c_lp_sm_c_lp_do_nm_genes.vcf | wc -l 

```
