---
title: "genetic_load_statistics"
output: html_document
---

#1: Derived allele frequency approaches.
##Separate callings:
###Define desired features.
```{r Define desired features, eval=FALSE, engine='bash'}

mkdir -p derived_allele_freq_ratio
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/
rm features_derived_allele_freq_ratio.txt
echo -e "synonymous_variant\tSYN" >> features_derived_allele_freq_ratio.txt
echo -e "missense_variant\tNSYN" >> features_derived_allele_freq_ratio.txt
echo -e "|HIGH|\tLOF" >> features_derived_allele_freq_ratio.txt
echo -e "intron_variant\tINTR" >> features_derived_allele_freq_ratio.txt
cat features_derived_allele_freq_ratio.txt

```

###Calculate derived allele frequency ratio.
####For variants within species:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_nm2_origcov) #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_nm2_origcov
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_derived_freq_ratio_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
GREP=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
cd $V_PATH/$CALLING/annotation
POP_VCFS=($(ls `find . -name *'_perpop_'$TYPE'.lr_ann.vcf' -print`))
echo "retrieving variants from each feature from each population"
for g in ${GREP[@]}
  do
  echo "feature: ${g}"
  FEATURE=$(grep "${g}" /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
  echo $FEATURE
  for p in ${POP_VCFS[@]}
    do
    echo "working with vcf ${p}"
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s_%s_%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_${TYPE}.lr_ann.vcf/perpop_${FEATURE}_${TYPE}.lr_ann.bed}
    done
  done
#Prepare file that will contain population and SYN ratio data:
mkdir -p derived_allele_freq_ratio
rm derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt
echo -e "population_1\tpopulation_2\tSYN_ratio" > derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt
DEL_FEATURES=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | grep -Ev 'synonymous_variant|intron_variant' | cut -f2)
#Prepare files that will contain DEL categories ratio data:
for f in ${DEL_FEATURES[@]}
  do
  rm derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt
  echo -e "${f}_ratio\t${f}_norm_ratio" > derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt
  done
POPS=($(ls `find . -name *'_perpop_'$TYPE'.lr_ann.vcf' -print` | cut -d'/' -f3 | cut -d'_' -f1,2,3,4,5,6))
echo "calculating ratio for each pair of populations"
for p1 in ${POPS[@]}
  do
  for p2 in ${POPS[@]}
    do
    echo "${p1} vs ${p2}"
    echo "feature: SYN"
    LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed #col1: scaffold, col2: pos_start, col3: pos_end, col4: p1_af, col5: p2_af, col6: p1_af*(1-p2_af), col7: p2_af*(1-p1_af)
    SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
    echo -e "$p1\t$p2\t$RATIO" >> derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt
    for f in ${DEL_FEATURES[@]}
      do
      echo "feature: ${f}"
      LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed
      SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
      echo -e "$RATIO" >> derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt
      done
    done
  done
cd $V_PATH/$CALLING/annotation/derived_allele_freq_ratio
#Make an ALL features file storing the population and SYN data
cat ${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt > ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt
for f in ${DEL_FEATURES[@]}
  do
  #First build a file with the feature's ratio and normalised ratio (i.e. divided by the SYN ratio):
  head ${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt -n1 > ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt ${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt | awk 'FNR == 1 {next}{print $4,$4/$3}' OFS="\t" >> ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar
  #Now join the ALL and the current DEL feature data
  paste <(head ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt -n1) <(head ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar -n1) > "${CALLING}"_joint_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar | awk 'FNR == 1 {next}{print $0}' OFS="\t" >> ${CALLING}_joint_derived_freq_ratio_${TYPE}.borrar
  #Replace the ALL table with the updated one
  mv ${CALLING}_joint_derived_freq_ratio_${TYPE}.borrar ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt
  rm ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar
  done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/*/annotation/derived_allele_freq_ratio/*ALL_derived_freq_ratio.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

####For substitutions between species:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_derived_freq_ratio_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
GREP=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
cd $V_PATH/$CALLING/annotation
POP_VCFS=($(ls `find . -name *'_perpop_'$TYPE'.lr_ann.vcf' -print`))
echo "retrieving variants from each feature from each population"
for g in ${GREP[@]}
  do
  echo "feature: ${g}"
  FEATURE=$(grep "${g}" /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
  echo $FEATURE
  for p in ${POP_VCFS[@]}
    do
    echo "working with vcf ${p}"
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s_%s_%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_${TYPE}.lr_ann.vcf/perpop_${FEATURE}_${TYPE}.lr_ann.bed}
    done
  done
#Prepare file that will contain population and SYN ratio data:
mkdir -p derived_allele_freq_ratio
rm derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt
echo -e "population_1\tpopulation_2\tSYN_ratio" > derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt
DEL_FEATURES=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | grep -Ev 'synonymous_variant|intron_variant' | cut -f2)
#Prepare files that will contain DEL categories ratio data:
for f in ${DEL_FEATURES[@]}
  do
  rm derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt
  echo -e "${f}_ratio\t${f}_norm_ratio" > derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt
  done
POPS=($(ls `find . -name *'_perpop_'$TYPE'.lr_ann.vcf' -print` | cut -d'/' -f3 | cut -d'_' -f1,2,3,4,5,6))
echo "calculating ratio for each pair of populations"
for p1 in ${POPS[@]}
  do
  for p2 in ${POPS[@]}
    do
    echo "${p1} vs ${p2}"
    echo "feature: SYN"
    LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed #col1: scaffold, col2: pos_start, col3: pos_end, col4: p1_af, col5: p2_af, col6: p1_af*(1-p2_af), col7: p2_af*(1-p1_af)
    SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    RATIO=$(if [ $SUM_P1 == 0 ] || [ $SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SUM_P1/$SUM_P2" | bc; fi)
    echo -e "$p1\t$p2\t$RATIO" >> derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt
    for f in ${DEL_FEATURES[@]}
      do
      echo "feature: ${f}"
      LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed
      SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      RATIO=$(if [ $SUM_P1 == 0 ] || [ $SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SUM_P1/$SUM_P2" | bc; fi)
      echo -e "$RATIO" >> derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt
      done
    done
  done
cd $V_PATH/$CALLING/annotation/derived_allele_freq_ratio
#Make an ALL features file storing the population and SYN data
cat ${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt > ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt
for f in ${DEL_FEATURES[@]}
  do
  #First build a file with the feature's ratio and normalised ratio (i.e. divided by the SYN ratio):
  head ${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt -n1 > ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_SYN_derived_freq_ratio_${TYPE}.txt ${CALLING}_${f}_derived_freq_ratio_${TYPE}.txt | awk 'FNR == 1 {next} $3==0 {print $4,0.0} $3>0 {print $4,$4/$3}' OFS="\t" >> ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar
  #Now join the ALL and the current DEL feature data
  paste <(head ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt -n1) <(head ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar -n1) > "${CALLING}"_joint_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar | awk 'FNR == 1 {next}{print $0}' OFS="\t" >> ${CALLING}_joint_derived_freq_ratio_${TYPE}.borrar
  #Replace the ALL table with the updated one
  mv ${CALLING}_joint_derived_freq_ratio_${TYPE}.borrar ${CALLING}_ALL_derived_freq_ratio_${TYPE}.txt
  rm ${CALLING}_temp_derived_freq_ratio_${TYPE}.borrar
  done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/*/annotation/derived_allele_freq_ratio/*ALL_derived_freq_ratio.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

####Global ratios (variants + substitutions) for the separate callings:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #use the substitutions calling, which is the more complete name
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_vars_subs_derived_freq_ratio_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path

N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
POPS=($(ls `find $V_PATH/$CALLING/annotation/ -name *'_perpop_'$TYPE'.lr_ann.vcf' -print` | rev | cut -d'/' -f1 | rev | cut -d'_' -f1,2,3,4,5,6))
DATASETS=$(for i in ${SPECIES[@]}; do ls /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered/${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
GREP=$(cat ./../features_derived_allele_freq_ratio.txt | cut -f1)

#Join the variants and substitutions from each population and feature:
echo "joining variants and substitutions from each feature from each population"
for g in ${GREP[@]}
  do
  echo "feature: ${g}"
  FEATURE=$(grep "${g}" ./../features_derived_allele_freq_ratio.txt | cut -f2)
  echo $FEATURE
  for p in ${POPS[@]}
    do
    echo "working with pop ${p}"
    CURRENT_SPECIES=$(echo ${p} | cut -d'_' -f5)
    POPS_IN_SPECIES=$(echo $CALLING | fold -w8 | grep $CURRENT_SPECIES | tr -d '[:space:]')
    SEP_CALLING=$(echo $POPS_IN_SPECIES"nm"*"_"$COVERAGE)
    SEP_NM_COV=$(realpath $V_PATH/$SEP_CALLING | rev |  cut -d'/' -f1 | cut -d'_' -f1,2 | rev)
    cat $V_PATH/$CALLING/annotation/*/${p}"_"${NM_COV}"_perpop_"${FEATURE}_${TYPE}".lr_ann.bed" $V_PATH/$SEP_CALLING/annotation/*/${p}"_"${SEP_NM_COV}"_perpop_"${FEATURE}_${TYPE}".lr_ann.bed" | sort > ${p}"_"${NM_COV}"_perpop_"${FEATURE}"_vars_subs"_${TYPE}".lr_ann.bed"
    done
  done
#Prepare file that will contain population and SYN derived allele frequency ratio data:
rm ${CALLING}_SYN_vars_subs_derived_freq_ratio_${TYPE}.txt
echo -e "population_1\tpopulation_2\tSYN_ratio" > ${CALLING}_SYN_vars_subs_derived_freq_ratio_${TYPE}.txt
DEL_FEATURES=$(cat ./../features_derived_allele_freq_ratio.txt | grep -v 'synonymous_variant' | cut -f2)
#Prepare files that will contain DEL categories derived allele frequency ratio data:
for f in ${DEL_FEATURES[@]}
  do
  rm ${CALLING}_${f}_vars_subs_derived_freq_ratio_${TYPE}.txt
  echo -e "${f}_ratio\t${f}_norm_ratio" > ${CALLING}_${f}_vars_subs_derived_freq_ratio_${TYPE}.txt
  done
#Calculate the derived allele frequency ratio for each feature for each pair of populations:
echo "calculating derived allele frequency ratio for each pair of populations"
for p1 in ${POPS[@]}
  do
  for p2 in ${POPS[@]}
    do
    echo "${p1} vs ${p2}"
    echo "feature: SYN"
    LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ${p1}"_"${NM_COV}"_perpop_SYN_vars_subs"_${TYPE}".lr_ann.bed") <(LANG=en_EN sort -k1 ${p2}"_"${NM_COV}"_perpop_SYN_vars_subs"_${TYPE}".lr_ann.bed") | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > ${p1}"_vs_"${p2}"_"${NM_COV}"_perpop_SYN_vars_subs"_${TYPE}".lr_ann.bed"
    SUM_P1=$(cut -f6 ${p1}"_vs_"${p2}"_"${NM_COV}"_perpop_SYN_vars_subs"_${TYPE}".lr_ann.bed" | paste -sd+ | bc)
    SUM_P2=$(cut -f7 ${p1}"_vs_"${p2}"_"${NM_COV}"_perpop_SYN_vars_subs"_${TYPE}".lr_ann.bed" | paste -sd+ | bc)
    RATIO=$(if [ $SUM_P1 == 0 ] || [ $SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SUM_P1/$SUM_P2" | bc; fi)
    echo -e "$p1\t$p2\t$RATIO" >> ${CALLING}_SYN_vars_subs_derived_freq_ratio_${TYPE}.txt
    for f in ${DEL_FEATURES[@]}
      do
      echo "feature: ${f}"
      LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ${p1}"_"${NM_COV}"_perpop_"${f}"_vars_subs"_${TYPE}".lr_ann.bed") <(LANG=en_EN sort -k1 ${p2}"_"${NM_COV}"_perpop_"${f}"_vars_subs"_${TYPE}".lr_ann.bed") | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > ${p1}"_vs_"${p2}"_"${NM_COV}"_perpop_"${f}"_vars_subs"_${TYPE}".lr_ann.bed"
      SUM_P1=$(cut -f6 ${p1}"_vs_"${p2}"_"${NM_COV}"_perpop_"${f}"_vars_subs"_${TYPE}".lr_ann.bed" | paste -sd+ | bc)
      SUM_P2=$(cut -f7 ${p1}"_vs_"${p2}"_"${NM_COV}"_perpop_"${f}"_vars_subs"_${TYPE}".lr_ann.bed" | paste -sd+ | bc)
      RATIO=$(if [ $SUM_P1 == 0 ] || [ $SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SUM_P1/$SUM_P2" | bc; fi)
      echo -e "$RATIO" >> ${CALLING}_${f}_vars_subs_derived_freq_ratio_${TYPE}.txt
      done
    done
  done
#Make an ALL features file that at first will only store the population and SYN data. Looping over DEL categories will add them:
cat ${CALLING}_SYN_vars_subs_derived_freq_ratio_${TYPE}.txt > ${CALLING}_ALL_vars_subs_derived_freq_ratio_${TYPE}.txt
for f in ${DEL_FEATURES[@]}
  do
  #First build a file with the feature's ratio and normalised ratio (i.e. divided by the SYN ratio):
  head ${CALLING}_${f}_vars_subs_derived_freq_ratio_${TYPE}.txt -n1 > ${CALLING}_temp_vars_subs_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_SYN_vars_subs_derived_freq_ratio_${TYPE}.txt ${CALLING}_${f}_vars_subs_derived_freq_ratio_${TYPE}.txt | awk 'FNR == 1 {next} $3==0 {print $4,0.0} $3>0 {print $4,$4/$3}' OFS="\t" >> ${CALLING}_temp_vars_subs_derived_freq_ratio_${TYPE}.borrar
  #Now join the ALL and the current DEL feature data
  paste <(head ${CALLING}_ALL_vars_subs_derived_freq_ratio_${TYPE}.txt -n1) <(head ${CALLING}_temp_vars_subs_derived_freq_ratio_${TYPE}.borrar -n1) > ${CALLING}_joint_vars_subs_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_ALL_vars_subs_derived_freq_ratio_${TYPE}.txt ${CALLING}_temp_vars_subs_derived_freq_ratio_${TYPE}.borrar | awk 'FNR == 1 {next}{print $0}' OFS="\t" >> ${CALLING}_joint_vars_subs_derived_freq_ratio_${TYPE}.borrar
  #Replace the ALL table with the updated one
  mv ${CALLING}_joint_vars_subs_derived_freq_ratio_${TYPE}.borrar ${CALLING}_ALL_vars_subs_derived_freq_ratio_${TYPE}.txt
  rm ${CALLING}_temp_vars_subs_derived_freq_ratio_${TYPE}.borrar
  done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/*/*ALL_vars_subs_derived_freq_ratio.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

####For variants within species (transversions only):
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov) #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_nm2_origcov
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_TVonly_derived_freq_ratio_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
GREP=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
cd $V_PATH/$CALLING/annotation
POP_VCFS=($(ls `find . -name *'_perpop_TVonly_'$TYPE'.lr_ann.vcf' -print`))
echo "retrieving variants from each feature from each population"
for g in ${GREP[@]}
  do
  echo "feature: ${g}"
  FEATURE=$(grep "${g}" /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
  echo $FEATURE
  for p in ${POP_VCFS[@]}
    do
    echo "working with vcf ${p}"
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s_%s_%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_TVonly_${TYPE}.lr_ann.vcf/perpop_TVonly_${FEATURE}_${TYPE}.lr_ann.bed}
    done
  done
#Prepare file that will contain population and SYN ratio data:
mkdir -p derived_allele_freq_ratio
rm derived_allele_freq_ratio/${CALLING}_SYN_TVonly_derived_freq_ratio_${TYPE}.txt
echo -e "population_1\tpopulation_2\tSYN_ratio" > derived_allele_freq_ratio/${CALLING}_SYN_TVonly_derived_freq_ratio_${TYPE}.txt
DEL_FEATURES=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | grep -Ev 'synonymous_variant|intron_variant' | cut -f2)
#Prepare files that will contain DEL categories ratio data:
for f in ${DEL_FEATURES[@]}
  do
  rm derived_allele_freq_ratio/${CALLING}_${f}_TVonly_derived_freq_ratio_${TYPE}.txt
  echo -e "${f}_ratio\t${f}_norm_ratio" > derived_allele_freq_ratio/${CALLING}_${f}_TVonly_derived_freq_ratio_${TYPE}.txt
  done
POPS=($(ls `find . -name *'_perpop_TVonly_'$TYPE'.lr_ann.vcf' -print` | cut -d'/' -f3 | cut -d'_' -f1,2,3,4,5,6))
echo "calculating ratio for each pair of populations"
for p1 in ${POPS[@]}
  do
  for p2 in ${POPS[@]}
    do
    echo "${p1} vs ${p2}"
    echo "feature: SYN"
    LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_TVonly_SYN_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_TVonly_SYN_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_TVonly_SYN_${TYPE}.lr_ann.bed #col1: scaffold, col2: pos_start, col3: pos_end, col4: p1_af, col5: p2_af, col6: p1_af*(1-p2_af), col7: p2_af*(1-p1_af)
    SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_TVonly_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_TVonly_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
    echo -e "$p1\t$p2\t$RATIO" >> derived_allele_freq_ratio/${CALLING}_SYN_TVonly_derived_freq_ratio_${TYPE}.txt
    for f in ${DEL_FEATURES[@]}
      do
      echo "feature: ${f}"
      LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_TVonly_${f}_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_TVonly_${f}_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_TVonly_${f}_${TYPE}.lr_ann.bed
      SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_TVonly_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_TVonly_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
      echo -e "$RATIO" >> derived_allele_freq_ratio/${CALLING}_${f}_TVonly_derived_freq_ratio_${TYPE}.txt
      done
    done
  done
cd $V_PATH/$CALLING/annotation/derived_allele_freq_ratio
#Make an ALL features file storing the population and SYN data
cat ${CALLING}_SYN_TVonly_derived_freq_ratio_${TYPE}.txt > ${CALLING}_ALL_TVonly_derived_freq_ratio_${TYPE}.txt
for f in ${DEL_FEATURES[@]}
  do
  #First build a file with the feature's ratio and normalised ratio (i.e. divided by the SYN ratio):
  head ${CALLING}_${f}_TVonly_derived_freq_ratio_${TYPE}.txt -n1 > ${CALLING}_temp_TVonly_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_SYN_TVonly_derived_freq_ratio_${TYPE}.txt ${CALLING}_${f}_TVonly_derived_freq_ratio_${TYPE}.txt | awk 'FNR == 1 {next}{print $4,$4/$3}' OFS="\t" >> ${CALLING}_temp_TVonly_derived_freq_ratio_${TYPE}.borrar
  #Now join the ALL and the current DEL feature data
  paste <(head ${CALLING}_ALL_TVonly_derived_freq_ratio_${TYPE}.txt -n1) <(head ${CALLING}_temp_TVonly_derived_freq_ratio_${TYPE}.borrar -n1) > "${CALLING}"_joint_TVonly_derived_freq_ratio_${TYPE}.borrar
  paste ${CALLING}_ALL_TVonly_derived_freq_ratio_${TYPE}.txt ${CALLING}_temp_TVonly_derived_freq_ratio_${TYPE}.borrar | awk 'FNR == 1 {next}{print $0}' OFS="\t" >> ${CALLING}_joint_TVonly_derived_freq_ratio_${TYPE}.borrar
  #Replace the ALL table with the updated one
  mv ${CALLING}_joint_TVonly_derived_freq_ratio_${TYPE}.borrar ${CALLING}_ALL_TVonly_derived_freq_ratio_${TYPE}.txt
  rm ${CALLING}_temp_TVonly_derived_freq_ratio_${TYPE}.borrar
  done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/*/annotation/derived_allele_freq_ratio/*ALL_TVonly_derived_freq_ratio.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

###Perform block-jackknife to estimate derived allele frequency ratio variance.
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

#Block-jackknife method was performed following: https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf

#Option 1: use the normalized ratio, remove consecutive SNPs:

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #use the substitutions calling, which is the more complete name
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_vars_subs_derived_freq_ratio_jackknife.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

#Prepare output file:
rm ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
echo -e "population_pair\tsynonymous_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
FILES=$(ls *_vs_*vars_subs_SNP.lr_ann.bed | grep -Ev '_INTR_|_SYN_') #Only perform jackknife for deleterious categories.
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1.temp}
  POP_PAIR=$(echo $FILE | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'_' -f5 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for synonymous variants:
  SYN_FILE=$(ls $POP_PAIR*.bed | grep '_SYN_')
  SYN_SUM_P1=$(cut -f6  ${SYN_FILE} | paste -sd+ | bc)
  SYN_SUM_P2=$(cut -f7  ${SYN_FILE} | paste -sd+ | bc)
  SYN_RATIO_ALL=$(if [ $SYN_SUM_P1 == 0 ] || [ $SYN_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SYN_SUM_P1/$SYN_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  #if [[ $FEATURE == "LOF" ]]
    #then
    #BLOCK_SIZE=(50)
    #else
    #BLOCK_SIZE=(1000)
  #fi
  VARIANT_N=$(cat $FILE | wc -l)
  echo $VARIANT_N
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  ROUND_N=$(($VARIANT_N+50)) #NEW
  echo $ROUND_N
  BLOCK_SIZE=$(echo "scale=0; $ROUND_N"/100 | bc) #NEW
  echo $BLOCK_SIZE
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  BLOCK_N=$(echo "scale=0; $VARIANT_N/$BLOCK_SIZE" | bc)
  echo $BLOCK_N
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in $(seq 1 $BLOCK_N)
    do
    #echo ${i}
    START=$((${i}*$BLOCK_SIZE-($BLOCK_SIZE-1)))
    #echo $START
    END=$((${i}*$BLOCK_SIZE))
    #echo $END
    sed $START','$END'd' $FILE > ${FILE/.bed/.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE_NORM >> ${FILE/.bed/.jack1.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife pseudovalues and variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Obtain the jackknife pseudovalues (for more info, see https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf)
  awk -v mean_ratio="$DEL_RATIO_SUBSAMPLE_MEAN" -v num_blocks="$BLOCK_N" -F "\t" '{printf "%.6f\t%.6f\n", $1, (num_blocks*mean_ratio)-(num_blocks-1)*$1}' ${FILE/.bed/.jack1.temp} > ${FILE/.bed/.jack2.temp}
  DEL_PSEUDOVALUES_SUM=$(cut -f2 ${FILE/.bed/.jack2.temp} | paste -sd+ | bc)
  DEL_PSEUDOVALUES_MEAN=$(echo "scale=6; $DEL_PSEUDOVALUES_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the pseudovalues:
  awk -v mean_psv="$DEL_PSEUDOVALUES_MEAN" -F "\t" '{printf "%.6f\t%.6f\t%.6f\n", $1, $2, ($2-mean_psv)*($2-mean_psv)}' ${FILE/.bed/.jack2.temp} > ${FILE/.bed/.jack} #col1: ratio for the block, col2: pseudovalue of the block, col3: numerator of the variance
  VARIANCE_NUMERATOR_SUM=$(cut -f3 ${FILE/.bed/.jack} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM/($BLOCK_N-1)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
  rm *.temp 
  done
POP1=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $1}'))
POP2=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $2}'))
NSYN_TWO_SD=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f8))
LOF_TWO_SD=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'LOF' | cut -f8))

echo -e "population_1\tpopulation_2\tNSYN_norm_ratio_2SD\tLOF_norm_ratio_2SD" > ${CALLING}"_vars_subs_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
for ((i=0; i<=${#POP1[@]}; i++))
  do
  printf '%s\t%s\t%s\t%s\n' "${POP1[i]}" "${POP2[i]}" "${NSYN_TWO_SD[i]}" "${LOF_TWO_SD[i]}" >> ${CALLING}"_vars_subs_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
done



#Option 2: use the normalized ratio, remove random SNPs:

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #use the substitutions calling, which is the more complete name
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_vars_subs_derived_freq_ratio_jackknife_shuffle.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

#Prepare output file:
rm ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt"
echo -e "population_pair\tsynonymous_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt"
"$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD"
FILES=$(ls *_vs_*vars_subs_SNP.lr_ann.bed | grep -Ev '_INTR_|_SYN_') #Only perform jackknife for deleterious categories.
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1.temp}
  POP_PAIR=$(echo $FILE | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'_' -f5 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for synonymous variants:
  SYN_FILE=$(ls $POP_PAIR*.bed | grep '_SYN_')
  SYN_SUM_P1=$(cut -f6  ${SYN_FILE} | paste -sd+ | bc)
  SYN_SUM_P2=$(cut -f7  ${SYN_FILE} | paste -sd+ | bc)
  SYN_RATIO_ALL=$(if [ $SYN_SUM_P1 == 0 ] || [ $SYN_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SYN_SUM_P1/$SYN_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  VARIANT_N=$(cat $FILE | wc -l)
  ROUND_N=$(($VARIANT_N+50))
  BLOCK_SIZE=$(echo "scale=0; $ROUND_N"/100 | bc)
  BLOCK_N=(100)
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in $(seq 1 $BLOCK_N)
    do
    shuf -n$(($VARIANT_N-$BLOCK_SIZE)) $FILE > ${FILE/.bed/.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE_NORM >> ${FILE/.bed/.jack1.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife pseudovalues and variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Obtain the jackknife pseudovalues (for more info, see https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf)
  awk -v mean_ratio="$DEL_RATIO_SUBSAMPLE_MEAN" -v num_blocks="$BLOCK_N" -F "\t" '{printf "%.6f\t%.6f\n", $1, (num_blocks*mean_ratio)-(num_blocks-1)*$1}' ${FILE/.bed/.jack1.temp} > ${FILE/.bed/.jack2.temp}
  DEL_PSEUDOVALUES_SUM=$(cut -f2 ${FILE/.bed/.jack2.temp} | paste -sd+ | bc)
  DEL_PSEUDOVALUES_MEAN=$(echo "scale=6; $DEL_PSEUDOVALUES_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the pseudovalues:
  awk -v mean_psv="$DEL_PSEUDOVALUES_MEAN" -F "\t" '{printf "%.6f\t%.6f\t%.6f\n", $1, $2, ($2-mean_psv)*($2-mean_psv)}' ${FILE/.bed/.jack2.temp} > ${FILE/.bed/.jack}
  VARIANCE_NUMERATOR_SUM=$(cut -f3 ${FILE/.bed/.jack} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM/($BLOCK_N-1)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt"
  rm *.temp 
  done
POP1=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $1}'))
POP2=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $2}'))
NSYN_TWO_SD=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f8))
LOF_TWO_SD=($(cat ${CALLING}"_vars_subs_derived_freq_ratio_variance_shuffle_"${TYPE}".lr_ann.txt" | grep 'LOF' | cut -f8))

echo -e "population_1\tpopulation_2\tNSYN_norm_ratio_2SD\tLOF_norm_ratio_2SD" > ${CALLING}"_vars_subs_derived_freq_ratio_2SD_shuffle_"${TYPE}".lr_ann.txt"
for ((i=0; i<=${#POP1[@]}; i++))
  do
  printf '%s\t%s\t%s\t%s\n' "${POP1[i]}" "${POP2[i]}" "${NSYN_TWO_SD[i]}" "${LOF_TWO_SD[i]}" >> ${CALLING}"_vars_subs_derived_freq_ratio_2SD_shuffle_"${TYPE}".lr_ann.txt"
done



#Option 3: use the non-normalized ratio, remove consecutive SNPs, and include the syn ratio in the variance estimation (it yields the same result as option 1).

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #use the substitutions calling, which is the more complete name
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_vars_subs_derived_freq_ratio_jackknife_bis.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

#Prepare output file:
rm ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}"_bis.lr_ann.txt"
echo -e "population_pair\tsynonymous_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}"_bis.lr_ann.txt"
"$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD"
FILES=$(ls *_vs_*vars_subs_SNP.lr_ann.bed | grep -Ev '_INTR_|_SYN_') #Only perform jackknife for deleterious categories.
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1_bis.temp}
  POP_PAIR=$(echo $FILE | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'_' -f5 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for synonymous variants:
  SYN_FILE=$(ls $POP_PAIR*.bed | grep '_SYN_')
  SYN_SUM_P1=$(cut -f6  ${SYN_FILE} | paste -sd+ | bc)
  SYN_SUM_P2=$(cut -f7  ${SYN_FILE} | paste -sd+ | bc)
  SYN_RATIO_ALL=$(if [ $SYN_SUM_P1 == 0 ] || [ $SYN_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SYN_SUM_P1/$SYN_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  INV_SYN_RATIO_ALL=$(echo "scale=6; 1/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  if [[ $FEATURE == "LOF" ]]
    then
    BLOCK_SIZE=(10)
    else
    BLOCK_SIZE=(100)
  fi
  VARIANT_N=$(cat $FILE | wc -l)
  MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in $(seq 1 $BLOCK_N)
    do
    #echo ${i}
    START=$((${i}*$BLOCK_SIZE-($BLOCK_SIZE-1)))
    #echo $START
    END=$((${i}*$BLOCK_SIZE))
    #echo $END
    sed $START','$END'd' $FILE > ${FILE/.bed/_bis.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/_bis.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/_bis.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE >> ${FILE/.bed/.jack1_bis.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife pseudovalues and variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1_bis.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Obtain the jackknife pseudovalues (for more info, see https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf)
  awk -v mean_ratio="$DEL_RATIO_SUBSAMPLE_MEAN" -v num_blocks="$BLOCK_N" -F "\t" '{printf "%.6f\t%.6f\n", $1, (num_blocks*mean_ratio)-(num_blocks-1)*$1}' ${FILE/.bed/.jack1_bis.temp} > ${FILE/.bed/.jack2_bis.temp}
  DEL_PSEUDOVALUES_SUM=$(cut -f2 ${FILE/.bed/.jack2_bis.temp} | paste -sd+ | bc)
  DEL_PSEUDOVALUES_MEAN=$(echo "scale=6; $DEL_PSEUDOVALUES_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the pseudovalues:
  awk -v syn_ratio="$INV_SYN_RATIO_ALL" -v mean_psv="$DEL_PSEUDOVALUES_MEAN" -F "\t" '{printf "%.6f\t%.6f\t%.6f\n", $1, $2, ($2*$2*syn_ratio*syn_ratio)-(mean_psv*syn_ratio)*(mean_psv*syn_ratio)}' ${FILE/.bed/.jack2_bis.temp} > ${FILE/.bed/.jack_bis}
  VARIANCE_NUMERATOR_SUM=$(cut -f3 ${FILE/.bed/.jack_bis} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM/($BLOCK_N-1)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_vars_subs_derived_freq_ratio_variance_"${TYPE}"_bis.lr_ann.txt"
  rm *.temp
  done

#awk -v mean_all="$DEL_RATIO_ALL_NORM" -F "\t" '{printf "%.6f\t%.6f\t%.6f\n", $1, ($1-mean_all), ($1-mean_all)*($1-mean_all)}' ${FILE/.bed/.jack} to obtain variance of original values

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/*/*_vars_subs_derived_freq_ratio_*_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

###Plot derived allele frequency ratio.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

global_derived_freq_ratios <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_vars_subs_derived_freq_ratio.txt",col_names=T,col_types=c("ffddddd")) %>% 
  dplyr::filter(., grepl('5x', population_1),grepl('5x', population_2))

global_derived_freq_ratios_errors <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_vars_subs_derived_freq_ratio_2SD_SNP.lr_ann.txt",col_names=T,col_types=c("ffdd")) %>% 
  dplyr::filter(., grepl('5x', population_1),grepl('5x', population_2))

global_derived_freq_ratios <- left_join(global_derived_freq_ratios,global_derived_freq_ratios_errors,by=c("population_1","population_2"))
global_derived_freq_ratios$population_1 <- factor(global_derived_freq_ratios$population_1,levels=unique(global_derived_freq_ratios$population_1))
global_derived_freq_ratios$population_2 <- factor(global_derived_freq_ratios$population_2,levels=unique(global_derived_freq_ratios$population_2))
global_derived_freq_ratios

filtered_derived_freq_ratios = data_frame()
levels <- c("c_lp_5x_c_lp_do","c_lp_5x_c_lp_sm","c_ll_5x_c_ll_po","c_ll_5x_c_ll_no","c_ll_5x_c_ll_ki")
counter=0
for (i in levels) {
  counter=which(levels==i)
  if (counter < length(levels)) {
    for (j in levels[c((counter+1):length(levels))]) {
    print(paste0(i," vs ",j))
    filtered_derived_freq_ratios <- rbind(filtered_derived_freq_ratios,filter(global_derived_freq_ratios,population_1==i&population_2==j))
    }
  }
}
filtered_derived_freq_ratios[] <- as.data.frame(lapply(filtered_derived_freq_ratios, function(x) gsub("c_lp_5x_|c_ll_5x_", "", x)),stringsAsFactors=F)
filtered_derived_freq_ratios
tidy_derived_freq_ratios <- filtered_derived_freq_ratios %>% mutate(populations=paste(population_1,population_2,sep="/")) %>% select(contains("_norm_"),populations,-contains("_2SD")) %>% gather(ratio_type,ratio_value,-populations)
tidy_derived_freq_ratios_errors <- filtered_derived_freq_ratios %>% mutate(populations=paste(population_1,population_2,sep="/")) %>%
select(populations,contains("_2SD")) %>% gather(error_type,error_value,-populations)
tidy_derived_freq_ratios_errors <- as_tibble(lapply(tidy_derived_freq_ratios_errors, function(x) {gsub("_2SD", "", x)}))
tidy_derived_freq_ratios <- left_join(tidy_derived_freq_ratios,tidy_derived_freq_ratios_errors,by=c("populations","ratio_type"="error_type"))
tidy_derived_freq_ratios$populations <- factor(tidy_derived_freq_ratios$populations,levels=unique(tidy_derived_freq_ratios$populations))
tidy_derived_freq_ratios$ratio_type <- factor(tidy_derived_freq_ratios$ratio_type,levels=unique(tidy_derived_freq_ratios$ratio_type))
tidy_derived_freq_ratios$ratio_value <- as.double(tidy_derived_freq_ratios$ratio_value)
tidy_derived_freq_ratios$error_value <- as.double(tidy_derived_freq_ratios$error_value)
tidy_derived_freq_ratios <- tidy_derived_freq_ratios %>% mutate(ratio_value_from1=ratio_value-1)
pop_pairs <- c("c_lp_do/c_lp_sm","c_lp_sm/c_ll_ki","c_ll_po/c_ll_no","c_ll_po/c_ll_ki","c_ll_no/c_ll_ki")
tidy_derived_freq_ratios <- tidy_derived_freq_ratios %>% filter(populations %in% pop_pairs)
tidy_derived_freq_ratios$populations <- as.factor(droplevels(tidy_derived_freq_ratios$populations))

derived_freq_ratios_ggplot <- ggplot(data=tidy_derived_freq_ratios,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.6,0.2,by=0.1),labels=seq(-0.6,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red","NSYN_norm_ratio"="orange"),labels=c("NSYN","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip() +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()) 
derived_freq_ratios_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_vars_subs_derived_freq_ratio.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_errors_ggplot <-
  ggplot(data=tidy_derived_freq_ratios,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-6,6,by=1),labels=seq(-6,6,by=1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red","NSYN_norm_ratio"="orange"),labels=c("NSYN","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip() +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()
  )
derived_freq_ratios_errors_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_vars_subs_derived_freq_ratio_errors.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")


```

###Study the distribution of the derived allele frequency ratio.
####Obtain file with values:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

#Here we'll be obtaining the distribution of the numerator of the derived allele frequency ratio. After plotting it and doing some numbers, it makes no sense to plot this distribution, as the parameter that measures the difference between populations is precisely the ratio of the numerator and the denominator, and this ratio has no distribution.

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #use the substitutions calling, which is the more complete name
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_ALL_derived_freq_ratio_distribution_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path

RATIO_FILES=$(ls *_5x_*_vs_*_5x_*_perpop_*${TYPE}.lr_ann.bed)
rm ${CALLING}"_ALL_derived_freq_ratio_distribution_"${TYPE}".lr_ann.txt"
echo -e "populations\tfeature\tfA*(1-fB)" > ${CALLING}"_ALL_derived_freq_ratio_distribution_"${TYPE}".lr_ann.txt"
for i in ${RATIO_FILES[@]}
  do
  echo "${i}"
  POPS=$(echo "${i}" | cut -d'_' -f1-13)
  FEATURE=$(echo "${i}" | rev | cut -d'_' -f5 | rev)
  awk -v pops=$POPS -v feat=$FEATURE -F "\t" '{printf "%s\t%s\t%s\n", pops, feat, $6}' ${i} >> ${CALLING}"_ALL_derived_freq_ratio_distribution_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/*/*_ALL_derived_freq_ratio_distribution*.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

####Draw derived frequency ratio distributions.
```{r Other stuff}

library(readr)
library(dplyr)
library(ggplot2)

#First draw the NM distribution for each individual:
derived_freq_ratio_distribution <- read_tsv("/Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/derived_allele_freq_ratio_distribution/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_distribution_SNP.lr_ann.txt",col_names=T)
derived_freq_ratio_distribution$feature = factor(derived_freq_ratio_distribution$feature,levels=c("INTR","SYN","NSYN","LOF")) #Reorder factor levels.
derived_freq_ratio_distribution

for (pops in unique(derived_freq_ratio_distribution$populations)) {
  popA <- substr(pops,1,15)
  popB <- substr(pops,20,34)
  plot_data <- derived_freq_ratio_distribution %>% filter(populations == pops) %>% group_by(`fA*(1-fB)`,feature) %>% tally()
  plot_data
  dfrd_ggplot <- ggplot(data=plot_data, aes(`fA*(1-fB)`,n)) +
    facet_grid(feature ~ ., scales="free") +
    geom_col() +
    ggtitle(paste0("fA*(1-fB) distr. for pops ",popA," and ",popB)) +
    ylab("count") +
    #xlab("heritability") +
    #scale_x_continuous(breaks=seq(0,nrow(plot_data),2)) +
    theme_bw() +
    theme(text=element_text(size=12,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=16),
          #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
          #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          #panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position=c(0.92,0.86),
          legend.title=element_blank()
    )
  dfrd_ggplot
  ggsave(paste0(pops,"_fAx(1-fB)_distribution.lr_ann.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/derived_allele_freq_ratio_distribution/")
}

```

###Plot derived allele frequency spectrum.
####Combine per population and feature allele frequency files:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}"
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
script "${CALLING}_derived_freq_spectrum.lr_ann.log"
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
GREP=$(cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
cd $V_PATH/$CALLING/annotation
POPS=($(ls `find . -name *'5x'*'_perpop_'*'.lr_ann.bed' ! -name '*vs*' -print` | rev | cut -d'/' -f1 | rev | cut -d'_' -f-6 | sort | uniq))
echo "joining frequency files from each feature from each population"
rm $CALLING"_5x_perpop_ALL_features.lr_ann.af"
for p in ${POPS[@]}
  do
  echo "working with pop ${p}"
  SHORT_POP=$(echo ${p} | cut -d'_' -f4-6)
  for g in ${GREP[@]}
    do
    echo "feature: ${g}"
    FEATURE=$(grep "${g}" /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
    echo $FEATURE
    cat $SHORT_POP"_"$NM_COV"_perpop"/${p}"_"$NM_COV"_perpop_"$FEATURE".lr_ann.bed" | awk -v pop="$SHORT_POP" -v feat="$FEATURE" '{print pop,feat,$0}' OFS="\t" >> $CALLING"_5x_perpop_ALL_features.lr_ann.af"
    done
  done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/*/annotation/*5x_perpop_ALL_features.lr_ann.af /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/

```

####Plot spectra:
```{r}

library(readr)
library(dplyr)
library(ggplot2)

callings_af_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/", pattern="*ALL_features.lr_ann.af"),pattern='nm2nm3',inv=T,value=T)
all_plot_data <- data_frame("pop"=character(0),"feat"=factor(0),"af"=double(0),"n"=numeric(0))
for (file in callings_af_files) {
  calling_af <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/",file),col_names=c("pop","feat","pos","af"))
  for (p in unique(calling_af$pop)) {
    print(p)
    pop_af_data <- filter(calling_af,pop==p)
    #popsize=ifelse(p=="c_lp_sm"|p=="c_ll_ki",12,8)
    pop_plot_data <- pop_af_data %>% group_by(af,feat) %>% summarize(n=n(),pop=pop[n]) %>% arrange(feat) %>% select(pop,feat,af,n) %>% as.data.frame(.)
    #pop_af_data %>% mutate(af_groups=cut(af,breaks=seq(0,1.1,by=1/(2*popsize)))) %>% group_by(af_groups,feat) %>% summarize(n=n()) #problem with the binning due to rounded allele frequencies
    all_plot_data <- rbind(all_plot_data,pop_plot_data)
  }
}  
all_plot_data$feat <- as.factor(all_plot_data$feat)
all_plot_data$feat = factor(all_plot_data$feat,levels=c("INTR","SYN","NSYN","LOF")) #Reorder factor levels.
all_plot_data$pop = factor(all_plot_data$pop,levels=c("c_ll_ki","c_ll_no","c_ll_po","c_lp_sm","c_lp_do")) #Reorder factor levels.
all_plot_data <- all_plot_data %>% arrange(pop,feat,af)
all_plot_data

for (f in unique(all_plot_data$feat)) {
  feat_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(af,n,fill=pop)) +
    geom_col(width=0.07) +
    facet_grid(~pop,scales="free_x",space="free_x") +
    ggtitle(paste0(f," derived allele frequency spectrum")) +
    ylab("N") +
    xlab("AF") +
    theme_bw() +
    theme(text=element_text(size=12,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=16),
          #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
          #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          #panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position="none",
          legend.title=element_blank()
    )
    feat_ggplot
    ggsave(paste0(f,"_derived_allele_frequency_spectrum.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")
}

```

##Joint calling:
###Define desired features.
```{r Define desired features, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/
mkdir -p derived_allele_freq_ratio
cd derived_allele_freq_ratio/
rm features_derived_allele_freq_ratio.txt
echo -e "intron_variant\tINTR" >> features_derived_allele_freq_ratio.txt
echo -e "synonymous_variant\tSYN" >> features_derived_allele_freq_ratio.txt
echo -e "missense_variant\tNSYN" >> features_derived_allele_freq_ratio.txt
echo -e "|HIGH|\tLOF" >> features_derived_allele_freq_ratio.txt
cat features_derived_allele_freq_ratio.txt

```

###Calculate derived allele frequency ratio (joint callings).
####Global ratios referred to INTR:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_derived_freq_ratio_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
cd $V_PATH/$CALLING/annotation
GREP=$(cat derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
POP_VCFS=($(ls `find . -path *"_perpop/*_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
echo "retrieving variants from each feature from each population"
for g in ${GREP[@]}
  do
  echo "feature: ${g}"
  FEATURE=$(grep "${g}" derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
  echo $FEATURE
  for p in ${POP_VCFS[@]}
    do
    echo "working with vcf ${p}"
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s_%s_%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.bed}
    if [ $FEATURE == "NSYN" ]
      then
      echo "NTOL"
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.bed}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | awk -F"\t" '{printf ("%s_%s_%s\t%s\n", $1,$2,$3,$4)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_NTOL_${TYPE}.lr_ann.bed}
      echo "NDEL"
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.bed}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | awk -F"\t" '{printf ("%s_%s_%s\t%s\n", $1,$2,$3,$4)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_NDEL_${TYPE}.lr_ann.bed}
    fi
    done
  done
#Prepare file that will contain population and INTR ratio data:
mkdir -p derived_allele_freq_ratio
rm derived_allele_freq_ratio/${CALLING}_INTR_derived_freq_ratio_${VAR}_${TYPE}.txt
echo -e "population_1\tpopulation_2\tINTR_ratio" > derived_allele_freq_ratio/${CALLING}_INTR_derived_freq_ratio_${VAR}_${TYPE}.txt
DEL_FEATURES=(SYN NSYN NTOL NDEL LOF) #cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | grep -Ev 'synonymous_variant|intron_variant' | cut -f2
#Prepare files that will contain DEL categories ratio data:
for f in ${DEL_FEATURES[@]}
  do
  rm derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt
  echo -e "${f}_ratio\t${f}_norm_ratio" > derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt
  done
POPS=($(ls `find . -name *'_perpop_'$VAR'_'$TYPE'.lr_ann.vcf' -print` | cut -d'/' -f3 | cut -d'_' -f1,2,3,4,5,6))
echo "calculating ratio for each pair of populations"
for p1 in ${POPS[@]}
  do
  for p2 in ${POPS[@]}
    do
    echo "${p1} vs ${p2}"
    echo "feature: INTR"
    LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_${VAR}_INTR_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_${VAR}_INTR_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_INTR_${TYPE}.lr_ann.bed #col1: scaffold, col2: pos_start, col3: pos_end, col4: p1_af, col5: p2_af, col6: p1_af*(1-p2_af), col7: p2_af*(1-p1_af)
    SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_INTR_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_INTR_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
    echo -e "$p1\t$p2\t$RATIO" >> derived_allele_freq_ratio/${CALLING}_INTR_derived_freq_ratio_${VAR}_${TYPE}.txt
    for f in ${DEL_FEATURES[@]}
      do
      echo "feature: ${f}"
      LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed
      SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
      echo -e "$RATIO" >> derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt
      done
    done
  done
cd $V_PATH/$CALLING/annotation/derived_allele_freq_ratio
#Make an ALL features file storing the population and INTR data
cat ${CALLING}_INTR_derived_freq_ratio_${VAR}_${TYPE}.txt > ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt
for f in ${DEL_FEATURES[@]}
  do
  #First build a file with the feature's ratio and normalised ratio (i.e. divided by the INTR ratio):
  head ${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt -n1 > ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar
  paste ${CALLING}_INTR_derived_freq_ratio_${VAR}_${TYPE}.txt ${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt | awk 'FNR == 1 {next}{print $4,$4/$3}' OFS="\t" >> ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar
  #Now join the ALL and the current DEL feature data
  paste <(head ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt -n1) <(head ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar -n1) > "${CALLING}"_joint_derived_freq_ratio_${VAR}_${TYPE}.borrar
  paste ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar | awk 'FNR == 1 {next}{print $0}' OFS="\t" >> ${CALLING}_joint_derived_freq_ratio_${VAR}_${TYPE}.borrar
  #Replace the ALL table with the updated one
  mv ${CALLING}_joint_derived_freq_ratio_${VAR}_${TYPE}.borrar ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt
  rm ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/derived_allele_freq_ratio/${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

####Global ratios referred to SYN:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_derived_freq_ratio_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)
cd $V_PATH/$CALLING/annotation
GREP=$(cat derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
POP_VCFS=($(ls `find . -path *"_perpop/*_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
echo "retrieving variants from each feature from each population"
for g in ${GREP[@]}
  do
  echo "feature: ${g}"
  FEATURE=$(grep "${g}" derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
  echo $FEATURE
  for p in ${POP_VCFS[@]}
    do
    echo "working with vcf ${p}"
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s_%s_%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.bed}
    if [ $FEATURE == "NSYN" ]
      then
      echo "NTOL"
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.bed}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | awk -F"\t" '{printf ("%s_%s_%s\t%s\n", $1,$2,$3,$4)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_NTOL_${TYPE}.lr_ann.bed}
      echo "NDEL"
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.bed}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | awk -F"\t" '{printf ("%s_%s_%s\t%s\n", $1,$2,$3,$4)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_NDEL_${TYPE}.lr_ann.bed}
    fi
    done
  done
#Prepare file that will contain population and SYN ratio data:
mkdir -p derived_allele_freq_ratio
rm derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${VAR}_${TYPE}.txt
echo -e "population_1\tpopulation_2\tSYN_ratio" > derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${VAR}_${TYPE}.txt
DEL_FEATURES=(NSYN NTOL NDEL LOF) #cat /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | grep -Ev 'synonymous_variant|intron_variant' | cut -f2
#Prepare files that will contain DEL categories ratio data:
for f in ${DEL_FEATURES[@]}
  do
  rm derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt
  echo -e "${f}_ratio\t${f}_norm_ratio" > derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt
  done
POPS=($(ls `find . -name *'_perpop_'$VAR'_'$TYPE'.lr_ann.vcf' -print` | cut -d'/' -f3 | cut -d'_' -f1,2,3,4,5,6))
echo "calculating ratio for each pair of populations"
for p1 in ${POPS[@]}
  do
  for p2 in ${POPS[@]}
    do
    echo "${p1} vs ${p2}"
    echo "feature: SYN"
    LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_${VAR}_SYN_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_${VAR}_SYN_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_SYN_${TYPE}.lr_ann.bed #col1: scaffold, col2: pos_start, col3: pos_end, col4: p1_af, col5: p2_af, col6: p1_af*(1-p2_af), col7: p2_af*(1-p1_af)
    SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_SYN_${TYPE}.lr_ann.bed | paste -sd+ | bc)
    RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
    echo -e "$p1\t$p2\t$RATIO" >> derived_allele_freq_ratio/${CALLING}_SYN_derived_freq_ratio_${VAR}_${TYPE}.txt
    for f in ${DEL_FEATURES[@]}
      do
      echo "feature: ${f}"
      LANG=en_EN join -a1 -a2 -e 0.00 -o auto -j 1 <(LANG=en_EN sort -k1 ./*/${p1}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed) <(LANG=en_EN sort -k1 ./*/${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed) | awk -F"\t|_" '{print $1,$2,$3,$4,$5}' OFS="\t" | LANG=en_EN sort -k1,1 -k2,2n | awk '{printf "%s\t%s\t%s\t%.3f\t%.3f\t%.3f\t%.3f\n", $1,$2,$3,$4,$5,$4*(1-$5),$5*(1-$4)}' > derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed
      SUM_P1=$(cut -f6 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      SUM_P2=$(cut -f7 derived_allele_freq_ratio/${p1}_vs_${p2}_${NM_COV}_perpop_${VAR}_${f}_${TYPE}.lr_ann.bed | paste -sd+ | bc)
      RATIO=$(echo "scale=6; $SUM_P1/$SUM_P2" | bc)
      echo -e "$RATIO" >> derived_allele_freq_ratio/${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt
      done
    done
  done
cd $V_PATH/$CALLING/annotation/derived_allele_freq_ratio
#Make an ALL features file storing the population and SYN data
cat ${CALLING}_SYN_derived_freq_ratio_${VAR}_${TYPE}.txt > ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt
for f in ${DEL_FEATURES[@]}
  do
  #First build a file with the feature's ratio and normalised ratio (i.e. divided by the SYN ratio):
  head ${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt -n1 > ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar
  paste ${CALLING}_SYN_derived_freq_ratio_${VAR}_${TYPE}.txt ${CALLING}_${f}_derived_freq_ratio_${VAR}_${TYPE}.txt | awk 'FNR == 1 {next}{print $4,$4/$3}' OFS="\t" >> ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar
  #Now join the ALL and the current DEL feature data
  paste <(head ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt -n1) <(head ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar -n1) > "${CALLING}"_joint_derived_freq_ratio_${VAR}_${TYPE}.borrar
  paste ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar | awk 'FNR == 1 {next}{print $0}' OFS="\t" >> ${CALLING}_joint_derived_freq_ratio_${VAR}_${TYPE}.borrar
  #Replace the ALL table with the updated one
  mv ${CALLING}_joint_derived_freq_ratio_${VAR}_${TYPE}.borrar ${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt
  rm ${CALLING}_temp_derived_freq_ratio_${VAR}_${TYPE}.borrar
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/derived_allele_freq_ratio/${CALLING}_ALL_derived_freq_ratio_${VAR}_${TYPE}.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

###Perform block-jackknife to estimate derived allele frequency ratio variance.
####Direct approach:
#####Remove blocks of constant size (referred to INTR):
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

#Block-jackknife method was performed following: https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/derived_allele_freq_ratio
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_${VAR}_derived_freq_ratio_jackknife.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

#Prepare output file:
rm ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
echo -e "population_pair\tintronic_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
FILES=$(realpath *_vs_*${VAR}_*_SNP.lr_ann.bed | grep -Ev '_INTR_') #Only perform jackknife for deleterious categories.
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1.temp}
  POP_PAIR=$(echo $FILE | rev | cut -d'/' -f1 | rev | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'/' -f1 | cut -d'_' -f3 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for INTR variants:
  INTR_FILE=$(ls ${POP_PAIR}_nm2nm3_origcov_perpop_${VAR}_INTR_${TYPE}.lr_ann.bed)
  INTR_SUM_P1=$(cut -f6  ${INTR_FILE} | paste -sd+ | bc)
  INTR_SUM_P2=$(cut -f7  ${INTR_FILE} | paste -sd+ | bc)
  INTR_RATIO_ALL=$(if [ $INTR_SUM_P1 == 0 ] || [ $INTR_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $INTR_SUM_P1/$INTR_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$INTR_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  #if [[ $FEATURE == "LOF" ]]
    #then
    #BLOCK_SIZE=(50)
    #else
    #BLOCK_SIZE=(1000)
  #fi
  VARIANT_N=$(cat $FILE | wc -l)
  echo $VARIANT_N
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  ROUND_N=$(($VARIANT_N+50)) #NEW
  echo $ROUND_N
  BLOCK_SIZE=$(echo "scale=0; $ROUND_N"/100 | bc) #NEW
  echo $BLOCK_SIZE
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  BLOCK_N=$(echo "scale=0; $VARIANT_N/$BLOCK_SIZE" | bc)
  echo $BLOCK_N
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in $(seq 1 $BLOCK_N)
    do
    #echo ${i}
    START=$((${i}*$BLOCK_SIZE-($BLOCK_SIZE-1)))
    #echo $START
    END=$((${i}*$BLOCK_SIZE))
    #echo $END
    sed $START','$END'd' $FILE > ${FILE/.bed/.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$INTR_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE_NORM >> ${FILE/.bed/.jack1.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the subsamples:
  awk -v mean_psv="$DEL_RATIO_SUBSAMPLE_MEAN" -F "\t" '{printf "%.6f\t%.6f\n", $1, ($1-mean_psv)*($1-mean_psv)}' ${FILE/.bed/.jack1.temp} > ${FILE/.bed/.jack} #col1: ratio for the block, col2: numerator of the variance
  VARIANCE_NUMERATOR_SUM=$(cut -f2 ${FILE/.bed/.jack} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM*($BLOCK_N-1)/($BLOCK_N)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$INTR_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
  #rm *.temp 
  done
  
POP1=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $1}'))
POP2=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $2}'))
SYN_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep -w 'SYN' | cut -f8))
NSYN_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f8))
NTOL_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NTOL' | cut -f8))
NDEL_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NDEL' | cut -f8))
LOF_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'LOF' | cut -f8))

echo -e "population_1\tpopulation_2\tSYN_norm_ratio_2SD\tNSYN_norm_ratio_2SD\tNTOL_norm_ratio_2SD\tNDEL_norm_ratio_2SD\tLOF_norm_ratio_2SD" > ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
for ((i=0; i<=${#POP1[@]}; i++))
  do
  printf '%s\t%s\t%s\t%s\t%s\t%s\t%s\n' "${POP1[i]}" "${POP2[i]}" "${SYN_TWO_SD[i]}" "${NSYN_TWO_SD[i]}" "${NTOL_TWO_SD[i]}" "${NDEL_TWO_SD[i]}" "${LOF_TWO_SD[i]}" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/${CALLING}/annotation/derived_allele_freq_ratio/${CALLING}_${VAR}_derived_freq_ratio_*_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

#####Remove blocks of constant size (referred to SYN):
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

#Block-jackknife method was performed following: https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/derived_allele_freq_ratio
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_${VAR}_derived_freq_ratio_jackknife.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

#Prepare output file:
rm ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
echo -e "population_pair\tsynonymous_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
FILES=$(realpath *_vs_*${VAR}_*_SNP.lr_ann.bed | grep -Ev '_INTR_|_SYN_') #Only perform jackknife for deleterious categories.
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1.temp}
  POP_PAIR=$(echo $FILE | rev | cut -d'/' -f1 | rev | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'/' -f1 | cut -d'_' -f3 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for synonymous variants:
  SYN_FILE=$(ls ${POP_PAIR}_nm2nm3_origcov_perpop_${VAR}_SYN_${TYPE}.lr_ann.bed)
  SYN_SUM_P1=$(cut -f6  ${SYN_FILE} | paste -sd+ | bc)
  SYN_SUM_P2=$(cut -f7  ${SYN_FILE} | paste -sd+ | bc)
  SYN_RATIO_ALL=$(if [ $SYN_SUM_P1 == 0 ] || [ $SYN_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SYN_SUM_P1/$SYN_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  #if [[ $FEATURE == "LOF" ]]
    #then
    #BLOCK_SIZE=(50)
    #else
    #BLOCK_SIZE=(1000)
  #fi
  VARIANT_N=$(cat $FILE | wc -l)
  echo $VARIANT_N
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  ROUND_N=$(($VARIANT_N+50)) #NEW
  echo $ROUND_N
  BLOCK_SIZE=$(echo "scale=0; $ROUND_N"/100 | bc) #NEW
  echo $BLOCK_SIZE
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  BLOCK_N=$(echo "scale=0; $VARIANT_N/$BLOCK_SIZE" | bc)
  echo $BLOCK_N
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in $(seq 1 $BLOCK_N)
    do
    #echo ${i}
    START=$((${i}*$BLOCK_SIZE-($BLOCK_SIZE-1)))
    #echo $START
    END=$((${i}*$BLOCK_SIZE))
    #echo $END
    sed $START','$END'd' $FILE > ${FILE/.bed/.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE_NORM >> ${FILE/.bed/.jack1.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the subsamples:
  awk -v mean_psv="$DEL_RATIO_SUBSAMPLE_MEAN" -F "\t" '{printf "%.6f\t%.6f\n", $1, ($1-mean_psv)*($1-mean_psv)}' ${FILE/.bed/.jack1.temp} > ${FILE/.bed/.jack} #col1: ratio for the block, col2: numerator of the variance
  VARIANCE_NUMERATOR_SUM=$(cut -f2 ${FILE/.bed/.jack} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM*($BLOCK_N-1)/($BLOCK_N)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
  #rm *.temp 
  done
  
POP1=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $1}'))
POP2=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $2}'))
NSYN_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f8))
NTOL_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NTOL' | cut -f8))
NDEL_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NDEL' | cut -f8))
LOF_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'LOF' | cut -f8))

echo -e "population_1\tpopulation_2\tNSYN_norm_ratio_2SD\tNTOL_norm_ratio_2SD\tNDEL_norm_ratio_2SD\tLOF_norm_ratio_2SD" > ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
for ((i=0; i<=${#POP1[@]}; i++))
  do
  printf '%s\t%s\t%s\t%s\t%s\t%s\n' "${POP1[i]}" "${POP2[i]}" "${NSYN_TWO_SD[i]}" "${NTOL_TWO_SD[i]}" "${NDEL_TWO_SD[i]}" "${LOF_TWO_SD[i]}" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/${CALLING}/annotation/derived_allele_freq_ratio/${CALLING}_${VAR}_derived_freq_ratio_*_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```


####Pseudo-values approach:
#####Remove blocks of constant size:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

#Block-jackknife method was performed following: https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_${VAR}_derived_freq_ratio_jackknife.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


#Prepare output file:
rm ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
echo -e "population_pair\tsynonymous_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/derived_allele_freq_ratio
FILES=$(realpath *_vs_*${VAR}_*_SNP.lr_ann.bed | grep -Ev '_INTR_|_SYN_') #Only perform jackknife for deleterious categories.
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1.temp}
  POP_PAIR=$(echo $FILE | rev | cut -d'/' -f1 | rev | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'/' -f1 | cut -d'_' -f3 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for synonymous variants:
  SYN_FILE=$(ls $POP_PAIR*.bed | grep '_SYN_')
  SYN_SUM_P1=$(cut -f6  ${SYN_FILE} | paste -sd+ | bc)
  SYN_SUM_P2=$(cut -f7  ${SYN_FILE} | paste -sd+ | bc)
  SYN_RATIO_ALL=$(if [ $SYN_SUM_P1 == 0 ] || [ $SYN_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SYN_SUM_P1/$SYN_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  #if [[ $FEATURE == "LOF" ]]
    #then
    #BLOCK_SIZE=(50)
    #else
    #BLOCK_SIZE=(1000)
  #fi
  VARIANT_N=$(cat $FILE | wc -l)
  echo $VARIANT_N
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  ROUND_N=$(($VARIANT_N+50)) #NEW
  echo $ROUND_N
  BLOCK_SIZE=$(echo "scale=0; $ROUND_N"/100 | bc) #NEW
  echo $BLOCK_SIZE
  #MODULE=$(($VARIANT_N % $BLOCK_SIZE))
  #BLOCK_N=$((($VARIANT_N-$MODULE)/$BLOCK_SIZE))
  BLOCK_N=$(echo "scale=0; $VARIANT_N/$BLOCK_SIZE" | bc)
  echo $BLOCK_N
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in $(seq 1 $BLOCK_N)
    do
    #echo ${i}
    START=$((${i}*$BLOCK_SIZE-($BLOCK_SIZE-1)))
    #echo $START
    END=$((${i}*$BLOCK_SIZE))
    #echo $END
    sed $START','$END'd' $FILE > ${FILE/.bed/.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE_NORM >> ${FILE/.bed/.jack1.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife pseudovalues and variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Obtain the jackknife pseudovalues (for more info, see https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf)
  awk -v mean_ratio="$DEL_RATIO_SUBSAMPLE_MEAN" -v num_blocks="$BLOCK_N" -F "\t" '{printf "%.6f\t%.6f\n", $1, (num_blocks*mean_ratio)-(num_blocks-1)*$1}' ${FILE/.bed/.jack1.temp} > ${FILE/.bed/.jack2.temp}
  DEL_PSEUDOVALUES_SUM=$(cut -f2 ${FILE/.bed/.jack2.temp} | paste -sd+ | bc)
  DEL_PSEUDOVALUES_MEAN=$(echo "scale=6; $DEL_PSEUDOVALUES_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the pseudovalues:
  awk -v mean_psv="$DEL_PSEUDOVALUES_MEAN" -F "\t" '{printf "%.6f\t%.6f\t%.6f\n", $1, $2, ($2-mean_psv)*($2-mean_psv)}' ${FILE/.bed/.jack2.temp} > ${FILE/.bed/.jack} #col1: ratio for the block, col2: pseudovalue of the block, col3: numerator of the variance
  VARIANCE_NUMERATOR_SUM=$(cut -f3 ${FILE/.bed/.jack} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM/($BLOCK_N-1)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
  rm *.temp 
  done
POP1=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $1}'))
POP2=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $2}'))
NSYN_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f8))
LOF_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'LOF' | cut -f8))

echo -e "population_1\tpopulation_2\tNSYN_norm_ratio_2SD\tLOF_norm_ratio_2SD" > ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
for ((i=0; i<=${#POP1[@]}; i++))
  do
  printf '%s\t%s\t%s\t%s\n' "${POP1[i]}" "${POP2[i]}" "${NSYN_TWO_SD[i]}" "${LOF_TWO_SD[i]}" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/*/*_vars_subs_derived_freq_ratio_*_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

#####Remove scaffolds:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

#Block-jackknife method was performed following: https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(variants) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_${VAR}_derived_freq_ratio_jackknife.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


#Prepare output file:
rm ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
echo -e "population_pair\tsynonymous_ratio\tfeature\tglobal_ratio\tglobal_ratio_norm\tmean_subsample_ratio_norm\tvariance_subsample_ratio_norm\t±2SD_subsample_ratio_norm" > ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/derived_allele_freq_ratio
FILES=$(realpath *_vs_*${VAR}_*_SNP.lr_ann.bed | grep -Ev '_SYN_') #Only perform jackknife for deleterious categories.
cd /GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/$CALLING
for FILE in ${FILES[@]}
  do
  #Define comparison file related variables:
  echo $FILE
  rm -f ${FILE/.bed/.jack1.temp}
  POP_PAIR=$(echo $FILE | rev | cut -d'/' -f1 | rev | cut -d'_' -f1-13)
  FEATURE=$(echo $FILE | rev | cut -d'/' -f1 | cut -d'_' -f3 | rev)
  echo "Working with comparison" $POP_PAIR "and feature" $FEATURE
  #Compute the ratio for all variants:
  DEL_SUM_P1=$(cut -f6  ${FILE} | paste -sd+ | bc)
  DEL_SUM_P2=$(cut -f7  ${FILE} | paste -sd+ | bc)
  DEL_RATIO_ALL=$(if [ $DEL_SUM_P1 == 0 ] || [ $DEL_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_P1/$DEL_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  #Compute the ratio for synonymous variants:
  SYN_FILE=$(ls $POP_PAIR*.bed | grep '_SYN_')
  SYN_SUM_P1=$(cut -f6  ${SYN_FILE} | paste -sd+ | bc)
  SYN_SUM_P2=$(cut -f7  ${SYN_FILE} | paste -sd+ | bc)
  SYN_RATIO_ALL=$(if [ $SYN_SUM_P1 == 0 ] || [ $SYN_SUM_P2 == 0 ]; then echo 0.00; else echo "scale=6; $SYN_SUM_P1/$SYN_SUM_P2" | bc | awk '{printf "%f", $0}'; fi)
  DEL_RATIO_ALL_NORM=$(echo "scale=6; $DEL_RATIO_ALL/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
  #Define jackknife block related variables:
  SCAFFOLDS=$(cat $FILE | cut -f1 | sort | uniq)
  BLOCK_N=$(cat $FILE | cut -f1 | sort | uniq | wc -l)
  #Compute the ratio for all block-less subsamples:
  echo "Performing jackknife"
  for i in ${SCAFFOLDS[@]}
    do
    #echo ${i}
    grep -v ${i} $FILE > ${FILE/.bed/.temp}
    DEL_SUM_SUBSAMPLE_P1=$(cut -f6  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_SUM_SUBSAMPLE_P2=$(cut -f7  ${FILE/.bed/.temp} | paste -sd+ | bc)
    DEL_RATIO_SUBSAMPLE=$(if [ $DEL_SUM_SUBSAMPLE_P1 == 0 ] || [ $DEL_SUM_SUBSAMPLE_P2 == 0 ]; then echo 0.00; else echo "scale=6; $DEL_SUM_SUBSAMPLE_P1/$DEL_SUM_SUBSAMPLE_P2" | bc | awk '{printf "%f", $0}'; fi)
    DEL_RATIO_SUBSAMPLE_NORM=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE/$SYN_RATIO_ALL" | bc | awk '{printf "%f", $0}')
    echo $DEL_RATIO_SUBSAMPLE_NORM >> ${FILE/.bed/.jack1.temp} #store all subsample ratios in a file
    done
  echo "Computing jackknife pseudovalues and variance"
  #Compute the subsamples average ratio:
  DEL_RATIO_SUBSAMPLE_SUM=$(cut -f1 ${FILE/.bed/.jack1.temp} | paste -sd+ | bc)
  DEL_RATIO_SUBSAMPLE_MEAN=$(echo "scale=6; $DEL_RATIO_SUBSAMPLE_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Obtain the jackknife pseudovalues (for more info, see https://www.math.wustl.edu/~sawyer/handouts/Jackknife.pdf)
  awk -v mean_ratio="$DEL_RATIO_SUBSAMPLE_MEAN" -v num_blocks="$BLOCK_N" -F "\t" '{printf "%.6f\t%.6f\n", $1, (num_blocks*mean_ratio)-(num_blocks-1)*$1}' ${FILE/.bed/.jack1.temp} > ${FILE/.bed/.jack2.temp}
  DEL_PSEUDOVALUES_SUM=$(cut -f2 ${FILE/.bed/.jack2.temp} | paste -sd+ | bc)
  DEL_PSEUDOVALUES_MEAN=$(echo "scale=6; $DEL_PSEUDOVALUES_SUM/$BLOCK_N" | bc | awk '{printf "%f", $0}')
  #Compute variance of the pseudovalues:
  awk -v mean_psv="$DEL_PSEUDOVALUES_MEAN" -F "\t" '{printf "%.6f\t%.6f\t%.6f\n", $1, $2, ($2-mean_psv)*($2-mean_psv)}' ${FILE/.bed/.jack2.temp} > ${FILE/.bed/.jack} #col1: ratio for the block, col2: pseudovalue of the block, col3: numerator of the variance
  VARIANCE_NUMERATOR_SUM=$(cut -f3 ${FILE/.bed/.jack} | paste -sd+ | bc)
  VARIANCE=$(echo "scale=6; $VARIANCE_NUMERATOR_SUM/($BLOCK_N-1)" | bc | awk '{printf "%f", $0}')
  TWO_SD=$(echo "scale=6; sqrt ($VARIANCE)" | bc | awk '{printf "%f", $0*2}')
  #Compile summary table:
  echo -e "$POP_PAIR\t$SYN_RATIO_ALL\t$FEATURE\t$DEL_RATIO_ALL\t$DEL_RATIO_ALL_NORM\t$DEL_RATIO_SUBSAMPLE_MEAN\t$VARIANCE\t$TWO_SD" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt"
  rm *.temp 
  done
POP1=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $1}'))
POP2=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f1 | awk 'BEGIN {FS="_vs_"} {print $2}'))
NSYN_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'NSYN' | cut -f8))
LOF_TWO_SD=($(cat ${CALLING}"_"${VAR}"_derived_freq_ratio_variance_"${TYPE}".lr_ann.txt" | grep 'LOF' | cut -f8))

echo -e "population_1\tpopulation_2\tNSYN_norm_ratio_2SD\tLOF_norm_ratio_2SD" > ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
for ((i=0; i<=${#POP1[@]}; i++))
  do
  printf '%s\t%s\t%s\t%s\n' "${POP1[i]}" "${POP2[i]}" "${NSYN_TWO_SD[i]}" "${LOF_TWO_SD[i]}" >> ${CALLING}"_"${VAR}"_derived_freq_ratio_2SD_"${TYPE}".lr_ann.txt"
done

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/genetic_load_analysis/derived_allele_freq_ratio/*/*_varssubs_derived_freq_ratio_*_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/

```

###Plot derived allele frequency ratio.
####Referred to INTR:
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

global_derived_freq_ratios <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.txt",col_names=T,col_types=c("ffddddddddddd")) %>% dplyr::filter(., grepl('5x', population_1),grepl('5x', population_2))

global_derived_freq_ratios_errors <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_varssubs_derived_freq_ratio_2SD_SNP.lr_ann.txt",col_names=T,col_types=c("ffddddd")) %>% dplyr::filter(., grepl('5x', population_1),grepl('5x', population_2))

global_derived_freq_ratios <- left_join(global_derived_freq_ratios,global_derived_freq_ratios_errors,by=c("population_1","population_2"))
global_derived_freq_ratios$population_1 <- factor(global_derived_freq_ratios$population_1,levels=unique(global_derived_freq_ratios$population_1))
global_derived_freq_ratios$population_2 <- factor(global_derived_freq_ratios$population_2,levels=unique(global_derived_freq_ratios$population_2))
global_derived_freq_ratios

filtered_derived_freq_ratios = data_frame()
levels <- c("c_lp_5x_c_lp_do","c_lp_5x_c_lp_sm","c_ll_5x_c_ll_po","c_ll_5x_c_ll_no","c_ll_5x_c_ll_ki")
counter=0
for (i in levels) {
  counter=which(levels==i)
  if (counter < length(levels)) {
    for (j in levels[c((counter+1):length(levels))]) {
    print(paste0(i," vs ",j))
    filtered_derived_freq_ratios <- rbind(filtered_derived_freq_ratios,filter(global_derived_freq_ratios,population_1==i&population_2==j))
    }
  }
}
filtered_derived_freq_ratios[] <- as.data.frame(lapply(filtered_derived_freq_ratios, function(x) gsub("c_lp_5x_|c_ll_5x_", "", x)),stringsAsFactors=F)
filtered_derived_freq_ratios
tidy_derived_freq_ratios <- filtered_derived_freq_ratios %>% mutate(populations=paste(population_1,population_2,sep="/")) %>% select(contains("_norm_"),populations,-contains("_2SD")) %>% gather(ratio_type,ratio_value,-populations)
tidy_derived_freq_ratios_errors <- filtered_derived_freq_ratios %>% mutate(populations=paste(population_1,population_2,sep="/")) %>%
select(populations,contains("_2SD")) %>% gather(error_type,error_value,-populations)
tidy_derived_freq_ratios_errors <- as_tibble(lapply(tidy_derived_freq_ratios_errors, function(x) {gsub("_2SD", "", x)}))
tidy_derived_freq_ratios <- left_join(tidy_derived_freq_ratios,tidy_derived_freq_ratios_errors,by=c("populations","ratio_type"="error_type"))
tidy_derived_freq_ratios$populations <- factor(tidy_derived_freq_ratios$populations,levels=unique(tidy_derived_freq_ratios$populations))
tidy_derived_freq_ratios$ratio_type <- factor(tidy_derived_freq_ratios$ratio_type,levels=c("SYN_norm_ratio","NSYN_norm_ratio","NTOL_norm_ratio","NDEL_norm_ratio","LOF_norm_ratio"))
tidy_derived_freq_ratios$ratio_value <- as.double(tidy_derived_freq_ratios$ratio_value)
tidy_derived_freq_ratios$error_value <- as.double(tidy_derived_freq_ratios$error_value)
tidy_derived_freq_ratios <- tidy_derived_freq_ratios %>% mutate(ratio_value_from1=ratio_value-1)
pop_pairs <- c("c_lp_do/c_lp_sm","c_lp_sm/c_ll_ki","c_ll_po/c_ll_no","c_ll_po/c_ll_ki","c_ll_no/c_ll_ki")
tidy_derived_freq_ratios <- tidy_derived_freq_ratios %>% filter(populations %in% pop_pairs)
tidy_derived_freq_ratios$populations <- as.factor(droplevels(tidy_derived_freq_ratios$populations))

tidy_derived_freq_ratios$populations <- gsub('c_ll_ki', 'KIR', tidy_derived_freq_ratios$populations)
tidy_derived_freq_ratios$populations <- gsub('c_ll_po', 'POL', tidy_derived_freq_ratios$populations)
tidy_derived_freq_ratios$populations <- gsub('c_ll_no', 'NOR', tidy_derived_freq_ratios$populations)
tidy_derived_freq_ratios$populations <- gsub('c_lp_sm', 'AND', tidy_derived_freq_ratios$populations)
tidy_derived_freq_ratios$populations <- gsub('c_lp_do', 'DON', tidy_derived_freq_ratios$populations)

tidy_derived_freq_ratios$populations = factor(tidy_derived_freq_ratios$populations,levels=c("DON/AND","AND/KIR","POL/NOR","POL/KIR","NOR/KIR"))


tidy_derived_freq_ratios_simple <- tidy_derived_freq_ratios[grep("SYN_norm_ratio|NSYN_norm_ratio|LOF_norm_ratio",tidy_derived_freq_ratios$ratio_type),]
tidy_derived_freq_ratios_complete <- tidy_derived_freq_ratios[grep("NSYN_norm_ratio",tidy_derived_freq_ratios$ratio_type,inv=T),]

derived_freq_ratios_simple_ggplot <- ggplot(data=tidy_derived_freq_ratios_simple,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.6,0.2,by=0.1),labels=seq(-0.6,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NSYN_norm_ratio"="orange","SYN_norm_ratio"="yellow"),labels=c("SYN","NSYN","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()) 
derived_freq_ratios_simple_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.simple.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_simple_errors_ggplot <-
  ggplot(data=tidy_derived_freq_ratios_simple,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.6,0.2,by=0.1),labels=seq(-0.6,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NSYN_norm_ratio"="orange","SYN_norm_ratio"="yellow"),labels=c("SYN","NSYN","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()
  )
derived_freq_ratios_simple_errors_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.simple_errors.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")


derived_freq_ratios_complete_ggplot <- ggplot(data=tidy_derived_freq_ratios_complete,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.4,0.2,by=0.1),labels=seq(-0.4,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NDEL_norm_ratio"="orangered","NTOL_norm_ratio"="orange","SYN_norm_ratio"="yellow"),labels=c("SYN","NTOL","NDEL","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()) 
derived_freq_ratios_complete_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.complete.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_complete_errors_ggplot <-
  ggplot(data=tidy_derived_freq_ratios_complete,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.4,0.3,by=0.1),labels=seq(-0.4,0.3,by=0.1)+1) +
  #scale_y_continuous(breaks=seq(-2,2,by=0.25),labels=seq(-2,2,by=0.25)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NDEL_norm_ratio"="orangered","NTOL_norm_ratio"="orange","SYN_norm_ratio"="yellow"),labels=c("synonymous","m. tolerated","m. deleterious","LoF"),guide=guide_legend(reverse=TRUE)) +
  geom_hline(yintercept = 0) +
  coord_flip(ylim=c(-0.45,0.35),expand=T,clip = "on") +
  #coord_cartesian(ylim=c(-0.3,0.2),expand=T,default=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.84,0.60),
      legend.title=element_blank()
  )
derived_freq_ratios_complete_errors_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.complete_errors.good.pdf", width=16, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_complete_errors_ugly_ggplot <-
  ggplot(data=tidy_derived_freq_ratios_complete,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.4,0.4,by=0.1),labels=seq(-0.4,0.4,by=0.1)+1) +
  #scale_y_continuous(breaks=seq(-2,2,by=0.25),labels=seq(-2,2,by=0.25)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NDEL_norm_ratio"="orangered","NTOL_norm_ratio"="orange"),labels=c("NTOL","NDEL","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip() +
  #coord_cartesian(ylim=c(-0.3,0.2),expand=T,default=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()
  )
derived_freq_ratios_complete_errors_ugly_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.complete_errors_ugly.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

```

####Referred to SYN:
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

global_derived_freq_ratios <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.txt",col_names=T,col_types=c("ffddddddddd")) %>% dplyr::filter(., grepl('5x', population_1),grepl('5x', population_2))

global_derived_freq_ratios_errors <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_varssubs_derived_freq_ratio_2SD_SNP.lr_ann.txt",col_names=T,col_types=c("ffdddd")) %>% dplyr::filter(., grepl('5x', population_1),grepl('5x', population_2))

global_derived_freq_ratios <- left_join(global_derived_freq_ratios,global_derived_freq_ratios_errors,by=c("population_1","population_2"))
global_derived_freq_ratios$population_1 <- factor(global_derived_freq_ratios$population_1,levels=unique(global_derived_freq_ratios$population_1))
global_derived_freq_ratios$population_2 <- factor(global_derived_freq_ratios$population_2,levels=unique(global_derived_freq_ratios$population_2))
global_derived_freq_ratios

filtered_derived_freq_ratios = data_frame()
levels <- c("c_lp_5x_c_lp_do","c_lp_5x_c_lp_sm","c_ll_5x_c_ll_po","c_ll_5x_c_ll_no","c_ll_5x_c_ll_ki")
counter=0
for (i in levels) {
  counter=which(levels==i)
  if (counter < length(levels)) {
    for (j in levels[c((counter+1):length(levels))]) {
    print(paste0(i," vs ",j))
    filtered_derived_freq_ratios <- rbind(filtered_derived_freq_ratios,filter(global_derived_freq_ratios,population_1==i&population_2==j))
    }
  }
}
filtered_derived_freq_ratios[] <- as.data.frame(lapply(filtered_derived_freq_ratios, function(x) gsub("c_lp_5x_|c_ll_5x_", "", x)),stringsAsFactors=F)
filtered_derived_freq_ratios
tidy_derived_freq_ratios <- filtered_derived_freq_ratios %>% mutate(populations=paste(population_1,population_2,sep="/")) %>% select(contains("_norm_"),populations,-contains("_2SD")) %>% gather(ratio_type,ratio_value,-populations)
tidy_derived_freq_ratios_errors <- filtered_derived_freq_ratios %>% mutate(populations=paste(population_1,population_2,sep="/")) %>%
select(populations,contains("_2SD")) %>% gather(error_type,error_value,-populations)
tidy_derived_freq_ratios_errors <- as_tibble(lapply(tidy_derived_freq_ratios_errors, function(x) {gsub("_2SD", "", x)}))
tidy_derived_freq_ratios <- left_join(tidy_derived_freq_ratios,tidy_derived_freq_ratios_errors,by=c("populations","ratio_type"="error_type"))
tidy_derived_freq_ratios$populations <- factor(tidy_derived_freq_ratios$populations,levels=unique(tidy_derived_freq_ratios$populations))
tidy_derived_freq_ratios$ratio_type <- factor(tidy_derived_freq_ratios$ratio_type,levels=c("NSYN_norm_ratio","NTOL_norm_ratio","NDEL_norm_ratio","LOF_norm_ratio"))
tidy_derived_freq_ratios$ratio_value <- as.double(tidy_derived_freq_ratios$ratio_value)
tidy_derived_freq_ratios$error_value <- as.double(tidy_derived_freq_ratios$error_value)
tidy_derived_freq_ratios <- tidy_derived_freq_ratios %>% mutate(ratio_value_from1=ratio_value-1)
pop_pairs <- c("c_lp_do/c_lp_sm","c_lp_sm/c_ll_ki","c_ll_po/c_ll_no","c_ll_po/c_ll_ki","c_ll_no/c_ll_ki")
tidy_derived_freq_ratios <- tidy_derived_freq_ratios %>% filter(populations %in% pop_pairs)
tidy_derived_freq_ratios$populations <- as.factor(droplevels(tidy_derived_freq_ratios$populations))

tidy_derived_freq_ratios_simple <- tidy_derived_freq_ratios[grep("NSYN_norm_ratio|LOF_norm_ratio",tidy_derived_freq_ratios$ratio_type),]
tidy_derived_freq_ratios_complete <- tidy_derived_freq_ratios[grep("NSYN_norm_ratio",tidy_derived_freq_ratios$ratio_type,inv=T),]

derived_freq_ratios_simple_ggplot <- ggplot(data=tidy_derived_freq_ratios_simple,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.6,0.2,by=0.1),labels=seq(-0.6,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NSYN_norm_ratio"="orange"),labels=c("NSYN","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()) 
derived_freq_ratios_simple_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.simple.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_simple_errors_ggplot <-
  ggplot(data=tidy_derived_freq_ratios_simple,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.6,0.2,by=0.1),labels=seq(-0.6,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NSYN_norm_ratio"="orange"),labels=c("NSYN","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()
  )
derived_freq_ratios_simple_errors_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.simple_errors.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")


derived_freq_ratios_complete_ggplot <- ggplot(data=tidy_derived_freq_ratios_complete,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.4,0.2,by=0.1),labels=seq(-0.4,0.2,by=0.1)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NDEL_norm_ratio"="orangered","NTOL_norm_ratio"="orange"),labels=c("NTOL","NDEL","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()) 
derived_freq_ratios_complete_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.complete.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_complete_errors_ggplot <-
  ggplot(data=tidy_derived_freq_ratios_complete,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.4,0.4,by=0.1),labels=seq(-0.4,0.4,by=0.1)+1) +
  #scale_y_continuous(breaks=seq(-2,2,by=0.25),labels=seq(-2,2,by=0.25)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NDEL_norm_ratio"="orangered","NTOL_norm_ratio"="orange"),labels=c("NTOL","NDEL","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip(ylim=c(-0.3,0.15),expand=T,clip = "on") +
  #coord_cartesian(ylim=c(-0.3,0.2),expand=T,default=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()
  )
derived_freq_ratios_complete_errors_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.complete_errors.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

derived_freq_ratios_complete_errors_ugly_ggplot <-
  ggplot(data=tidy_derived_freq_ratios_complete,aes(x=populations,y=ratio_value_from1,fill=ratio_type)) +
  geom_bar(stat='identity', position=position_dodge()) +
  geom_errorbar(aes(ymin=ratio_value_from1-error_value, ymax=ratio_value_from1+error_value), position=position_dodge()) +
  scale_x_discrete(limits = rev(levels(tidy_derived_freq_ratios$populations))) +
  scale_y_continuous(breaks=seq(-0.4,0.4,by=0.1),labels=seq(-0.4,0.4,by=0.1)+1) +
  #scale_y_continuous(breaks=seq(-2,2,by=0.25),labels=seq(-2,2,by=0.25)+1) +
  scale_fill_manual("legend",values=c("LOF_norm_ratio"="red3","NDEL_norm_ratio"="orangered","NTOL_norm_ratio"="orange"),labels=c("NTOL","NDEL","LoF"),guide=guide_legend(reverse=TRUE)) +
  coord_flip() +
  #coord_cartesian(ylim=c(-0.3,0.2),expand=T,default=T,clip = "on") +
  xlab("Populations") +
  ylab(expression(bold(R["X/Y"]))) +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #panel.background=element_blank(),
      #panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.12,0.12),
      legend.title=element_blank()
  )
derived_freq_ratios_complete_errors_ugly_ggplot
ggsave("c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ALL_derived_freq_ratio_varssubs_SNP.complete_errors_ugly.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_ratio")

```

###Plot derived allele frequency spectrum.
####Combine per population and feature allele frequency files:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions #segregating
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_derived_freq_spectrum_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


cd /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
GREP=$(cat derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
POP_VCFS=($(ls `find . -path *"_perpop/c_*5x*_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
rm ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.af"
rm ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.maf.txt"
echo -e "population\tfeature\tmaf" > ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.maf.txt"
for p in ${POP_VCFS[@]}
  do
  SP_DATASET=$(echo ${p} | cut -d'/' -f3 | cut -d'_' -f1-3)
  SHORT_POP=$(echo ${p} | cut -d'/' -f3 | cut -d'_' -f4-6)
  echo "working with pop" $SHORT_POP "and vcf" $p
  echo "extracting derived allele frequencies from" $SHORT_POP
  for g in ${GREP[@]}
    do
    #echo "grepping: ${g}"
    FEATURE=$(grep "${g}" derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
    echo "extracting from feature:" $FEATURE
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s\t%s\t%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.af}
    N_SITES=$(wc -l < ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.af})
    SUM_AF=$(cut -f4 ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.af} | paste -sd+ | bc)
    AVG_AF=$(awk -v sum=$SUM_AF -v total=$N_SITES 'BEGIN{printf "%.3f\n", sum/total}')
    echo -e "$SHORT_POP\t$FEATURE\t$AVG_AF" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.maf.txt"
    echo "joining frequency files"
    awk -v pop="$SHORT_POP" -v feat="$FEATURE" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_"$VAR"_"$FEATURE"_"$TYPE".lr_ann.af" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.af"
    if [ $FEATURE == "NSYN" ]
      then
      FEATURE2=(NTOL)
      echo "extracting from feature:" $FEATURE2
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.af}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af}
      N_SITES=$(wc -l < ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af})
      SUM_AF=$(cut -f4 ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af} | paste -sd+ | bc)
      AVG_AF=$(awk -v sum=$SUM_AF -v total=$N_SITES 'BEGIN{printf "%.3f\n", sum/total}')
      echo -e "$SHORT_POP\t$FEATURE2\t$AVG_AF" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.maf.txt"
      echo "joining frequency files"
      awk -v pop="$SHORT_POP" -v feat="$FEATURE2" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_"$VAR"_"$FEATURE2"_"$TYPE".lr_ann.af" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.af"
      FEATURE2=(NDEL)
      echo "extracting from feature:" $FEATURE2
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.af}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af}
      N_SITES=$(wc -l < ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af})
      SUM_AF=$(cut -f4 ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af} | paste -sd+ | bc)
      AVG_AF=$(awk -v sum=$SUM_AF -v total=$N_SITES 'BEGIN{printf "%.3f\n", sum/total}')
      echo -e "$SHORT_POP\t$FEATURE2\t$AVG_AF" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.maf.txt"
      echo "joining frequency files"
      awk -v pop="$SHORT_POP" -v feat="$FEATURE2" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_"$VAR"_"$FEATURE2"_"$TYPE".lr_ann.af" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.af"
    fi
    done
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*5x_perpop_ALL_features_${VAR}*.lr_ann.af /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/
#scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*5x_perpop_ALL_features*.lr_ann.maf.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/

```

####Plot spectra (old):
```{r}

library(readr)
library(dplyr)
library(ggplot2)

callings_af_files <- intersect(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/", pattern="*ALL_features*"),list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/", pattern="_varssubs_SNP.|_variants_SNP.|_substitutions_SNP."))

#var_type <- c("varssubs","variants","substitutions")
var_type <- c("varssubs")

for (type in var_type) {
  print(type)
  all_plot_data <- data_frame("pop"=character(0),"feat"=factor(0),"af"=double(0),"n"=numeric(0))
  calling_af <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/",callings_af_files[grep(type,callings_af_files)]),col_names=c("pop","feat","pos","af"))
  for (p in unique(calling_af$pop)) {
    print(p)
    pop_af_data <- filter(calling_af,pop==p)
    #popsize=ifelse(p=="c_lp_sm"|p=="c_ll_ki",12,8)
    pop_plot_data <- pop_af_data %>% group_by(af,feat) %>% summarize(n=n(),pop=pop[n]) %>% arrange(feat) %>% select(pop,feat,af,n) %>% as.data.frame(.)
    #pop_af_data %>% mutate(af_groups=cut(af,breaks=seq(0,1.1,by=1/(2*popsize)))) %>% group_by(af_groups,feat) %>% summarize(n=n()) #problem with the binning due to rounded allele frequencies
    all_plot_data <- rbind(all_plot_data,pop_plot_data)
    }
  all_plot_data$feat <- as.factor(all_plot_data$feat)
  all_plot_data$feat = factor(all_plot_data$feat,levels=c("INTR","SYN","NSYN","NTOL","NDEL","LOF")) #Reorder factor levels.
  all_plot_data$pop = factor(all_plot_data$pop,levels=c("c_ll_ki","c_ll_no","c_ll_po","c_lp_sm","c_lp_do")) #Reorder factor levels.
  all_plot_data <- all_plot_data %>% arrange(pop,feat,af)
  all_plot_data

  for (f in unique(all_plot_data$feat)) {
    feat_thin_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(af,n,fill=pop)) +
      geom_col() +
      facet_grid(~pop,scales="free_x",space="free_x") +
      ggtitle(paste(type,f,"derived allele frequency spectrum")) +
      ylab("N") +
      xlab("AF") +
      theme_bw() +
      theme(text=element_text(size=12,face="bold"),
            rect=element_rect(size=1),
            axis.line=element_line(colour="black"),
            axis.title=element_text(size=16),
            #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
            #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
            #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
            panel.background=element_blank(),
            panel.border=element_rect(colour="black"),
            #panel.grid=element_blank(),
            #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
            plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
            #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
            legend.background=element_rect(linetype="solid", colour="black", size=.5),
            #legend.justification=c(0,0),
            legend.key=element_rect(colour="white"),
            #legend.key.size=unit(1.3,"cm"),
            legend.position="none",
            legend.title=element_blank()
      )
      feat_thin_ggplot
      ggsave(paste0(type,"_",f,"_derived_allele_frequency_spectrum_thin.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")
    
    feat_thick_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(af,n,fill=pop)) +
      geom_col(width=0.07) +
      facet_grid(~pop,scales="free_x",space="free_x") +
      ggtitle(paste(type,f,"derived allele frequency spectrum")) +
      ylab("N") +
      xlab("AF") +
      theme_bw() +
      theme(text=element_text(size=12,face="bold"),
            rect=element_rect(size=1),
            axis.line=element_line(colour="black"),
            axis.title=element_text(size=16),
            #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
            #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
            #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
            panel.background=element_blank(),
            panel.border=element_rect(colour="black"),
            #panel.grid=element_blank(),
            #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
            plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
            #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
            legend.background=element_rect(linetype="solid", colour="black", size=.5),
            #legend.justification=c(0,0),
            legend.key=element_rect(colour="white"),
            #legend.key.size=unit(1.3,"cm"),
            legend.position="none",
            legend.title=element_blank()
      )
      feat_thick_ggplot
      ggsave(paste0(type,"_",f,"_derived_allele_frequency_spectrum_thick.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")
  }
}

```

####Combine per population and feature allele frequency files (only sites with no missingness):
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions #segregating
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_derived_freq_spectrum_nomissing_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


cd /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
GREP=$(cat derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f1)
POP_VCFS=($(ls `find . -path *"_perpop/c_*5x*_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
rm ${CALLING}"_5x_perpop_ALL_features_nomissing_"${VAR}"_"${TYPE}".lr_ann.af"
for p in ${POP_VCFS[@]}
  do
  SP_DATASET=$(echo ${p} | cut -d'/' -f3 | cut -d'_' -f1-3)
  SHORT_POP=$(echo ${p} | cut -d'/' -f3 | cut -d'_' -f4-6)
  N_IND=$(($(bcftools query -l $p | wc -l) *2))
  echo "working with pop" $SHORT_POP "and vcf" $p
  echo "extracting derived allele frequencies from" $SHORT_POP
  for g in ${GREP[@]}
    do
    #echo "grepping: ${g}"
    FEATURE=$(grep "${g}" derived_allele_freq_ratio/features_derived_allele_freq_ratio.txt | cut -f2)
    echo "extracting from feature:" $FEATURE
    grep "${g}" "${p}" | grep ";AN=${N_IND};" | awk -F"\t|;|=" '{printf ("%s\t%s\t%s\t%s\n", $1,$2-1,$2,$13)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_nomissing_${VAR}_${FEATURE}_${TYPE}.lr_ann.af}
    echo "joining frequency files"
    awk -v pop="$SHORT_POP" -v feat="$FEATURE" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_nomissing_"$VAR"_"$FEATURE"_"$TYPE".lr_ann.af" >> ${CALLING}"_5x_perpop_ALL_features_nomissing_"${VAR}"_"${TYPE}".lr_ann.af"
    if [ $FEATURE == "NSYN" ]
      then
      FEATURE2=(NTOL)
      echo "extracting from feature:" $FEATURE2
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_nomissing_${VAR}_${FEATURE}_${TYPE}.lr_ann.af}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_nomissing_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af}
      echo "joining frequency files"
      awk -v pop="$SHORT_POP" -v feat="$FEATURE2" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_nomissing_"$VAR"_"$FEATURE2"_"$TYPE".lr_ann.af" >> ${CALLING}"_5x_perpop_ALL_features_nomissing_"${VAR}"_"${TYPE}".lr_ann.af"
      FEATURE2=(NDEL)
      echo "extracting from feature:" $FEATURE2
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_nomissing_${VAR}_${FEATURE}_${TYPE}.lr_ann.af}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_nomissing_${VAR}_${FEATURE2}_${TYPE}.lr_ann.af}
      echo "joining frequency files"
      awk -v pop="$SHORT_POP" -v feat="$FEATURE2" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_nomissing_"$VAR"_"$FEATURE2"_"$TYPE".lr_ann.af" >> ${CALLING}"_5x_perpop_ALL_features_nomissing_"${VAR}"_"${TYPE}".lr_ann.af"
    fi
    done
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*5x_perpop_ALL_features_nomissing_${VAR}*.lr_ann.af /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/
#scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*5x_perpop_ALL_features*.lr_ann.maf.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/

```

####Plot spectra (new):
```{r}

library(readr)
library(dplyr)
library(ggplot2)

callings_ac_files <- intersect(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/", pattern="*ALL_features*"),list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/", pattern=".af$"))

callings_ac_files <- callings_ac_files[grep("nomissing",callings_ac_files)]

#var_type <- c("varssubs","variants","substitutions")
var_type <- c("segregating")

for (type in var_type) {
  print(type)
  all_plot_data <- data_frame("pop"=character(0),"feat"=factor(0),"ac"=double(0),"n"=numeric(0),"n_bis"=numeric(0))
  calling_ac <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum/",callings_ac_files[grep(type,callings_ac_files)]),col_names=c("pop","feat","sc","start","stop","ac"))
  pop_size_data <- data_frame("pop"=unique(calling_ac$pop)) %>% mutate(size=ifelse(pop=="c_ll_ki" | pop=="c_lp_sm",12,8))
  min_size <- min(pop_size_data$size)
  for (p in unique(calling_ac$pop)) {
    print(p)
    p_size <- pop_size_data[which(pop_size_data$pop==p),2] %>% unlist(use.names=F)
    pop_ac_data <- filter(calling_ac,pop==p)
    #popsize=ifelse(p=="c_lp_sm"|p=="c_ll_ki",12,8)
    pop_plot_data <- pop_ac_data %>% group_by(ac,feat) %>% summarize(n=n(),pop=pop[n]) %>% arrange(feat) %>% mutate(n_bis=n*p_size/min_size) %>% select(pop,feat,ac,n,n_bis) %>% as.data.frame(.)
    #pop_ac_data %>% mutate(ac_groups=cut(ac,breaks=seq(0,1.1,by=1/(2*popsize)))) %>% group_by(ac_groups,feat) %>% summarize(n=n()) #problem with the binning due to rounded allele frequencies
    all_plot_data <- rbind(all_plot_data,pop_plot_data)
    }
  all_plot_data$feat <- as.factor(all_plot_data$feat)
  all_plot_data$feat = factor(all_plot_data$feat,levels=c("INTER","INTR","SYN","NSYN","NTOL","NDEL","LOF","UCNE")) #Reorder factor levels.
  levels(all_plot_data$feat) <- c("intergenic","introns","syn.","missense","m. tol.","m. del.","LoF","UCNE")
  all_plot_data$pop = factor(all_plot_data$pop,levels=c("c_ll_ki","c_ll_po","c_ll_no","c_lp_sm","c_lp_do")) #Reorder factor levels.
  levels(all_plot_data$pop) <- c("KIR","POL","NOR","AND","DON")
  all_plot_data <- all_plot_data %>% arrange(pop,feat,ac) %>% mutate(species=ifelse(pop=="KIR" | pop=="POL" | pop=="NOR","ll","lp"),size=ifelse(pop=="KIR" | pop=="AND","large","small"))
  all_plot_data$species = factor(all_plot_data$species,levels=c("ll","lp"))
  all_plot_data$size = factor(all_plot_data$size,levels=c("large","small"))
  all_plot_data
  
  all_nbis_ggplot <- ggplot(data=all_plot_data,aes(ac,n_bis,fill=interaction(species,size),alpha=interaction(species,size))) +
    geom_col() +
    facet_grid(feat~pop,scales="free",space="free_x") +
    #ggtitle(paste(type,"derived allele frequency spectrum")) +
    ylab("Number of sites") +
    xlab("Derived allele frequency") +
    scale_fill_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
    scale_alpha_manual(values=c(1,1,0.4,0.4)) +
    scale_x_continuous(breaks=seq(0,1,by=0.2)) +
    theme_bw() +
    theme(text=element_text(size=9,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=10),
          axis.text.x=element_text(colour="black",face="bold"),
          axis.text.y=element_text(colour="black",face="bold"),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          panel.spacing=unit(0.075,"cm"),
          strip.background=element_rect(colour="black"),
          strip.text=element_text(size=8),
          panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          #plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position="none",
          legend.title=element_blank()
    )
    all_nbis_ggplot
    ggsave(paste0(type,"_ALL_derived_allele_frequency_spectrum_nbis.pdf"), width=16.9, height=24, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")

    all_n_ggplot <- ggplot(data=all_plot_data,aes(ac,n,fill=interaction(species,size),alpha=interaction(species,size))) +
    geom_col() +
    facet_grid(feat~pop,scales="free",space="free_x") +
    #ggtitle(paste(type,"derived allele frequency spectrum")) +
    ylab("Number of sites") +
    xlab("Derived allele frequency") +
    scale_fill_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
    scale_alpha_manual(values=c(1,1,0.5,0.5)) +
    theme_bw() +
    theme(text=element_text(size=14,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=18),
          axis.text.x=element_text(angle=30,hjust=1,size=14,colour="black"),
          axis.text.y=element_text(size=14,colour="black"),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          strip.background=element_rect(colour="black",size=1.5),
          strip.text=element_text(size=14),
          panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position="none",
          legend.title=element_blank()
    )
    all_n_ggplot
    ggsave(paste0(type,"_ALL_derived_allele_frequency_spectrum_n.pdf"), width=30, height=40, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")

  for (f in unique(all_plot_data$feat)) {
    feat_nbis_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(ac,n_bis,fill=pop)) +
      geom_col() +
      facet_grid(~pop,scales="free_x",space="free_x") +
      ggtitle(paste(type,f,"derived allele frequency spectrum")) +
      ylab("N") +
      xlab("AC") +
      theme_bw() +
      theme(text=element_text(size=14,face="bold"),
            rect=element_rect(size=1),
            axis.line=element_line(colour="black"),
            axis.title=element_text(size=18),
            axis.text.x=element_text(angle=30,hjust=1,size=14,colour="black"),
            axis.text.y=element_text(size=14,colour="black"),
            #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
            panel.background=element_blank(),
            panel.border=element_rect(colour="black"),
            strip.background=element_rect(colour="black",size=1.5),
            strip.text=element_text(size=14),
            panel.grid=element_blank(),
            #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
            plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
            #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
            legend.background=element_rect(linetype="solid", colour="black", size=.5),
            #legend.justification=c(0,0),
            legend.key=element_rect(colour="white"),
            #legend.key.size=unit(1.3,"cm"),
            legend.position="none",
            legend.title=element_blank()
      )
      feat_nbis_ggplot
      ggsave(paste0(type,"_",f,"_derived_allele_frequency_spectrum_nbis.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")
    
    feat_n_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(ac,n,fill=pop)) +
      geom_col() +
      facet_grid(~pop,scales="free_x",space="free_x") +
      ggtitle(paste(type,f,"derived allele frequency spectrum")) +
      ylab("N") +
      xlab("AC") +
      theme_bw() +
      theme(text=element_text(size=14,face="bold"),
            rect=element_rect(size=1),
            axis.line=element_line(colour="black"),
            axis.title=element_text(size=18),
            axis.text.x=element_text(angle=30,hjust=1,size=14,colour="black"),
            axis.text.y=element_text(size=14,colour="black"),
            #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
            panel.background=element_blank(),
            panel.border=element_rect(colour="black"),
            strip.background=element_rect(colour="black",size=1.5),
            strip.text=element_text(size=14),
            panel.grid=element_blank(),
            #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
            plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
            #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
            legend.background=element_rect(linetype="solid", colour="black", size=.5),
            #legend.justification=c(0,0),
            legend.key=element_rect(colour="white"),
            #legend.key.size=unit(1.3,"cm"),
            legend.position="none",
            legend.title=element_blank()
      )
      feat_n_ggplot
      ggsave(paste0(type,"_",f,"_derived_allele_frequency_spectrum_n.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_freq_spectrum")
  }
}

```

#2: Derived allele counts approaches.
#2A: Variant and allele level.
##At the individual level.
###Whole-genome (separate callings).
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_ann_individual_summary_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_individual_summary_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_ll_ki_c_ll_no_c_ll_po_nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_individual_summary_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/

```

####Plot individual variant count results.
#####Visualise derived allele counts.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_SNP.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*nm2nm3_origcov_ann_individual_summary_SNP.lr_ann.txt"),pattern='wrong',inv=T,value=T)
variants_wg <- rbind(read_tsv(paste0(wd_path,grep(variant_files,pattern='c_ll',value=T))),read_tsv(paste0(wd_path,grep(variant_files,pattern='c_lp',value=T))))
subst_wg <- read_tsv(paste0(wd_path,substitution_files))
variants_and_subst_wg <- cbind(variants_wg[c(1:4)],variants_wg[-c(1:4,20:23)]+subst_wg[-c(1:4,20:23)])

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_varN <- variants_and_subst_wg %>% select(c(1:5,9,12,14,16,18)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_varN

derived_allele_variant_counts_ggplot <- ggplot(data=variants_and_subst_wg_varN, aes(population,N,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N variants") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_variant_counts_ggplot
ggsave("derived_allele_variant_counts.pdf", width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")


variants_and_subst_wg_alleleN <- variants_and_subst_wg %>% select(c(1:4,6,10,13,15,17,19)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleN

derived_allele_allele_counts_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N alleles") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_ggplot
ggsave("derived_allele_allele_counts.pdf", width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

derived_allele_allele_counts_sep_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  #facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N alleles") +
  theme_bw() +
  theme(strip.text.x = element_blank(),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_sep_ggplot
ggsave("derived_allele_allele_counts_sep.pdf", width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

variants_and_subst_wg_alleleN <- variants_and_subst_wg %>% select(c(1:4,6,10,13,15,17,19)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleN


variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(syn_intr_R=synonymous_A/intronic_A,mis_intr_R=missense_A/intronic_A,non_intr_R=nonsense_A/intronic_A,UCNE_intr_R=UCNE_A/intronic_A) %>% select(c(1:4,20:23)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

derived_allele_allele_ratio_ggplot <- ggplot(data=variants_and_subst_wg_alleleR, aes(population,value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_ratio_ggplot
ggsave("derived_allele_allele_ratio.pdf", width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")


```

#####Visualise derived allele counts (variant transversions only).

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/", pattern="*origcov_ann_individual_summary_Tvonly.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
#substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*nm2nm3_origcov_ann_individual_summary_Tvonly.lr_ann.txt"),pattern='wrong',inv=T,value=T)
variants_wg <- rbind(read_tsv(paste0(wd_path,variant_files)))
#subst_wg <- read_tsv(paste0(wd_path,substitution_files))
#variants_and_subst_wg <- cbind(variants_wg[c(1:4)],variants_wg[-c(1:4,20:23)]+subst_wg[-c(1:4,20:23)])

variants_wg$dataset <- as.factor(variants_wg$dataset)
variants_wg$dataset = factor(variants_wg$dataset,levels=c("5x","MG","LD")) #Reorder factor levels
variants_wg$population = factor(variants_wg$population,levels=c("ki","no","po","pv"))
print.data.frame(variants_wg)

variants_wg_varN <- variants_wg %>% select(c(1:5,9,12,14,16,18)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_wg_varN

derived_allele_variant_counts_ggplot <- ggplot(data=variants_wg_varN, aes(population,N,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N variants") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_variant_counts_ggplot
ggsave("derived_allele_variant_counts.pdf", width=15, height=20, units="cm", device="pdf", path=wd_path)


variants_wg_alleleN <- variants_wg %>% select(c(1:4,6,10,13,15,17,19)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_wg_alleleN

derived_allele_allele_counts_ggplot <- ggplot(data=variants_wg_alleleN, aes(population,N,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N alleles") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_ggplot
ggsave("derived_allele_allele_counts.pdf", width=15, height=20, units="cm", device="pdf", path=wd_path)

derived_allele_allele_counts_sep_ggplot <- ggplot(data=variants_wg_alleleN, aes(population,N,colour=population)) +
  facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  #facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N alleles") +
  theme_bw() +
  theme(strip.text.x = element_blank(),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_sep_ggplot
ggsave("derived_allele_allele_counts_sep.pdf", width=15, height=20, units="cm", device="pdf", path=wd_path)

variants_wg_alleleN <- variants_wg %>% select(c(1:4,6,10,13,15,17,19)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_wg_alleleN


variants_wg_alleleR <- variants_wg %>% select(-c(20:24)) %>% mutate(syn_intr_R=synonymous_A/intronic_A,mis_intr_R=missense_A/intronic_A,non_intr_R=nonsense_A/intronic_A,UCNE_intr_R=UCNE_A/intronic_A) %>% select(c(1:4,20:23)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_wg_alleleR

derived_allele_allele_ratio_ggplot <- ggplot(data=variants_wg_alleleR, aes(population,value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_ratio_ggplot
ggsave("derived_allele_allele_ratio.pdf", width=15, height=15, units="cm", device="pdf", path=wd_path)

```

#####Visualise counts and count ratios with depth effect.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)

#First draw the NM distribution for each individual:
sample_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*_ann_individual_summary.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
snpeff_filters_summary <- data_frame()
for (file in sample_files) {
  snpeff_individual_summary <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/",file))
  snpeff_individual_summary <- mutate(snpeff_individual_summary, filter=strsplit(file,"_")[[1]][length(strsplit(file,"_")[[1]])-4])
  snpeff_filters_summary <- rbind(snpeff_filters_summary,snpeff_individual_summary)
}
snpeff_filters_summary$dataset <- as.factor(snpeff_filters_summary$dataset)
snpeff_filters_summary$dataset = factor(snpeff_filters_summary$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
print.data.frame(snpeff_filters_summary)

syn_int_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`synonymous/intronic_V`,colour=population)) +
  facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("SYN/INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  syn_int_ggplot
ggsave("SYN_vs_INTR_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

nsyn_int_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`missense/intronic_V`,colour=population)) +
  facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  nsyn_int_ggplot
ggsave("NSYN_vs_INTR_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

nsyn_syn_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`missense/synonymous_V`,colour=population)) +
  facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/SYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  nsyn_syn_ggplot
ggsave("NSYN_vs_SYN_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

INTR_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,intronic_V,colour=population)) +
  facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  INTR_ggplot
ggsave("INTR_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

SYN_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,synonymous_V,colour=population)) +
  facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("SYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  SYN_ggplot
ggsave("SYN_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

NSYN_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,missense_V,colour=population)) +
  facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  NSYN_ggplot
ggsave("NSYN_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

#####Visualise counts and count ratios for the calling that includes h_ll_pv with depth effect.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)

#First draw the NM distribution for each individual:
sample_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/", pattern="*_ann_individual_summary.lr_ann.txt"),pattern='wrong',inv=T,value=T)
snpeff_filters_summary <- data_frame()
for (file in sample_files) {
  snpeff_individual_summary <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/",file))
  snpeff_individual_summary <- mutate(snpeff_individual_summary, filter=strsplit(file,"_")[[1]][length(strsplit(file,"_")[[1]])-4])
  snpeff_filters_summary <- rbind(snpeff_filters_summary,snpeff_individual_summary)
}
snpeff_filters_summary$dataset <- as.factor(snpeff_filters_summary$dataset)
snpeff_filters_summary$dataset = factor(snpeff_filters_summary$dataset,levels=c("REF","GP","5x","MG","LD")) #Reorder factor levels to: REF, GP, 5x, MG
print.data.frame(snpeff_filters_summary)

syn_int_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`synonymous/intronic_V`,colour=population)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("SYN/INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.90,0.84),
        legend.title=element_blank()
  )
  syn_int_ggplot
ggsave("SYN_vs_INTR_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")

nsyn_int_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`missense/intronic_V`,colour=population)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.90,0.84),
        legend.title=element_blank()
  )
  nsyn_int_ggplot
ggsave("NSYN_vs_INTR_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")

nsyn_syn_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`missense/synonymous_V`,colour=population)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/SYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.90,0.84),
        legend.title=element_blank()
  )
  nsyn_syn_ggplot
ggsave("NSYN_vs_SYN_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")

INTR_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,intronic_V,colour=population)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.90,0.84),
        legend.title=element_blank()
  )
  INTR_ggplot
ggsave("INTR_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")

SYN_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,synonymous_V,colour=population)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("SYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.90,0.84),
        legend.title=element_blank()
  )
  SYN_ggplot
ggsave("SYN_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")

NSYN_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,missense_V,colour=population)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_point(position = position_jitter(w=0.2,h=0)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.90,0.84),
        legend.title=element_blank()
  )
  NSYN_ggplot
ggsave("NSYN_cov_filters.pdf", width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov/")

```

#####Compare counts with different NM filterings.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)

#First draw the NM distribution for each individual:
sample_files <- list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/old/sep_calling/", pattern="*.genes.lr_ann.txt")
snpeff_filters_summary <- data_frame()
for (file in sample_files) {
  snpeff_individual_summary <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/old/sep_calling/",file))
  snpeff_individual_summary <- mutate(snpeff_individual_summary, filter=ifelse(strsplit(file,"_")[[1]][4]=="genes.lr","no_filter",strsplit(file,"_")[[1]][4]))
  snpeff_filters_summary <- rbind(snpeff_filters_summary,snpeff_individual_summary)
}
snpeff_filters_summary

syn_int_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`synonymous/intronic_V`,colour=population)) +
  facet_grid(. ~ filter) +
  geom_point(position="jitter") +
  #ggtitle("Proportion of reads at different NM") +
  ylab("SYN/INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.08,0.86),
        legend.title=element_blank()
  )
  syn_int_ggplot
ggsave("SYN_vs_INTR_nm_filters.pdf", width=30, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/sep_calling/")

nsyn_int_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`missense/intronic_V`,colour=population)) +
  facet_grid(. ~ filter) +
  geom_point(position="jitter") +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.08,0.86),
        legend.title=element_blank()
  )
  nsyn_int_ggplot
ggsave("NSYN_vs_INTR_nm_filters.pdf", width=30, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/sep_calling/")

nsyn_syn_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,`missense/synonymous_V`,colour=population)) +
  facet_grid(. ~ filter) +
  geom_point(position="jitter") +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/SYN") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.08,0.16),
        legend.title=element_blank()
  )
  nsyn_syn_ggplot
ggsave("NSYN_vs_SYN_nm_filters.pdf", width=30, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/sep_calling/")

INTR_ggplot <- ggplot(data=snpeff_filters_summary, aes(dataset,intronic_V,colour=population)) +
  facet_grid(. ~ filter) +
  geom_point(position="jitter") +
  #ggtitle("Proportion of reads at different NM") +
  ylab("INTR") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.12,0.82),
        legend.title=element_blank()
  )
  INTR_ggplot
ggsave("INTR_nm_filters.pdf", width=30, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/sep_calling/")

```

#####Plot NS/S ratio.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_SNP.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*nm2nm3_origcov_ann_individual_summary_SNP.lr_ann.txt"),pattern='wrong',inv=T,value=T)

variants_wg <- rbind(read_tsv(paste0(wd_path,grep(variant_files,pattern='c_ll',value=T))),read_tsv(paste0(wd_path,grep(variant_files,pattern='c_lp',value=T))))
subst_wg <- read_tsv(paste0(wd_path,substitution_files))
variants_and_subst_wg <- cbind(variants_wg[c(1:4)],variants_wg[-c(1:4,20:23)]+subst_wg[-c(1:4,20:23)]) %>% mutate(`missense/synonymous_V`=missense_V/synonymous_V)

NSYN_SYN_plot_data <- rbind(variants_wg %>% select(c(1:4,20)) %>% mutate(category="variants"),subst_wg %>% select(c(1:4,20)) %>% mutate(category="substitutions"),variants_and_subst_wg %>% select(c(1:4,20)) %>% mutate(category="all"))

NSYN_SYN_plot_data$population = factor(NSYN_SYN_plot_data$population,levels=c("ki","no","po","sm","do"))
NSYN_SYN_plot_data$category = factor(NSYN_SYN_plot_data$category,levels=c("variants","substitutions","all"))
NSYN_SYN_plot_data

NSYN_SYN_ggplot <- ggplot(data=NSYN_SYN_plot_data, aes(population,`missense/synonymous_V`,colour=category)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_boxplot(position=position_dodge(width=0.5)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/SYN ratio") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        legend.position=c(0.16,0.82),
        legend.title=element_blank()
  )
  NSYN_SYN_ggplot
ggsave("NSYN_SYN_ratio.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

###Whole-genome (joint calling).
####Get counts (of variants, substitutions or vars+subs).
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_samecov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_individual_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${VAR}_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${VAR}_syn_pref.temp.borrar)
  SYN_PREF_A=$(cut -f8 ${VAR}_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${VAR}_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${VAR}_syn_unpref.temp.borrar)
  SYN_UNPREF_A=$(cut -f8 ${VAR}_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${VAR}_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${VAR}_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${VAR}_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${VAR}_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${VAR}_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${VAR}_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${VAR}_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${VAR}_ucne_low.temp.borrar)
  UCNE_LOW_A=$(cut -f8 ${VAR}_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${VAR}_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${VAR}_ucne_mid.temp.borrar)
  UCNE_MID_A=$(cut -f8 ${VAR}_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${VAR}_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${VAR}_ucne_high.temp.borrar)
  UCNE_HIGH_A=$(cut -f8 ${VAR}_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
  done
rm ${VAR}_*.temp.borrar

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_samecov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/
unset SSHPASS

```

####Plot individual variant count results.
#####Visualise derived allele counts.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_varN <- variants_and_subst_wg %>% select(c(1:5,7,9,12,14,16,18,20,22,24,26,28)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_and_subst_wg_varN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_varN$feature))
variants_and_subst_wg_varN

derived_allele_variant_counts_ggplot <- ggplot(data=variants_and_subst_wg_varN, aes(population,N,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N variants") +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_variant_counts_ggplot
ggsave(paste0(type,"_derived_allele_variant_counts.pdf"), width=15, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

variants_and_subst_wg_alleleN <- variants_and_subst_wg %>% select(c(1:4,6,8,10,13,15,17,19,21,23,25,27,29)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_and_subst_wg_alleleN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_alleleN$feature))
variants_and_subst_wg_alleleN

derived_allele_allele_counts_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N alleles") +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_ggplot
ggsave(paste0(type,"_derived_allele_allele_counts.pdf"), width=15, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

derived_allele_allele_counts_sep_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  facet_wrap(feature ~ species,nrow=length(levels(variants_and_subst_wg_alleleN$feature)),ncol=2,scales="free") +
  #facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N alleles") +
  theme_bw() +
  theme(strip.text.x = element_blank(),
        strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_sep_ggplot
ggsave(paste0(type,"_derived_allele_allele_counts_sep.pdf"), width=15, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(syn_intr_R=synonymous_A/intronic_A,mis_intr_R=missense_A/intronic_A,mistol_intr_R=missense_tol_A/intronic_A,misdel_intr_R=missense_del_A/intronic_A,non_intr_R=nonsense_A/intronic_A,UCNE_intr_R=UCNE_A/intronic_A,UCNElow_intr_R=UCNE_low_A/intronic_A,UCNEmid_intr_R=UCNE_mid_A/intronic_A,UCNEhigh_intr_R=UCNE_high_A/intronic_A) %>% select(c(1:4,34:42)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

derived_allele_allele_ratio_ggplot <- ggplot(data=variants_and_subst_wg_alleleR, aes(population,value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

#####Compare origcov vs. samecov.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
origcov <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt")) %>% mutate(cov="origcov")
samecov <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_samecov_ann_individual_summary_",type,"_SNP.lr_ann.txt")) %>% mutate(cov="samecov")

variants_and_subst_wg <- rbind(origcov,samecov)

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
levels(variants_and_subst_wg$population) <- c("KIR","POL","NOR","AND","DON")
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg$cov <- as.factor(variants_and_subst_wg$cov)
variants_and_subst_wg$cov = factor(variants_and_subst_wg$cov,levels=c("origcov","samecov")) #Reorder factor levels to: REF, GP, 5x, MG

variants_and_subst_wg_varN <- variants_and_subst_wg %>% select(.,species,population,dataset,sample,cov,contains("_A"),-contains("/")) %>% gather(feature,N,-species,-population,-dataset,-sample,-cov,factor_key=T)
levels(variants_and_subst_wg_varN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_varN$feature))
variants_and_subst_wg_varN

derived_allele_variant_counts_ggplot <- ggplot(data=filter(variants_and_subst_wg_varN,grepl('intergenic|intronic|synonymous|missense|nonsense',feature) & (!grepl('REF',dataset))), aes(interaction(dataset,population),N,colour=population,alpha=dataset)) +
  facet_grid(feature ~ cov,scales="free",space="free_x") +
  geom_point(position = position_jitter(width=0.2,height=0)) +
  geom_vline(xintercept=4.5,colour="black") +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population x dataset") +
  ylab("Derived allele count") +
  scale_alpha_manual(values=c(1,0.2,1)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        axis.text.x=element_text(angle=90,vjust=0.5),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
derived_allele_variant_counts_ggplot
ggsave(paste0(type,"_c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_depthtest_counts.pdf"), width=24, height=28, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

#####Relative to Kirov, definitive version.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt")) %>% select(.,species,population,dataset,sample,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("total","intergenic","intronic","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio) <- c("total","intergenic","introns","synonymous","synonymous_pref","synonymous_unpref","missense","missense_tolerated","missense_deleterious","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR
#(average_relativised_variants_and_subst_wg_alleleR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/",type,"_derived_allele_allele_ratio_relative2introns_mean.csv"))

#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.75,0.95,0.75,0.5),"max"=c(1.1,1.2,1.75,3.5),"breaks"=c(0.05,0.1,0.2,0.5))

average_relativised_derived_allele_allele_ratio_up <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|nonsense|\\<UCNE\\>',ratio)), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_up

average_relativised_derived_allele_allele_ratio_down <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,grepl('synonymous_pref|synonymous_unpref|missense_tolerated|missense_deleterious|UCNE_mid|UCNE_high',ratio)), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_down

average_relativised_derived_allele_allele_ratio_other <- ggplot(average_relativised_variants_and_subst_wg_alleleR[average_relativised_variants_and_subst_wg_alleleR$ratio=="UCNE_low",], aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.text=element_text(colour="black",face="bold"),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_other


if(type=="varssubs") {
  average_relativised_derived_allele_allele_ratio_combined <- grid.arrange(set_panel_size(average_relativised_derived_allele_allele_ratio_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(average_relativised_derived_allele_allele_ratio_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_relativised_derived_allele_allele_ratio_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Average genetic load relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    average_relativised_derived_allele_allele_ratio_combined <- grid.arrange(set_panel_size(average_relativised_derived_allele_allele_ratio_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(average_relativised_derived_allele_allele_ratio_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_relativised_derived_allele_allele_ratio_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_relative2Kirov_definitive.pdf"), width=41, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios",average_relativised_derived_allele_allele_ratio_combined)

```

#####Relative to introns (population average version).
######All individuals.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,synpref=syn_pref_A/intronic_A,synunpref=syn_unpref_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A,UCNElow=UCNE_low_A/intronic_A,UCNEmid=UCNE_mid_A/intronic_A,UCNEhigh=UCNE_high_A/intronic_A) %>% select(c(1:4,38:48)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synpref"] <- "synonymous_pref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synunpref"] <- "synonymous_unpref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR
write_csv(average_relativised_variants_and_subst_wg_alleleR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/",type,"_derived_allele_allele_ratio_relative2introns_mean.csv"))

#Plot these means and standard errors:
average_relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=average_relativised_variants_and_subst_wg_alleleR, aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  average_relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_short_mean.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "synonymous" | ratio == "synonymous_pref" | ratio == "synonymous_unpref" | ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.9, 1.05, by = 0.05)) +
  ylim(0.8,1.1) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_part1_mean.pdf"), width=35, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 2, layout_matrix = rbind(c(rep(1,20)), c(rep(2,18),rep(NA,3))))
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_all.pdf"), width=40, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

######5x only.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,synpref=syn_pref_A/intronic_A,synunpref=syn_unpref_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A,UCNElow=UCNE_low_A/intronic_A,UCNEmid=UCNE_mid_A/intronic_A,UCNEhigh=UCNE_high_A/intronic_A) %>% select(c(1:4,38:48)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki" & dataset!="MG") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop & dataset=="5x") %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop & dataset=="5x") %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop & dataset=="5x") %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synpref"] <- "synonymous_pref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synunpref"] <- "synonymous_unpref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR
write_csv(average_relativised_variants_and_subst_wg_alleleR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/",type,"_derived_allele_allele_ratio_5xonly_relative2introns_mean.csv"))

#Plot these means and standard errors:
average_relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=average_relativised_variants_and_subst_wg_alleleR, aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  average_relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_5xonly_relative2introns_short_mean.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "synonymous" | ratio == "synonymous_pref" | ratio == "synonymous_unpref" | ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.85, 1.05, by = 0.05)) +
  ylim(0.85,1.05) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_5xonly_relative2introns_part1_mean.pdf"), width=35, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_5xonly_relative2introns_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 2, layout_matrix = rbind(c(rep(1,20)), c(rep(2,18),rep(NA,3))))
ggsave(paste0(type,"_derived_allele_allele_ratio_5xonly_relative2introns_all.pdf"), width=40, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

#####Obsolete: relative to introns.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="substitutions" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A) %>% select(c(1:4,28:33)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=relativised_variants_and_subst_wg_alleleR, aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="fixed") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.2, by = 0.2)) +
  ylim(0.5,1.2) +
  ggtitle(paste0("ratio of ",type," relative to intronic and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

#####Obsolete: relative to intergenic.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="substitutions" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intergenic_A,missense=missense_A/intergenic_A,mistol=missense_tol_A/intergenic_A,misdel=missense_del_A/intergenic_A,nonsense=nonsense_A/intergenic_A,UCNE=UCNE_A/intergenic_A) %>% select(c(1:4,28:33)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=relativised_variants_and_subst_wg_alleleR, aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.2, by = 0.2)) +
  ylim(0.5,1.2) +
  ggtitle(paste0("ratio of ",type," relative to intergenic and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2interg.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

#####Obsolete: relative to synonymous.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="substitutions" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/synonymous_A,missense=missense_A/synonymous_A,mistol=missense_tol_A/synonymous_A,misdel=missense_del_A/synonymous_A,nonsense=nonsense_A/synonymous_A,UCNE=UCNE_A/synonymous_A) %>% select(c(1:4,28:33)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=relativised_variants_and_subst_wg_alleleR, aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.2, by = 0.2)) +
  ylim(0.5,1.2) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

#####Relative to synonymous (per individual version).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(gridExtra)

type="segregating" #varssubs #variants #substitutions #segregating #fixed

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/synonymous_A,missense=missense_A/synonymous_A,mistol=missense_tol_A/synonymous_A,misdel=missense_del_A/synonymous_A,nonsense=nonsense_A/synonymous_A,UCNE=UCNE_A/synonymous_A,UCNElow=UCNE_low_A/synonymous_A,UCNEmid=UCNE_mid_A/synonymous_A,UCNEhigh=UCNE_high_A/synonymous_A) %>% select(c(1:4,34:42)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=filter(relativised_variants_and_subst_wg_alleleR,ratio != "synonymous" & ratio != "missense"), aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.5, by = 0.2)) +
  ylim(0.5,1.5) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_short.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")


relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(relativised_variants_and_subst_wg_alleleR,ratio == "missense" | ratio == "mistol" | ratio == "misdel"), aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.9, 1.2, by = 0.05)) +
  ylim(0.9,1.2) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
relativised_derived_allele_allele_ratio_ggplot1

relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.5, by = 0.2)) +
  ylim(0.5,1.5) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
relativised_derived_allele_allele_ratio_ggplot2

relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(relativised_derived_allele_allele_ratio_ggplot1, relativised_derived_allele_allele_ratio_ggplot2, ncol = 2, nrow = 1, widths = c(3.5,5))
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_separate.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", relativised_derived_allele_allele_ratio_ggplotcombined)

```

#####Relative to synonymous (population average version).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="segregating" #varssubs #variants #substitutions #segregating #fixed

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/synonymous_A,missense=missense_A/synonymous_A,mistol=missense_tol_A/synonymous_A,misdel=missense_del_A/synonymous_A,nonsense=nonsense_A/synonymous_A,UCNE=UCNE_A/synonymous_A,UCNElow=UCNE_low_A/synonymous_A,UCNEmid=UCNE_mid_A/synonymous_A,UCNEhigh=UCNE_high_A/synonymous_A) %>% select(c(1:4,34:42)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("synonymous","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR
write_csv(average_relativised_variants_and_subst_wg_alleleR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/",type,"_derived_allele_allele_ratio_relative2synonymous_mean.csv"))

#Plot these means and standard errors:
average_relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio != "synonymous" & ratio != "missense"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  average_relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_short_mean.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.9, 1.05, by = 0.05)) +
  ylim(0.9,1.05) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_part1_mean.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 1)
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_separate_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

#####Plot NS/S ratio.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

variants_wg <- read_tsv(paste0(wd_path,list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_variants_SNP.lr_ann.txt")))
subst_wg <- read_tsv(paste0(wd_path,list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_substitutions_SNP.lr_ann.txt")))
variants_and_subst_wg <- read_tsv(paste0(wd_path,list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_varssubs_SNP.lr_ann.txt")))

NSYN_SYN_plot_data <- rbind(variants_wg %>% select(c(1:4,24,25)) %>% mutate(category="variants"),subst_wg %>% select(c(1:4,24,25)) %>% mutate(category="substitutions"),variants_and_subst_wg %>% select(c(1:4,24,25)) %>% mutate(category="all"))

NSYN_SYN_plot_data$population = factor(NSYN_SYN_plot_data$population,levels=c("ki","no","po","sm","do"))
NSYN_SYN_plot_data$category = factor(NSYN_SYN_plot_data$category,levels=c("variants","substitutions","all"))
NSYN_SYN_plot_data

NSYN_SYN_ggplot <- ggplot(data=NSYN_SYN_plot_data, aes(population,`missense/synonymous_V`,colour=category)) +
  #facet_grid(species ~ filter, scales="free") +
  geom_boxplot(position=position_dodge(width=0.5)) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("NSYN/SYN ratio") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.16,0.82),
        legend.title=element_blank()
  )
  NSYN_SYN_ggplot
ggsave("NSYN_SYN_ratio.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

####Get counts for the less filtered dataset (filtered3).
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/testing_variant_numbers
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_ann_individual_summary_varssubs_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation/testing_variant_numbers
rm ${CALLING}"_ann_individual_summary_varssubs_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_varssubs_"${TYPE}".lr_ann.txt"
cd $V_PATH/$CALLING/annotation/
INDLIST=($(ls `find . -name *"_individual_varssubs_"${TYPE}".lr_ann.vcf" -path '*testing_variant_numbers*' -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ./testing_variant_numbers/${CALLING}"_ann_individual_summary_varssubs_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_ll_ki_c_ll_no_c_ll_po_nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_individual_summary_varssubs_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/

```

###gBGC.
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed
gBGC=(same) #WtoS #StoW #same
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${gBGC}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
gBGC=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $gBGC == "same" ]; then gBGC=(StoSandWtoW); fi
VAR=$(echo ${STY#*.} | cut -d'-' -f3)
TYPE=$(echo ${STY#*.} | cut -d'-' -f4)
script "${CALLING}_ann_individual_summary_${gBGC}_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
gBGC=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $gBGC == "same" ]; then gBGC=(StoSandWtoW); fi
VAR=$(echo ${STY#*.} | cut -d'-' -f3)
TYPE=$(echo ${STY#*.} | cut -d'-' -f4)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm -f ${CALLING}"_ann_individual_summary_"${gBGC}"_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_"${gBGC}"_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${gBGC}"_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${gBGC}_${VAR}_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${gBGC}_${VAR}_syn_pref.temp.borrar)
  if [ -z "$SYN_PREF_V" ]; then SYN_PREF_V=0; fi
  SYN_PREF_A=$(cut -f8 ${gBGC}_${VAR}_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$SYN_PREF_A" ]; then SYN_PREF_A=0; fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${gBGC}_${VAR}_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${gBGC}_${VAR}_syn_unpref.temp.borrar)
  if [ -z "$SYN_UNPREF_V" ]; then SYN_UNPREF_V=0; fi
  SYN_UNPREF_A=$(cut -f8 ${gBGC}_${VAR}_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$SYN_UNPREF_A" ]; then SYN_UNPREF_A=0; fi
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${gBGC}_${VAR}_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${gBGC}_${VAR}_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${gBGC}_${VAR}_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${gBGC}_${VAR}_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${gBGC}_${VAR}_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${gBGC}_${VAR}_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  if [ -z "$NONSENSE_V" ]; then NONSENSE_V=0; fi
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$NONSENSE_A" ]; then NONSENSE_A=0; fi
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  if [ -z "$UCNE_V" ]; then UCNE_V=0; fi
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$UCNE_A" ]; then UCNE_A=0; fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${gBGC}_${VAR}_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${gBGC}_${VAR}_ucne_low.temp.borrar)
  if [ -z "$UCNE_LOW_V" ]; then UCNE_LOW_V=0; fi
  UCNE_LOW_A=$(cut -f8 ${gBGC}_${VAR}_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$UCNE_LOW_A" ]; then UCNE_LOW_A=0; fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${gBGC}_${VAR}_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${gBGC}_${VAR}_ucne_mid.temp.borrar)
  if [ -z "$UCNE_MID_V" ]; then UCNE_MID_V=0; fi
  UCNE_MID_A=$(cut -f8 ${gBGC}_${VAR}_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$UCNE_MID_A" ]; then UCNE_MID_A=0; fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${gBGC}_${VAR}_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${gBGC}_${VAR}_ucne_high.temp.borrar)
  if [ -z "$UCNE_HIGH_V" ]; then UCNE_HIGH_V=0; fi
  UCNE_HIGH_A=$(cut -f8 ${gBGC}_${VAR}_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  if [ -z "$UCNE_HIGH_A" ]; then UCNE_HIGH_A=0; fi
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_"${gBGC}"_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed #private_segregating
gBGC=(StoSandWtoW) #WtoS #StoW #StoSandWtoW
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_${gBGC}_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/gBGC/
unset SSHPASS

```

####Plot individual variant count results.
#####Relative to Kirov, definitive version.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

gBGC="StoSandWtoW" #WtoS #StoW #StoSandWtoW
type="fixed" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/gBGC/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_",gBGC,"_",type,"_SNP.lr_ann.txt")) %>% select(.,species,population,dataset,sample,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("total","intergenic","intronic","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio) <- c("total","intergenic","introns","synonymous","synonymous_pref","synonymous_unpref","missense","missense_tolerated","missense_deleterious","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR


#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.70,0.70,0.50,0.50),"max"=c(1.70,1.70,1.75,3.5),"breaks"=c(0.1,0.1,0.2,0.5))

average_relativised_derived_allele_allele_ratio_up <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|nonsense|\\<UCNE\\>',ratio)), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_up

average_relativised_derived_allele_allele_ratio_down <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,grepl('synonymous_pref|synonymous_unpref|missense_tolerated|missense_deleterious|UCNE_mid|UCNE_high',ratio)), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_down

average_relativised_derived_allele_allele_ratio_other <- ggplot(average_relativised_variants_and_subst_wg_alleleR[average_relativised_variants_and_subst_wg_alleleR$ratio=="UCNE_low",], aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.text=element_text(colour="black",face="bold"),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_other


if(gBGC=="WtoS") {
  average_relativised_derived_allele_allele_ratio_combined <- grid.arrange(set_panel_size(average_relativised_derived_allele_allele_ratio_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(average_relativised_derived_allele_allele_ratio_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_relativised_derived_allele_allele_ratio_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("WtoS average genetic load relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (gBGC=="StoW") {
  average_relativised_derived_allele_allele_ratio_combined <- grid.arrange(set_panel_size(average_relativised_derived_allele_allele_ratio_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(average_relativised_derived_allele_allele_ratio_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_relativised_derived_allele_allele_ratio_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("StoW average genetic load relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (gBGC=="StoSandWtoW") {
  average_relativised_derived_allele_allele_ratio_combined <- grid.arrange(set_panel_size(average_relativised_derived_allele_allele_ratio_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(average_relativised_derived_allele_allele_ratio_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_relativised_derived_allele_allele_ratio_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("GC-conservative average genetic load relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(gBGC,"_",type,"_genetic_load_relative2Kirov_definitive.pdf"), width=41, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/gBGC",average_relativised_derived_allele_allele_ratio_combined)

```

###IBD tracts and non IBD tracts (joint calling).
####Good coordinates (nm_filtered):
#####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions #segregating #fixed
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_IBD_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm -f ${CALLING}"_ann_individual_summary_IBD_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\tVAR_type\tIBD_type\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_IBD_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"*"IBD"*"_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  IBD=$(echo ${i} | rev | cut -d'_' -f4 | rev)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${VAR}_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${VAR}_syn_pref.temp.borrar)
  SYN_PREF_A=$(cut -f8 ${VAR}_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${VAR}_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${VAR}_syn_unpref.temp.borrar)
  SYN_UNPREF_A=$(cut -f8 ${VAR}_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${VAR}_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${VAR}_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${VAR}_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${VAR}_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${VAR}_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${VAR}_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${VAR}_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${VAR}_ucne_low.temp.borrar)
  UCNE_LOW_A=$(cut -f8 ${VAR}_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${VAR}_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${VAR}_ucne_mid.temp.borrar)
  UCNE_MID_A=$(cut -f8 ${VAR}_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${VAR}_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${VAR}_ucne_high.temp.borrar)
  UCNE_HIGH_A=$(cut -f8 ${VAR}_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$VAR\t$IBD\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_IBD_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_IBD_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/
unset SSHPASS

```

#####Plot individual variant count results.
```{r Plot variant count results}
library(readr)
library(dplyr)
library(ggplot2)

type="varssubs" #varssubs #variants #substitutions
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")
#variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',inv=T,value=T)
#substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',value=T)
IBD_summary <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_IBD_",type,"_SNP.lr_ann.txt"))
IBD_summary[is.na(IBD_summary)] <- 0

LOF_SYN_ratio <- IBD_summary %>% mutate(LOF_SYN_ratio=nonsense_A/synonymous_A) %>% select(c(1:6,(ncol(IBD_summary)+1)))
LOF_SYN_ratio$population = factor(LOF_SYN_ratio$population,levels=c("ki","no","po","sm","do"))
#LOF_SYN_ratio$VAR_type = factor(LOF_SYN_ratio$VAR_type,levels=c("varssubs","variants","substitutions"))
LOF_SYN_ratio$IBD_type = factor(LOF_SYN_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
LOF_SYN_ratio
for (pop in unique(LOF_SYN_ratio$population)) {
  print(pop)
  pop_df <- filter(LOF_SYN_ratio,population==pop)
  ks_test <- ks.test(unlist(filter(pop_df,IBD_type=="IBD") %>% select(LOF_SYN_ratio),use.names=F), unlist(filter(pop_df,IBD_type=="notIBD") %>% select(LOF_SYN_ratio),use.names=F), alternative="g") #one-tailed ks test (significant result means that the IBD LOF_SYN_ratio is higher than the notIBD one)
  print(ks_test$p.value)
}
#Add $VAR and plot
ggplot_LOF_SYN_ratio_IBD <- ggplot(data=filter(LOF_SYN_ratio,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=LOF_SYN_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.6)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("nonsense/synonymous ",type)) +
  scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_SYN_ratio_IBD
ggsave(paste0(type,"_LOF_SYN_ratio_IBD_and_notIBD.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

ggplot_LOF_SYN_ratio_IBD_detail <- ggplot(data=filter(LOF_SYN_ratio,IBD_type!="IBD"),aes(x=population,y=LOF_SYN_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5, position=position_dodge(width=0.6)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("nonsense/synonymous ",type)) +
  scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_SYN_ratio_IBD_detail
ggsave(paste0(type,"_LOF_SYN_ratio_IBD_and_notIBD_detailed.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

LOF_INTR_ratio <- IBD_summary %>% mutate(LOF_INTR_ratio=nonsense_A/intronic_A) %>% select(c(1:6,(ncol(IBD_summary)+1)))
LOF_INTR_ratio$population = factor(LOF_INTR_ratio$population,levels=c("ki","no","po","sm","do"))
LOF_INTR_ratio$IBD_type = factor(LOF_INTR_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
LOF_INTR_ratio

for (pop in unique(LOF_INTR_ratio$population)) {
  print(pop)
  pop_df <- filter(LOF_INTR_ratio,population==pop)
  ks_test <- ks.test(unlist(filter(pop_df,IBD_type=="IBD") %>% select(LOF_INTR_ratio),use.names=F), unlist(filter(pop_df,IBD_type=="notIBD") %>% select(LOF_INTR_ratio),use.names=F), alternative="g")
  print(ks_test$p.value)
}

ggplot_LOF_INTR_ratio_IBD <- ggplot(data=filter(LOF_INTR_ratio,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=LOF_INTR_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("nonsense/intronic ",type)) +
  #scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_INTR_ratio_IBD
ggsave(paste0(type,"_LOF_INTR_ratio_IBD_and_notIBD.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

ggplot_LOF_INTR_ratio_IBD_detail <- ggplot(data=filter(LOF_INTR_ratio,IBD_type!="IBD"),aes(x=population,y=LOF_INTR_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("nonsense/intronic ",type)) +
  #scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_INTR_ratio_IBD_detail
ggsave(paste0(type,"_LOF_INTR_ratio_IBD_and_notIBD_detailed.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

UCNE_SYN_ratio <- IBD_summary %>% mutate(UCNE_SYN_ratio=UCNE_A/synonymous_A) %>% select(c(1:6,(ncol(IBD_summary)+1)))
UCNE_SYN_ratio$population = factor(UCNE_SYN_ratio$population,levels=c("ki","no","po","sm","do"))
#UCNE_SYN_ratio$VAR_type = factor(UCNE_SYN_ratio$VAR_type,levels=c("varssubs","variants","substitutions"))
UCNE_SYN_ratio$IBD_type = factor(UCNE_SYN_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
UCNE_SYN_ratio
for (pop in unique(UCNE_SYN_ratio$population)) {
  print(pop)
  pop_df <- filter(UCNE_SYN_ratio,population==pop)
  ks_test <- ks.test(unlist(filter(pop_df,IBD_type=="IBD") %>% select(UCNE_SYN_ratio),use.names=F), unlist(filter(pop_df,IBD_type=="notIBD") %>% select(UCNE_SYN_ratio),use.names=F), alternative="g") #one-tailed ks test (significant result means that the IBD UCNE_SYN_ratio is higher than the notIBD one)
  print(ks_test$p.value)
}
#Add $VAR and plot
ggplot_UCNE_SYN_ratio_IBD <- ggplot(data=filter(UCNE_SYN_ratio,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=UCNE_SYN_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.6)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("UCNE/synonymous ",type)) +
  scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_UCNE_SYN_ratio_IBD
ggsave(paste0(type,"_UCNE_SYN_ratio_IBD_and_notIBD.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

```

#####Relative to introns (population average version).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")
IBD_summary <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_IBD_",type,"_SNP.lr_ann.txt"))
IBD_summary[is.na(IBD_summary)] <- 0

IBD_summary$dataset <- as.factor(IBD_summary$dataset)
IBD_summary$dataset = factor(IBD_summary$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
IBD_summary$population = factor(IBD_summary$population,levels=c("ki","no","po","sm","do"))
IBD_summary$IBD_type = factor(IBD_summary$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
IBD_summary
print.data.frame(IBD_summary)

IBD_summary_alleleR <- IBD_summary %>% mutate(synonymous=synonymous_A/intronic_A,synpref=syn_pref_A/intronic_A,synunpref=syn_unpref_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A,UCNElow=UCNE_low_A/intronic_A,UCNEmid=UCNE_mid_A/intronic_A,UCNEhigh=UCNE_high_A/intronic_A) %>% select(c(1:4,6,40:50)) %>% gather(ratio,value,-IBD_type,-species,-population,-dataset,-sample,factor_key=T)
IBD_summary_alleleR

r_average_vector <- c()
r_ibd_average_vector <- c()
for (r in unique(IBD_summary_alleleR$ratio)) {
  print(r)
  for (ibd in unique(IBD_summary_alleleR$IBD_type)) {
    print(ibd)
    r_ibd_average <- filter(IBD_summary_alleleR,r==ratio & IBD_type==ibd & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
    r_ibd_average_vector <- c(r_ibd_average_vector,r_ibd_average)
  }
  r_average_vector <- c(r_average_vector,rep(r_ibd_average_vector,nrow(filter(IBD_summary_alleleR,r==ratio))/length(levels(IBD_summary_alleleR$IBD_type))))
  r_ibd_average_vector <- c()
}
print(r_average_vector)

relativised_IBD_summary_alleleR <- mutate(IBD_summary_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_IBD_summary_alleleR <- data_frame("species"=character(0),"population"=character(0),"IBD_type"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_IBD_summary_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  #print(pop)
  species <- filter(relativised_IBD_summary_alleleR,ratio==r & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_IBD_summary_alleleR$ratio)) {
    #print(r)
    for (ibd in unique(relativised_IBD_summary_alleleR$IBD_type)) {
      pop_mean <- filter(relativised_IBD_summary_alleleR,ratio==r & population==pop & IBD_type==ibd) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
      print(paste0(pop," feature ",r," average for ",ibd," is ",pop_mean))
      pop_se <- filter(relativised_IBD_summary_alleleR,ratio==r & population==pop & IBD_type==ibd) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
      print(paste0(pop," feature ",r," std error for ",ibd," is ",pop_mean))
      row_data <- cbind(species,pop,ibd,r,pop_mean,pop_se)
      colnames(row_data) <- c("species","population","IBD_type", "ratio","avg_ki_relative_value","se_ki_relative_value")
      average_relativised_IBD_summary_alleleR <- rbind(average_relativised_IBD_summary_alleleR,row_data,stringsAsFactors=F)
    }
  }
}
average_relativised_IBD_summary_alleleR$population = factor(average_relativised_IBD_summary_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_IBD_summary_alleleR$population)[levels(average_relativised_IBD_summary_alleleR$population)=="sm"] <- "an"
average_relativised_IBD_summary_alleleR$IBD_type = factor(average_relativised_IBD_summary_alleleR$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
average_relativised_IBD_summary_alleleR$ratio = factor(average_relativised_IBD_summary_alleleR$ratio,levels=c("synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_IBD_summary_alleleR$ratio)[levels(average_relativised_IBD_summary_alleleR$ratio)=="synpref"] <- "synonymous_pref"
levels(average_relativised_IBD_summary_alleleR$ratio)[levels(average_relativised_IBD_summary_alleleR$ratio)=="synunpref"] <- "synonymous_unpref"
levels(average_relativised_IBD_summary_alleleR$ratio)[levels(average_relativised_IBD_summary_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_IBD_summary_alleleR$ratio)[levels(average_relativised_IBD_summary_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_IBD_summary_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_IBD_summary_alleleR$avg_ki_relative_value)
average_relativised_IBD_summary_alleleR$se_ki_relative_value <- as.numeric(average_relativised_IBD_summary_alleleR$se_ki_relative_value)
average_relativised_IBD_summary_alleleR


#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_IBD_summary_alleleR,(IBD_type == "IBD" | IBD_type == "notIBD") & (ratio == "synonymous" | ratio == "synonymous_pref" | ratio == "synonymous_unpref" | ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious")), aes(population,avg_ki_relative_value,colour=IBD_type)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(position=position_dodge(width=0.6)) +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.85, 1.15, by = 0.1)) +
  ylim(0.85,1.15) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_IBD_and_notIBD_part1_mean.pdf"), width=35, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")


average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_IBD_summary_alleleR,(IBD_type == "IBD" | IBD_type == "notIBD") & (ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh")), aes(population,avg_ki_relative_value,colour=IBD_type)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(position=position_dodge(width=0.6)) +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_IBD_and_notIBD_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 2, layout_matrix = rbind(c(rep(1,20)), c(rep(2,18),rep(NA,3))))
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_IBD_and_notIBD_all.pdf"), width=40, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

#####Plot population average variant count results.
######Nonsense/Syn:
```{r Plot variant count results}
library(readr)
library(dplyr)
library(ggplot2)

type="varssubs" #varssubs #variants #substitutions
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")
#variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',inv=T,value=T)
#substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',value=T)
IBD_summary <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_IBD_",type,"_SNP.lr_ann.txt"))
IBD_summary[is.na(IBD_summary)] <- 0

LOF_SYN_ratio <- IBD_summary %>% mutate(LOF_SYN_ratio=nonsense_A/synonymous_A) %>% select(c(1:6,(ncol(IBD_summary)+1)))
LOF_SYN_ratio$population = factor(LOF_SYN_ratio$population,levels=c("ki","no","po","sm","do"))
#LOF_SYN_ratio$VAR_type = factor(LOF_SYN_ratio$VAR_type,levels=c("varssubs","variants","substitutions"))
LOF_SYN_ratio$IBD_type = factor(LOF_SYN_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
LOF_SYN_ratio

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_LOF_SYN_ratioR <- data_frame("species"=character(0),"population"=character(0),"IBD_type"=character(0),"avg_LOF_SYN_ratio"=character(0),"se_LOF_SYN_ratio"=character(0)) #next, create the empty dataframe

for (pop in unique(LOF_SYN_ratio$population)) { #then loop over each population and feature to get the mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(LOF_SYN_ratio,IBD_type=="IBD" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(LOF_SYN_ratio$IBD_type)) {
    print(r)
    pop_mean <- filter(LOF_SYN_ratio,IBD_type==r & population==pop) %>% select(LOF_SYN_ratio) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(LOF_SYN_ratio,IBD_type==r & population==pop) %>% select(LOF_SYN_ratio) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","IBD_type","avg_LOF_SYN_ratio","se_LOF_SYN_ratio")
    average_LOF_SYN_ratioR <- rbind(average_LOF_SYN_ratioR,row_data,stringsAsFactors=F)
  }
}

average_LOF_SYN_ratioR$population = factor(average_LOF_SYN_ratioR$population,levels=c("ki","no","po","sm","do"))
average_LOF_SYN_ratioR$IBD_type = factor(average_LOF_SYN_ratioR$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
average_LOF_SYN_ratioR$avg_LOF_SYN_ratio <- as.numeric(average_LOF_SYN_ratioR$avg_LOF_SYN_ratio)
average_LOF_SYN_ratioR$se_LOF_SYN_ratio <- as.numeric(average_LOF_SYN_ratioR$se_LOF_SYN_ratio)
average_LOF_SYN_ratioR
write_csv(average_LOF_SYN_ratioR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/",type,"_LOF_SYN_ratio_mean.csv"))

#Add $VAR and plot
pd <- position_dodge(0.5)
average_ggplot_LOF_SYN_ratio_IBD <- ggplot(data=filter(average_LOF_SYN_ratioR,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=avg_LOF_SYN_ratio,colour=IBD_type)) +
  geom_point(position=pd) +
  geom_errorbar(aes(ymin=avg_LOF_SYN_ratio-se_LOF_SYN_ratio, ymax=avg_LOF_SYN_ratio+se_LOF_SYN_ratio), position=pd, width=0.5) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("nonsense/synonymous ",type)) +
  scale_y_continuous(limits = c(0, 0.025), breaks=seq(0,0.025,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
average_ggplot_LOF_SYN_ratio_IBD
ggsave(paste0(type,"_LOF_SYN_ratio_IBD_and_notIBD_mean.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

pd <- position_dodge(0.5)
average_ggplot_LOF_SYN_ratio_IBD_detail <- ggplot(data=filter(average_LOF_SYN_ratioR,IBD_type!="IBD"),aes(x=population,y=avg_LOF_SYN_ratio,colour=IBD_type)) +
  geom_point(position=pd) +
  geom_errorbar(aes(ymin=avg_LOF_SYN_ratio-se_LOF_SYN_ratio, ymax=avg_LOF_SYN_ratio+se_LOF_SYN_ratio), position=pd, width=0.2) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("nonsense/synonymous ",type)) +
  scale_y_continuous(limits = c(0, 0.025), breaks=seq(0,0.025,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
average_ggplot_LOF_SYN_ratio_IBD_detail
ggsave(paste0(type,"_LOF_SYN_ratio_IBD_and_notIBD_detailed_mean.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

```

######UCNE/Syn:
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)

type="varssubs" #varssubs #variants #substitutions
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")
#variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',inv=T,value=T)
#substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',value=T)
IBD_summary <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_IBD_",type,"_SNP.lr_ann.txt"))
IBD_summary[is.na(IBD_summary)] <- 0

UCNE_SYN_ratio <- IBD_summary %>% mutate(UCNE_SYN_ratio=UCNE_A/synonymous_A) %>% select(c(1:6,(ncol(IBD_summary)+1)))
UCNE_SYN_ratio$population = factor(UCNE_SYN_ratio$population,levels=c("ki","no","po","sm","do"))
#UCNE_SYN_ratio$VAR_type = factor(UCNE_SYN_ratio$VAR_type,levels=c("varssubs","variants","substitutions"))
UCNE_SYN_ratio$IBD_type = factor(UCNE_SYN_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
UCNE_SYN_ratio

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_UCNE_SYN_ratioR <- data_frame("species"=character(0),"population"=character(0),"IBD_type"=character(0),"avg_UCNE_SYN_ratio"=character(0),"se_UCNE_SYN_ratio"=character(0)) #next, create the empty dataframe

for (pop in unique(UCNE_SYN_ratio$population)) { #then loop over each population and feature to get the mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(UCNE_SYN_ratio,IBD_type=="IBD" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(UCNE_SYN_ratio$IBD_type)) {
    print(r)
    pop_mean <- filter(UCNE_SYN_ratio,IBD_type==r & population==pop) %>% select(UCNE_SYN_ratio) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(UCNE_SYN_ratio,IBD_type==r & population==pop) %>% select(UCNE_SYN_ratio) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","IBD_type","avg_UCNE_SYN_ratio","se_UCNE_SYN_ratio")
    average_UCNE_SYN_ratioR <- rbind(average_UCNE_SYN_ratioR,row_data,stringsAsFactors=F)
  }
}

average_UCNE_SYN_ratioR$population = factor(average_UCNE_SYN_ratioR$population,levels=c("ki","no","po","sm","do"))
average_UCNE_SYN_ratioR$IBD_type = factor(average_UCNE_SYN_ratioR$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
average_UCNE_SYN_ratioR$avg_UCNE_SYN_ratio <- as.numeric(average_UCNE_SYN_ratioR$avg_UCNE_SYN_ratio)
average_UCNE_SYN_ratioR$se_UCNE_SYN_ratio <- as.numeric(average_UCNE_SYN_ratioR$se_UCNE_SYN_ratio)
average_UCNE_SYN_ratioR
write_csv(average_UCNE_SYN_ratioR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/",type,"_UCNE_SYN_ratio_mean.csv"))

#Add $VAR and plot
pd <- position_dodge(0.5)
average_ggplot_UCNE_SYN_ratio_IBD <- ggplot(data=filter(average_UCNE_SYN_ratioR,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=avg_UCNE_SYN_ratio,colour=IBD_type)) +
  geom_point(position=pd) +
  geom_errorbar(aes(ymin=avg_UCNE_SYN_ratio-se_UCNE_SYN_ratio, ymax=avg_UCNE_SYN_ratio+se_UCNE_SYN_ratio), position=pd, width=0.5) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("UCNE/synonymous ",type)) +
  scale_y_continuous(limits = c(0, 0.025), breaks=seq(0,0.025,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
average_ggplot_UCNE_SYN_ratio_IBD
ggsave(paste0(type,"_UCNE_SYN_ratio_IBD_and_notIBD_mean.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

pd <- position_dodge(0.5)
average_ggplot_UCNE_SYN_ratio_IBD_detail <- ggplot(data=filter(average_UCNE_SYN_ratioR,IBD_type!="IBD"),aes(x=population,y=avg_UCNE_SYN_ratio,colour=IBD_type)) +
  geom_point(position=pd) +
  geom_errorbar(aes(ymin=avg_UCNE_SYN_ratio-se_UCNE_SYN_ratio, ymax=avg_UCNE_SYN_ratio+se_UCNE_SYN_ratio), position=pd, width=0.2) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("UCNE/synonymous ",type)) +
  scale_y_continuous(limits = c(0, 0.025), breaks=seq(0,0.025,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
average_ggplot_UCNE_SYN_ratio_IBD_detail
ggsave(paste0(type,"_UCNE_SYN_ratio_IBD_and_notIBD_detailed_mean.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

```

####Bad coordinates (not nm_filtered):
#####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(variants) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_IBD_${VAR}_${TYPE}.woutnm.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm -f ${CALLING}"_ann_individual_summary_IBD_"${VAR}"_"${TYPE}".woutnm.lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\tVAR_type\tIBD_type\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_IBD_"${VAR}"_"${TYPE}".woutnm.lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"*"IBD"*"_"${VAR}"_"${TYPE}".woutnm.lr_ann.vcf" ! -path '*testing_variant_numbers*' -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  IBD=$(echo ${i} | rev | cut -d'_' -f4 | rev)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_TOL_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep -v '#' | wc -l)
  MISSENSE_TOL_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_DEL_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep -v '#' | wc -l)
  MISSENSE_DEL_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$VAR\t$IBD\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_IBD_"${VAR}"_"${TYPE}".woutnm.lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/$CALLING"_ann_individual_summary_IBD_"*"_SNP.woutnm.lr_ann.txt" /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD

```

#####Plot individual variant count results.
```{r Plot variant count results}
library(readr)
library(dplyr)
library(ggplot2)

type="variants" #varssubs #variants #substitutions
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")
#variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',inv=T,value=T)
#substitution_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/", pattern="_ann_individual_summary_IBD_SNP.lr_ann.txt"),pattern='nm2nm3',value=T)
IBD_summary <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_IBD_",type,"_SNP.woutnm.lr_ann.txt"))
IBD_summary[is.na(IBD_summary)] <- 0

LOF_SYN_ratio <- IBD_summary %>% mutate(LOF_SYN_ratio=nonsense_V/synonymous_V) %>% select(c(1:6,(ncol(IBD_summary)+1)))
LOF_SYN_ratio$population = factor(LOF_SYN_ratio$population,levels=c("ki","no","po","sm","do"))
#LOF_SYN_ratio$VAR_type = factor(LOF_SYN_ratio$VAR_type,levels=c("varssubs","variants","substitutions"))
LOF_SYN_ratio$IBD_type = factor(LOF_SYN_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
LOF_SYN_ratio
for (pop in unique(LOF_SYN_ratio$population)) {
  print(pop)
  pop_df <- filter(LOF_SYN_ratio,population==pop)
  ks_test <- ks.test(unlist(filter(pop_df,IBD_type=="IBD") %>% select(LOF_SYN_ratio),use.names=F), unlist(filter(pop_df,IBD_type=="notIBD") %>% select(LOF_SYN_ratio),use.names=F), alternative="g") #one-tailed ks test (significant result means that the IBD LOF_SYN_ratio is higher than the notIBD one)
  print(ks_test$p.value)
}
#Add $VAR and plot
ggplot_LOF_SYN_ratio_IBD <- ggplot(data=filter(LOF_SYN_ratio,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=LOF_SYN_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.6)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("LoF/synonymous ",type," ratio")) +
  scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_SYN_ratio_IBD
ggsave(paste0(type,"_LOF_SYN_ratio_IBD_and_notIBD.woutnm.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

ggplot_LOF_SYN_ratio_IBD_detail <- ggplot(data=filter(LOF_SYN_ratio,IBD_type!="IBD"),aes(x=population,y=LOF_SYN_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5, position=position_dodge(width=0.6)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("LoF/synonymous ",type," ratio")) +
  scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_SYN_ratio_IBD_detail
ggsave(paste0(type,"_LOF_SYN_ratio_IBD_and_notIBD_detailed.woutnm.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

LOF_INTR_ratio <- IBD_summary %>% mutate(LOF_INTR_ratio=nonsense_V/intronic_V) %>% select(c(1:6,(ncol(IBD_summary)+1)))
LOF_INTR_ratio$population = factor(LOF_INTR_ratio$population,levels=c("ki","no","po","sm","do"))
LOF_INTR_ratio$IBD_type = factor(LOF_INTR_ratio$IBD_type,levels=c("IBD10k","IBD100k","IBD1000k","IBD","notIBD"))
LOF_INTR_ratio

for (pop in unique(LOF_INTR_ratio$population)) {
  print(pop)
  pop_df <- filter(LOF_INTR_ratio,population==pop)
  ks_test <- ks.test(unlist(filter(pop_df,IBD_type=="IBD") %>% select(LOF_INTR_ratio),use.names=F), unlist(filter(pop_df,IBD_type=="notIBD") %>% select(LOF_INTR_ratio),use.names=F), alternative="g")
  print(ks_test$p.value)
}

ggplot_LOF_INTR_ratio_IBD <- ggplot(data=filter(LOF_INTR_ratio,IBD_type=="IBD" | IBD_type=="notIBD"),aes(x=population,y=LOF_INTR_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("LoF/intronic ",type," ratio")) +
  #scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_INTR_ratio_IBD
ggsave(paste0(type,"_LOF_INTR_ratio_IBD_and_notIBD.woutnm.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

ggplot_LOF_INTR_ratio_IBD_detail <- ggplot(data=filter(LOF_INTR_ratio,IBD_type!="IBD"),aes(x=population,y=LOF_INTR_ratio,colour=IBD_type)) +
  geom_boxplot(width=0.5,position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous sites") +
  xlab("Populations") +
  ylab(paste0("LoF/intronic ",type," ratio")) +
  #scale_y_continuous(limits = c(0.005, 0.030), breaks=seq(0.005,0.03,by=0.005)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.08,0.10),
      legend.title=element_blank()
  )
ggplot_LOF_INTR_ratio_IBD_detail
ggsave(paste0(type,"_LOF_INTR_ratio_IBD_and_notIBD_detailed.woutnm.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/IBD_and_notIBD/")

```

###X-CHR (joint calling).
####Define Xchr regions:
```{bash}

#The Xchr was already defined by Abascal. But it includes the pseudoautosomic region (PAR) which we don't want to include. In order to be more conservative, I'll follow María's criterion and exclude the telomeric regions from the X chromosome.

cd /GRUPOS/grupolince/copia_fabascal/MAPPINGS/

bedtools subtract -a lynx2cat_wTiger_Xchr.sorted.merged.bed -b /GRUPOS/grupolince/telomers_centromers_definition/tel0-10m_regions_based_on_synteny_1000bp.bed > lynx2cat_wTiger_Xchr_woutPAR.sorted.merged.bed

```

####Create file with sex determination:
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation

#I use nano to paste the following information (generated by María as part of her diversity paper, based on the coverage-based sex determination that Elena designed).
Sample 	Sex
c_ll_ki_0090	Female
c_ll_ki_0091	Male
c_ll_ki_0092	Female
c_ll_ki_0093	Male
c_ll_ki_0094	Male
c_ll_ki_0095	Male
c_ll_ki_0096	Male
c_ll_ki_0097	Female
c_ll_ki_0098	Female
c_ll_ki_0099	Female
c_ll_ki_0100	Male
c_ll_ki_0101	Female
c_ll_ki_0102	Male
c_ll_no_0075	Male
c_ll_no_0076	Male
c_ll_no_0077	Male
c_ll_no_0078	Male
c_ll_no_0079	Male
c_ll_no_0080	Female
c_ll_no_0081	Female
c_ll_no_0082	Male
c_ll_po_0001	Male
c_ll_po_0002	Female
c_ll_po_0003	Female
c_ll_po_0011	Male
c_ll_po_0014	Male
c_ll_po_0019	Male
c_ll_po_0105	Male
c_ll_po_0106	Female
c_lp_do_0007	Male
c_lp_sm_0134	Female
c_lp_sm_0138	Male
c_lp_sm_0140	Male
c_lp_do_0141	Male
c_lp_do_0144	Male
c_lp_do_0153	Male
c_lp_sm_0155	Female
c_lp_sm_0156	Female
c_lp_sm_0161	Female
c_lp_do_0162	Female
c_lp_do_0163	Female
c_lp_do_0173	Male
c_lp_sm_0185	Male
c_lp_sm_0186	Male
c_lp_sm_0206	Female
c_lp_sm_0208	Female
c_lp_sm_0213	Female
c_lp_sm_0221	Male
c_lp_sm_0226	Male
c_lp_sm_0276	Female
c_lp_sm_0298	Male
c_lp_do_0300	Male
c_lp_sm_0320	Female
c_lp_sm_0325	Female
c_lp_do_0333	Female
c_lp_do_0335	Female
c_lp_sm_0359	Male
c_lp_do_0443	Male
c_lp_do_0444	Female
c_lp_sm_0450	Male

#The information is stored in:
samples_sex.txt

```

####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_Xchr_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm -f ${CALLING}"_ann_individual_summary_Xchr_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\tsex\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_Xchr_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  SEX=$(grep ${ind} samples_sex.txt | cut -f2)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/copia_fabascal/MAPPINGS/lynx2cat_wTiger_Xchr_woutPAR.sorted.merged.bed -header > ${i/_individual_${VAR}_${TYPE}.lr_ann.vcf/_individual_${VAR}_Xchr_${TYPE}.lr_ann.vcf}
  j=(${i/_individual_${VAR}_${TYPE}.lr_ann.vcf/_individual_${VAR}_Xchr_${TYPE}.lr_ann.vcf})
  TOTAL_V=$(grep -v '#' ${j} | wc -l)
  TOTAL_A=$(grep -v '#' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${j} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${j} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${j} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${j} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${VAR}_Xchr_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${VAR}_Xchr_syn_pref.temp.borrar)
  SYN_PREF_A=$(cut -f8 ${VAR}_Xchr_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${VAR}_Xchr_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${VAR}_Xchr_syn_unpref.temp.borrar)
  SYN_UNPREF_A=$(cut -f8 ${VAR}_Xchr_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${j} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${VAR}_Xchr_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${VAR}_Xchr_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${VAR}_Xchr_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${VAR}_Xchr_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${VAR}_Xchr_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${VAR}_Xchr_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${j} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${j} | wc -l)
  UCNE_A=$(grep 'UCNE' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
    bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${VAR}_Xchr_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${VAR}_Xchr_ucne_low.temp.borrar)
  UCNE_LOW_A=$(cut -f8 ${VAR}_Xchr_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${VAR}_Xchr_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${VAR}_Xchr_ucne_mid.temp.borrar)
  UCNE_MID_A=$(cut -f8 ${VAR}_Xchr_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${VAR}_Xchr_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${VAR}_Xchr_ucne_high.temp.borrar)
  UCNE_HIGH_A=$(cut -f8 ${VAR}_Xchr_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$SEX\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_Xchr_"${VAR}"_"${TYPE}".lr_ann.txt"
  done
rm ${VAR}_*.temp.borrar

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/$CALLING"_ann_individual_summary_Xchr_"*"_SNP.lr_ann.txt" /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/Xchr/

```

####Plot results.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/Xchr/")

female_data <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_Xchr_",type,"_SNP.lr_ann.txt")) %>% filter(sex=="Female", dataset=="5x") %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
female_data$population = factor(female_data$population,levels=c("ki","no","po","sm","do"))
print.data.frame(female_data)

male_data <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_Xchr_",type,"_SNP.lr_ann.txt")) %>% filter(sex=="Male", dataset=="5x") %>% select(.,population,contains("_V"),-contains("/"),-contains("coding")) %>% rename_at(vars(ends_with("_V")),funs(gsub("_V","",.)))
male_data$population = factor(male_data$population,levels=c("ki","no","po","sm","do"))
print.data.frame(male_data)

female_data_tidy <- female_data %>% gather(ratio,value,-population,factor_key=T)
female_data_tidy

male_data_tidy <- male_data %>% gather(ratio,value,-population,factor_key=T)
male_data_tidy

female_average_vector <- c()
for (r in unique(female_data_tidy$ratio)) {
  print(r)
  female_average <- filter(female_data_tidy,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  female_average_vector <- c(female_average_vector,rep(female_average,nrow(filter(female_data_tidy,r==ratio))))
}
print(female_average_vector)

relativised_female_data_tidy <- mutate(female_data_tidy, ki_relative_value=value/female_average_vector, sex="female")

male_average_vector <- c()
for (r in unique(male_data_tidy$ratio)) {
  print(r)
  male_average <- filter(male_data_tidy,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  male_average_vector <- c(male_average_vector,rep(male_average,nrow(filter(male_data_tidy,r==ratio))))
}
print(male_average_vector)

relativised_male_data_tidy <- mutate(male_data_tidy, ki_relative_value=value/male_average_vector, sex="male")

average_relativised_female_data_tidy <- data_frame("population"=character(0),"sex"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_female_data_tidy$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  for (r in unique(relativised_female_data_tidy$ratio)) {
    print(r)
    pop_mean <- filter(relativised_female_data_tidy,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    row_data <- cbind(pop,"female",r,pop_mean)
    colnames(row_data) <- c("population","sex","ratio","avg_ki_relative_value")
    average_relativised_female_data_tidy <- rbind(average_relativised_female_data_tidy,row_data,stringsAsFactors=F)
  }
}


average_relativised_male_data_tidy <- data_frame("population"=character(0),"sex"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_male_data_tidy$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  for (r in unique(relativised_male_data_tidy$ratio)) {
    print(r)
    pop_mean <- filter(relativised_male_data_tidy,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    row_data <- cbind(pop,"male",r,pop_mean)
    colnames(row_data) <- c("population","sex","ratio","avg_ki_relative_value")
    average_relativised_male_data_tidy <- rbind(average_relativised_male_data_tidy,row_data,stringsAsFactors=F)
  }
}

average_relativised_all_data_tidy <- rbind(average_relativised_female_data_tidy,average_relativised_male_data_tidy)

average_relativised_all_data_tidy$population = factor(average_relativised_all_data_tidy$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_all_data_tidy$population)[levels(average_relativised_all_data_tidy$population)=="sm"] <- "an"
average_relativised_all_data_tidy$ratio = factor(average_relativised_all_data_tidy$ratio,levels=c("total","intergenic","intronic","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE"))
levels(average_relativised_all_data_tidy$ratio) <- c("total","intergenic","introns","synonymous","synonymous_pref","synonymous_unpref","missense","missense_tolerated","missense_deleterious","nonsense","UCNE")
average_relativised_all_data_tidy$avg_ki_relative_value <- as.numeric(average_relativised_all_data_tidy$avg_ki_relative_value)
average_relativised_all_data_tidy






#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.2,0.2,0.2,0.2),"max"=c(1.9,1.9,1.9,1.9),"breaks"=c(0.3,0.3,0.3,0.3))

ggplot_up <- ggplot(data=filter(average_relativised_all_data_tidy,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|nonsense|\\<UCNE\\>',ratio)), aes(population,avg_ki_relative_value,colour=population,shape=sex)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average Xchr genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_down <- ggplot(data=filter(average_relativised_all_data_tidy,grepl('synonymous_pref|synonymous_unpref|missense_tolerated|missense_deleterious',ratio)), aes(population,avg_ki_relative_value,colour=population,shape=sex)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average Xchr genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down


if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,1),c(2,2,NA)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Average chrX genetic load relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_Xchr_relative2Kirov.pdf"), width=41, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/Xchr/",ggplot_combined)

```
###Centromere (joint calling).
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm -f ${CALLING}"_ann_individual_summary_centr_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\tVAR_type\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_centr_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/telomers_centromers_definition/centr_regions_based_on_synteny_1000bp.bed -header > "ind_centr_"${VAR}"_"${TYPE}".temporary.vcf"
  j=("ind_centr_"${VAR}"_"${TYPE}".temporary.vcf")
  TOTAL_V=$(grep -v '#' ${j} | wc -l)
  TOTAL_A=$(grep -v '#' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${j} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${j} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${j} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${j} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${VAR}_centr_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${VAR}_centr_syn_pref.temp.borrar)
  SYN_PREF_A=$(cut -f8 ${VAR}_centr_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${VAR}_centr_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${VAR}_centr_syn_unpref.temp.borrar)
  SYN_UNPREF_A=$(cut -f8 ${VAR}_centr_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${j} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${VAR}_centr_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${VAR}_centr_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${VAR}_centr_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${VAR}_centr_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${VAR}_centr_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${VAR}_centr_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${j} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${j} | wc -l)
  UCNE_A=$(grep 'UCNE' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${VAR}_centr_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${VAR}_centr_ucne_low.temp.borrar)
  UCNE_LOW_A=$(cut -f8 ${VAR}_centr_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${VAR}_centr_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${VAR}_centr_ucne_mid.temp.borrar)
  UCNE_MID_A=$(cut -f8 ${VAR}_centr_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${VAR}_centr_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${VAR}_centr_ucne_high.temp.borrar)
  UCNE_HIGH_A=$(cut -f8 ${VAR}_centr_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$VAR\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_centr_"${VAR}"_"${TYPE}".lr_ann.txt"
  rm "ind_centr_"${VAR}"_"${TYPE}".temporary.vcf"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere
unset SSHPASS

```

####Relative to introns (population average version).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_centr_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,synpref=syn_pref_A/intronic_A,synunpref=syn_unpref_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A,UCNElow=UCNE_low_A/intronic_A,UCNEmid=UCNE_mid_A/intronic_A,UCNEhigh=UCNE_high_A/intronic_A) %>% select(c(1:4,39:49)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & dataset=="5x" & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & dataset=="5x" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & dataset=="5x" & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & dataset=="5x" & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synpref"] <- "synonymous_pref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synunpref"] <- "synonymous_unpref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR

#Plot these means and standard errors:
average_relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=average_relativised_variants_and_subst_wg_alleleR, aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  average_relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_centr_relative2introns_short_mean.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")

#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "synonymous" | ratio == "synonymous_pref" | ratio == "synonymous_unpref" | ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.9, 1.05, by = 0.05)) +
  ylim(0.8,1.1) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_centr_relative2introns_part1_mean.pdf"), width=35, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")

average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.7, by = 0.2)) +
  ylim(0.5,1.7) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_centr_relative2introns_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 2, layout_matrix = rbind(c(rep(1,20)), c(rep(2,18),rep(NA,3))))
ggsave(paste0(type,"_derived_allele_allele_ratio_centr_5xrelative2introns_all.pdf"), width=40, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

####Visualise derived allele counts.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")
file <- "c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_centr_varssubs_SNP.lr_ann.txt"
variants_and_subst_wg <- read_tsv(paste0(wd_path,file)) %>% select(-c(25:28))

type <- strsplit(file,"_")[[1]][length(strsplit(file,"_")[[1]])-2]

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_varN <- variants_and_subst_wg %>% select(c(1:4,6,10,13,15,17,19,21,23)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_and_subst_wg_varN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_varN$feature))
variants_and_subst_wg_varN

derived_allele_varssubs_counts_ggplot <- ggplot(data=variants_and_subst_wg_varN, aes(population,N,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste("N",type)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_varssubs_counts_ggplot
ggsave(paste0(type,"_derived_allele_variant_counts.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")


variants_and_subst_wg_alleleN <- variants_and_subst_wg %>% select(c(1:4,7,11,14,16,18,20,22,24)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_and_subst_wg_alleleN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_alleleN$feature))
variants_and_subst_wg_alleleN

derived_allele_allele_counts_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste("N alleles",type)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
derived_allele_allele_counts_ggplot
ggsave(paste0(type,"_derived_allele_allele_counts.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")

derived_allele_allele_counts_sep_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  facet_wrap(feature ~ species,nrow=8,ncol=2,scales="free") +
  #facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste("N alleles",type)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        strip.text.x = element_blank(),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
derived_allele_allele_counts_sep_ggplot
ggsave(paste0(type,"_derived_allele_allele_counts_sep.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")


variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(syn_intr_R=synonymous_A/intronic_A,mis_intr_R=missense_A/intronic_A,mistol_intr_R=missense_tol_A/intronic_A,misdel_intr_R=missense_del_A/intronic_A,non_intr_R=nonsense_A/intronic_A,UCNE_intr_R=UCNE_A/intronic_A) %>% select(c(1:4,25:30)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

derived_allele_allele_ratio_ggplot <- ggplot(data=variants_and_subst_wg_alleleR, aes(population,value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste(type,"ratio")) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")

```

####Relative to synonymous (per individual version for Aurora).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_centr_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/synonymous_A,missense=missense_A/synonymous_A,mistol=missense_tol_A/synonymous_A,misdel=missense_del_A/synonymous_A,nonsense=nonsense_A/synonymous_A,UCNE=UCNE_A/synonymous_A) %>% select(c(1:4,29:34)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=filter(relativised_variants_and_subst_wg_alleleR,ratio != "synonymous" & ratio != "missense"), aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.7, 1.2, by = 0.2)) +
  ylim(0.7,1.2) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_short.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/centromere/")

```

###Telomere (joint calling).
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm -f ${CALLING}"_ann_individual_summary_tel_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\tVAR_type\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_tel_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/telomers_centromers_definition/tel0-10m_regions_based_on_synteny_1000bp.bed -header > "ind_tel_"${VAR}"_"${TYPE}".temporary.vcf"
  j=("ind_tel_"${VAR}"_"${TYPE}".temporary.vcf")
  TOTAL_V=$(grep -v '#' ${j} | wc -l)
  TOTAL_A=$(grep -v '#' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${j} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${j} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${j} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${j} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${VAR}_tel_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${VAR}_tel_syn_pref.temp.borrar)
  SYN_PREF_A=$(cut -f8 ${VAR}_tel_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${VAR}_tel_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${VAR}_tel_syn_unpref.temp.borrar)
  SYN_UNPREF_A=$(cut -f8 ${VAR}_tel_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${j} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${VAR}_tel_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${VAR}_tel_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${VAR}_tel_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${VAR}_tel_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${VAR}_tel_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${VAR}_tel_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${j} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${j} | wc -l)
  UCNE_A=$(grep 'UCNE' ${j} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${VAR}_tel_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${VAR}_tel_ucne_low.temp.borrar)
  UCNE_LOW_A=$(cut -f8 ${VAR}_tel_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${VAR}_tel_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${VAR}_tel_ucne_mid.temp.borrar)
  UCNE_MID_A=$(cut -f8 ${VAR}_tel_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${j} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${VAR}_tel_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${VAR}_tel_ucne_high.temp.borrar)
  UCNE_HIGH_A=$(cut -f8 ${VAR}_tel_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$VAR\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_tel_"${VAR}"_"${TYPE}".lr_ann.txt"
  rm "ind_tel_"${VAR}"_"${TYPE}".temporary.vcf"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere
unset SSHPASS

```

####Relative to introns (population average version).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_tel_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,synpref=syn_pref_A/intronic_A,synunpref=syn_unpref_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A,UCNElow=UCNE_low_A/intronic_A,UCNEmid=UCNE_mid_A/intronic_A,UCNEhigh=UCNE_high_A/intronic_A) %>% select(c(1:4,39:49)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & dataset=="5x" & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & dataset=="5x" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & dataset=="5x" & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & dataset=="5x" & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synpref"] <- "synonymous_pref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synunpref"] <- "synonymous_unpref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR

#Plot these means and standard errors:
average_relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=average_relativised_variants_and_subst_wg_alleleR, aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.6, 1.4, by = 0.2)) +
  ylim(0.6,1.4) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  average_relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_tel_relative2introns_short_mean.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")

#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "synonymous" | ratio == "synonymous_pref" | ratio == "synonymous_unpref" | ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.7, 1.2, by = 0.05)) +
  ylim(0.7,1.2) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_tel_relative2introns_part1_mean.pdf"), width=35, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")

average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.4, 2.2, by = 0.2)) +
  ylim(0.4,2.2) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_tel_relative2introns_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 2, layout_matrix = rbind(c(rep(1,20)), c(rep(2,18),rep(NA,3))))
ggsave(paste0(type,"_derived_allele_allele_ratio_tel_5xrelative2introns_all.pdf"), width=40, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

####Visualise derived allele counts.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")
file <- "c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_tel_varssubs_SNP.lr_ann.txt"
variants_and_subst_wg <- read_tsv(paste0(wd_path,file)) %>% select(-c(25:28))

type <- strsplit(file,"_")[[1]][length(strsplit(file,"_")[[1]])-2]

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_varN <- variants_and_subst_wg %>% select(c(1:4,6,10,13,15,17,19,21,23)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_and_subst_wg_varN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_varN$feature))
variants_and_subst_wg_varN

derived_allele_varssubs_counts_ggplot <- ggplot(data=variants_and_subst_wg_varN, aes(population,N,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste("N",type)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_varssubs_counts_ggplot
ggsave(paste0(type,"_derived_allele_variant_counts.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")


variants_and_subst_wg_alleleN <- variants_and_subst_wg %>% select(c(1:4,7,11,14,16,18,20,22,24)) %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_and_subst_wg_alleleN$feature) <- gsub('.{2}$', '', levels(variants_and_subst_wg_alleleN$feature))
variants_and_subst_wg_alleleN

derived_allele_allele_counts_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste("N alleles",type)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_ggplot
ggsave(paste0(type,"_derived_allele_allele_counts.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")

derived_allele_allele_counts_sep_ggplot <- ggplot(data=variants_and_subst_wg_alleleN, aes(population,N,colour=population)) +
  facet_wrap(feature ~ species,nrow=8,ncol=2,scales="free") +
  #facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste("N alleles",type)) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        strip.text.x = element_blank(),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_counts_sep_ggplot
ggsave(paste0(type,"_derived_allele_allele_counts_sep.pdf"), width=15, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")


variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(syn_intr_R=synonymous_A/intronic_A,mis_intr_R=missense_A/intronic_A,mistol_intr_R=missense_tol_A/intronic_A,misdel_intr_R=missense_del_A/intronic_A,non_intr_R=nonsense_A/intronic_A,UCNE_intr_R=UCNE_A/intronic_A) %>% select(c(1:4,25:30)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

derived_allele_allele_ratio_ggplot <- ggplot(data=variants_and_subst_wg_alleleR, aes(population,value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab(paste(type,"ratio")) +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")

```

####Relative to synonymous (per individual version for Aurora).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_tel_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/synonymous_A,missense=missense_A/synonymous_A,mistol=missense_tol_A/synonymous_A,misdel=missense_del_A/synonymous_A,nonsense=nonsense_A/synonymous_A,UCNE=UCNE_A/synonymous_A) %>% select(c(1:4,29:34)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=filter(relativised_variants_and_subst_wg_alleleR,ratio != "synonymous" & ratio != "missense"), aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.2, 2.2, by = 0.4)) +
  ylim(0.2,2.2) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2synonymous_short.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/telomere/")

```

###Transversions only.
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_ann_individual_summary_TVonly_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_individual_summary_TVonly_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_summary_TVonly_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_TVonly_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_summary_TVonly_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_ll_ki_c_ll_no_c_ll_po_h_ll_pv_nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_individual_summary_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/

```

##At the population level.
###Whole-genome (joint calling).
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions #segregating #fixed
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_population_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_population_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -path *"_perpop/c_*5x*_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${POPLIST[@]}"
  do
  echo "${i}"
  vcf=$(echo "${i}" | rev | cut -d'/' -f1 | rev)
  echo "${vcf}"
  SPECIES=$(echo "${vcf}" | cut -c3-4)
  POPULATION=$(echo "${vcf}" | cut -c14-15)
  DATASET=$(echo "${vcf}" | cut -c6-7)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_TOL_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep -v '#' | wc -l)
  MISSENSE_TOL_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_DEL_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep -v '#' | wc -l)
  MISSENSE_DEL_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_LOW_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed | grep -v '#' | wc -l)
  UCNE_LOW_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_MID_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed | grep -v '#' | wc -l)
  UCNE_MID_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_HIGH_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed | grep -v '#' | wc -l)
  UCNE_HIGH_A=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed | grep -v '#' | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_population_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_population_summary_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/
unset SSHPASS

```

##At the subsampled population level. N=8 populations will be used.
###Whole-genome.
####Get counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_nm2_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_ann_population_summary_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_population_summary_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_population_summary_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_subsample_n008_"${TYPE}".lr_ann.vcf" -print`))
for i in "${POPLIST[@]}"
  do
  echo "${i}"
  vcf=$(echo "${i}" | rev | cut -d'/' -f1 | rev)
  echo "${vcf}"
  SPECIES=$(echo "${vcf}" | cut -c3-4)
  POPULATION=$(echo "${vcf}" | cut -c14-15)
  DATASET=$(echo "${vcf}" | cut -c6-7)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_population_summary_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_nm2_origcov)
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary*.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/

```

####Plot population variant count results.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

calling_counts_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*population_summary_SNP.lr_ann.txt"),pattern='nm2nm3',inv=T,value=T)
perpop_counts <- data_frame()
for (file in calling_counts_files) {
  calling_counts <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/",file),col_names=T) %>% select(.,population,matches("synonymous|missense|nonsense|UCNE"),-contains("/")) %>% gather(feature,N,-population,factor_key=T)
  perpop_counts <- rbind(perpop_counts,calling_counts)
}
perpop_counts$population <- as.factor(perpop_counts$population)
perpop_counts$population = factor(perpop_counts$population,levels=c("ki","no","po","sm","do"))

counter=0
for (i in levels(perpop_counts$population)) {
  counter=which(levels(perpop_counts$population)==i)
  if (counter < length(levels(perpop_counts$population))) {
    for (j in levels(perpop_counts$population)[c((counter+1):length(levels(perpop_counts$population)))]) {
    print(paste0(i," vs ",j))
    i_vs_j_counts <- full_join(filter(perpop_counts,population==i),filter(perpop_counts,population==j),by=c("feature"),suffix=c(paste0("_",i),paste0("_",j))) %>% select(-contains("population")) %>% filter(!grepl("/",feature)&!grepl("_V",feature)&!grepl("coding",feature))
    i_vs_j_ggplot <- ggplot(data=as.data.frame(i_vs_j_counts)) +
      geom_point(aes_string(x=paste0("N_",i),y=paste0("N_",j),colour="feature")) +
      #coord_fixed() +
      ggtitle(paste0(i, " vs ",j)) +
      xlab(paste0("N_",i)) +
      xlim(0,100000) +
      ylab(paste0("N_",j)) +
      ylim(0,100000) +
      geom_abline(intercept=0,slope=1) +
      theme_bw() +
      theme(text=element_text(size=12,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=16),
          #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
          #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          #panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position=c(0.2,0.84),
          legend.title=element_blank()
      )
    i_vs_j_ggplot
    ggsave(paste0("derived_allele_counts_",i,"_vs_",j,".pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts")
    }
  }
}


#Plot log N:
counter=0
for (i in levels(perpop_counts$population)) {
  counter=which(levels(perpop_counts$population)==i)
  if (counter < length(levels(perpop_counts$population))) {
    for (j in levels(perpop_counts$population)[c((counter+1):length(levels(perpop_counts$population)))]) {
    print(paste0(i," vs ",j))
    i_vs_j_counts <- full_join(filter(perpop_counts,population==i),filter(perpop_counts,population==j),by=c("feature"),suffix=c(paste0("_",i),paste0("_",j))) %>% select(-contains("population"))
    i_vs_j_counts[paste0("log_N_",i)] <- log(i_vs_j_counts[,which(colnames(i_vs_j_counts)==paste0("N_",i))])
    i_vs_j_counts[paste0("log_N_",j)] <- log(i_vs_j_counts[,which(colnames(i_vs_j_counts)==paste0("N_",j))])
    i_vs_j_counts <- i_vs_j_counts %>% filter(!grepl("/",feature)&!grepl("_V",feature)&!grepl("coding",feature))
    i_vs_j_ggplot <- ggplot(data=as.data.frame(i_vs_j_counts)) +
      geom_point(aes_string(x=paste0("log_N_",i),y=paste0("log_N_",j),colour="feature")) +
      ggtitle(paste0(i, " vs ",j)) +
      xlab(paste0("log_N_",i)) +
      ylab(paste0("log_N_",j)) +
      geom_abline() +
      theme_bw() +
      theme(text=element_text(size=12,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=16),
          #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
          #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          #panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position=c(0.18,0.78),
          legend.title=element_blank()
      )
    i_vs_j_ggplot
    ggsave(paste0("derived_allele_counts_",i,"_vs_",j,".pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts")
    }
  }
}

```

#2B: Homozygous variant level.
##At the individual level (no depth threshold)
###Get derived allele homozygous counts.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_individual_summary_homozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_H\tintergenic_H\tintronic_H\tcoding_H\tsynonymous_H\tsyn_pref_H\tsyn_unpref_H\tmissense_H\tmistol_H\tmisdel_H\tnonsense_H\tUCNE_H\tUCNE_low_H\tUCNE_mid_H\tUCNE_high_H" > ${CALLING}"_ann_individual_summary_homozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_H=$(grep -v '#' ${i} | grep ';AF=1.00;' | wc -l)
  INTERGENIC_H=$(grep 'intergenic' ${i} | grep ';AF=1.00;' | wc -l)
  INTRONIC_H=$(grep 'intron_variant' ${i} | grep ';AF=1.00;' | wc -l)
  CODING_H=$(grep 'CDS' ${i} | grep ';AF=1.00;' | wc -l)
  SYNONYMOUS_H=$(grep 'synonymous_variant' ${i} | grep ';AF=1.00;' | wc -l)
  SYN_PREF_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed | grep ';AF=1.00;' | wc -l)
  SYN_UNPREF_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed | grep ';AF=1.00;' | wc -l)
  MISSENSE_H=$(grep 'missense_variant' ${i} | grep ';AF=1.00;' | wc -l)
  MISTOL_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep ';AF=1.00;' | wc -l)
  MISDEL_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep ';AF=1.00;' | wc -l)
  NONSENSE_H=$(grep '|HIGH|' ${i} | grep ';AF=1.00;' | wc -l)
  UCNE_H=$(grep 'UCNE' ${i} | grep ';AF=1.00;' | wc -l)
  UCNE_LOW_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed | grep ';AF=1.00;' | wc -l)
  UCNE_MID_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed | grep ';AF=1.00;' | wc -l)
  UCNE_HIGH_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed | grep ';AF=1.00;' | wc -l)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_H\t$INTERGENIC_H\t$INTRONIC_H\t$CODING_H\t$SYNONYMOUS_H\t$SYN_PREF_H\t$SYN_UNPREF_H\t$MISSENSE_H\t$MISTOL_H\t$MISDEL_H\t$NONSENSE_H\t$UCNE_H\t$UCNE_LOW_H\t$UCNE_MID_H\t$UCNE_HIGH_H" >> ${CALLING}"_ann_individual_summary_homozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/
unset SSHPASS

```

###Plot individual homozygous proportion.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

path <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/"

homoz_df <- read_tsv(paste0(path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_homozygous_varssubs_SNP.lr_ann.txt"),col_names=T) %>% select(.,species,population,dataset,sample,contains("_H")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) 

all_df <- read_tsv(paste0(path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_varssubs_SNP.lr_ann.txt"),col_names=T) %>% select(.,species,population,dataset,sample,contains("_V"),-contains("/")) %>% rename_at(vars(ends_with("_V")),funs(gsub("_V","",.))) 

homoz_prop_df <- cbind(homoz_df[c(1:4)],round(homoz_df[-c(1:4)]/all_df[-c(1:4)],3))
homoz_prop_df

homoz_prop_tidy <- homoz_prop_df %>% gather(feature,homoz_prop,-species,-population,-dataset,-sample,factor_key=T) %>% mutate(heteroz_prop = 1 - homoz_prop)
homoz_prop_tidy

all_prop_tidy <- homoz_prop_tidy %>% gather(proportion,value,-species,-population,-dataset,-sample,-feature,factor_key=T)

all_prop_tidy$population = factor(all_prop_tidy$population,levels=c("ki","no","po","sm","do"))
levels(all_prop_tidy$population)[levels(all_prop_tidy$population)=="sm"] <- "an"
all_prop_tidy$feature = factor(all_prop_tidy$feature,levels=c("total","intergenic","intronic","coding","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
levels(all_prop_tidy$feature) <- c("total","intergenic","intronic","coding","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")
all_prop_tidy$proportion = factor(all_prop_tidy$proportion,levels=c("heteroz_prop","homoz_prop"))


all_prop_tidy_subset <- filter(all_prop_tidy,feature=="intronic" | feature=="synonymous" | feature=="mistol" | feature=="misdel" | feature=="nonsense")


ggplot_homoz_prop <- ggplot(data=filter(all_prop_tidy_subset,dataset=="5x"), aes(fill=proportion, y=value, x=feature)) + 
  geom_bar(position="fill", stat="identity") +
  facet_wrap(population ~ sample) +
  ylab("Proportion of sites with a derived allele that are in homozygous state") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=90,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      #legend.position=c(0.9,0.84),
      legend.title=element_blank()
  )
ggplot_homoz_prop
ggsave("individual_homozygous_derived_proportion_new.pdf", width=40, height=50, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous")

```

##At the individual level DD/(AA+AD+DD).
###Extract counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_homozygous_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.all.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.pre.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.unp.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.tol.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.del.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.ucl.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.ucm.table"
bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.uch.table"

rm ${CALLING}"_ann_individual_homozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_H\ttotal_HP\tintergenic_V\tintergenic_H\tintergenic_HP\tintronic_V\tintronic_H\tintronic_HP\tsynonymous_V\tsynonymous_H\tsynonymous_HP\tsynpref_V\tsynpref_H\tsynpref_HP\tsynunpref_V\tsynunpref_H\tsynunpref_HP\tmissense_V\tmissense_H\tmissense_HP\tmistol_V\tmistol_H\tmistol_HP\tmisdel_V\tmisdel_H\tmisdel_HP\tnonsense_V\tnonsense_H\tnonsense_HP\tUCNE_V\tUCNE_H\tUCNE_HP\tUCNElow_V\tUCNElow_H\tUCNElow_HP\tUCNEmid_V\tUCNEmid_H\tUCNEmid_HP\tUCNEhigh_V\tUCNEhigh_H\tUCNEhigh_HP" > ${CALLING}"_ann_individual_homozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  #echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "counting variants from individual" ${ind}
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  FILE=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.all.table")
  FILE_PRE=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.pre.table")
  FILE_UNP=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.unp.table")
  FILE_TOL=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.tol.table")
  FILE_DEL=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.del.table")
  FILE_UCL=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.ucl.table")
  FILE_UCM=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.ucm.table")
  FILE_UCH=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.uch.table")
  GENO_COLUMN=$(head -n1 $FILE | tr "\t" "\n" | grep -n $ind":GT" | cut -d':' -f1) #get index for the column storing the current individual's genotypes
  DEPTH_COLUMN=$(head -n1 $FILE | tr "\t" "\n" | grep -n $ind":DP" | cut -d':' -f1) #get index for the column storing the current individual's depth
  TOTAL_V=$(grep -v '#' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  TOTAL_H=$(grep -v '#' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  TOTAL_HP=$(echo "scale=4; $TOTAL_H/$TOTAL_V" | bc)
  INTERGENIC_V=$(grep 'intergenic' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTERGENIC_H=$(grep 'intergenic' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTERGENIC_HP=$(echo "scale=4; $INTERGENIC_H/$INTERGENIC_V" | bc)
  INTRONIC_V=$(grep 'intron_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTRONIC_H=$(grep 'intron_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTRONIC_HP=$(echo "scale=4; $INTRONIC_H/$INTRONIC_V" | bc)
  SYNONYMOUS_V=$(grep 'synonymous_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNONYMOUS_H=$(grep 'synonymous_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNONYMOUS_HP=$(echo "scale=4; $SYNONYMOUS_H/$SYNONYMOUS_V" | bc)
  SYNPRE_V=$(cat $FILE_PRE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNPRE_H=$(cat $FILE_PRE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNPRE_HP=$(echo "scale=4; $SYNPRE_H/$SYNPRE_V" | bc)
  SYNUNP_V=$(cat $FILE_UNP | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNUNP_H=$(cat $FILE_UNP | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNUNP_HP=$(echo "scale=4; $SYNUNP_H/$SYNUNP_V" | bc)
  MISSENSE_V=$(grep 'missense_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISSENSE_H=$(grep 'missense_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISSENSE_HP=$(echo "scale=4; $MISSENSE_H/$MISSENSE_V" | bc)
  MISTOL_V=$(cat $FILE_TOL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISTOL_H=$(cat $FILE_TOL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISTOL_HP=$(echo "scale=4; $MISTOL_H/$MISTOL_V" | bc)
  MISDEL_V=$(cat $FILE_DEL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISDEL_H=$(cat $FILE_DEL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISDEL_HP=$(echo "scale=4; $MISDEL_H/$MISDEL_V" | bc)
  NONSENSE_V=$(grep '|HIGH|' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  NONSENSE_H=$(grep '|HIGH|' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  NONSENSE_HP=$(echo "scale=4; $NONSENSE_H/$NONSENSE_V" | bc)
  UCNE_V=$(grep 'UCNE' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNE_H=$(grep 'UCNE' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNE_HP=$(echo "scale=4; $UCNE_H/$UCNE_V" | bc)
  UCNELOW_V=$(cat $FILE_UCL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNELOW_H=$(cat $FILE_UCL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNELOW_HP=$(echo "scale=4; $UCNELOW_H/$UCNELOW_V" | bc)
  UCNEMID_V=$(cat $FILE_UCM | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNEMID_H=$(cat $FILE_UCM | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNEMID_HP=$(echo "scale=4; $UCNEMID_H/$UCNEMID_V" | bc)
  UCNEHIGH_V=$(cat $FILE_UCH | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 0) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNEHIGH_H=$(cat $FILE_UCH | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 0) && ($geno_col == "1/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNEHIGH_HP=$(echo "scale=4; $UCNEHIGH_H/$UCNEHIGH_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_H\t$TOTAL_HP\t$INTERGENIC_V\t$INTERGENIC_H\t$INTERGENIC_HP\t$INTRONIC_V\t$INTRONIC_H\t$INTRONIC_HP\t$SYNONYMOUS_V\t$SYNONYMOUS_H\t$SYNONYMOUS_HP\t$SYNPRE_V\t$SYNPRE_H\t$SYNPRE_HP\t$SYNUNP_V\t$SYNUNP_H\t$SYNUNP_HP\t$MISSENSE_V\t$MISSENSE_H\t$MISSENSE_HP\t$MISTOL_V\t$MISTOL_H\t$MISTOL_HP\t$MISDEL_V\t$MISDEL_H\t$MISDEL_HP\t$NONSENSE_V\t$NONSENSE_H\t$NONSENSE_HP\t$UCNE_V\t$UCNE_H\t$UCNE_HP\t$UCNELOW_V\t$UCNELOW_H\t$UCNELOW_HP\t$UCNEMID_V\t$UCNEMID_H\t$UCNEMID_HP\t$UCNEHIGH_V\t$UCNEHIGH_H\t$UCNEHIGH_HP" >> ${CALLING}"_ann_individual_homozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_homozygous_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/

```

###Plot population average homozygous raw.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

var_type <- c("varssubs") #variants varssubs

homozygous_raw <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_homozygous_",var_type,"_SNP.lr_ann.txt"),col_names=T) %>% select(.,species,population,dataset,sample,ends_with("_H")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% gather(feature,homoz_raw,-species,-population,-dataset,-sample,factor_key=T)
homozygous_raw$species = factor(homozygous_raw$species,levels=c("ll","lp"))
homozygous_raw$population = factor(homozygous_raw$population,levels=c("ki","no","po","sm","do"))

se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_homozygous_raw <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_homo_raw"=character(0),"se_homo_raw"=character(0)) #next, create the empty dataframe

for (pop in unique(homozygous_raw$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(homozygous_raw,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique() %>% as.character()
  for (r in unique(homozygous_raw$feature)) {
    print(r)
    pop_mean <- filter(homozygous_raw,feature==r & population==pop & dataset=="5x") %>% select(homoz_raw) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(homozygous_raw,feature==r & population==pop & dataset=="5x") %>% select(homoz_raw) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_homo_raw","se_homo_raw")
    average_homozygous_raw <- rbind(average_homozygous_raw,row_data,stringsAsFactors=F)
  }
}
average_homozygous_raw

average_homozygous_raw$population = factor(average_homozygous_raw$population,levels=c("ki","no","po","sm","do"))
average_homozygous_raw$feature = factor(average_homozygous_raw$feature,levels=c("total","intergenic","intronic","CDS","synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
average_homozygous_raw$avg_homo_raw <- as.double(average_homozygous_raw$avg_homo_raw)
average_homozygous_raw$se_homo_raw <- as.double(average_homozygous_raw$se_homo_raw)
average_homozygous_raw


homoz_raw_ggplot <- ggplot(data=filter(average_homozygous_raw,grepl('intronic|synonymous|missense|mistol|misdel|nonsense',feature)),aes(x=population,y=avg_homo_raw,colour=population)) +
  facet_wrap(. ~ feature, scales="free_y", nrow=1) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_homo_raw-se_homo_raw, ymax=avg_homo_raw+se_homo_raw), position=position_dodge(), width=0.5) +
  xlab("Features") +
  ylab("Individual derived homozygote count") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
homoz_raw_ggplot

ggsave(paste0(var_type,"_individual_homozygous_derived_proportion_raw.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous")

```

###Plot individual homozygous proportion.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

var_type <- c("varssubs") #variants varssubs

homozygous_proportion <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_homozygous_",var_type,"_SNP.lr_ann.txt"),col_names=T) %>% select(.,population,sample,contains("_HP")) %>% rename_at(vars(ends_with("_HP")),funs(gsub("_HP","",.))) %>% gather(feature,homoz_prop,-population,-sample,factor_key=T)
homozygous_proportion$population <- as.factor(homozygous_proportion$population)
homozygous_proportion$population = factor(homozygous_proportion$population,levels=c("ki","no","po","sm","do"))

homoz_prop_ggplot <- ggplot(data=as.data.frame(homozygous_proportion),aes(x=feature,y=homoz_prop,colour=population)) +
  geom_boxplot() +
  #geom_point(position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous variants (DP ≥ 4)") +
  xlab("Features") +
  ylab(paste("Proportion of derived homozygous\n",var_type,"(DP > 3)")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.06,0.15),
      legend.title=element_blank()
  )
homoz_prop_ggplot
ggsave(paste0(var_type,"_individual_homozygous_derived_proportion_boxplot_dp4.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous")

derived_allele_homo_prop_ggplot <- ggplot(data=as.data.frame(homozygous_proportion),aes(x=population,y=homoz_prop,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  scale_y_continuous(breaks = seq(0.12, 0.35, by = 0.06)) +
  ylim(0.12,0.35) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("Derived homozygous proportion") +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_homo_prop_ggplot
ggsave(paste0(type,"_derived_allele_homozygous_proportion.pdf"), width=15, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

##Heterozygosity at the individual level AD/(AA+AD+DD).
###Extract counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_heterozygous_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

#cd $V_PATH/$CALLING/annotation
#bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.all.table"
#bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.tol.table"
#bedtools intersect -a ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf" -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt -header | bcftools query -H -f '%INFO/ANN[\t%GT\t%DP]\n' > ${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.del.table"

rm ${CALLING}"_ann_individual_heterozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_H\ttotal_HP\tintergenic_V\tintergenic_H\tintergenic_HP\tintronic_V\tintronic_H\tintronic_HP\tsynonymous_V\tsynonymous_H\tsynonymous_HP\tmissense_V\tmissense_H\tmissense_HP\tmistol_V\tmistol_H\tmistol_HP\tmisdel_V\tmisdel_H\tmisdel_HP\tnonsense_V\tnonsense_H\tnonsense_HP\tUCNE_V\tUCNE_H\tUCNE_HP" > ${CALLING}"_ann_individual_heterozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" ! -path '*testing_variant_numbers*' -print`))
for i in "${INDLIST[@]}"
  do
  #echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "counting variants from individual" ${ind}
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  FILE=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.all.table")
  FILE_TOL=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.tol.table")
  FILE_DEL=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.del.table")
  GENO_COLUMN=$(head -n1 $FILE | tr "\t" "\n" | grep -n $ind":GT" | cut -d':' -f1) #get index for the column storing the current individual's genotypes
  DEPTH_COLUMN=$(head -n1 $FILE | tr "\t" "\n" | grep -n $ind":DP" | cut -d':' -f1) #get index for the column storing the current individual's depth
  TOTAL_V=$(grep -v '#' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  TOTAL_H=$(grep -v '#' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  TOTAL_HP=$(echo "scale=4; $TOTAL_H/$TOTAL_V" | bc)
  INTERGENIC_V=$(grep 'intergenic' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTERGENIC_H=$(grep 'intergenic' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTERGENIC_HP=$(echo "scale=4; $INTERGENIC_H/$INTERGENIC_V" | bc)
  INTRONIC_V=$(grep 'intron_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTRONIC_H=$(grep 'intron_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  INTRONIC_HP=$(echo "scale=4; $INTRONIC_H/$INTRONIC_V" | bc)
  SYNONYMOUS_V=$(grep 'synonymous_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNONYMOUS_H=$(grep 'synonymous_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  SYNONYMOUS_HP=$(echo "scale=4; $SYNONYMOUS_H/$SYNONYMOUS_V" | bc)
  MISSENSE_V=$(grep 'missense_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISSENSE_H=$(grep 'missense_variant' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISSENSE_HP=$(echo "scale=4; $MISSENSE_H/$MISSENSE_V" | bc)
  MISTOL_V=$(cat $FILE_TOL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISTOL_H=$(cat $FILE_TOL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISTOL_HP=$(echo "scale=4; $MISTOL_H/$MISTOL_V" | bc)
  MISDEL_V=$(cat $FILE_DEL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISDEL_H=$(cat $FILE_DEL | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  MISDEL_HP=$(echo "scale=4; $MISDEL_H/$MISDEL_V" | bc)
  NONSENSE_V=$(grep '|HIGH|' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  NONSENSE_H=$(grep '|HIGH|' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  NONSENSE_HP=$(echo "scale=4; $NONSENSE_H/$NONSENSE_V" | bc)
  UCNE_V=$(grep 'UCNE' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if($dp_col >= 4) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNE_H=$(grep 'UCNE' $FILE | awk -v geno_col=$GENO_COLUMN -v dp_col=$DEPTH_COLUMN -F "\t" '{ if(($dp_col >= 4) && ($geno_col == "0/1")) {printf "%s\t%s\n", $geno_col, $dp_col} }' | wc -l)
  UCNE_HP=$(echo "scale=4; $UCNE_H/$UCNE_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_H\t$TOTAL_HP\t$INTERGENIC_V\t$INTERGENIC_H\t$INTERGENIC_HP\t$INTRONIC_V\t$INTRONIC_H\t$INTRONIC_HP\t$SYNONYMOUS_V\t$SYNONYMOUS_H\t$SYNONYMOUS_HP\t$MISSENSE_V\t$MISSENSE_H\t$MISSENSE_HP\t$MISTOL_V\t$MISTOL_H\t$MISTOL_HP\t$MISDEL_V\t$MISDEL_H\t$MISDEL_HP\t$NONSENSE_V\t$NONSENSE_H\t$NONSENSE_HP\t$UCNE_V\t$UCNE_H\t$UCNE_HP" >> ${CALLING}"_ann_individual_heterozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_heterozygous_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/

```

###Plot individual heterozygous proportion.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

var_type <- c("varssubs") #variants varssubs

heterozygous_proportion <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_heterozygous_",var_type,"_SNP.lr_ann.txt"),col_names=T) %>% select(.,population,sample,contains("_HP")) %>% rename_at(vars(ends_with("_HP")),funs(gsub("_HP","",.))) %>% gather(feature,heteroz_prop,-population,-sample,factor_key=T)
heterozygous_proportion$population <- as.factor(heterozygous_proportion$population)
heterozygous_proportion$population = factor(heterozygous_proportion$population,levels=c("ki","no","po","sm","do"))

heteroz_prop_ggplot <- ggplot(data=as.data.frame(heterozygous_proportion),aes(x=feature,y=heteroz_prop,colour=population)) +
  geom_boxplot() +
  #geom_point(position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived heterozygous variants (DP ≥ 4)") +
  xlab("Features") +
  ylab(paste("Proportion of derived heterozygous\n",var_type,"(DP > 3)")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.06,0.15),
      legend.title=element_blank()
  )
heteroz_prop_ggplot
ggsave(paste0(var_type,"_individual_heterozygous_derived_proportion_boxplot_dp4.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous")

derived_allele_hetero_prop_ggplot <- ggplot(data=as.data.frame(heterozygous_proportion),aes(x=population,y=heteroz_prop,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  scale_y_continuous(breaks = seq(0.01, 0.15, by = 0.03)) +
  ylim(0.01,0.15) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("Derived heterozygous proportion") +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_hetero_prop_ggplot
ggsave(paste0(type,"_derived_allele_heterozygous_proportion.pdf"), width=15, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

##At the individual level DD/(AD+DD).
###Extract counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(substitutions) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_homozygous_bis_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)


S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation

rm ${CALLING}"_ann_individual_homozygous_bis_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_H\ttotal_HP\tintergenic_V\tintergenic_H\tintergenic_HP\tintronic_V\tintronic_H\tintronic_HP\tsynonymous_V\tsynonymous_H\tsynonymous_HP\tmissense_V\tmissense_H\tmissense_HP\tmistol_V\tmistol_H\tmistol_HP\tmisdel_V\tmisdel_H\tmisdel_HP\tnonsense_V\tnonsense_H\tnonsense_HP\tUCNE_V\tUCNE_H\tUCNE_HP" > ${CALLING}"_ann_individual_homozygous_bis_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" ! -path '*testing_variant_numbers*' -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_H=$(grep -v '#' ${i} | grep ';AF=1.00;' | wc -l)
  TOTAL_HP=$(echo "scale=4; $TOTAL_H/$TOTAL_V" | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_H=$(grep ';AF=1.00;.*intergenic' ${i} | wc -l)
  INTERGENIC_HP=$(echo "scale=4; $INTERGENIC_H/$INTERGENIC_V" | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_H=$(grep ';AF=1.00;.*intron_variant' ${i} | wc -l)
  INTRONIC_HP=$(echo "scale=4; $INTRONIC_H/$INTRONIC_V" | bc)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_H=$(grep ';AF=1.00;.*synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_HP=$(echo "scale=4; $SYNONYMOUS_H/$SYNONYMOUS_V" | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_H=$(grep ';AF=1.00;.*missense_variant' ${i} | wc -l)
  MISSENSE_HP=$(echo "scale=4; $MISSENSE_H/$MISSENSE_V" | bc)
  MISTOL_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep -v '#' | wc -l)
  MISTOL_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep ';AF=1.00;'| wc -l)
  MISTOL_HP=$(echo "scale=4; $MISTOL_H/$MISTOL_V" | bc)
  MISDEL_V=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep -v '#' | wc -l)
  MISDEL_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep ';AF=1.00;'| wc -l)
  MISDEL_HP=$(echo "scale=4; $MISDEL_H/$MISDEL_V" | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_H=$(grep ';AF=1.00;.*|HIGH|' ${i} | wc -l)
  NONSENSE_HP=$(echo "scale=4; $NONSENSE_H/$NONSENSE_V" | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_H=$(grep ';AF=1.00;.*UCNE' ${i} | wc -l)
  UCNE_HP=$(echo "scale=4; $UCNE_H/$UCNE_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_H\t$TOTAL_HP\t$INTERGENIC_V\t$INTERGENIC_H\t$INTERGENIC_HP\t$INTRONIC_V\t$INTRONIC_H\t$INTRONIC_HP\t$SYNONYMOUS_V\t$SYNONYMOUS_H\t$SYNONYMOUS_HP\t$MISSENSE_V\t$MISSENSE_H\t$MISSENSE_HP\t$MISTOL_V\t$MISTOL_H\t$MISTOL_HP\t$MISDEL_V\t$MISDEL_H\t$MISDEL_HP\t$NONSENSE_V\t$NONSENSE_H\t$NONSENSE_HP\t$UCNE_V\t$UCNE_H\t$UCNE_HP" >> ${CALLING}"_ann_individual_homozygous_bis_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(substitutions) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_homozygous_bis_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/

```

###Plot individual homozygous proportion.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

var_type <- c("variants") #varssubs #variants #substitutions

homozygous_proportion <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_homozygous_bis_",var_type,"_SNP.lr_ann.txt"),col_names=T) %>% select(.,population,sample,contains("_HP")) %>% rename_at(vars(ends_with("_HP")),funs(gsub("_HP","",.))) %>% gather(feature,homoz_prop,-population,-sample,factor_key=T)
homozygous_proportion$population <- as.factor(homozygous_proportion$population)
homozygous_proportion$population = factor(homozygous_proportion$population,levels=c("ki","no","po","sm","do"))

homoz_prop_ggplot <- ggplot(data=as.data.frame(homozygous_proportion),aes(x=feature,y=homoz_prop,colour=population)) +
  geom_boxplot() +
  #geom_point(position=position_dodge(width=0.5)) +
  #geom_line() +
  #ggtitle("Proportion of derived homozygous variants (DP ≥ 4)") +
  xlab("Features") +
  ylab(paste("Proportion of derived homozygous\n",var_type,"(DP > 3)")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      #axis.line=element_line(colour="black"),
      axis.title=element_text(size=14),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.06,0.15),
      legend.title=element_blank()
  )
homoz_prop_ggplot
ggsave(paste0(var_type,"_individual_homozygous_derived_proportion_bis_boxplot.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous")

```

##At the population level DD/(AD+DD). Subsampled populations (N=8) will be used.
###Extract counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_ll_ki_c_ll_no_c_ll_po_nm3_origcov) #write down name of the calling
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${TYPE}"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}
script "${CALLING}_ann_population_homozygous_${TYPE}.lr_ann.log"
CALLING=${STY#*.}
CALLING=${CALLING%-*}
TYPE=${STY#*-}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_population_homozygous_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\ttotal_V\ttotal_H\ttotal_HP\tintergenic_V\tintergenic_H\tintergenic_HP\tintronic_V\tintronic_H\tintronic_HP\tsynonymous_V\tsynonymous_H\tsynonymous_HP\tmissense_V\tmissense_H\tmissense_HP\tnonsense_V\tnonsense_H\tnonsense_HP\tUCNE_V\tUCNE_H\tUCNE_HP" > ${CALLING}"_ann_population_homozygous_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_subsample_n008_"${TYPE}".lr_ann.vcf" -print`))
for i in "${POPLIST[@]}"
  do
  echo "${i}"
  vcf=$(echo "${i}" | rev | cut -d'/' -f1 | rev)
  echo "${vcf}"
  SPECIES=$(echo "${vcf}" | cut -c3-4)
  POPULATION=$(echo "${vcf}" | cut -c14-15)
  DATASET=$(echo "${vcf}" | cut -c6-7)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_H=$(grep -v '#' ${i} | grep ';AF=1.00;' | wc -l)
  TOTAL_HP=$(echo "scale=4; $TOTAL_H/($TOTAL_V+$TOTAL_H)" | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_H=$(grep ';AF=1.00;.*intergenic' ${i} | wc -l)
  INTERGENIC_HP=$(echo "scale=4; $INTERGENIC_H/($INTERGENIC_V+$INTERGENIC_H)" | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_H=$(grep ';AF=1.00;.*intron_variant' ${i} | wc -l)
  INTRONIC_HP=$(echo "scale=4; $INTRONIC_H/($INTRONIC_V+$INTRONIC_H)" | bc)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_H=$(grep ';AF=1.00;.*synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_HP=$(echo "scale=4; $SYNONYMOUS_H/($SYNONYMOUS_V+$SYNONYMOUS_H)" | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_H=$(grep ';AF=1.00;.*missense_variant' ${i} | wc -l)
  MISSENSE_HP=$(echo "scale=4; $MISSENSE_H/($MISSENSE_V+$MISSENSE_H)" | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  NONSENSE_H=$(grep ';AF=1.00;.*|HIGH|' ${i} | wc -l)
  NONSENSE_HP=$(echo "scale=4; $NONSENSE_H/($NONSENSE_V+$NONSENSE_H)" | bc)
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  UCNE_H=$(grep ';AF=1.00;.*UCNE' ${i} | wc -l)
  UCNE_HP=$(echo "scale=4; $UCNE_H/($UCNE_V+$UCNE_H)" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$TOTAL_V\t$TOTAL_H\t$TOTAL_HP\t$INTERGENIC_V\t$INTERGENIC_H\t$INTERGENIC_HP\t$INTRONIC_V\t$INTRONIC_H\t$INTRONIC_HP\t$SYNONYMOUS_V\t$SYNONYMOUS_H\t$SYNONYMOUS_HP\t$MISSENSE_V\t$MISSENSE_H\t$MISSENSE_HP\t$NONSENSE_V\t$NONSENSE_H\t$NONSENSE_HP\t$UCNE_V\t$UCNE_H\t$UCNE_HP" >> ${CALLING}"_ann_population_homozygous_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_ll_ki_c_ll_no_c_ll_po_nm3_origcov) #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_nm2_origcov
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_homozygous*.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/

```

###Plot population homozygous proportion.
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

calling_counts_files <- grep(list.files("/Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/", pattern="*population_homozygous_SNP.lr_ann.txt"),pattern='nm2nm3',inv=T,value=T)
homozygous_proportion <- data_frame()
for (file in calling_counts_files) {
  calling_counts <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous/",file),col_names=T) %>% select(.,population,contains("_HP")) %>% rename_at(vars(ends_with("_HP")),funs(gsub("_HP","",.))) %>% gather(feature,homoz_prop,-population,factor_key=T)
  homozygous_proportion <- rbind(homozygous_proportion,calling_counts)
}
homozygous_proportion$population <- as.factor(homozygous_proportion$population)
homozygous_proportion$population = factor(homozygous_proportion$population,levels=c("ki","no","po","sm","do"))

homoz_prop_ggplot <- ggplot(data=as.data.frame(homozygous_proportion),aes(x=feature,y=homoz_prop,colour=population,group=population)) +
  geom_point() +
  geom_line() +
  #¢ggtitle("Proportion of derived homozygous sites") +
  xlab("Features") +
  ylab("Proportion of derived homozygous variants") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.9,0.84),
      legend.title=element_blank()
  )
homoz_prop_ggplot
ggsave("population_homozygous_derived_proportion.pdf", width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_homozygous")


```

#2C: derived allele count spectrum.
####Combine per population and feature allele count files:
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions #segregating
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_derived_counts_spectrum_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
 
cd /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered
N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
SPECIES=$(echo $CALLING | fold -w8 | cut -c1-4 | head -n$N_POPS | sort | uniq)
DATASETS=$(for i in ${SPECIES[@]}; do ls ${i}*_samples | cut -d'_' -f1,2,3; done)
COVERAGE=$(echo "${CALLING}" | rev | cut -d'_' -f1 | rev)
NM_COV=$(echo "${CALLING}" | rev | cut -d'_' -f1,2 | rev)

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
mkdir -p derived_allele_counts_spectrum
GREP=$(cat derived_allele_counts_spectrum/features_derived_allele_freq_ratio.txt | cut -f1)
POP_VCFS=($(ls `find . -path *"_perpop/c_*5x*_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
rm ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.ac"
for p in ${POP_VCFS[@]}
  do
  SP_DATASET=$(echo ${p} | cut -d'/' -f3 | cut -d'_' -f1-3)
  SHORT_POP=$(echo ${p} | cut -d'/' -f3 | cut -d'_' -f4-6)
  echo "working with pop" $SHORT_POP "and vcf" $p
  echo "extracting derived allele counts from" $SHORT_POP
  for g in ${GREP[@]}
    do
    #echo "grepping: ${g}"
    FEATURE=$(grep "${g}" derived_allele_counts_spectrum/features_derived_allele_freq_ratio.txt | cut -f2)
    echo "extracting from feature:" $FEATURE
    grep "${g}" "${p}" | awk -F"\t|;|=" '{printf ("%s\t%s\t%s\t%s\n", $1,$2-1,$2,$11)}' > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.ac}
    echo "joining frequency files"
    awk -v pop="$SHORT_POP" -v feat="$FEATURE" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_"$VAR"_"$FEATURE"_"$TYPE".lr_ann.ac" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.ac"
    if [ $FEATURE == "NSYN" ]
      then
      FEATURE2=(NTOL)
      echo "extracting from feature:" $FEATURE2
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.ac}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.ac}
      echo "joining frequency files"
      awk -v pop="$SHORT_POP" -v feat="$FEATURE2" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_"$VAR"_"$FEATURE2"_"$TYPE".lr_ann.ac" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.ac"
      FEATURE2=(NDEL)
      echo "extracting from feature:" $FEATURE2
      bedtools intersect -a <(awk -F"\t|_" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE}_${TYPE}.lr_ann.ac}) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${p/perpop_${VAR}_${TYPE}.lr_ann.vcf/perpop_${VAR}_${FEATURE2}_${TYPE}.lr_ann.ac}
      echo "joining frequency files"
      awk -v pop="$SHORT_POP" -v feat="$FEATURE2" '{print pop,feat,$0}' OFS="\t" $SHORT_POP"_"$NM_COV"_perpop"/$SP_DATASET"_"$SHORT_POP"_"$NM_COV"_perpop_"$VAR"_"$FEATURE2"_"$TYPE".lr_ann.ac" >> ${CALLING}"_5x_perpop_ALL_features_"${VAR}"_"${TYPE}".lr_ann.ac"
    fi
    done
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(segregating) #varssubs #variants #substitutions #segregating
TYPE=(SNP) #write down SNP or INDEL
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*5x_perpop_ALL_features_${VAR}.lr_ann.ac /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum/

```

####Plot spectra:
```{r}

library(readr)
library(dplyr)
library(ggplot2)

callings_ac_files <- intersect(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum/", pattern="*ALL_features*"),list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum/", pattern="_varssubs_SNP.|_variants_SNP.|_substitutions_SNP.|_segregating_SNP."))

#var_type <- c("varssubs","variants","substitutions")
var_type <- c("segregating")

for (type in var_type) {
  print(type)
  all_plot_data <- data_frame("pop"=character(0),"feat"=factor(0),"ac"=double(0),"n"=numeric(0),"n_bis"=numeric(0))
  calling_ac <- read_tsv(paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum/",callings_ac_files[grep(type,callings_ac_files)]),col_names=c("pop","feat","sc","start","stop","ac"))
  pop_size_data <- data_frame("pop"=unique(calling_ac$pop)) %>% mutate(size=ifelse(pop=="c_ll_ki" | pop=="c_lp_sm",12,8))
  min_size <- min(pop_size_data$size)
  for (p in unique(calling_ac$pop)) {
    print(p)
    p_size <- pop_size_data[which(pop_size_data$pop==p),2] %>% unlist(use.names=F)
    pop_ac_data <- filter(calling_ac,pop==p)
    #popsize=ifelse(p=="c_lp_sm"|p=="c_ll_ki",12,8)
    pop_plot_data <- pop_ac_data %>% group_by(ac,feat) %>% summarize(n=n(),pop=pop[n]) %>% arrange(feat) %>% mutate(n_bis=n*p_size/min_size) %>% select(pop,feat,ac,n,n_bis) %>% as.data.frame(.)
    #pop_ac_data %>% mutate(ac_groups=cut(ac,breaks=seq(0,1.1,by=1/(2*popsize)))) %>% group_by(ac_groups,feat) %>% summarize(n=n()) #problem with the binning due to rounded allele frequencies
    all_plot_data <- rbind(all_plot_data,pop_plot_data)
    }
  all_plot_data$feat <- as.factor(all_plot_data$feat)
  all_plot_data$feat = factor(all_plot_data$feat,levels=c("INTER","INTR","SYN","NSYN","NTOL","NDEL","LOF","UCNE")) #Reorder factor levels.
  levels(all_plot_data$feat) <- c("intergenic","introns","syn.","missense","m. tol.","m. del.","LoF","UCNE")
  all_plot_data$pop = factor(all_plot_data$pop,levels=c("c_ll_ki","c_ll_po","c_ll_no","c_lp_sm","c_lp_do")) #Reorder factor levels.
  levels(all_plot_data$pop) <- c("KIR","POL","NOR","AND","DON")
  all_plot_data <- all_plot_data %>% arrange(pop,feat,ac) %>% mutate(species=ifelse(pop=="KIR" | pop=="POL" | pop=="NOR","ll","lp"),size=ifelse(pop=="KIR" | pop=="AND","large","small"))
  all_plot_data$species = factor(all_plot_data$species,levels=c("ll","lp"))
  all_plot_data$size = factor(all_plot_data$size,levels=c("large","small"))
  all_plot_data
  
  all_nbis_ggplot <- ggplot(data=all_plot_data,aes(ac,n_bis,fill=interaction(species,size),alpha=interaction(species,size))) +
    geom_col() +
    facet_grid(feat~pop,scales="free",space="free_x") +
    #ggtitle(paste(type,"derived allele counts spectrum")) +
    ylab("Number of sites") +
    xlab("Derived allele count") +
    scale_fill_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
    scale_alpha_manual(values=c(1,1,0.4,0.4)) +
    scale_x_continuous(breaks=seq(4,24,by=4)) +
    theme_bw() +
    theme(text=element_text(size=9,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=10),
          axis.text.x=element_text(colour="black",face="bold"),
          axis.text.y=element_text(colour="black",face="bold"),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          panel.spacing=unit(0.075,"cm"),
          strip.background=element_rect(colour="black"),
          strip.text=element_text(size=8),
          panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          #plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position="none",
          legend.title=element_blank()
    )
    all_nbis_ggplot
    ggsave(paste0(type,"_ALL_derived_allele_counts_spectrum_nbis.pdf"), width=16.9, height=24, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum")

    all_n_ggplot <- ggplot(data=all_plot_data,aes(ac,n,fill=interaction(species,size),alpha=interaction(species,size))) +
    geom_col() +
    facet_grid(feat~pop,scales="free",space="free_x") +
    #ggtitle(paste(type,"derived allele counts spectrum")) +
    ylab("Number of sites") +
    xlab("Derived allele count") +
    scale_fill_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
    scale_alpha_manual(values=c(1,1,0.5,0.5)) +
    theme_bw() +
    theme(text=element_text(size=14,face="bold"),
          rect=element_rect(size=1),
          axis.line=element_line(colour="black"),
          axis.title=element_text(size=18),
          axis.text.x=element_text(angle=30,hjust=1,size=14,colour="black"),
          axis.text.y=element_text(size=14,colour="black"),
          #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
          panel.background=element_blank(),
          panel.border=element_rect(colour="black"),
          strip.background=element_rect(colour="black",size=1.5),
          strip.text=element_text(size=14),
          panel.grid=element_blank(),
          #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
          plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
          #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
          legend.background=element_rect(linetype="solid", colour="black", size=.5),
          #legend.justification=c(0,0),
          legend.key=element_rect(colour="white"),
          #legend.key.size=unit(1.3,"cm"),
          legend.position="none",
          legend.title=element_blank()
    )
    all_n_ggplot
    ggsave(paste0(type,"_ALL_derived_allele_counts_spectrum_n.pdf"), width=30, height=40, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum")

  for (f in unique(all_plot_data$feat)) {
    feat_nbis_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(ac,n_bis,fill=pop)) +
      geom_col() +
      facet_grid(~pop,scales="free_x",space="free_x") +
      ggtitle(paste(type,f,"derived allele counts spectrum")) +
      ylab("N") +
      xlab("AC") +
      theme_bw() +
      theme(text=element_text(size=14,face="bold"),
            rect=element_rect(size=1),
            axis.line=element_line(colour="black"),
            axis.title=element_text(size=18),
            axis.text.x=element_text(angle=30,hjust=1,size=14,colour="black"),
            axis.text.y=element_text(size=14,colour="black"),
            #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
            panel.background=element_blank(),
            panel.border=element_rect(colour="black"),
            strip.background=element_rect(colour="black",size=1.5),
            strip.text=element_text(size=14),
            panel.grid=element_blank(),
            #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
            plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
            #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
            legend.background=element_rect(linetype="solid", colour="black", size=.5),
            #legend.justification=c(0,0),
            legend.key=element_rect(colour="white"),
            #legend.key.size=unit(1.3,"cm"),
            legend.position="none",
            legend.title=element_blank()
      )
      feat_nbis_ggplot
      ggsave(paste0(type,"_",f,"_derived_allele_counts_spectrum_nbis.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum")
    
    feat_n_ggplot <- ggplot(data=filter(all_plot_data,feat==f),aes(ac,n,fill=pop)) +
      geom_col() +
      facet_grid(~pop,scales="free_x",space="free_x") +
      ggtitle(paste(type,f,"derived allele counts spectrum")) +
      ylab("N") +
      xlab("AC") +
      theme_bw() +
      theme(text=element_text(size=14,face="bold"),
            rect=element_rect(size=1),
            axis.line=element_line(colour="black"),
            axis.title=element_text(size=18),
            axis.text.x=element_text(angle=30,hjust=1,size=14,colour="black"),
            axis.text.y=element_text(size=14,colour="black"),
            #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
            panel.background=element_blank(),
            panel.border=element_rect(colour="black"),
            strip.background=element_rect(colour="black",size=1.5),
            strip.text=element_text(size=14),
            panel.grid=element_blank(),
            #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
            plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
            #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
            legend.background=element_rect(linetype="solid", colour="black", size=.5),
            #legend.justification=c(0,0),
            legend.key=element_rect(colour="white"),
            #legend.key.size=unit(1.3,"cm"),
            legend.position="none",
            legend.title=element_blank()
      )
      feat_n_ggplot
      ggsave(paste0(type,"_",f,"_derived_allele_counts_spectrum_n.pdf"), width=30, height=8, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/derived_allele_counts_spectrum")
  }
}

```

#2D: generate table of deleterious variants.
```{r Calculate derived allele frequency ratio, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation


grep '|HIGH|' ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.lr_ann.vcf > ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.LoF.lr_ann.vcf
bedtools intersect -a <(grep -E '#|missense_variant' ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.lr_ann.vcf) -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.misDEL.lr_ann.vcf

cat ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.LoF.lr_ann.vcf ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.misDEL.lr_ann.vcf | sort -k1,1 -k2,2n > ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.mDEL_and_LoF.lr_ann.vcf

awk -F"\t|;|=|\\\\|" '{printf ("%s\t%s\t%s\t%s\t%s\t%s\t%s\t%s\n", $1,$2,$4,$5,$13,$18,$20,$27)}' ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.mDEL_and_LoF.lr_ann.vcf > ${CALLING}_polarized_filteredall_${VAR}_${TYPE}.mDEL_and_LoF.lr_ann.table

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_polarized_filteredall_${VAR}_${TYPE}.mDEL_and_LoF.lr_ann.table /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/deleterious_db 
unset SSHPASS

```



#3: Diversity approaches.
##At the individual level.
###Get derived allele heterozygous counts.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(private) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_individual_summary_heterozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_H\tintergenic_H\tintronic_H\tcoding_H\tsynonymous_H\tsyn_pref_H\tsyn_unpref_H\tmissense_H\tmistol_H\tmisdel_H\tnonsense_H\tUCNE_H\tUCNE_low_H\tUCNE_mid_H\tUCNE_high_H" > ${CALLING}"_ann_individual_summary_heterozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_H=$(grep -v '#' ${i} | grep ';AF=0.500;' | wc -l)
  INTERGENIC_H=$(grep 'intergenic' ${i} | grep ';AF=0.500;' | wc -l)
  INTRONIC_H=$(grep 'intron_variant' ${i} | grep ';AF=0.500;' | wc -l)
  CODING_H=$(grep 'CDS' ${i} | grep ';AF=0.500;' | wc -l)
  SYNONYMOUS_H=$(grep 'synonymous_variant' ${i} | grep ';AF=0.500;' | wc -l)
  SYN_PREF_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed | grep ';AF=0.500;' | wc -l)
  SYN_UNPREF_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed | grep ';AF=0.500;' | wc -l)
  MISSENSE_H=$(grep 'missense_variant' ${i} | grep ';AF=0.500;' | wc -l)
  MISTOL_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep ';AF=0.500;' | wc -l)
  MISDEL_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep ';AF=0.500;' | wc -l)
  NONSENSE_H=$(grep '|HIGH|' ${i} | grep ';AF=0.500;' | wc -l)
  UCNE_H=$(grep 'UCNE' ${i} | grep ';AF=0.500;' | wc -l)
  UCNE_LOW_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed | grep ';AF=0.500;' | wc -l)
  UCNE_MID_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed | grep ';AF=0.500;' | wc -l)
  UCNE_HIGH_H=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed | grep ';AF=0.500;' | wc -l)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_H\t$INTERGENIC_H\t$INTRONIC_H\t$CODING_H\t$SYNONYMOUS_H\t$SYN_PREF_H\t$SYN_UNPREF_H\t$MISSENSE_H\t$MISTOL_H\t$MISDEL_H\t$NONSENSE_H\t$UCNE_H\t$UCNE_LOW_H\t$UCNE_MID_H\t$UCNE_HIGH_H" >> ${CALLING}"_ann_individual_summary_heterozygous_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(private_segregating) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/
unset SSHPASS

```

###Visualise derived allele heterozygous counts.

```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_het <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt"))

colnames(variants_het)[which(colnames(variants_het) == "mistol_H")] <- "missense_tol_H"
colnames(variants_het)[which(colnames(variants_het) == "misdel_H")] <- "missense_del_H"
variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
levels(variants_het_tidy$feature) <- gsub('.{2}$', '', levels(variants_het_tidy$feature))
variants_het_tidy

derived_allele_hetero_counts_ggplot <- ggplot(data=variants_het_tidy, aes(population,N,colour=population)) +
  facet_grid(feature ~ .,scales="free") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("N heterozygous variants") +
  theme_bw() +
  theme(strip.text=element_text(face="bold",size=9),
        text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  derived_allele_hetero_counts_ggplot
ggsave(paste0(type,"_derived_allele_heterozygous_counts.pdf"), width=15, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

###Calculate individual heterozygosity per category (joint calling; estimated joint totals from the variants only filters and from published proportions for CDS).
####All individuals.
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

#Define the following variables:
type="substitutions" #varssubs #variants #substitutions #segregating #private_segregating
plot_variable <- "het_by_total_filter" #het_by_total_filter #het_by_total (for a filtered or unfiltered denominator)

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_het <- read_tsv(paste0(wd_path,variant_files))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt") 
total_sites

#For coding categories, we'll follow the proportions estimated by Yamaguchi-Kagata et al (see here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561068/): 28.5% SYN, 68.1% NSYN, and 3.4% NONSENSE. From my own classification of SYN, 16.9% were PREF and 50.8% were UNPREF. From my own classification of NSYN, 60.8% were NTOL and 39.2% were NDEL. So the totals would be: 28.5% SYN (4.8% PREF, 14.5% UNPREF), 68.1% NSYN (41.4% NTOL, 26.7% NDEL), and 3.4% NONSENSE.
coding_proportions <- data_frame("coding_category"=c("synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense"),"proportion"=c(0.285,0.048,0.145,0.681,0.414,0.267,0.034)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(coding_proportions))) {
  coding_category <- unlist(coding_proportions[row,1],use.names=F)
  print(coding_category)
  proportion <- unlist(coding_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(coding_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

#For UCNE categories, I calculated the proportion of total sites under each GERP category in the ucne_database script (section named: "Calculate proportion of sites for each GERP category"). These are: 3.0% UCNE_low, 26.3% UCNE_mid, 70.7% UCNE_high.
ucne_proportions <- data_frame("ucne_category"=c("UCNE_low","UCNE_mid","UCNE_high"),"proportion"=c(0.030,0.263,0.707)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(ucne_proportions))) {
  ucne_category <- unlist(ucne_proportions[row,1],use.names=F)
  print(ucne_category)
  proportion <- unlist(ucne_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(ucne_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

total_sites$category <- factor(total_sites$category,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
total_sites <- total_sites[order(total_sites$category),]
#write_tsv(total_sites,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt"))

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("feature"="category")) %>% mutate(het_by_total=N/callable_N,het_by_total_filter=N/callable_N_postfilter)
variants_het_joined_all$feature = factor(variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
variants_het_joined_all

#Plot:
ggplot_all_pi <- ggplot(data=variants_het_joined_all,aes(x=population,y=colnames(variants_het_joined_all)[which(colnames(variants_het_joined_all) == plot_variable)],colour=population)) +
  geom_point() +
  facet_grid(feature ~ .) +
  xlab("Population") +
  ylab(paste(type," individual ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_all_pi
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate.pdf"), width=10, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")


#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_variants_het_joined_all <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_het_by_total"=character(0),"se_het_by_total"=character(0)) #next, create the empty dataframe

for (pop in unique(variants_het_joined_all$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(variants_het_joined_all,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(variants_het_joined_all$feature)) {
    print(r)
    pop_mean <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_het_by_total","se_het_by_total")
    average_variants_het_joined_all <- rbind(average_variants_het_joined_all,row_data,stringsAsFactors=F)
  }
}
average_variants_het_joined_all
average_variants_het_joined_all$population = factor(average_variants_het_joined_all$population,levels=c("ki","no","po","sm","do"))
average_variants_het_joined_all$feature = factor(average_variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
average_variants_het_joined_all$avg_het_by_total <- as.numeric(average_variants_het_joined_all$avg_het_by_total)
average_variants_het_joined_all$se_het_by_total <- as.numeric(average_variants_het_joined_all$se_het_by_total)
average_variants_het_joined_all
write_csv(average_variants_het_joined_all,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/",type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean.csv"))
  
#Plot these means and standard errors (categories with higher values):
average_ggplot_all_pi_part1 <- ggplot(data=filter(average_variants_het_joined_all,grepl('total|intergenic|intron|CDS|synonymous|syn_pref|syn_unpref|UCNE_low',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part1
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_part1.pdf"), width=10, height=28, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with intermediate values):
average_ggplot_all_pi_part2 <- ggplot(data=filter(average_variants_het_joined_all,grepl('missense|mistol|misdel|UCNE_mid',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part2
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_part2.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with low values):
average_ggplot_all_pi_part3 <- ggplot(data=filter(average_variants_het_joined_all,grepl('nonsense|\\<UCNE\\>|UCNE_high',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part3
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_part3.pdf"), width=10, height=12, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

average_ggplot_all_pi_combined <- grid.arrange(average_ggplot_all_pi_part1,average_ggplot_all_pi_part2,average_ggplot_all_pi_part3, ncol = 3, layout_matrix = cbind(c(rep(1,20)), c(rep(2,11),rep(NA,9)), c(rep(3,9),rep(NA,11))))
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_all.pdf"), width=35, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity",average_ggplot_all_pi_combined)

```

####All individuals: SYN and NSYN breakdowns relative to totals.
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

#Define the following variables:
type="varssubs" #varssubs #variants #substitutions #segregating #private_segregating
plot_variable <- "het_by_total_filter" #het_by_total_filter #het_by_total (for a filtered or unfiltered denominator)

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_het <- read_tsv(paste0(wd_path,variant_files))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt") 
total_sites

#For coding categories, we'll follow the proportions estimated by Yamaguchi-Kagata et al (see here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561068/): 28.5% SYN, 68.1% NSYN, and 3.4% NONSENSE. From my own classification of SYN, 16.9% were PREF and 50.8% were UNPREF. From my own classification of NSYN, 60.8% were NTOL and 39.2% were NDEL. So the totals would be: 28.5% SYN (4.8% PREF, 14.5% UNPREF), 68.1% NSYN (41.4% NTOL, 26.7% NDEL), and 3.4% NONSENSE.
coding_proportions <- data_frame("coding_category"=c("synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense"),"proportion"=c(0.285,0.285,0.285,0.681,0.681,0.681,0.034)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(coding_proportions))) {
  coding_category <- unlist(coding_proportions[row,1],use.names=F)
  print(coding_category)
  proportion <- unlist(coding_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(coding_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

#For UCNE categories, I calculated the proportion of total sites under each GERP category in the ucne_database script (section named: "Calculate proportion of sites for each GERP category"). These are: 3.0% UCNE_low, 26.3% UCNE_mid, 70.7% UCNE_high.
ucne_proportions <- data_frame("ucne_category"=c("UCNE_low","UCNE_mid","UCNE_high"),"proportion"=c(0.030,0.263,0.707)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(ucne_proportions))) {
  ucne_category <- unlist(ucne_proportions[row,1],use.names=F)
  print(ucne_category)
  proportion <- unlist(ucne_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(ucne_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

total_sites$category <- factor(total_sites$category,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
total_sites <- total_sites[order(total_sites$category),]
#write_tsv(total_sites,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt"))

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("feature"="category")) %>% mutate(het_by_total=N/callable_N,het_by_total_filter=N/callable_N_postfilter)
variants_het_joined_all$feature = factor(variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
variants_het_joined_all


#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_variants_het_joined_all <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_het_by_total"=character(0),"se_het_by_total"=character(0)) #next, create the empty dataframe

for (pop in unique(variants_het_joined_all$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(variants_het_joined_all,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(variants_het_joined_all$feature)) {
    print(r)
    pop_mean <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_het_by_total","se_het_by_total")
    average_variants_het_joined_all <- rbind(average_variants_het_joined_all,row_data,stringsAsFactors=F)
  }
}
average_variants_het_joined_all
average_variants_het_joined_all$population = factor(average_variants_het_joined_all$population,levels=c("ki","no","po","sm","do"))
average_variants_het_joined_all$feature = factor(average_variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
average_variants_het_joined_all$avg_het_by_total <- as.numeric(average_variants_het_joined_all$avg_het_by_total)
average_variants_het_joined_all$se_het_by_total <- as.numeric(average_variants_het_joined_all$se_het_by_total)
average_variants_het_joined_all
#write_csv(average_variants_het_joined_all,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/",type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean.csv"))
  
#Plot these means and standard errors (categories with higher values):
average_ggplot_SYN <- ggplot(data=filter(average_variants_het_joined_all,grepl('synonymous|syn_pref|syn_unpref',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_SYN
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_SYN.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

average_ggplot_MIS <- ggplot(data=filter(average_variants_het_joined_all,grepl('missense|mistol|misdel',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_MIS
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_MIS.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

average_ggplot_SYN_MIS_combined <- grid.arrange(average_ggplot_SYN,average_ggplot_MIS, ncol = 2)
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_SYN&MIS.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity",average_ggplot_SYN_MIS_combined)

```

####5x only.
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

#Define the following variables:
type="varssubs" #varssubs #variants #segregating #private_segregating
plot_variable <- "het_by_total_filter" #het_by_total_filter #het_by_total (for a filtered or unfiltered denominator)

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_het <- read_tsv(paste0(wd_path,variant_files)) %>% filter(dataset=="5x")

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt") 
total_sites

#For coding categories, we'll follow the proportions estimated by Yamaguchi-Kagata et al (see here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561068/): 28.5% SYN, 68.1% NSYN, and 3.4% NONSENSE. From my own classification of SYN, 16.9% were PREF and 50.8% were UNPREF. From my own classification of NSYN, 60.8% were NTOL and 39.2% were NDEL. So the totals would be: 28.5% SYN (4.8% PREF, 14.5% UNPREF), 68.1% NSYN (41.4% NTOL, 26.7% NDEL), and 3.4% NONSENSE.
coding_proportions <- data_frame("coding_category"=c("synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense"),"proportion"=c(0.285,0.048,0.145,0.681,0.414,0.267,0.034)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(coding_proportions))) {
  coding_category <- unlist(coding_proportions[row,1],use.names=F)
  print(coding_category)
  proportion <- unlist(coding_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(coding_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

#For UCNE categories, I calculated the proportion of total sites under each GERP category in the ucne_database script (section named: "Calculate proportion of sites for each GERP category"). These are: 3.0% UCNE_low, 26.3% UCNE_mid, 70.7% UCNE_high.
ucne_proportions <- data_frame("ucne_category"=c("UCNE_low","UCNE_mid","UCNE_high"),"proportion"=c(0.030,0.263,0.707)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(ucne_proportions))) {
  ucne_category <- unlist(ucne_proportions[row,1],use.names=F)
  print(ucne_category)
  proportion <- unlist(ucne_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(ucne_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

total_sites$category <- factor(total_sites$category,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
total_sites <- total_sites[order(total_sites$category),]
#write_tsv(total_sites,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt"))

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("feature"="category")) %>% mutate(het_by_total=N/callable_N,het_by_total_filter=N/callable_N_postfilter)
variants_het_joined_all$feature = factor(variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
variants_het_joined_all

#Plot:
ggplot_all_pi <- ggplot(data=variants_het_joined_all,aes(x=population,y=colnames(variants_het_joined_all)[which(colnames(variants_het_joined_all) == plot_variable)],colour=population)) +
  geom_point() +
  facet_grid(feature ~ .) +
  xlab("Population") +
  ylab(paste(type," individual ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_all_pi
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_5xonly.pdf"), width=10, height=30, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")


#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_variants_het_joined_all <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_het_by_total"=character(0),"se_het_by_total"=character(0)) #next, create the empty dataframe

for (pop in unique(variants_het_joined_all$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(variants_het_joined_all,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(variants_het_joined_all$feature)) {
    print(r)
    pop_mean <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_het_by_total","se_het_by_total")
    average_variants_het_joined_all <- rbind(average_variants_het_joined_all,row_data,stringsAsFactors=F)
  }
}
average_variants_het_joined_all
average_variants_het_joined_all$population = factor(average_variants_het_joined_all$population,levels=c("ki","no","po","sm","do"))
average_variants_het_joined_all$feature = factor(average_variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
average_variants_het_joined_all$avg_het_by_total <- as.numeric(average_variants_het_joined_all$avg_het_by_total)
average_variants_het_joined_all$se_het_by_total <- as.numeric(average_variants_het_joined_all$se_het_by_total)
average_variants_het_joined_all
#write_csv(average_variants_het_joined_all,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/",type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean.csv"))
  
#Plot these means and standard errors (categories with higher values):
average_ggplot_all_pi_part1 <- ggplot(data=filter(average_variants_het_joined_all,grepl('total|intergenic|intron|CDS|synonymous|syn_pref|syn_unpref|UCNE_low',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part1
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_5xonly_mean_part1.pdf"), width=10, height=28, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with intermediate values):
average_ggplot_all_pi_part2 <- ggplot(data=filter(average_variants_het_joined_all,grepl('missense|mistol|misdel|UCNE_mid',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part2
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_5xonly_mean_part2.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with low values):
average_ggplot_all_pi_part3 <- ggplot(data=filter(average_variants_het_joined_all,grepl('nonsense|\\<UCNE\\>|UCNE_high',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part3
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_5xonly_mean_part3.pdf"), width=10, height=12, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

average_ggplot_all_pi_combined <- grid.arrange(average_ggplot_all_pi_part1,average_ggplot_all_pi_part2,average_ggplot_all_pi_part3, ncol = 3, layout_matrix = cbind(c(rep(1,20)), c(rep(2,11),rep(NA,9)), c(rep(3,9),rep(NA,11))))
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_5xonly_mean_all.pdf"), width=35, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity",average_ggplot_all_pi_combined)

```

####5x only, definitive version.
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(egg)

#Define the following variables:
type="varssubs" #varssubs #variants #segregating #private_segregating
plot_variable <- "het_by_total_filter" #het_by_total_filter #het_by_total (for a filtered or unfiltered denominator)

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_het <- read_tsv(paste0(wd_path,variant_files)) %>% filter(dataset=="5x")

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt") 
total_sites

#For coding categories, we'll follow the proportions estimated by Yamaguchi-Kagata et al (see here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561068/): 28.5% SYN, 68.1% NSYN, and 3.4% NONSENSE. From my own classification of SYN, 16.9% were PREF and 50.8% were UNPREF. From my own classification of NSYN, 60.8% were NTOL and 39.2% were NDEL. So the totals would be: 28.5% SYN (4.8% PREF, 14.5% UNPREF), 68.1% NSYN (41.4% NTOL, 26.7% NDEL), and 3.4% NONSENSE.
coding_proportions <- data_frame("coding_category"=c("synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense"),"proportion"=c(0.285,0.048,0.145,0.681,0.414,0.267,0.034)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(coding_proportions))) {
  coding_category <- unlist(coding_proportions[row,1],use.names=F)
  print(coding_category)
  proportion <- unlist(coding_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(coding_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

#For UCNE categories, I calculated the proportion of total sites under each GERP category in the ucne_database script (section named: "Calculate proportion of sites for each GERP category"). These are: 3.0% UCNE_low, 26.3% UCNE_mid, 70.7% UCNE_high.
ucne_proportions <- data_frame("ucne_category"=c("UCNE_low","UCNE_mid","UCNE_high"),"proportion"=c(0.030,0.263,0.707)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(ucne_proportions))) {
  ucne_category <- unlist(ucne_proportions[row,1],use.names=F)
  print(ucne_category)
  proportion <- unlist(ucne_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(ucne_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

total_sites$category <- factor(total_sites$category,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
total_sites <- total_sites[order(total_sites$category),]
#write_tsv(total_sites,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt"))

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("feature"="category")) %>% mutate(het_by_total=N/callable_N,het_by_total_filter=N/callable_N_postfilter)
variants_het_joined_all$feature = factor(variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
variants_het_joined_all

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_variants_het_joined_all <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_het_by_total"=character(0),"se_het_by_total"=character(0)) #next, create the empty dataframe

for (pop in unique(variants_het_joined_all$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(variants_het_joined_all,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(variants_het_joined_all$feature)) {
    print(r)
    pop_mean <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(variants_het_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_het_by_total","se_het_by_total")
    average_variants_het_joined_all <- rbind(average_variants_het_joined_all,row_data,stringsAsFactors=F)
  }
}
average_variants_het_joined_all
average_variants_het_joined_all$population = factor(average_variants_het_joined_all$population,levels=c("ki","no","po","sm","do"))
average_variants_het_joined_all$feature = factor(average_variants_het_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
average_variants_het_joined_all$avg_het_by_total <- as.numeric(average_variants_het_joined_all$avg_het_by_total)
average_variants_het_joined_all$se_het_by_total <- as.numeric(average_variants_het_joined_all$se_het_by_total)
average_variants_het_joined_all
#write_csv(average_variants_het_joined_all,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/",type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean.csv"))

#Plot these means and standard errors (categories with higher values):
scaleFUN <- function(x) sprintf("%7.1e", x)

average_ggplot_all_pi_up <- ggplot(data=filter(average_variants_het_joined_all,grepl('total|intergenic|intron|synonymous|missense',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ feature) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,0,0.5,1),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

#upright_lims <- layer_scales(average_ggplot_all_pi_upright)$y$range$range

# average_ggplot_all_pi_upleft <- ggplot(data=filter(average_variants_het_joined_all,grepl('total|intergenic',feature)), aes(population,avg_het_by_total,colour=population)) +
#   #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
#   facet_grid(. ~ feature) +
#   geom_point() +
#   geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
#   #ggtitle("Proportion of reads at different NM") +
#   xlab("Population") +
#   ylab("Average nucleotide heterozygosity") +
#   scale_y_continuous(labels=scaleFUN,limits=c(upright_lims[1],upright_lims[2])) +
#   theme_bw() +
#   theme(text=element_text(size=12,face="bold"),
#       rect=element_rect(size=1),
#       axis.line=element_blank(),
#       axis.title=element_text(size=16),
#       axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
#       axis.title.x=element_blank(),
#       axis.text.y=element_text(size=12,colour="black"),
#       axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
#       panel.background=element_blank(),
#       panel.border=element_rect(colour="black",fill=NA,size=1.5),
#       strip.background=element_rect(colour="black",size=1.5),
#       #panel.grid=element_blank(),
#       #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
#       plot.margin=unit(c(0.5,0,0.5,-2),"cm"),
#       #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
#       legend.background=element_rect(linetype="solid", colour="black", size=.5),
#       #legend.justification=c(0,0),
#       legend.key=element_rect(colour="white"),
#       legend.key.size=unit(0.5,"cm"),
#       legend.position="none",
#       legend.title=element_blank()
#   )
# average_ggplot_all_pi_upleft

average_ggplot_all_pi_downleft <- ggplot(data=filter(average_variants_het_joined_all,grepl('nonsense|\\<UCNE\\>|UCNE_mid|UCNE_high',feature)), aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ feature) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,0,0.5,-2.8),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downleft

#downright_lims <- layer_scales(average_ggplot_all_pi_up)$y$range$range

average_ggplot_all_pi_downright <- ggplot(average_variants_het_joined_all[average_variants_het_joined_all$feature=="UCNE_low",], aes(population,avg_het_by_total,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ feature) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_het_by_total-se_het_by_total, ymax=avg_het_by_total+se_het_by_total), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  scale_y_continuous(position="right",labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,2,0.5,-4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downright

average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_downleft,average_ggplot_all_pi_downright), set_panel_size,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,0.2,1),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")))

ggsave(paste0(type,"_",plot_variable,"_5xonly_mean_definitive.pdf"), width=32, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity",average_ggplot_all_pi_combined)

```

###Calculate relative amount of derived alleles per category (joint calling; estimated joint totals from the variants only filters and from published proportions for CDS).
####All individuals.
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(gridExtra)

type="fixed" #varssubs #variants #substitutions #private_segregating #fixed
plot_variable <- "rda_by_total_filter" #rda_by_total_filter #rda_by_total (for a filtered or unfiltered denominator)

#Import derived counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_rda <- read_tsv(paste0(wd_path,variant_files)) %>% select(-c(34:37))

variants_rda$dataset <- as.factor(variants_rda$dataset)
variants_rda$dataset = factor(variants_rda$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_rda$population = factor(variants_rda$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_rda)

variants_rda_tidy <- variants_rda %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T) %>% filter(!grepl("_V",feature))
variants_rda_tidy$feature <- gsub('_A', '', variants_rda_tidy$feature)
variants_rda_tidy$feature <- gsub('intronic', 'intron', variants_rda_tidy$feature)
variants_rda_tidy$feature <- gsub('coding', 'CDS', variants_rda_tidy$feature)
variants_rda_tidy

#Import total sites data and cross it with the derived counts to estimate RDA:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt") 
total_sites

#For coding categories, we'll follow the proportions estimated by Yamaguchi-Kagata et al (see here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561068/): 28.5% SYN, 68.1% NSYN, and 3.4% NONSENSE. From my own classification of SYN, 16.9% were PREF and 50.8% were UNPREF. From my own classification of NSYN, 60.8% were NTOL and 39.2% were NDEL. So the totals would be: 28.5% SYN (4.8% PREF, 14.5% UNPREF), 68.1% NSYN (41.4% NTOL, 26.7% NDEL), and 3.4% NONSENSE.
coding_proportions <- data_frame("coding_category"=c("synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense"),"proportion"=c(0.285,0.048,0.145,0.681,0.414,0.267,0.034)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(coding_proportions))) {
  coding_category <- unlist(coding_proportions[row,1],use.names=F)
  print(coding_category)
  proportion <- unlist(coding_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(coding_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

#For UCNE categories, I calculated the proportion of total sites under each GERP category in the ucne_database script (section named: "Calculate proportion of sites for each GERP category"). These are: 3.0% UCNE_low, 26.3% UCNE_mid, 70.7% UCNE_high.
ucne_proportions <- data_frame("ucne_category"=c("UCNE_low","UCNE_mid","UCNE_high"),"proportion"=c(0.030,0.263,0.707)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(ucne_proportions))) {
  ucne_category <- unlist(ucne_proportions[row,1],use.names=F)
  print(ucne_category)
  proportion <- unlist(ucne_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(ucne_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

total_sites$category <- factor(total_sites$category,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
total_sites <- total_sites[order(total_sites$category),]
#write_tsv(total_sites,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt"))

variants_rda_joined_all <- left_join(variants_rda_tidy,total_sites,by=c("feature"="category")) %>% mutate(rda_by_total=N/callable_N,rda_by_total_filter=N/callable_N_postfilter)
variants_rda_joined_all$feature = factor(variants_rda_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
variants_rda_joined_all

#Plot:
ggplot_all_pi <- ggplot(data=as.data.frame(variants_rda_joined_all),aes(x=population,y=plot_variable,colour=population)) +
  geom_point() +
  facet_grid(feature ~ .,scales="free_y") +
  xlab("Population") +
  ylab(paste(type," individual ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_all_pi
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate.pdf"), width=10, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_variants_rda_joined_all <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_rda"=character(0),"se_rda"=character(0)) #next, create the empty dataframe
for (pop in unique(variants_rda_joined_all$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(variants_rda_joined_all,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(variants_rda_joined_all$feature)) {
    print(r)
    pop_mean <- filter(variants_rda_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(variants_rda_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_rda","se_rda")
    average_variants_rda_joined_all <- rbind(average_variants_rda_joined_all,row_data,stringsAsFactors=F)
  }
}
average_variants_rda_joined_all
average_variants_rda_joined_all$population = factor(average_variants_rda_joined_all$population,levels=c("ki","no","po","sm","do"))
average_variants_rda_joined_all$feature = factor(average_variants_rda_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
average_variants_rda_joined_all$avg_rda <- as.numeric(average_variants_rda_joined_all$avg_rda)
average_variants_rda_joined_all$se_rda <- as.numeric(average_variants_rda_joined_all$se_rda)
average_variants_rda_joined_all
write_csv(average_variants_rda_joined_all,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/",type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean.csv"))

#Plot these means and standard errors (categories with higher values):
average_ggplot_all_pi_part1 <- ggplot(data=filter(average_variants_rda_joined_all,grepl('total|intergenic|intron|CDS|synonymous|syn_pref|syn_unpref|UCNE_low',feature)), aes(population,avg_rda,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_rda-se_rda, ymax=avg_rda+se_rda), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part1
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_part1.pdf"), width=10, height=28, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with intermediate values):
average_ggplot_all_pi_part2 <- ggplot(data=filter(average_variants_rda_joined_all,grepl('missense|missense_tol|missense_del|UCNE_mid',feature)), aes(population,avg_rda,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_rda-se_rda, ymax=avg_rda+se_rda), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part2
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_part2.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with low values):
average_ggplot_all_pi_part3 <- ggplot(data=filter(average_variants_rda_joined_all,grepl('nonsense|\\<UCNE\\>|UCNE_high',feature)), aes(population,avg_rda,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_rda-se_rda, ymax=avg_rda+se_rda), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_part3
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_part3.pdf"), width=10, height=12, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

average_ggplot_all_pi_combined <- grid.arrange(average_ggplot_all_pi_part1,average_ggplot_all_pi_part2,average_ggplot_all_pi_part3, ncol = 3, layout_matrix = cbind(c(rep(1,20)), c(rep(2,11),rep(NA,9)), c(rep(3,9),rep(NA,11))))
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_all.pdf"), width=35, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity",average_ggplot_all_pi_combined)

```

####All individuals: SYN and NSYN breakdowns relative to totals.
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(gridExtra)

type="varssubs" #varssubs #variants #substitutions #private_segregating #fixed
plot_variable <- "rda_by_total_filter" #rda_by_total_filter #rda_by_total (for a filtered or unfiltered denominator)

#Import derived counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_rda <- read_tsv(paste0(wd_path,variant_files)) %>% select(-c(34:37))

variants_rda$dataset <- as.factor(variants_rda$dataset)
variants_rda$dataset = factor(variants_rda$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_rda$population = factor(variants_rda$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_rda)

variants_rda_tidy <- variants_rda %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T) %>% filter(!grepl("_V",feature))
variants_rda_tidy$feature <- gsub('_A', '', variants_rda_tidy$feature)
variants_rda_tidy$feature <- gsub('intronic', 'intron', variants_rda_tidy$feature)
variants_rda_tidy$feature <- gsub('coding', 'CDS', variants_rda_tidy$feature)
variants_rda_tidy

#Import total sites data and cross it with the derived counts to estimate RDA:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt") 
total_sites

#For coding categories, we'll follow the proportions estimated by Yamaguchi-Kagata et al (see here: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2561068/): 28.5% SYN, 68.1% NSYN, and 3.4% NONSENSE. From my own classification of SYN, 16.9% were PREF and 50.8% were UNPREF. From my own classification of NSYN, 60.8% were NTOL and 39.2% were NDEL. So the totals would be: 28.5% SYN (4.8% PREF, 14.5% UNPREF), 68.1% NSYN (41.4% NTOL, 26.7% NDEL), and 3.4% NONSENSE.
coding_proportions <- data_frame("coding_category"=c("synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense"),"proportion"=c(0.285,0.285,0.285,0.681,0.681,0.681,0.034)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(coding_proportions))) {
  coding_category <- unlist(coding_proportions[row,1],use.names=F)
  print(coding_category)
  proportion <- unlist(coding_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(coding_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="CDS"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

#For UCNE categories, I calculated the proportion of total sites under each GERP category in the ucne_database script (section named: "Calculate proportion of sites for each GERP category"). These are: 3.0% UCNE_low, 26.3% UCNE_mid, 70.7% UCNE_high.
ucne_proportions <- data_frame("ucne_category"=c("UCNE_low","UCNE_mid","UCNE_high"),"proportion"=c(0.030,0.263,0.707)) #create dataframe with these numbers and loop over it to add their estimated total counts to total_sites:
for (row in seq(1,nrow(ucne_proportions))) {
  ucne_category <- unlist(ucne_proportions[row,1],use.names=F)
  print(ucne_category)
  proportion <- unlist(ucne_proportions[row,2],use.names=F)
  print(proportion)
  total_sites <- rbind(total_sites,c(ucne_category,NA,NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),4],use.names=F)),NA,round(proportion*unlist(total_sites[which(total_sites$category=="UCNE"),6],use.names=F))))
  total_sites$callable_N <- as.numeric(total_sites$callable_N)
  total_sites$callable_N_postfilter <- as.numeric(total_sites$callable_N_postfilter)
}
total_sites

total_sites$category <- factor(total_sites$category,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
total_sites <- total_sites[order(total_sites$category),]
#write_tsv(total_sites,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt"))

variants_rda_joined_all <- left_join(variants_rda_tidy,total_sites,by=c("feature"="category")) %>% mutate(rda_by_total=N/callable_N,rda_by_total_filter=N/callable_N_postfilter)
variants_rda_joined_all$feature = factor(variants_rda_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
variants_rda_joined_all

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_variants_rda_joined_all <- data_frame("species"=character(0),"population"=character(0),"feature"=character(0),"avg_rda"=character(0),"se_rda"=character(0)) #next, create the empty dataframe
for (pop in unique(variants_rda_joined_all$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(variants_rda_joined_all,feature=="total" & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(variants_rda_joined_all$feature)) {
    print(r)
    pop_mean <- filter(variants_rda_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(variants_rda_joined_all,feature==r & population==pop) %>% select(plot_variable) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","feature","avg_rda","se_rda")
    average_variants_rda_joined_all <- rbind(average_variants_rda_joined_all,row_data,stringsAsFactors=F)
  }
}
average_variants_rda_joined_all
average_variants_rda_joined_all$population = factor(average_variants_rda_joined_all$population,levels=c("ki","no","po","sm","do"))
average_variants_rda_joined_all$feature = factor(average_variants_rda_joined_all$feature,levels=c("total","intergenic","intron","CDS","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
average_variants_rda_joined_all$avg_rda <- as.numeric(average_variants_rda_joined_all$avg_rda)
average_variants_rda_joined_all$se_rda <- as.numeric(average_variants_rda_joined_all$se_rda)
average_variants_rda_joined_all
#write_csv(average_variants_rda_joined_all,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/",type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean.csv"))

#Plot these means and standard errors (categories with higher values):
average_ggplot_SYN <- ggplot(data=filter(average_variants_rda_joined_all,grepl('synonymous|syn_pref|syn_unpref',feature)), aes(population,avg_rda,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_rda-se_rda, ymax=avg_rda+se_rda), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_SYN
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_SYN.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot these means and standard errors (categories with intermediate values):
average_ggplot_MIS <- ggplot(data=filter(average_variants_rda_joined_all,grepl('missense|missense_tol|missense_del',feature)), aes(population,avg_rda,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(feature ~ .) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_rda-se_rda, ymax=avg_rda+se_rda), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(paste0(type," population average ",plot_variable)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_MIS
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_MIS.pdf"), width=10, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

average_ggplot_SYN_MIS_combined <- grid.arrange(average_ggplot_SYN,average_ggplot_MIS, ncol = 2)
ggsave(paste0(type,"_all_",plot_variable,"_joint_calling_joint_estimate_mean_SYN&MIS.pdf"), width=20, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity",average_ggplot_SYN_MIS_combined)

```

###Obsolete:
####Calculate individual heterozygosity per category (joint calling; split totals).
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="variants" #varssubs #variants

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_het <- read_tsv(paste0(wd_path,variant_files))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_SNP.lr_ann.txt") %>% mutate(species=ifelse(calling=="c_lp_sm_c_lp_do_all","lp","ll")) %>% select(1,5,2:4)
total_sites

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("species","feature"="category")) %>% select(-c(7,8)) %>% mutate(PIx1000=1000*N/total_N_depth_correction)

#Calculate PI per site for coding categories using the whole CDS category as denominator:
variants_het_joined_coding <- left_join(variants_het_joined_all %>% filter(feature=="CDS" | feature=="synonymous" | feature=="missense" | feature=="mistol" | feature=="misdel" | feature=="nonsense") %>% select(-c(7:8)),variants_het_joined_all %>% filter(feature=="CDS") %>% select(-c(5:6,8)),by=c("species","population","dataset","sample")) %>% mutate(PI=N/total_N_depth_correction)
variants_het_joined_coding$feature = factor(variants_het_joined_coding$feature,levels=c("CDS","synonymous","missense","mistol","misdel","nonsense"))
variants_het_joined_coding

ggplot_coding_pi <- ggplot(data=as.data.frame(variants_het_joined_coding),aes(x=population,y=PI,colour=population)) +
  geom_point() +
  facet_grid(feature ~ .,scales="free_y") +
  xlab("Feature") +
  ylab(paste0(type," PI")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_coding_pi
ggsave(paste0(type,"_coding_PI_joint_calling.pdf"), width=10, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

```

####Calculate individual heterozygosity per category (separate calling).
```{r Calculate PI per category}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_heterozygous_SNP.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
variants_het <- rbind(read_tsv(paste0(wd_path,grep(variant_files,pattern='c_ll',value=T))),read_tsv(paste0(wd_path,grep(variant_files,pattern='c_lp',value=T))))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_SNP.lr_ann.txt") %>% mutate(species=ifelse(calling=="c_lp_sm_c_lp_do_all","lp","ll")) %>% select(1,5,2:4)
total_sites

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("species","feature"="category")) %>% select(-c(7,8)) %>% mutate(PIx1000=1000*N/total_N_depth_correction)

#Calculate PI per site for coding categories using the whole CDS category as denominator:
variants_het_joined_coding <- left_join(variants_het_joined_all %>% filter(feature=="CDS" | feature=="synonymous" | feature=="missense" | feature=="nonsense") %>% select(-c(7:8)),variants_het_joined_all %>% filter(feature=="CDS") %>% select(-c(5:6,8)),by=c("species","population","dataset","sample")) %>% mutate(PI=N/total_N_depth_correction)
variants_het_joined_coding$feature = factor(variants_het_joined_coding$feature,levels=c("CDS","synonymous","missense","nonsense"))
variants_het_joined_coding

ggplot_coding_pi <- ggplot(data=as.data.frame(variants_het_joined_coding),aes(x=population,y=PI,colour=population)) +
  geom_point() +
  facet_grid(feature ~ .,scales="free_y") +
  xlab("Feature") +
  ylab("PI") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_coding_pi
ggsave("coding_PI.pdf", width=10, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

```

####Plot genetic load vs. diversity graphs (joint callings).
```{r Plot genetic load vs. diversity graphs}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_heterozygous_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_het <- read_tsv(paste0(wd_path,variant_files))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_SNP.lr_ann.txt") %>% mutate(species=ifelse(calling=="c_lp_sm_c_lp_do_all","lp","ll")) %>% select(1,5,2:4)
total_sites

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("species","feature"="category")) %>% select(-c(7,8)) %>% mutate(PIx1000=1000*N/total_N_depth_correction)
variants_het_joined <- variants_het_joined_all %>% drop_na() %>% filter(feature=="intron")
variants_het_joined

#Plot heterozygosity of NS/S sites vs neutral sites
variants_het_joined_coding <- left_join(variants_het_joined_all %>% filter(feature=="CDS" | feature=="synonymous" | feature=="missense" | feature=="mistol" | feature=="misdel" | feature=="nonsense") %>% select(-c(7:8)),variants_het_joined_all %>% filter(feature=="CDS") %>% select(-c(5:6,8)),by=c("species","population","dataset","sample")) %>% mutate(PI=N/total_N_depth_correction)
variants_het_joined_coding$feature = factor(variants_het_joined_coding$feature,levels=c("CDS","synonymous","missense","mistol","misdel","nonsense"))
variants_het_joined_coding
variants_het_joined_coding <- mutate(variants_het_joined_coding,PIx1000=1000*N/total_N_depth_correction)
variants_het_joined_missense <- filter(variants_het_joined_coding,feature=="missense")
variants_het_joined_synonymous <- filter(variants_het_joined_coding,feature=="synonymous")

variants_het_load_het <- left_join(left_join(variants_het_joined,variants_het_joined_missense,by=c("species","population","dataset","sample")),variants_het_joined_synonymous,by=c("species","population","dataset","sample")) %>% select(1:5,8:9,13:14,18) %>% mutate(PIx1000_NS_S=PIx1000.y/PIx1000)

variants_het_load_plot <- ggplot(data=as.data.frame(variants_het_load_het),aes(x=PIx1000.x,y=PIx1000_NS_S,colour=population)) +
  geom_point() +
  xlab("Neutral heterozygosity (per kb)") +
  ylab(paste0(type," NS/S heterozygosity")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.9,0.76),
      legend.title=element_blank()
  )
variants_het_load_plot
ggsave(paste0(type,"_NS_S_vs_neutral_diversity_joint_calling.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")


#Plot the NS/S ratio vs the neutral heterozygosity:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern=paste0("*origcov_ann_individual_summary_",type,"_SNP.lr_ann.txt")),pattern='wrong',inv=T,value=T)
variants_wg <- read_tsv(paste0(wd_path,variant_files))

variants_wg$dataset <- as.factor(variants_wg$dataset)
variants_wg$dataset = factor(variants_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_wg$population = factor(variants_wg$population,levels=c("ki","no","po","sm","do"))
variants_wg <- variants_wg %>% mutate(`nonsense/synonymous_V`=nonsense_V/synonymous_V)
print.data.frame(variants_wg)

#Plot NS/S ratio:
variants_het_nssratio_plot_db <- left_join(variants_het_joined,select(variants_wg,c(1:4,24)),by=c("species","population","dataset","sample"))
variants_het_nssratio_plot_db

ggplot_variants_het_nssratio <- ggplot(data=as.data.frame(variants_het_nssratio_plot_db),aes(x=PIx1000,y=`missense/synonymous_V`,colour=population)) +
  geom_point() +
  xlab("Neutral heterozygosity (per kb)") +
  ylab(paste0(type," NS/S ratio")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.9,0.76),
      legend.title=element_blank()
  )
ggplot_variants_het_nssratio
ggsave(paste0(type,"_NS_S_ratio_vs_diversity_joint_calling.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot LOF/S ratio:
variants_het_lofsratio_plot_db <- left_join(variants_het_joined,select(variants_wg,c(1:4,28)),by=c("species","population","dataset","sample"))
variants_het_lofsratio_plot_db

ggplot_variants_het_lofsratio <- ggplot(data=as.data.frame(variants_het_lofsratio_plot_db),aes(x=PIx1000,y=`nonsense/synonymous_V`,colour=population)) +
  geom_point() +
  xlab("Neutral heterozygosity (per kb)") +
  ylab(paste0(type," LOF/S ratio")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.9,0.76),
      legend.title=element_blank()
  )
ggplot_variants_het_lofsratio
ggsave(paste0(type,"_LOF_S_ratio_vs_diversity_joint_calling.pdf"), width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

```

####Plot genetic load vs. diversity graphs (separate callings).
```{r Plot genetic load vs. diversity graphs}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

#Import heterozygous counts and change to tidy format:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_heterozygous_SNP.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
variants_het <- rbind(read_tsv(paste0(wd_path,grep(variant_files,pattern='c_ll',value=T))),read_tsv(paste0(wd_path,grep(variant_files,pattern='c_lp',value=T))))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,N,-species,-population,-dataset,-sample,factor_key=T)
variants_het_tidy$feature <- gsub('_H', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy

#Import total sites data and cross it with the heterozygous counts to estimate neutral PI:
total_sites <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/category_sites_total_counts_SNP.lr_ann.txt") %>% mutate(species=ifelse(calling=="c_lp_sm_c_lp_do_all","lp","ll")) %>% select(1,5,2:4)
total_sites

variants_het_joined_all <- left_join(variants_het_tidy,total_sites,by=c("species","feature"="category")) %>% select(-c(7,8)) %>% mutate(PIx1000=1000*N/total_N_depth_correction)
variants_het_joined <- variants_het_joined_all %>% drop_na() %>% filter(feature=="intron")
variants_het_joined

#Import the genetic load estimate and cross it with the previous data:
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variant_files <- grep(list.files("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/", pattern="*origcov_ann_individual_summary_SNP.lr_ann.txt"),pattern='wrong|nm2nm3',inv=T,value=T)
variants_wg <- rbind(read_tsv(paste0(wd_path,grep(variant_files,pattern='c_ll',value=T))),read_tsv(paste0(wd_path,grep(variant_files,pattern='c_lp',value=T))))

variants_wg$dataset <- as.factor(variants_wg$dataset)
variants_wg$dataset = factor(variants_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_wg$population = factor(variants_wg$population,levels=c("ki","no","po","sm","do"))
variants_wg <- variants_wg %>% mutate(`nonsense/synonymous_V`=nonsense_V/synonymous_V)
print.data.frame(variants_wg)

#Plot NS/S ratio:
variants_het_nssratio_plot_db <- left_join(variants_het_joined,select(variants_wg,c(1:4,20)),by=c("species","population","dataset","sample"))
variants_het_nssratio_plot_db

ggplot_variants_het_nssratio <- ggplot(data=as.data.frame(variants_het_nssratio_plot_db),aes(x=PIx1000,y=`missense/synonymous_V`,colour=population)) +
  geom_point() +
  xlab("Neutral heterozygosity (per kb)") +
  ylab("NS/S ratio") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.9,0.76),
      legend.title=element_blank()
  )
ggplot_variants_het_load
ggsave("NS_S_ratio_vs_diversity.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

#Plot LOF/S ratio:
variants_het_lofsratio_plot_db <- left_join(variants_het_joined,select(variants_wg,c(1:4,24)),by=c("species","population","dataset","sample"))
variants_het_lofsratio_plot_db

ggplot_variants_het_lofsratio <- ggplot(data=as.data.frame(variants_het_lofsratio_plot_db),aes(x=PIx1000,y=`nonsense/synonymous_V`,colour=population)) +
  geom_point() +
  xlab("Neutral heterozygosity (per kb)") +
  ylab("LOF/S ratio") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position=c(0.9,0.76),
      legend.title=element_blank()
  )
ggplot_variants_het_lofsratio
ggsave("LOF_S_ratio_vs_diversity.pdf", width=15, height=10, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

```

####Obtain heterozygosity reduction (HR).
```{r Obtain heterozygosity reduction (HR)}

HR_table <- left_join(variants_het_joined_coding,variants_het_joined_all %>% filter(feature=="intron") %>% mutate(PI=N/total_N_depth_correction) %>% select(-c(5:8)),by=c("species","population","dataset","sample"),suffix=c("_coding", "_intron")) %>% mutate(HR=(PI_intron-PI_coding)/PI_intron)

```


##At the population level.
###Get heterozygosis calculus.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(variants) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_heterozygosis_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
#First, prepare the depth-filtered per species vcf:
N_SITES_FILE=(${CALLING}"_polarized_filteredall_"${VAR}"_"${TYPE}".lr_ann.vcf")

rm ${CALLING}"_ann_population_summary_heterozygosis_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\ttotal_HET_AVG\tintergenic_HET_AVG\tintronic_HET_AVG\tcoding_HET_AVG\tsynonymous_HET_AVG\tmissense_HET_AVG\tnonsense_HET_AVG\tUCNE_HET_AVG" > ${CALLING}"_ann_population_summary_heterozygosis_"${VAR}"_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${POPLIST[@]}"
  do
  echo "${i}"
  vcf=$(echo "${i}" | rev | cut -d'/' -f1 | rev)
  echo "${vcf}"
  SPECIES=$(echo "${vcf}" | cut -c3-4)
  POPULATION=$(echo "${vcf}" | cut -c14-15)
  DATASET=$(echo "${vcf}" | cut -c6-7)
  TOTAL_N_SITES=$(grep -v '#' ${N_SITES_FILE} | wc -l)
  TOTAL_HET_SUM=$(grep -v '#' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  TOTAL_HET_AVG=$(echo "scale=4; $TOTAL_HET_SUM/$TOTAL_N_SITES" | bc)
  INTERGENIC_N_SITES=$(grep 'intergenic' ${N_SITES_FILE} | wc -l)
  INTERGENIC_HET_SUM=$(grep 'intergenic' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  INTERGENIC_HET_AVG=$(echo "scale=4; $INTERGENIC_HET_SUM/$INTERGENIC_N_SITES" | bc)
  INTRONIC_N_SITES=$(grep 'intron_variant' ${N_SITES_FILE} | wc -l)
  INTRONIC_HET_SUM=$(grep 'intron_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  INTRONIC_HET_AVG=$(echo "scale=4; $INTRONIC_HET_SUM/$INTRONIC_N_SITES" | bc)
  CODING_N_SITES=$(grep 'CDS' ${N_SITES_FILE} | wc -l)
  CODING_HET_SUM=$(grep 'CDS' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  CODING_HET_AVG=$(echo "scale=4; $CODING_HET_SUM/$CODING_N_SITES" | bc)
  SYNONYMOUS_N_SITES=$(grep 'synonymous_variant' ${N_SITES_FILE} | wc -l)
  SYNONYMOUS_HET_SUM=$(grep 'synonymous_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  SYNONYMOUS_HET_AVG=$(echo "scale=4; $SYNONYMOUS_HET_SUM/$SYNONYMOUS_N_SITES" | bc)
  MISSENSE_N_SITES=$(grep 'missense_variant' ${N_SITES_FILE} | wc -l)
  MISSENSE_HET_SUM=$(grep 'missense_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  MISSENSE_HET_AVG=$(echo "scale=4; $MISSENSE_HET_SUM/$MISSENSE_N_SITES" | bc)
  NONSENSE_N_SITES=$(grep '|HIGH|' ${N_SITES_FILE} | wc -l)
  NONSENSE_HET_SUM=$(grep '|HIGH|' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  NONSENSE_HET_AVG=$(echo "scale=4; $NONSENSE_HET_SUM/$NONSENSE_N_SITES" | bc)
  UCNE_N_SITES=$(grep 'UCNE' ${N_SITES_FILE} | wc -l)
  UCNE_HET_SUM=$(grep 'UCNE' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  UCNE_HET_AVG=$(echo "scale=4; $UCNE_HET_SUM/$UCNE_N_SITES" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$TOTAL_HET_AVG\t$INTERGENIC_HET_AVG\t$INTRONIC_HET_AVG\t$CODING_HET_AVG\t$SYNONYMOUS_HET_AVG\t$MISSENSE_HET_AVG\t$NONSENSE_HET_AVG\t$UCNE_HET_AVG" >> ${CALLING}"_ann_population_summary_heterozygosis_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_nm2_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_heterozygosis_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity
CALLING=(c_ll_ki_c_ll_no_c_ll_po_nm3_origcov)
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_heterozygosis_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity
unset SSHPASS

```

###Get heterozygosity per site calculus (joint calling).
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(variants) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_PI_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

ALL_SITES_FILE=$(echo /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/${CALLING}"_category_sites_total_counts_all.lr_ann.txt")
rm ${CALLING}"_ann_population_summary_PI_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsize\ttotal_PI_AVG\tintergenic_PI_AVG\tintronic_PI_AVG\tcoding_PI_AVG\tsynonymous_PI_AVG\tmissense_PI_AVG\tmistol_PI_AVG\tmisdel_PI_AVG\tnonsense_PI_AVG\tUCNE_PI_AVG\tPI_NS_S" > ${CALLING}"_ann_population_summary_PI_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${POPLIST[@]}"
  do
  echo "${i}"
  vcf=$(echo "${i}" | rev | cut -d'/' -f1 | rev)
  echo "${vcf}"
  SPECIES=$(echo "${vcf}" | cut -c3-4)
  POPULATION=$(echo "${vcf}" | cut -c14-15)
  DATASET=$(echo "${vcf}" | cut -c6-7)
  SIZE=$(bcftools query -l ${i} | wc -l)
  TOTAL_N_SITES=$(grep "all" $ALL_SITES_FILE | cut -f3)
  TOTAL_HET_SUM=$(grep -v '#' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  TOTAL_PI_AVG=$(echo "scale=8; $TOTAL_HET_SUM/$TOTAL_N_SITES" | bc)
  INTERGENIC_N_SITES=$(grep "intergenic" $ALL_SITES_FILE | cut -f3)
  INTERGENIC_HET_SUM=$(grep 'intergenic' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  INTERGENIC_PI_AVG=$(echo "scale=8; $INTERGENIC_HET_SUM/$INTERGENIC_N_SITES" | bc)
  INTRONIC_N_SITES=$(grep "intron" $ALL_SITES_FILE | cut -f3)
  INTRONIC_HET_SUM=$(grep 'intron_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  INTRONIC_PI_AVG=$(echo "scale=8; $INTRONIC_HET_SUM/$INTRONIC_N_SITES" | bc)
  CODING_N_SITES=$(grep "CDS" $ALL_SITES_FILE | cut -f3)
  CODING_HET_SUM=$(grep 'CDS' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  CODING_PI_AVG=$(echo "scale=8; $CODING_HET_SUM/$CODING_N_SITES" | bc)
  SYNONYMOUS_N_SITES=$(grep "CDS" $ALL_SITES_FILE | cut -f3)
  SYNONYMOUS_HET_SUM=$(grep 'synonymous_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  SYNONYMOUS_PI_AVG=$(echo "scale=8; $SYNONYMOUS_HET_SUM/($SYNONYMOUS_N_SITES)" | bc) #*1/4
  MISSENSE_N_SITES=$(grep "CDS" $ALL_SITES_FILE | cut -f3)
  MISSENSE_HET_SUM=$(grep 'missense_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  MISSENSE_PI_AVG=$(echo "scale=8; $MISSENSE_HET_SUM/($MISSENSE_N_SITES)" | bc) #*3/4
  MISTOL_N_SITES=$(grep "CDS" $ALL_SITES_FILE | cut -f3)
  MISTOL_HET_SUM=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  MISTOL_PI_AVG=$(echo "scale=8; $MISTOL_HET_SUM/($MISTOL_N_SITES)" | bc) #*(3/4)*(3/5)
  MISDEL_N_SITES=$(grep "CDS" $ALL_SITES_FILE | cut -f3)
  MISDEL_HET_SUM=$(bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  MISDEL_PI_AVG=$(echo "scale=8; $MISDEL_HET_SUM/($MISDEL_N_SITES)" | bc) #*(3/4)*(2/5)
  NONSENSE_N_SITES=$(grep "CDS" $ALL_SITES_FILE | cut -f3)
  NONSENSE_HET_SUM=$(grep '|HIGH|' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  NONSENSE_PI_AVG=$(echo "scale=8; $NONSENSE_HET_SUM/($NONSENSE_N_SITES)" | bc)
  UCNE_N_SITES=$(grep "UCNE" $ALL_SITES_FILE | cut -f3)
  UCNE_HET_SUM=$(grep 'UCNE' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  UCNE_PI_AVG=$(echo "scale=8; $UCNE_HET_SUM/$UCNE_N_SITES" | bc)
  PI_NS_S=$(echo "NA") #(echo "scale=4; $MISSENSE_PI_AVG/$SYNONYMOUS_PI_AVG" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SIZE\t$TOTAL_PI_AVG\t$INTERGENIC_PI_AVG\t$INTRONIC_PI_AVG\t$CODING_PI_AVG\t$SYNONYMOUS_PI_AVG\t$MISSENSE_PI_AVG\t$MISTOL_PI_AVG\t$MISDEL_PI_AVG\t$NONSENSE_PI_AVG\t$UCNE_PI_AVG\t$PI_NS_S" >> ${CALLING}"_ann_population_summary_PI_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_PI_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity
unset SSHPASS

<!-- for i in "${CALLING[@]}"  -->
<!--   do -->
<!--   echo "${i}" -->
<!--   export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt) -->
<!--   cd /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity -->
<!--   sshpass -e scp -o StrictHostKeyChecking=no dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_PI_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity -->
<!--   unset SSHPASS -->
<!--   echo "${i} done" -->
<!--   cd -->
<!--   done -->

```

###Get heterozygosity per site calculus (separate callings).
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(variants) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_PI_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
if [ $CALLING == "c_ll_ki_c_ll_no_c_ll_po_nm3_origcov" ]
  then ALL_SITES_CODE=$(echo "c_ll_ki_c_ll_no_c_ll_po_all")
  elif [ $CALLING == "c_lp_sm_c_lp_do_nm2_origcov" ]
  then ALL_SITES_CODE=$(echo "c_lp_sm_c_lp_do_all")
fi

ALL_SITES_FILE=$(echo /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/annotation/${ALL_SITES_CODE}"_category_sites_total_counts_"${TYPE}".lr_ann.txt")
rm ${CALLING}"_ann_population_summary_PI_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\ttotal_PI_AVG\tintergenic_PI_AVG\tintronic_PI_AVG\tcoding_PI_AVG\tsynonymous_PI_AVG\tmissense_PI_AVG\tnonsense_PI_AVG\tUCNE_PI_AVG\tPI_NS_S" > ${CALLING}"_ann_population_summary_PI_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_"${TYPE}".lr_ann.vcf" -print`))
for i in "${POPLIST[@]}"
  do
  echo "${i}"
  vcf=$(echo "${i}" | rev | cut -d'/' -f1 | rev)
  echo "${vcf}"
  SPECIES=$(echo "${vcf}" | cut -c3-4)
  POPULATION=$(echo "${vcf}" | cut -c14-15)
  DATASET=$(echo "${vcf}" | cut -c6-7)
  #TOTAL_N_SITES=$(grep -v '#' ${i} | wc -l)
  #TOTAL_HET_SUM=$(grep -v '#' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  TOTAL_PI_AVG=$(echo "NA")
  INTERGENIC_N_SITES=$(grep 'intergenic' $ALL_SITES_FILE | cut -f4)
  INTERGENIC_HET_SUM=$(grep 'intergenic' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  INTERGENIC_PI_AVG=$(echo "scale=8; $INTERGENIC_HET_SUM/$INTERGENIC_N_SITES" | bc)
  INTRONIC_N_SITES=$(grep 'intron' $ALL_SITES_FILE | cut -f4)
  INTRONIC_HET_SUM=$(grep 'intron_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  INTRONIC_PI_AVG=$(echo "scale=8; $INTRONIC_HET_SUM/$INTRONIC_N_SITES" | bc)
  CODING_N_SITES=$(grep 'CDS' $ALL_SITES_FILE | cut -f4)
  CODING_HET_SUM=$(grep 'CDS' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  CODING_PI_AVG=$(echo "scale=8; $CODING_HET_SUM/$CODING_N_SITES" | bc)
  SYNONYMOUS_N_SITES=$(grep 'CDS' $ALL_SITES_FILE | cut -f4)
  SYNONYMOUS_HET_SUM=$(grep 'synonymous_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  SYNONYMOUS_PI_AVG=$(echo "scale=8; $SYNONYMOUS_HET_SUM/($SYNONYMOUS_N_SITES*1/4)" | bc)
  MISSENSE_N_SITES=$(grep 'CDS' $ALL_SITES_FILE | cut -f4)
  MISSENSE_HET_SUM=$(grep 'missense_variant' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  MISSENSE_PI_AVG=$(echo "scale=8; $MISSENSE_HET_SUM/($MISSENSE_N_SITES*3/4)" | bc)
  NONSENSE_N_SITES=$(grep 'CDS' $ALL_SITES_FILE | cut -f4)
  NONSENSE_HET_SUM=$(grep '|HIGH|' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  NONSENSE_PI_AVG=$(echo "NA")
  UCNE_N_SITES=$(grep 'UCNE' $ALL_SITES_FILE | cut -f4)
  UCNE_HET_SUM=$(grep 'UCNE' ${i} | awk -F"\t|;|=" '{printf ("%f\n", 2*$13*(1-$13))}' | paste -sd+ | bc)
  UCNE_PI_AVG=$(echo "scale=8; $UCNE_HET_SUM/$UCNE_N_SITES" | bc)
  PI_NS_S=$(echo "scale=4; $MISSENSE_PI_AVG/$SYNONYMOUS_PI_AVG" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$TOTAL_PI_AVG\t$INTERGENIC_PI_AVG\t$INTRONIC_PI_AVG\t$CODING_PI_AVG\t$SYNONYMOUS_PI_AVG\t$MISSENSE_PI_AVG\t$NONSENSE_PI_AVG\t$UCNE_PI_AVG\t$PI_NS_S" >> ${CALLING}"_ann_population_summary_PI_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_nm2_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_PI_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity
CALLING=(c_ll_ki_c_ll_no_c_ll_po_nm3_origcov)
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_PI_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity
unset SSHPASS

<!-- for i in "${CALLING[@]}"  -->
<!--   do -->
<!--   echo "${i}" -->
<!--   export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt) -->
<!--   cd /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity -->
<!--   sshpass -e scp -o StrictHostKeyChecking=no dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_PI_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity -->
<!--   unset SSHPASS -->
<!--   echo "${i} done" -->
<!--   cd -->
<!--   done -->

```

###Plot heterozygosity per site per category.
```{r Plot genetic load vs. diversity graphs}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

#Import heterozygous counts and change to tidy format:
variants_het <- read_tsv("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_population_summary_PI_SNP.lr_ann.txt") %>% select(-c(PI_NS_S))

variants_het$dataset <- as.factor(variants_het$dataset)
variants_het$dataset = factor(variants_het$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_het$population = factor(variants_het$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_het)

variants_het_tidy <- variants_het %>% gather(feature,PI,-species,-population,-dataset,-size,factor_key=T)
variants_het_tidy$feature <- gsub('_PI_AVG', '', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('intronic', 'intron', variants_het_tidy$feature)
variants_het_tidy$feature <- gsub('coding', 'CDS', variants_het_tidy$feature)
variants_het_tidy$feature = factor(variants_het_tidy$feature,levels=c("total","intergenic","intron","CDS","synonymous","missense","mistol","misdel","nonsense","UCNE"))
variants_het_tidy

#Plot:
ggplot_pi <- ggplot(data=filter(as.data.frame(variants_het_tidy),dataset=="5x"),aes(x=population,y=PI,colour=population)) +
  geom_point() +
  facet_grid(feature ~ .,scales="free_y") +
  xlab("Feature") +
  ylab(paste0(type," PI")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_line(colour="black"),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black"),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_pi
ggsave("variants_population_PI_joint_calling_joint_estimate.pdf", width=10, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity")

```

###Obtain heterozygosity reduction (HR).
```{r Obtain heterozygosity reduction (HR)}

left_join(variants_het_joined_coding,variants_het_joined_all %>% filter(feature=="intron") %>% mutate(PI=N/total_N_depth_correction) %>% select(-c(5:8)),by=c("species","population","dataset","sample"),suffix=c("_coding", "_intron")) %>% mutate(HR=(PI_intron-PI_coding)/PI_intron)

```

###Get Hardy-Weinberg equilibrium measurements.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_HWE_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_population_summary_HWE_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tcategory\tpolymorphic_N\tHWE_p_phred_mean\tHWE_p_phred_var\tHWE_p_significant_proportion" > ${CALLING}"_ann_population_summary_HWE_"${VAR}"_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" ! -path '*testing_variant_numbers*' -print`))
for p in "${POPLIST[@]}"
  do
  echo "${p}"
  pop=$(echo "${p}" | awk -F'[/]' '{print $3}' | cut -d'_' -f6)
  echo "${pop}"
  SPECIES=$(echo "${p}" | awk -F'[/]' '{print $3}' | cut -d'_' -f5)
  POPULATION=$(echo "${pop}")
  DATASET=$(echo "${p}" | awk -F'[/]' '{print $3}' | cut -d'_' -f3)
  CATEGORIES=$(cut -f1 categories.txt) #This file was manually defined beforehand (first column: category name; second column: grep term (when necessary)).
  for c in ${CATEGORIES[@]}
    do
    echo "${c}"
    if [ $c == "TOTAL" ]
      then
      grep -v '#' ${p} | cut -f8 | tr ';' '\t' | cut -f9 | cut -d'=' -f2 | awk '{printf "%f\t%f\n",$1,$1*$1}' > ${c}_data.borrar
      elif [ $c == "MISTOL" ]
      then
      bedtools intersect -a ${p} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | cut -f8 | tr ';' '\t' | cut -f9 | cut -d'=' -f2 | awk '{printf "%f\t%f\n",$1,$1*$1}' > ${c}_data.borrar
      elif [ $c == "MISDEL" ]
      then
      bedtools intersect -a ${p} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | cut -f8 | tr ';' '\t' | cut -f9 | cut -d'=' -f2 | awk '{printf "%f\t%f\n",$1,$1*$1}' > ${c}_data.borrar
      else
      GREP=$(grep "${c}" categories.txt | cut -f2)
      echo "$GREP"
      grep "$GREP" ${p} | cut -f8 | tr ';' '\t' | cut -f9 | cut -d'=' -f2 | awk '{printf "%f\t%f\n",$1,$1*$1}' > ${c}_data.borrar
    fi
    DATA_N=$(wc -l < ${c}_data.borrar)
    DATA_SUM=$(cut -f1 ${c}_data.borrar | paste -sd+ | bc)
    DATA_MEAN=$(echo "scale=4; $DATA_SUM/$DATA_N" | bc)
    DATA_SQUARE_SUM=$(cut -f2 ${c}_data.borrar | paste -sd+ | bc)
    DATA_VAR_1ST_TERM=$(echo "scale=6; $DATA_SQUARE_SUM/($DATA_N-1)" | bc | awk '{printf "%f", $0}')
    DATA_VAR_2ND_TERM=$(echo "scale=6; ($DATA_N*$DATA_MEAN)/($DATA_N-1)" | bc | awk '{printf "%f", $0}')
    DATA_VARIANCE=$(echo "$DATA_VAR_1ST_TERM-$DATA_VAR_2ND_TERM" | bc)
    DATA_SIGNIFICANT_N=$(awk '$1 >= 54.69' ${c}_data.borrar | wc -l)
    DATA_SIGNIFICANT_PROP=$(echo "scale=4; $DATA_SIGNIFICANT_N/$DATA_N" | bc)
    #Copy calculations to a file:
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$c\t$DATA_N\t$DATA_MEAN\t$DATA_VARIANCE\t$DATA_SIGNIFICANT_PROP" >> ${CALLING}"_ann_population_summary_HWE_"${VAR}"_"${TYPE}".lr_ann.txt"
    done
  done
rm *_data.borrar

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_HWE_*_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/

```

###Get Watterson theta.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_watterson_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
ALL_SITES_FILE=$(echo /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt)
rm ${CALLING}"_ann_population_summary_watterson_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsize\tcategory\tsegregating_N\tharmonic_N\tglobal_theta\ttotal_N\tnucleotide_theta" > ${CALLING}"_ann_population_summary_watterson_"${VAR}"_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -name *"_perpop_"${VAR}"_"${TYPE}".lr_ann.vcf" ! -path '*testing_variant_numbers*' -print`))
for p in "${POPLIST[@]}"
  do
  echo "working with file: ${p}"
  pop=$(echo "${p}" | awk -F'[/]' '{print $3}' | cut -d'_' -f6)
  echo "${pop} population"
  SPECIES=$(echo "${p}" | awk -F'[/]' '{print $3}' | cut -d'_' -f5)
  POPULATION=$(echo "${pop}")
  SIZE=$(bcftools query -l "${p}" | wc -l)
  DATASET=$(echo "${p}" | awk -F'[/]' '{print $3}' | cut -d'_' -f3)
  CATEGORIES=$(cut -f1 categories.txt | tail -n+1) #This file was manually defined beforehand (first column: category name; second column: grep term (when necessary)).
  #For each population and category, we need to retrieve 3 numbers: the number of ploymorphic sites (global theta numerator), the harmonic number of the sample size minus 1, which is a sample size correction factor (global theta denominator), and the number of total sites (per site theta denominator).
  for c in ${CATEGORIES[@]}
    do
    echo "${c} category"
    if [ $c == "total" ]
      then
      SEGREGATING_N=$(grep -v '#' ${p} | grep -v ';AF=1.00;' | wc -l)
      elif [ $c == "mistol" ]
      then
      SEGREGATING_N=$(bedtools intersect -a ${p} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt | grep -v ';AF=1.00;' | wc -l)
      elif [ $c == "misdel" ]
      then
      SEGREGATING_N=$(bedtools intersect -a ${p} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt | grep -v ';AF=1.00;' | wc -l)
      else
      GREP=$(grep "${c}" categories.txt | cut -f2)
      SEGREGATING_N=$(grep "$GREP" ${p} | grep -v ';AF=1.00;' | wc -l)
      fi
    echo $SEGREGATING_N "segregating sites" #this is the numerator of the global theta
    HARMONIC_N=0
    COUNTER=1
    while [ $COUNTER -lt $((SIZE*2)) ] #for every number from 1 to 2N-1, calculate the reciprocal and update the harmonic number
      do
      #echo $COUNTER
      RECIPROCAL_N=$(echo "scale=5; 1/$COUNTER" | bc)
      HARMONIC_N=$(echo "$HARMONIC_N + $RECIPROCAL_N" | bc)
      ((COUNTER++))
      done
    #for i in $(seq 1 $TOTAL_N) it doesn't work; seq is too big!
      #do
      #echo ${i}
      #RECIPROCAL_N=$(echo "scale=5; 1/$i" | bc)
      #echo $RECIPROCAL_N
      #HARMONIC_N=$(echo "$HARMONIC_N + $RECIPROCAL_N" | bc)
      #done
    echo $HARMONIC_N "harmonic number" #this is the denominator of the global theta
    GLOBAL_THETA=$(echo "scale=8; $SEGREGATING_N/$HARMONIC_N" | bc)
    TOTAL_N=$(grep -w "${c}" "$ALL_SITES_FILE" | cut -f6)
    echo $TOTAL_N "total sites" #this is the denominator of the per site theta
    NUCLEOTIDE_THETA=$(echo "scale=8; $GLOBAL_THETA/$TOTAL_N" | bc)
    echo $NUCLEOTIDE_THETA "nucleotide average Watterson's theta" #this is the nucleotide average theta
    #Copy calculations to a file:
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SIZE\t$c\t$SEGREGATING_N\t$HARMONIC_N\t$GLOBAL_THETA\t$TOTAL_N\t$NUCLEOTIDE_THETA" >> ${CALLING}"_ann_population_summary_watterson_"${VAR}"_"${TYPE}".lr_ann.txt"
    done
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_watterson_*_SNP.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity/

```

###Get πN and πS per site.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_population_summary_πNπSdNdS_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path


ALL_SITES_FILE=$(echo /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt) 
rm ${CALLING}"_ann_population_summary_πNπS_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsize\tsynonymous_PI_AVG\tmissense_PI_AVG\tPI_NS_S" > ${CALLING}"_ann_population_summary_πNπS_"${TYPE}".lr_ann.txt"
POPLIST=($(ls `find . -path *"_perpop/*_perpop_segregating_"${TYPE}".lr_ann.vcf" -print`))
for pop in "${POPLIST[@]}"
  do
  echo "${pop}"
  grep -E '#|synonymous_variant' ${pop} | vcftools --vcf - --site-pi --out ${pop/.vcf/.synonymous}
  grep -E '#|missense_variant' ${pop} | vcftools --vcf - --site-pi --out ${pop/.vcf/.missense}
  vcf=$(echo ${pop} | rev | cut -d'/' -f1 | rev)
  echo ${vcf}
  SPECIES=$(echo ${vcf} | cut -c3-4)
  POPULATION=$(echo ${vcf} | cut -c14-15)
  DATASET=$(echo ${vcf} | cut -c6-7)
  SIZE=$(bcftools query -l ${pop} | wc -l)
  SYNONYMOUS_N_SITES=$(grep "synonymous" $ALL_SITES_FILE | cut -f6)
  SYNONYMOUS_PI_SUM=$(tail -n+2 ${pop/.vcf/.synonymous.sites.pi} | cut -f3 | paste -sd+ | bc)
  SYNONYMOUS_PI_AVG=$(echo "scale=8; $SYNONYMOUS_PI_SUM/$SYNONYMOUS_N_SITES" | bc)
  SYNONYMOUS_SUBS_N=$(grep "synonymous_variant" ${pop/_segregating_/_substitutions_} | wc -l)
  SYNONYMOUS_SUBS_PROP=$(echo "scale=8; $SYNONYMOUS_SUBS_N/$SYNONYMOUS_N_SITES" | bc)
  
  MISSENSE_N_SITES=$(grep "missense" $ALL_SITES_FILE | cut -f6)
  MISSENSE_PI_SUM=$(tail -n+2 ${pop/.vcf/.missense.sites.pi} | cut -f3 | paste -sd+ | bc)
  MISSENSE_PI_AVG=$(echo "scale=8; $MISSENSE_PI_SUM/$MISSENSE_N_SITES" | bc)
  MISSENSE_SUBS_N=$(grep "missense_variant" ${pop/_segregating_/_substitutions_} | wc -l)
  MISSENSE_SUBS_PROP=$(echo "scale=8; $MISSENSE_SUBS_N/$MISSENSE_N_SITES" | bc)

  PI_NS_S=$(echo "scale=5; $MISSENSE_PI_AVG/$SYNONYMOUS_PI_AVG" | bc)
  D_NS_S=$(echo "scale=5; $MISSENSE_SUBS_PROP/$SYNONYMOUS_SUBS_PROP" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SIZE\t$SYNONYMOUS_PI_AVG\t$MISSENSE_PI_AVG\t$SYNONYMOUS_SUBS_PROP\t$MISSENSE_SUBS_PROP\t$PI_NS_S\t$D_NS_S" >> ${CALLING}"_ann_population_summary_πNπSdNdS_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov) #c_lp_sm_c_lp_do_nm2_origcov #c_ll_ki_c_ll_no_c_ll_po_nm3_origcov #c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/*_ann_population_summary_PI_SNP.lr_ann.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/genetic_load_vs_diversity
unset SSHPASS

```


#KAKA
```{r}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

scaffolds_length <- read_tsv("/Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts/Length_scaffolds_lp23",col_names=c("scaffold","length"),col_types=c("cd")) %>% arrange(desc(length))

scaffolds_length <- scaffolds_length %>% mutate(cumlength=cumsum(scaffolds_length$length))

scaffolds_length$scaffold <- factor(scaffolds_length$scaffold,levels=unique(scaffolds_length$scaffold))

xint1 <- scaffolds_length %>% filter(length>2400000) %>% nrow()
xint2 <- scaffolds_length %>% filter(length>1200000) %>% nrow()

scaffolds_length_plot <- ggplot(scaffolds_length,aes(x=scaffold,y=length)) +
  #geom_point() +
  geom_point(size=0.1) +
  geom_vline(xintercept=xint1,colour="red") +
  geom_vline(xintercept=xint2,colour="blue") +
  scale_y_continuous(breaks=c(200,200000,2000000,10000000)) +
  xlab("Scaffolds") +
  ylab("Length") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
scaffolds_length_plot

ggsave("scaffolds_length_plot.pdf", width=30, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts")



scaffolds_cumlength_plot <- ggplot(scaffolds_length,aes(x=scaffold,y=cumlength)) +
  #geom_point() +
  geom_point(size=0.1) +
  geom_vline(xintercept=xint1,colour="red") +
  geom_vline(xintercept=xint2,colour="blue") +
  scale_y_continuous(breaks=c(0,1200000000,2400000000)) +
  xlab("Scaffolds") +
  ylab("Length") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
scaffolds_cumlength_plot

ggsave("scaffolds_cumlength_plot.pdf", width=30, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/whole_genome_counts")



```

