---
title: "genome_bootstrap"
output: html_document
---

#1. Define bootstrap blocks:
##Whole-genome.
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/


#Obtain the length of the whole genome:
cut -f2 /GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/Length_scaffolds_lp23 | paste -sd+ | bc #2413209059

#Since the genome measures 2.4Gb, we'll split it into 2400 blocks of 1005504b. To this end, we'll be using only those scaffolds larger than 1005504b (after discarding the regions that were associated to chromosomes X and Y). Obtain length of the scaffolds:
BLOCK_LENGTH=1005504
awk -F"\t" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$3-$2)}' /GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/bed_file_all_the_genome_without_Y_X_chr.bed | awk -F"\t" -v bl=$BLOCK_LENGTH '$4 >= bl {printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$4)}' > bed_file_long_scaffolds_without_Y_X_chr.bed

cut -f4 bed_file_long_scaffolds_without_Y_X_chr.bed | paste -sd+ | bc #1503187977 (which constitutes 62.3% of the genome)

#Obtain the number of subdivision blocks for each scaffold, as well as the starting point of the last block:
awk -F"\t" -v bl=$BLOCK_LENGTH '{printf ("%s\t%s\t%s\t%s\t%s\t%s\n", $1,$2,$3,$4,$4-bl,int($4/bl))}' bed_file_long_scaffolds_without_Y_X_chr.bed | sort -k4,4nr > bed_file_long_scaffolds_without_Y_X_chr_modified.bed

#For each scaffold, calculate $DIFF (the average distance between the starting points of different blocks i.e. the length of a block + the interblock distance, without considering the last block). Then define the starting and finishing point of each block (except for the last one) accounting for the interblock distance. Then define the last block at the very end of the scaffold. Finally, output all blocks to a new file.
rm bed_file_bootstrap_blocks.bed
while read -r SCAFFOLD BEGIN END LENGTH WOUT_LAST N_BLOCKS; do
  echo $SCAFFOLD "has" $N_BLOCKS "block(s)"
  DIFF=$(echo "scale=6; $WOUT_LAST/($N_BLOCKS-1)" | bc | awk '{printf "%s", int($0)}')
  for ((b=1; b<N_BLOCKS; b++))
    do
    CURRENT_DIFF=$((DIFF*(b-1)))
    START=$((BEGIN+1+CURRENT_DIFF))
    STOP=$((START+BLOCK_LENGTH-1))
    echo -e "$SCAFFOLD\t$START\t$STOP" >> bed_file_bootstrap_blocks.bed
    done
  echo -e "$SCAFFOLD\t$((WOUT_LAST+1))\t$LENGTH" >> bed_file_bootstrap_blocks.bed
done < bed_file_long_scaffolds_without_Y_X_chr_modified.bed

#Sanity check: test whether all blocks measure exactly 1005504b. They do!
awk -F"\t" '{printf ("%s\t%s\t%s\t%s\n", $1,$2,$3,$3-$2+1)}' bed_file_bootstrap_blocks.bed | cut -f4 | sort | uniq

#Sort the bed by scaffold and position.
sort -k1,1 -k2,2n bed_file_bootstrap_blocks.bed | awk -F"\t" '{printf ("%s\t%s\t%s\t%s%04d_%s_%s_%s\n", $1,$2,$3,"bl",NR,$1,$2,$3)}' > bed_file_bootstrap_blocks_sorted.bed
mv bed_file_bootstrap_blocks_sorted.bed bed_file_bootstrap_blocks.bed

#Finally, generate a .bed file for each block (row).
mkdir -p block_beds
cd block_beds
awk '{filename = sprintf("%s.bed", $4); print >filename; close(filename)}' ./../bed_file_bootstrap_blocks.bed

```

##Telomere blocks.
###2MB.
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/
mkdir -p tel_blocks

cut -f1 /GRUPOS/grupolince/telomers_centromers_definition/tel2m_regions_based_on_synteny_1000bp.bed | sort | uniq > tel_scaffolds #there are 359 scaffolds with tel regions

TEL_BLOCKS=$(grep -f tel_scaffolds ../bed_file_bootstrap_blocks.bed | cut -f4) #there are 20 scaffolds with tel regions in the bootstrap dataset

echo -e "block\ttelomere_%" > tel_blocks.txt
for BLOCK in ${TEL_BLOCKS[@]}
  do 
  TEL_BASES=$(bedtools intersect -a ../block_beds/${BLOCK}.bed -b /GRUPOS/grupolince/telomers_centromers_definition/tel2m_regions_based_on_synteny_1000bp.bed | bedtools sort | awk -F"\t" '{printf ("%s\n", $3-$2)}' | paste -sd+ | bc)
  TEL_SHARE=$(echo "scale=4; $TEL_BASES/1005504" | bc)
  if [ -z "$TEL_SHARE" ]; then TEL_SHARE=0; fi
  echo -e "$BLOCK\t$TEL_SHARE" >> tel_blocks.txt
  done
  
tail -n+2 temp_tel_blocks.txt | awk -F"\t" '$2>0.85 {print $0}' | cut -f1 > def_tel_blocks.txt

```

###10MB.
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/tel_blocks/
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/tel_blocks/

cut -f1 /GRUPOS/grupolince/telomers_centromers_definition/tel0-10m_regions_based_on_synteny_1000bp.bed | sort | uniq > tel_scaffolds #there are 989 scaffolds with tel regions

TEL_BLOCKS=$(grep -f tel_scaffolds ../bed_file_bootstrap_blocks.bed | cut -f4) #there are 235 scaffolds with tel regions in the bootstrap dataset

echo -e "block\ttelomere_%" > temp_tel_blocks.txt
for BLOCK in ${TEL_BLOCKS[@]}
  do  
  TEL_BASES=$(bedtools intersect -a ../block_beds/${BLOCK}.bed -b /GRUPOS/grupolince/telomers_centromers_definition/tel0-10m_regions_based_on_synteny_1000bp.bed | bedtools sort | awk -F"\t" '{printf ("%s\n", $3-$2)}' | paste -sd+ | bc)
  TEL_SHARE=$(echo "scale=4; $TEL_BASES/1005504" | bc)
  if [ -z "$TEL_SHARE" ]; then TEL_SHARE=0; fi
  echo -e "$BLOCK\t$TEL_SHARE" >> temp_tel_blocks.txt
  done

tail -n+2 temp_tel_blocks.txt | awk -F"\t" '$2>0.85 {print $0}' | cut -f1 > def_tel_blocks.txt #there are 141 blocks with > 85% telomeric content

```

##Centromere blocks.
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/centr_blocks/
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/centr_blocks/

cut -f1 /GRUPOS/grupolince/telomers_centromers_definition/centr_regions_based_on_synteny_1000bp.bed | sort | uniq > centr_scaffolds #there are 1081 scaffolds with centr regions

CENTR_BLOCKS=$(grep -f centr_scaffolds ../bed_file_bootstrap_blocks.bed | cut -f4) #there are 253 scaffolds with centr regions in the bootstrap dataset

echo -e "block\tcentromere_%" > temp_centr_blocks.txt
for BLOCK in ${CENTR_BLOCKS[@]}
  do  
  CENTR_BASES=$(bedtools intersect -a ../block_beds/${BLOCK}.bed -b /GRUPOS/grupolince/telomers_centromers_definition/centr_regions_based_on_synteny_1000bp.bed | bedtools sort | awk -F"\t" '{printf ("%s\n", $3-$2)}' | paste -sd+ | bc)
  CENTR_SHARE=$(echo "scale=4; $CENTR_BASES/1005504" | bc)
  if [ -z "$CENTR_SHARE" ]; then CENTR_SHARE=0; fi
  echo -e "$BLOCK\t$CENTR_SHARE" >> temp_centr_blocks.txt
  done

tail -n+2 temp_centr_blocks.txt | awk -F"\t" '$2>0.85 {print $0}' | cut -f1 > def_centr_blocks.txt #there are 162 blocks with > 85% centromeric content

```

##Xchr blocks.
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/Xchr_blocks/
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/Xchr_blocks/

cut -f1 /GRUPOS/grupolince/copia_fabascal/MAPPINGS/lynx2cat_wTiger_Xchr_woutPAR.sorted.merged.bed | sort | uniq > Xchr_scaffolds #

XCHR_BLOCKS=$(grep -f Xchr_scaffolds ../bed_file_bootstrap_blocks.bed | cut -f4) #
echo -e "block\tXchr_%" > temp_Xchr_blocks.txt
for BLOCK in ${XCHR_BLOCKS[@]}
  do  
  XCHR_BASES=$(bedtools intersect -a ../block_beds/${BLOCK}.bed -b /GRUPOS/grupolince/copia_fabascal/MAPPINGS/lynx2cat_wTiger_Xchr_woutPAR.sorted.merged.bed | bedtools sort | awk -F"\t" '{printf ("%s\n", $3-$2)}' | paste -sd+ | bc)
  XCHR_SHARE=$(echo "scale=4; $XCHR_BASES/1005504" | bc)
  if [ -z "$XCHR_SHARE" ]; then XCHR_SHARE=0; fi
  echo -e "$BLOCK\t$XCHR_SHARE" >> temp_Xchr_blocks.txt
  done

tail -n+2 temp_Xchr_blocks.txt | awk -F"\t" '$2>0.85 {print $0}' | cut -f1 > def_Xchr_blocks.txt #there are 0 blocks with > 85% centromeric content. We can't continue :(

```

#2. Define parallelisation parameters (this should be performed for each variable for each polarization method for each calling).
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
POLARIZATION=(TCRLP_outerparsimony) #annotation #TCRLP_polarizedfixed #TCRLP_outerparsimony
STATISTIC=(derived_homozygous_counts) #derived_allele_counts #derived_heterozygous_counts #derived_homozygous_counts
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
PARTITIONS=20

mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/$POLARIZATION/bootstrap/$STATISTIC/$VAR/block_list
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/$POLARIZATION/bootstrap/

rm $STATISTIC/$VAR/block_list/block_list_*.txt
BLOCK_N=$(ll -rth block_beds/bl*.bed | wc -l)
PARTITION_N=$(echo "scale=3; $BLOCK_N/$PARTITIONS" | bc | awk '{printf "%s", int($0)}')
for ((part=1; part < ${PARTITIONS[@]}; part++));
  do
  echo $part
  PARTbis=$(printf "%02d\n" $part)
  START=$(((part-1)*PARTITION_N+1))
  ls block_beds/ | tail -n+$START | head -n$PARTITION_N > $STATISTIC/$VAR/block_list/block_list_${PARTbis}.txt
  done
part=${PARTITIONS[@]}
echo $part
PARTbis=$(printf "%02d\n" $part)
START=$(((part-1)*PARTITION_N+1)) 
ls block_beds/ | tail -n+$START > $STATISTIC/$VAR/block_list/block_list_${PARTbis}.txt

```

#3. Write scripts for each statistic.
##Rufus polarisation:
###Derived allele counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation

echo "generating parallel derived allele counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/block_list_${CHUNK}.txt)
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A" > bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/.${BLOCK}.vcf}
    i=${n/.vcf/.${BLOCK}.vcf}
    TOTAL_V=$(grep -v '#' ${i} | wc -l)
    if [ "$TOTAL_V" -eq 0 ]; then TOTAL_A=0; else TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
    if [ "$INTERGENIC_V" -eq 0 ]; then INTERGENIC_A=0; else INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
    if [ "$INTRONIC_V" -eq 0 ]; then INTRONIC_A=0; else INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    CODING_V=$(grep 'CDS' ${i} | wc -l)
    SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
    if [ "$SYNONYMOUS_V" -eq 0 ]; then SYNONYMOUS_A=0; else SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${BLOCK}_${ind}_${VAR}_syn_pref.temp.borrar
    SYN_PREF_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_syn_pref.temp.borrar)
    if [ "$SYN_PREF_V" -eq 0 ]; then SYN_PREF_A=0; else SYN_PREF_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${BLOCK}_${ind}_${VAR}_syn_unpref.temp.borrar
    SYN_UNPREF_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_syn_unpref.temp.borrar)
    if [ "$SYN_UNPREF_V" -eq 0 ]; then SYN_UNPREF_A=0; else SYN_UNPREF_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
    if [ "$MISSENSE_V" -eq 0 ]; then MISSENSE_A=0; else MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_${VAR}_mis_tol.temp.borrar
    MISSENSE_TOL_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_mis_tol.temp.borrar)
    if [ "$MISSENSE_TOL_V" -eq 0 ]; then MISSENSE_TOL_A=0; else MISSENSE_TOL_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_${VAR}_mis_del.temp.borrar
    MISSENSE_DEL_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_mis_del.temp.borrar)
    if [ "$MISSENSE_DEL_V" -eq 0 ]; then MISSENSE_DEL_A=0; else MISSENSE_DEL_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
    if [ "$NONSENSE_V" -eq 0 ]; then NONSENSE_A=0; else NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    UCNE_V=$(grep 'UCNE' ${i} | wc -l)
    if [ "$UCNE_V" -eq 0 ]; then UCNE_A=0; else UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${BLOCK}_${ind}_${VAR}_ucne_low.temp.borrar
    UCNE_LOW_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_ucne_low.temp.borrar)
    if [ "$UCNE_LOW_V" -eq 0 ]; then UCNE_LOW_A=0; else UCNE_LOW_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_${VAR}_ucne_mid.temp.borrar
    UCNE_MID_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_ucne_mid.temp.borrar)
    if [ "$UCNE_MID_V" -eq 0 ]; then UCNE_MID_A=0; else UCNE_MID_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_${VAR}_ucne_high.temp.borrar
    UCNE_HIGH_V=$(wc -l < ${BLOCK}_${ind}_${VAR}_ucne_high.temp.borrar)
    if [ "$UCNE_HIGH_V" -eq 0 ]; then UCNE_HIGH_A=0; else UCNE_HIGH_A=$(cut -f8 ${BLOCK}_${ind}_${VAR}_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A" >> bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
    rm ${BLOCK}_${ind}_${VAR}_*.temp.borrar
    rm ${n/.vcf/.${BLOCK}.vcf}
    done
done

```

###Derived allele heterozygous counts.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation

echo "generating parallel derived heterozygous counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/block_list_${CHUNK}.txt)
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_heterozygous_counts/$VAR/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_H\tintergenic_H\tintronic_H\tcoding_H\tsynonymous_H\tsyn_pref_H\tsyn_unpref_H\tmissense_H\tmistol_H\tmisdel_H\tnonsense_H\tUCNE_H\tUCNE_low_H\tUCNE_mid_H\tUCNE_high_H" > bootstrap/derived_heterozygous_counts/$VAR/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/.${BLOCK}.vcf}
    i=${n/.vcf/.${BLOCK}.vcf}
    TOTAL_H=$(grep -v '#' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$TOTAL_H" ]; then TOTAL_H=0; fi
    INTERGENIC_H=$(grep 'intergenic' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$INTERGENIC_H" ]; then INTERGENIC_H=0; fi
    INTRONIC_H=$(grep 'intron_variant' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$INTRONIC_H" ]; then INTRONIC_H=0; fi
    CODING_H=$(grep 'CDS' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$CODING_H" ]; then CODING_H=0; fi
    SYNONYMOUS_H=$(grep 'synonymous_variant' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$SYNONYMOUS_H" ]; then SYNONYMOUS_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${BLOCK}_${ind}_${VAR}_het_syn_pref.temp.borrar
    SYN_PREF_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_syn_pref.temp.borrar | wc -l)
    if [ -z "$SYN_PREF_H" ]; then SYN_PREF_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${BLOCK}_${ind}_${VAR}_het_syn_unpref.temp.borrar
    SYN_UNPREF_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_syn_unpref.temp.borrar | wc -l)
    if [ -z "$SYN_UNPREF_H" ]; then SYN_UNPREF_H=0; fi
    MISSENSE_H=$(grep 'missense_variant' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$MISSENSE_H" ]; then MISSENSE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_${VAR}_het_mis_tol.temp.borrar
    MISTOL_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_mis_tol.temp.borrar | wc -l)
    if [ -z "$MISTOL_H" ]; then MISTOL_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_${VAR}_het_mis_del.temp.borrar
    MISDEL_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_mis_del.temp.borrar | wc -l)
    if [ -z "$MISDEL_H" ]; then MISDEL_H=0; fi
    NONSENSE_H=$(grep '|HIGH|' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$NONSENSE_H" ]; then NONSENSE_H=0; fi
    UCNE_H=$(grep 'UCNE' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$UCNE_H" ]; then UCNE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${BLOCK}_${ind}_${VAR}_het_ucne_low.temp.borrar
    UCNE_LOW_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_ucne_low.temp.borrar | wc -l)
    if [ -z "$UCNE_LOW_H" ]; then UCNE_LOW_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_${VAR}_het_ucne_mid.temp.borrar
    UCNE_MID_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_ucne_mid.temp.borrar | wc -l)
    if [ -z "$UCNE_MID_H" ]; then UCNE_MID_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_${VAR}_het_ucne_high.temp.borrar
    UCNE_HIGH_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_${VAR}_het_ucne_high.temp.borrar | wc -l)
    if [ -z "$UCNE_HIGH_H" ]; then UCNE_HIGH_H=0; fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_H\t$INTERGENIC_H\t$INTRONIC_H\t$CODING_H\t$SYNONYMOUS_H\t$SYN_PREF_H\t$SYN_UNPREF_H\t$MISSENSE_H\t$MISTOL_H\t$MISDEL_H\t$NONSENSE_H\t$UCNE_H\t$UCNE_LOW_H\t$UCNE_MID_H\t$UCNE_HIGH_H" >> bootstrap/derived_heterozygous_counts/$VAR/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
    rm ${BLOCK}_${ind}_${VAR}_het_*.temp.borrar
    rm ${n/.vcf/.${BLOCK}.vcf}
    done
  done

```

###Derived allele homozygous counts.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation

echo "generating parallel derived homozygous counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/block_list_${CHUNK}.txt)
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_homozygous_counts/$VAR/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_H\tintergenic_H\tintronic_H\tcoding_H\tsynonymous_H\tsyn_pref_H\tsyn_unpref_H\tmissense_H\tmistol_H\tmisdel_H\tnonsense_H\tUCNE_H\tUCNE_low_H\tUCNE_mid_H\tUCNE_high_H" > bootstrap/derived_homozygous_counts/$VAR/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/.${BLOCK}.vcf}
    i=${n/.vcf/.${BLOCK}.vcf}
    TOTAL_H=$(grep -v '#' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$TOTAL_H" ]; then TOTAL_H=0; fi
    INTERGENIC_H=$(grep 'intergenic' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$INTERGENIC_H" ]; then INTERGENIC_H=0; fi
    INTRONIC_H=$(grep 'intron_variant' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$INTRONIC_H" ]; then INTRONIC_H=0; fi
    CODING_H=$(grep 'CDS' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$CODING_H" ]; then CODING_H=0; fi
    SYNONYMOUS_H=$(grep 'synonymous_variant' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$SYNONYMOUS_H" ]; then SYNONYMOUS_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${BLOCK}_${ind}_${VAR}_hom_syn_pref.temp.borrar
    SYN_PREF_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_syn_pref.temp.borrar | wc -l)
    if [ -z "$SYN_PREF_H" ]; then SYN_PREF_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${BLOCK}_${ind}_${VAR}_hom_syn_unpref.temp.borrar
    SYN_UNPREF_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_syn_unpref.temp.borrar | wc -l)
    if [ -z "$SYN_UNPREF_H" ]; then SYN_UNPREF_H=0; fi
    MISSENSE_H=$(grep 'missense_variant' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$MISSENSE_H" ]; then MISSENSE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_${VAR}_hom_mis_tol.temp.borrar
    MISTOL_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_mis_tol.temp.borrar | wc -l)
    if [ -z "$MISTOL_H" ]; then MISTOL_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_${VAR}_hom_mis_del.temp.borrar
    MISDEL_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_mis_del.temp.borrar | wc -l)
    if [ -z "$MISDEL_H" ]; then MISDEL_H=0; fi
    NONSENSE_H=$(grep '|HIGH|' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$NONSENSE_H" ]; then NONSENSE_H=0; fi
    UCNE_H=$(grep 'UCNE' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$UCNE_H" ]; then UCNE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${BLOCK}_${ind}_${VAR}_hom_ucne_low.temp.borrar
    UCNE_LOW_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_ucne_low.temp.borrar | wc -l)
    if [ -z "$UCNE_LOW_H" ]; then UCNE_LOW_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_${VAR}_hom_ucne_mid.temp.borrar
    UCNE_MID_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_ucne_mid.temp.borrar | wc -l)
    if [ -z "$UCNE_MID_H" ]; then UCNE_MID_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_${VAR}_hom_ucne_high.temp.borrar
    UCNE_HIGH_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_${VAR}_hom_ucne_high.temp.borrar | wc -l)
    if [ -z "$UCNE_HIGH_H" ]; then UCNE_HIGH_H=0; fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_H\t$INTERGENIC_H\t$INTRONIC_H\t$CODING_H\t$SYNONYMOUS_H\t$SYN_PREF_H\t$SYN_UNPREF_H\t$MISSENSE_H\t$MISTOL_H\t$MISDEL_H\t$NONSENSE_H\t$UCNE_H\t$UCNE_LOW_H\t$UCNE_MID_H\t$UCNE_HIGH_H" >> bootstrap/derived_homozygous_counts/$VAR/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.${BLOCK}.lr_ann.txt
    rm ${BLOCK}_${ind}_${VAR}_hom_*.temp.borrar
    rm ${n/.vcf/.${BLOCK}.vcf}
    done
  done

```

##TCRLP outerparsimony:
###Derived allele counts.
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony

echo "generating parallel derived allele counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/derived_allele_counts/$VAR/block_list/block_list_${CHUNK}.txt)
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".anc_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A" > bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/_derived_allele_counts_$VAR.${BLOCK}.vcf}
    i=${n/.vcf/_derived_allele_counts_$VAR.${BLOCK}.vcf}
    TOTAL_V=$(grep -v '#' ${i} | wc -l)
    if [ "$TOTAL_V" -eq 0 ]; then TOTAL_A=0; else TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
    if [ "$INTERGENIC_V" -eq 0 ]; then INTERGENIC_A=0; else INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
    if [ "$INTRONIC_V" -eq 0 ]; then INTRONIC_A=0; else INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    CODING_V=$(grep 'CDS' ${i} | wc -l)
    SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
    if [ "$SYNONYMOUS_V" -eq 0 ]; then SYNONYMOUS_A=0; else SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
    if [ "$MISSENSE_V" -eq 0 ]; then MISSENSE_A=0; else MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_tol.temp.borrar
    MISSENSE_TOL_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_tol.temp.borrar)
    if [ "$MISSENSE_TOL_V" -eq 0 ]; then MISSENSE_TOL_A=0; else MISSENSE_TOL_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_del.temp.borrar
    MISSENSE_DEL_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_del.temp.borrar)
    if [ "$MISSENSE_DEL_V" -eq 0 ]; then MISSENSE_DEL_A=0; else MISSENSE_DEL_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
    if [ "$NONSENSE_V" -eq 0 ]; then NONSENSE_A=0; else NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    UCNE_V=$(grep 'UCNE' ${i} | wc -l)
    if [ "$UCNE_V" -eq 0 ]; then UCNE_A=0; else UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_mid.temp.borrar
    UCNE_MID_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_mid.temp.borrar)
    if [ "$UCNE_MID_V" -eq 0 ]; then UCNE_MID_A=0; else UCNE_MID_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_high.temp.borrar
    UCNE_HIGH_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_high.temp.borrar)
    if [ "$UCNE_HIGH_V" -eq 0 ]; then UCNE_HIGH_A=0; else UCNE_HIGH_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A" >> bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
    rm ${BLOCK}_${ind}_derived_allele_counts_${VAR}_*.temp.borrar
    rm ${n/.vcf/_derived_allele_counts_$VAR.${BLOCK}.vcf}
    done
done

```

###Derived allele counts (only particular blocks).
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony

echo "generating parallel derived allele counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
if [ $CHUNK == 40 ]
  then
  BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/derived_allele_counts/$VAR/block_list/block_list_${CHUNK}.txt | tail -n4)
  else
  BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/derived_allele_counts/$VAR/block_list/block_list_${CHUNK}.txt | tail -n3)
fi
echo "working with blocks:"
echo $BLOCKLIST
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".anc_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A" > bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/_derived_allele_counts_$VAR.${BLOCK}.vcf}
    i=${n/.vcf/_derived_allele_counts_$VAR.${BLOCK}.vcf}
    TOTAL_V=$(grep -v '#' ${i} | wc -l)
    if [ "$TOTAL_V" -eq 0 ]; then TOTAL_A=0; else TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
    if [ "$INTERGENIC_V" -eq 0 ]; then INTERGENIC_A=0; else INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
    if [ "$INTRONIC_V" -eq 0 ]; then INTRONIC_A=0; else INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    CODING_V=$(grep 'CDS' ${i} | wc -l)
    SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
    if [ "$SYNONYMOUS_V" -eq 0 ]; then SYNONYMOUS_A=0; else SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
    if [ "$MISSENSE_V" -eq 0 ]; then MISSENSE_A=0; else MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_tol.temp.borrar
    MISSENSE_TOL_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_tol.temp.borrar)
    if [ "$MISSENSE_TOL_V" -eq 0 ]; then MISSENSE_TOL_A=0; else MISSENSE_TOL_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_del.temp.borrar
    MISSENSE_DEL_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_del.temp.borrar)
    if [ "$MISSENSE_DEL_V" -eq 0 ]; then MISSENSE_DEL_A=0; else MISSENSE_DEL_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
    if [ "$NONSENSE_V" -eq 0 ]; then NONSENSE_A=0; else NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    UCNE_V=$(grep 'UCNE' ${i} | wc -l)
    if [ "$UCNE_V" -eq 0 ]; then UCNE_A=0; else UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_mid.temp.borrar
    UCNE_MID_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_mid.temp.borrar)
    if [ "$UCNE_MID_V" -eq 0 ]; then UCNE_MID_A=0; else UCNE_MID_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_high.temp.borrar
    UCNE_HIGH_V=$(wc -l < ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_high.temp.borrar)
    if [ "$UCNE_HIGH_V" -eq 0 ]; then UCNE_HIGH_A=0; else UCNE_HIGH_A=$(cut -f8 ${BLOCK}_${ind}_derived_allele_counts_${VAR}_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A" >> bootstrap/derived_allele_counts/$VAR/${CALLING}_ann_individual_summary_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
    rm ${BLOCK}_${ind}_derived_allele_counts_${VAR}_*.temp.borrar
    rm ${n/.vcf/_derived_allele_counts_$VAR.${BLOCK}.vcf}
    done
done

```

###Derived allele heterozygous counts.
```{r Get heterozygous counts, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony

echo "generating parallel derived heterozygous counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/derived_heterozygous_counts/$VAR/block_list/block_list_${CHUNK}.txt)
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".anc_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_heterozygous_counts/$VAR/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_H\tintergenic_H\tintronic_H\tcoding_H\tsynonymous_H\tmissense_H\tmistol_H\tmisdel_H\tnonsense_H\tUCNE_H\tUCNE_mid_H\tUCNE_high_H" > bootstrap/derived_heterozygous_counts/$VAR/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/_derived_heterozygous_counts_$VAR.${BLOCK}.vcf}
    i=${n/.vcf/_derived_heterozygous_counts_$VAR.${BLOCK}.vcf}
    TOTAL_H=$(grep -v '#' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$TOTAL_H" ]; then TOTAL_H=0; fi
    INTERGENIC_H=$(grep 'intergenic' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$INTERGENIC_H" ]; then INTERGENIC_H=0; fi
    INTRONIC_H=$(grep 'intron_variant' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$INTRONIC_H" ]; then INTRONIC_H=0; fi
    CODING_H=$(grep 'CDS' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$CODING_H" ]; then CODING_H=0; fi
    SYNONYMOUS_H=$(grep 'synonymous_variant' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$SYNONYMOUS_H" ]; then SYNONYMOUS_H=0; fi
    MISSENSE_H=$(grep 'missense_variant' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$MISSENSE_H" ]; then MISSENSE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_mis_tol.temp.borrar
    MISTOL_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_mis_tol.temp.borrar | wc -l)
    if [ -z "$MISTOL_H" ]; then MISTOL_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_mis_del.temp.borrar
    MISDEL_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_mis_del.temp.borrar | wc -l)
    if [ -z "$MISDEL_H" ]; then MISDEL_H=0; fi
    NONSENSE_H=$(grep '|HIGH|' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$NONSENSE_H" ]; then NONSENSE_H=0; fi
    UCNE_H=$(grep 'UCNE' ${i} | grep ';AF=0.500;' | wc -l)
    if [ -z "$UCNE_H" ]; then UCNE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_ucne_mid.temp.borrar
    UCNE_MID_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_ucne_mid.temp.borrar | wc -l)
    if [ -z "$UCNE_MID_H" ]; then UCNE_MID_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_ucne_high.temp.borrar
    UCNE_HIGH_H=$(grep ';AF=0.500;' ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_ucne_high.temp.borrar | wc -l)
    if [ -z "$UCNE_HIGH_H" ]; then UCNE_HIGH_H=0; fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_H\t$INTERGENIC_H\t$INTRONIC_H\t$CODING_H\t$SYNONYMOUS_H\t$MISSENSE_H\t$MISTOL_H\t$MISDEL_H\t$NONSENSE_H\t$UCNE_H\t$UCNE_MID_H\t$UCNE_HIGH_H" >> bootstrap/derived_heterozygous_counts/$VAR/${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
    rm ${BLOCK}_${ind}_derived_heterozygous_counts_${VAR}_het_*.temp.borrar
    rm ${n/.vcf/_derived_heterozygous_counts_$VAR.${BLOCK}.vcf}
    done
  done

```

###Derived allele homozygous counts.
```{r Get homozygous counts, eval=FALSE, engine='bash'}

CALLING=$(echo ${STY#*.} | cut -d'-' -f1) #name of the calling
VAR=$(echo ${STY#*.} | cut -d'-' -f2) #behaviour of variant
if [ $VAR == "private" ]
  then
  VAR="private_segregating"
fi
TYPE=$(echo ${STY#*.} | cut -d'-' -f3) #type of variant
CHUNK=$(echo ${STY#*.} | cut -d'-' -f4) #chunk of parallelisation
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony

echo "generating parallel derived homozygous counts for" $VAR "and chunk" $CHUNK
echo "retrieving blocks in chunk" $CHUNK
BLOCKLIST=$(cut -d'.' -f1,2 bootstrap/derived_homozygous_counts/$VAR/block_list/block_list_${CHUNK}.txt)
echo "retrieving individual VCFs"
INDLIST=($(ls `find . -path *"_individuals/*_individual_"${VAR}"_"${TYPE}".anc_ann.vcf" -print`))
for BLOCK in ${BLOCKLIST[@]}
  do 
  echo "working with block" ${BLOCK}
  BED=$(realpath bootstrap/block_beds/${BLOCK}.bed)
  rm bootstrap/derived_homozygous_counts/$VAR/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  echo -e "species\tpopulation\tdataset\tsample\ttotal_H\tintergenic_H\tintronic_H\tcoding_H\tsynonymous_H\tmissense_H\tmistol_H\tmisdel_H\tnonsense_H\tUCNE_H\tUCNE_mid_H\tUCNE_high_H" > bootstrap/derived_homozygous_counts/$VAR/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
  for n in "${INDLIST[@]}"
    do
    echo ${n}
    ind=$(echo ${n} | awk -F'[/]' '{print $3}' | cut -c1-12)
    echo ${ind}
    SPECIES=$(echo ${ind} | cut -c3-4)
    POPULATION=$(echo ${ind} | cut -c6-7)
    DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
    SAMPLE=$(echo ${ind} | cut -c9-12)
    echo "subsetting VCF for block" $BLOCK "and individual" $ind
    bedtools intersect -a ${n} -b ${BED} -header > ${n/.vcf/_derived_homozygous_counts_$VAR.${BLOCK}.vcf}
    i=${n/.vcf/_derived_homozygous_counts_$VAR.${BLOCK}.vcf}
    TOTAL_H=$(grep -v '#' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$TOTAL_H" ]; then TOTAL_H=0; fi
    INTERGENIC_H=$(grep 'intergenic' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$INTERGENIC_H" ]; then INTERGENIC_H=0; fi
    INTRONIC_H=$(grep 'intron_variant' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$INTRONIC_H" ]; then INTRONIC_H=0; fi
    CODING_H=$(grep 'CDS' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$CODING_H" ]; then CODING_H=0; fi
    SYNONYMOUS_H=$(grep 'synonymous_variant' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$SYNONYMOUS_H" ]; then SYNONYMOUS_H=0; fi
    MISSENSE_H=$(grep 'missense_variant' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$MISSENSE_H" ]; then MISSENSE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_tolerated.txt > ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_mis_tol.temp.borrar
    MISTOL_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_mis_tol.temp.borrar | wc -l)
    if [ -z "$MISTOL_H" ]; then MISTOL_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/provean/missense_variants_provean_scores_deleterious.txt > ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_mis_del.temp.borrar
    MISDEL_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_mis_del.temp.borrar | wc -l)
    if [ -z "$MISDEL_H" ]; then MISDEL_H=0; fi
    NONSENSE_H=$(grep '|HIGH|' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$NONSENSE_H" ]; then NONSENSE_H=0; fi
    UCNE_H=$(grep 'UCNE' ${i} | grep ';AF=1.00;' | wc -l)
    if [ -z "$UCNE_H" ]; then UCNE_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_ucne_mid.temp.borrar
    UCNE_MID_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_ucne_mid.temp.borrar | wc -l)
    if [ -z "$UCNE_MID_H" ]; then UCNE_MID_H=0; fi
    bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_ucne_high.temp.borrar
    UCNE_HIGH_H=$(grep ';AF=1.00;' ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_ucne_high.temp.borrar | wc -l)
    if [ -z "$UCNE_HIGH_H" ]; then UCNE_HIGH_H=0; fi
    echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_H\t$INTERGENIC_H\t$INTRONIC_H\t$CODING_H\t$SYNONYMOUS_H\t$MISSENSE_H\t$MISTOL_H\t$MISDEL_H\t$NONSENSE_H\t$UCNE_H\t$UCNE_MID_H\t$UCNE_HIGH_H" >> bootstrap/derived_homozygous_counts/$VAR/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.${BLOCK}.anc_ann.txt
    rm ${BLOCK}_${ind}_derived_homozygous_counts_${VAR}_hom_*.temp.borrar
    rm ${n/.vcf/_derived_homozygous_counts_$VAR.${BLOCK}.vcf}
    done
  done

```

#4. Upload scripts to the server.
##Rufus polarisation:
###Derived allele counts.
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_individual_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_individual_counts.sh dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/
chmod +x retrieve_individual_counts.sh

```

###Derived allele heterozygous counts.
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_heterozygous_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_heterozygous_counts.sh dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/
chmod +x retrieve_heterozygous_counts.sh

```

###Derived allele homozygous counts.
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_homozygous_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_homozygous_counts.sh dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_homozygous_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_homozygous_counts/
chmod +x retrieve_homozygous_counts.sh

```

##TCRLP outerparsimony:
###Derived allele counts.
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_individual_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_individual_counts.sh dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/
chmod +x retrieve_individual_counts.sh

```

###Derived allele counts (only particular blocks).
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_individual_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_individual_counts_subset.sh dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/
chmod +x retrieve_individual_counts_subset.sh

```

###Derived allele heterozygous counts.
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_heterozygous_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_heterozygous_counts.sh dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/
chmod +x retrieve_heterozygous_counts.sh

```

###Derived allele homozygous counts.
```{r, eval=FALSE, engine='bash'}

#Save it in a text editor as /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/retrieve_homozygous_counts.sh and upload it to the server:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/retrieve_homozygous_counts.sh dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_homozygous_counts/

#Then inside the server change permissions to run it:
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_homozygous_counts/
chmod +x retrieve_homozygous_counts.sh

```

#5. Obtain statistics for each block in parallel runs.
##Rufus polarisation:
###Derived allele counts.
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/
PARALLEL=$(ls block_list_*.txt | cut -d'_' -f3 | cut -d'.' -f1)
mkdir -p derived_allele_counts/$VAR
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_individual_counts.sh; exec bash$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```

###Derived allele heterozygous counts.
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/
PARALLEL=$(ls block_list_*.txt | cut -d'_' -f3 | cut -d'.' -f1)
mkdir -p derived_heterozygous_counts/$VAR
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/$VAR
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_heterozygous_counts.sh; exec bash$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```

###Derived allele homozygous counts.
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/
PARALLEL=$(ls block_list_*.txt | cut -d'_' -f3 | cut -d'.' -f1)
mkdir -p derived_homozygous_counts/$VAR
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_homozygous_counts/$VAR
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_homozygous_counts.sh; exec bash$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```


##TCRLP outerparsimony:
###Derived allele counts.
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR
PARALLEL=$(ls block_list/block_list_*.txt | cut -d'/' -f2 | cut -d'_' -f3 | cut -d'.' -f1)
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_individual_counts.sh; exec bash$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the scripts:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the screens:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```

###Derived allele counts (only particular blocks).
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR
PARALLEL=$(ls block_list/block_list_*.txt | cut -d'/' -f2 | cut -d'_' -f3 | cut -d'.' -f1)
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_individual_counts_subset.sh; exec bash$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the scripts:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the screens:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```


###Derived allele heterozygous counts.
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/$VAR
PARALLEL=$(ls block_list/block_list_*.txt | cut -d'/' -f2 | cut -d'_' -f3 | cut -d'.' -f1)
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_heterozygous_counts.sh; exec bash$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the scripts:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the screens:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```

###Derived allele homozygous counts.
```{r, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_homozygous_counts/$VAR
PARALLEL=$(ls block_list/block_list_*.txt | cut -d'/' -f2 | cut -d'_' -f3 | cut -d'.' -f1)
NL=$'\n'

for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -dmS "${CALLING}-${VAR}-${TYPE}-${p}"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "script ${CALLING}-${VAR}-${TYPE}-${p}.log$NL"
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "time ./../retrieve_homozygous_counts.sh; exec bash$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  #screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the scripts:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

#Once it has finished, run this loop to close the screens:
for p in ${PARALLEL[@]}
  do
  echo ${p}
  screen -S "${CALLING}-${VAR}-${TYPE}-${p}" -p 0 -X stuff "exit$NL"
  done

```

#6. Perform the bootstrap, obtain the bootstrap error, and relativise it.
##Draw coordinates for each bootstrapped genome. This should only be performed once (for each N_BOOT size).
###Whole-genome.
```{bash}

#This should only be performed once (for each N_BOOT size). The same coordinates will be carried over to other sections (fixed, 5x, heterozygous...)
N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/block_beds/
BLOCK_N=$(ll -rth bl*.bed | wc -l)

#Generate a huge pool of genome blocks (50 of each):
rm ../blocks_bootstrap.list
for i in {1..50}
  do
  ls bl*.bed >> ../blocks_bootstrap.list #List each block 50 times to generate the pool for the randomisation
  done
sed -i -e 's/.bed/.lr_ann.txt/g' ../blocks_bootstrap.list

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/
#2400 blocks from the pool will be drawn randomly to generate a bootstrap genome, and this will be performed N_BOOT times:
mkdir -p bootstrapped_genomes_coordinates
for boot in $(seq 1 $N_BOOT)
  do
  #echo "drawing blocks for bootstrapped genome number" $boot
  shuf -n2400 blocks_bootstrap.list > bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list
  done

#For the new polarisation, copy these lists to the new folder:
scp -pr /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/bootstrap/bootstrapped_genomes_coordinates /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/TCRLP_outerparsimony/bootstrap/
#Then fix the file names included in each list:
sed -i 's/lr_ann/anc_ann/g' *

```

###Telomeres.
```{bash}

#This should only be performed once (for each N_BOOT size). The same coordinates will be carried over to other sections (fixed, 5x, heterozygous...)
N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/tel_blocks/

N_TEL_BLOCKS=$(wc -l < def_tel_blocks.txt)
N_BLOCKS=$(ll -rth ../block_beds/bl*.bed | wc -l)
EXPECTED_TEL_BLOCKS=$(echo "scale=4; $N_TEL_BLOCKS*2400/$N_BLOCKS" | bc)
EXPECTED_TEL_BLOCKS=$(awk -v exp_tel_blocks=$EXPECTED_TEL_BLOCKS 'BEGIN{printf "%.0f\n", exp_tel_blocks}')
echo $EXPECTED_TEL_BLOCKS

grep -f def_tel_blocks.txt ../blocks_bootstrap.list > tel_blocks_bootstrap.list

#282 blocks from the telomere pool will be drawn randomly to generate a bootstrapped "telomeric genome", and this will be performed N_BOOT times:
for boot in $(seq 1 $N_BOOT)
  do
  #echo "drawing blocks for bootstrapped genome number" $boot
  shuf -n$EXPECTED_TEL_BLOCKS tel_blocks_bootstrap.list > bootstrapped_tel_${boot}.list
  done

#For the new polarisation, copy these lists to the new folder:
scp -pr /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/bootstrap/tel_blocks /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/TCRLP_outerparsimony/bootstrap/
#Then fix the file names included in each list:
sed -i 's/lr_ann/anc_ann/g' *.list

```

###Centromeres.
```{bash}

#This should only be performed once (for each N_BOOT size). The same coordinates will be carried over to other sections (fixed, 5x, heterozygous...)
N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/centr_blocks/

N_CENTR_BLOCKS=$(wc -l < def_centr_blocks.txt)
N_BLOCKS=$(ll -rth ../block_beds/bl*.bed | wc -l)
EXPECTED_CENTR_BLOCKS=$(echo "scale=4; $N_CENTR_BLOCKS*2400/$N_BLOCKS" | bc)
EXPECTED_CENTR_BLOCKS=$(awk -v exp_centr_blocks=$EXPECTED_CENTR_BLOCKS 'BEGIN{printf "%.0f\n", exp_centr_blocks}')
echo $EXPECTED_CENTR_BLOCKS

grep -f def_centr_blocks.txt ../blocks_bootstrap.list > centr_blocks_bootstrap.list

#324 blocks from the centromere pool will be drawn randomly to generate a bootstrapped "centromeric genome", and this will be performed N_BOOT times:
for boot in $(seq 1 $N_BOOT)
  do
  #echo "drawing blocks for bootstrapped genome number" $boot
  shuf -n$EXPECTED_CENTR_BLOCKS centr_blocks_bootstrap.list > bootstrapped_centr_${boot}.list
  done
  
#For the new polarisation, copy these lists to the new folder:
scp -pr /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/bootstrap/centr_blocks /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/TCRLP_outerparsimony/bootstrap/
#Then fix the file names included in each list:
sed -i 's/lr_ann/anc_ann/g' *.list


```

##Rufus polarisation:
###Population level.
####Derived allele counts.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#First, generate some headers and info files.
head -n1 <$(ls ${CALLING}_ann_individual_summary_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-33 > ind_headers.txt #Retrieve headers for files with individuals
cut -f2,5- ind_headers.txt > pop_headers.txt #Retrieve headers for files with populations
tail -n+2 <$(ls ${CALLING}_ann_individual_summary_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-4 > ids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data:
##Average:
cat pop_headers.txt <(gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt

##Standard error: sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_individual_summary_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_genome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste ids.txt <(awk '{for (i=5;i<=33;i++) total[FNR","i]+=$i;} END {for (j=1;j<=FNR;j++) {for (i=5;i<=33;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_genome_${boot}.list) | tail -n+2) > ${CALLING}_ann_individual_summary_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_individual_summary_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt > ${CALLING}_ann_population_average_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_population_average_${VAR}_${TYPE}.boot_1.lr_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_population_average_${VAR}_${TYPE}.boot_*[[:digit:]]*.lr_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$24 <= nboot {print $0}' > ${CALLING}_ann_population_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_population_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.bootmean.lr_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_population_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror.lr_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.bootmean.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_population_average_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt))) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.totalerror.lr_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt


#Other files can also be relativised:
##Empirical errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror_ki_rel.lr_ann.txt
##Bootstrap errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror_ki_rel.lr_ann.txt
##Bootstrap corrected errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected_ki_rel.lr_ann.txt

```

####Derived allele counts 5xonly.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate heterozygosity.
BLOCK_FILES=$(ls *bl*.lr_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#Next, generate some headers and info files.
tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.lr_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirerror.lr_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_5xgenome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=33;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=33;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xgenome_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.boot_1.lr_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.boot_*[[:digit:]]*.lr_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$24 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootmean.lr_ann.txt

#sed 's/ 0.0/ 1.1/g' OR sed 's/\t0.0/\t1.1/g'
#^ use this to remove 0s

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterror.lr_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootmean.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirerror.lr_ann.txt))) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror.lr_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt


#Other files can also be relativised:
##Empirical errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror_ki_rel.lr_ann.txt
##Bootstrap errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror_ki_rel.lr_ann.txt
##Bootstrap corrected errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected_ki_rel.lr_ann.txt

```

####Derived allele heterozygous counts (per site).
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate heterozygosity.
BLOCK_FILES=$(ls *bl*.lr_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#First, generate some headers and info files.
head -n1 <$(ls ${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-33 > ind_headers.txt #Retrieve headers for files with individuals
cut -f2,5- ind_headers.txt > pop_headers.txt #Retrieve headers for files with populations
tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_heterozygous_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.lr_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.lr_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirerror.lr_ann.txt


#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages (using 5x individuals only):
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_heterozygous_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_5xgenome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=19;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=19;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xgenome_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_heterozygous_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_heterozygous_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.boot_1.lr_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.boot_*[[:digit:]]*.lr_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",total[j","i]/nboot; printf "%.5f\n",total[j","NF]/nboot;}}' $(<${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootmean.lr_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; printf "%.5f\n",((sumsq[j","NF]-total[j","NF]*total[j","NF]/nboot)/(nboot-1))**0.5;}}' $(<${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterror.lr_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootmean.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirerror.lr_ann.txt))) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.lr_ann.txt

#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt. This file will be transposed to mimic the format of the population_average_heterozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):
awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt | grep 'callable_N_postfilter' > category_sizes.txt

#In order to relativise by the category size, divide both the population average and the total error by the size of the respective category:
##Population average:
cat pop_headers.txt <(paste pops.txt <(cat <(cut -f2- category_sizes.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_size_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(cut -f2- category_sizes.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_size_rel.lr_ann.txt

```

####Derived allele heterozygous counts (relative to Kirov).
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/$VAR


#All files used here were created in the previous section.

#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt

```

####Derived allele homozygous counts (relative to Kirov).
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_homozygous_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate homozygosity.
BLOCK_FILES=$(ls *bl*.lr_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#First, generate some headers and info files.
head -n1 <$(ls ${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-33 > ind_headers.txt #Retrieve headers for files with individuals
cut -f2,5- ind_headers.txt > pop_headers.txt #Retrieve headers for files with populations
tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_homozygous_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.lr_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.lr_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirerror.lr_ann.txt


#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages (using 5x individuals only):
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_homozygous_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_5xgenome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=19;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=19;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xgenome_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_homozygous_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_homozygous_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.boot_1.lr_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.boot_*[[:digit:]]*.lr_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",total[j","i]/nboot; printf "%.5f\n",total[j","NF]/nboot;}}' $(<${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootmean.lr_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; printf "%.5f\n",((sumsq[j","NF]-total[j","NF]*total[j","NF]/nboot)/(nboot-1))**0.5;}}' $(<${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterror.lr_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootmean.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirerror.lr_ann.txt))) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror.lr_ann.txt

#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt. This file will be transposed to mimic the format of the population_average_homozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):
awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt | grep 'callable_N_postfilter' > category_sizes.txt

#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt

```

####Telomere derived allele counts 5xonly.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate heterozygosity.
BLOCK_FILES=$(ls *bl*.lr_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#Next, generate some headers and info files.
#tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=6;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.lr_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=6;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirerror.lr_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of the telomere blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_'"$VAR"'_'"$TYPE"'./' ../../tel_blocks/bootstrapped_tel_${boot}.list > bootstrapped_5xtel_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=33;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=33;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xtel_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_tel_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_tel_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

#cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_tel_${VAR}_${TYPE}.boot_1.lr_ann.txt > pops.txt #Retrieve column with population data

#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.boot_*[[:digit:]]*.lr_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootmean.lr_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterror.lr_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootmean.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirerror.lr_ann.txt))) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.totalerror.lr_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt


#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt. This file will be transposed to mimic the format of the population_average_heterozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):
awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/category_sites_total_counts_telcentrfilter_corrected.lr_ann.txt | grep 'callable_N_postfilter' > category_sizes.txt

#In order to relativise by the category size, divide both the population average and the total error by the size of the respective category:
##Population average:
cat pop_headers.txt <(paste pops.txt <(cat <(cut -f2- category_sizes.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_size_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(cut -f2- category_sizes.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_size_rel.lr_ann.txt

```

####Centromere derived allele counts 5xonly.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate heterozygosity.
BLOCK_FILES=$(ls *bl*.lr_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#Next, generate some headers and info files.
#tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.bl*.lr_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=6;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.lr_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=6;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirerror.lr_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of the centromere blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 590 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_'"$VAR"'_'"$TYPE"'./' ../../centr_blocks/bootstrapped_centr_${boot}.list > bootstrapped_5xcentr_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=33;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=33;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xcentr_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_centr_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_centr_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.boot_${boot}.lr_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

#cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_centr_${VAR}_${TYPE}.boot_1.lr_ann.txt > pops.txt #Retrieve column with population data

#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.boot_*[[:digit:]]*.lr_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootmean.lr_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterror.lr_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootmean.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootcorrectionfactor.lr_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirerror.lr_ann.txt))) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.totalerror.lr_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.totalerror.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt

```

###Species level (run population level first).
####Derived allele counts.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#OBSOLETE SECTION. Weighted per species empirical mean: M = Sum(Mi*(1/ET^2i)) / Sum(1/ET2i), where i refers to each population in the species, and ET is the standard error.
##Create intermediate files with the numerator and the denominator components of the weighted mean for each population:
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", $i/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", 1/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt

##Sum the numerator and denominator over populations and divide both to obtain the weighted mean for each species:
#cat sp_headers.txt <(paste ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt | tail -n+2 | awk '{h=NF/2;N[$1]++; for (i=3;i<=NF;i++) {sum[$1","i] += $i;}} END {for (p in N) {printf "%s\t", p; for (i=3;i<h;i++) printf("%.8f\t", sum[p","i]/sum[p","i+h]); printf("%.8f\n", sum[p","h]/sum[p","h+h]);}}') > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirvarN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirvarN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirvarweight.lr_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirvarweight.lr_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirvarweight.lr_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirvarN.lr_ann.txt) ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_species_average_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_species_average_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_species_average_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt))) > ${CALLING}_ann_species_average_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_species_average_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_species_average_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt

```

####Derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#OBSOLETE SECTION. Weighted per species empirical mean: M = Sum(Mi*(1/ET^2i)) / Sum(1/ET2i), where i refers to each population in the species, and ET is the standard error.
##Create intermediate files with the numerator and the denominator components of the weighted mean for each population:
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", $i/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", 1/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt

##Sum the numerator and denominator over populations and divide both to obtain the weighted mean for each species:
#cat sp_headers.txt <(paste ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt | tail -n+2 | awk '{h=NF/2;N[$1]++; for (i=3;i<=NF;i++) {sum[$1","i] += $i;}} END {for (p in N) {printf "%s\t", p; for (i=3;i<h;i++) printf("%.8f\t", sum[p","i]/sum[p","i+h]); printf("%.8f\n", sum[p","h]/sum[p","h+h]);}}') > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirvarN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirvarN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirvarweight.lr_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirvarweight.lr_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirvarweight.lr_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirvarN.lr_ann.txt) ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt))) > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt

```

####Derived allele heterozygous counts.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#OBSOLETE SECTION. Weighted per species empirical mean: M = Sum(Mi*(1/ET^2i)) / Sum(1/ET2i), where i refers to each population in the species, and ET is the standard error.
##Create intermediate files with the numerator and the denominator components of the weighted mean for each population:
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", $i/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", 1/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt

##Sum the numerator and denominator over populations and divide both to obtain the weighted mean for each species:
#cat sp_headers.txt <(paste ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt | tail -n+2 | awk '{h=NF/2;N[$1]++; for (i=3;i<=NF;i++) {sum[$1","i] += $i;}} END {for (p in N) {printf "%s\t", p; for (i=3;i<h;i++) printf("%.8f\t", sum[p","i]/sum[p","i+h]); printf("%.8f\n", sum[p","h]/sum[p","h+h]);}}') > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirvarN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirvarN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirvarweight.lr_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_species_average_heterozygous_${VAR}_${TYPE}.empirvarweight.lr_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirvarweight.lr_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirvarN.lr_ann.txt) ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt))) > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt


#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt. This file will be transposed to mimic the format of the population_average_heterozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):
awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt | grep 'callable_N_postfilter' > category_sizes.txt


#In order to relativise by the category size, divide both the species average and the total error by the size of the respective category:
##Species average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight_size_rel.lr_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight_size_rel.lr_ann.txt

```

####Derived allele homozygous counts.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_homozygous_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#OBSOLETE SECTION. Weighted per species empirical mean: M = Sum(Mi*(1/ET^2i)) / Sum(1/ET2i), where i refers to each population in the species, and ET is the standard error.
##Create intermediate files with the numerator and the denominator components of the weighted mean for each population:
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", $i/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt
#paste species_column.txt <(cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=sprintf ("%.15f", 1/($(i+h)*$(i+h)));NF=h};1' | tr " " "\t")) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt

##Sum the numerator and denominator over populations and divide both to obtain the weighted mean for each species:
#cat sp_headers.txt <(paste ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.numerator.lr_ann.txt ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmeanweight.denominator.lr_ann.txt | tail -n+2 | awk '{h=NF/2;N[$1]++; for (i=3;i<=NF;i++) {sum[$1","i] += $i;}} END {for (p in N) {printf "%s\t", p; for (i=3;i<h;i++) printf("%.8f\t", sum[p","i]/sum[p","i+h]); printf("%.8f\n", sum[p","h]/sum[p","h+h]);}}') > ../../../${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.lr_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirvarN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirvarN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirvarweight.lr_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_species_average_homozygous_${VAR}_${TYPE}.empirvarweight.lr_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirvarweight.lr_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirvarN.lr_ann.txt) ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt))) > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt


#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt. This file will be transposed to mimic the format of the population_average_homozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):
awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt | grep 'callable_N_postfilter' > category_sizes.txt


#In order to relativise by the category size, divide both the species average and the total error by the size of the respective category:
##Species average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight_size_rel.lr_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.totalerrorweight_size_rel.lr_ann.txt

```

####Telomere derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
#echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=6;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=6;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirvarN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirvarN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirvarweight.lr_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirvarweight.lr_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirvarweight.lr_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirvarN.lr_ann.txt) ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt))) > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt

```

####Centromere derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
#echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=6;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=6;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirvarN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirvarN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirvarweight.lr_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirvarweight.lr_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirvarweight.lr_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirvarN.lr_ann.txt) ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.booterrorcorrectedweight.lr_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirerrorweight.lr_ann.txt))) > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt

```

####Xchr derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/$VAR

#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=6;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_Xchr_${VAR}_${TYPE}.lr_ann.txt | cut -f-34)) > ../../../${CALLING}_ann_5xpopulation_average_Xchr_${VAR}_${TYPE}.empirmeanN.lr_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_Xchr_${VAR}_${TYPE}.empirmeanN.lr_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_Xchr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt

#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_Xchr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_Xchr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_Xchr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_Xchr_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt

```

##TCRLP outerparsimony:
###Population level.
####Derived allele counts 5xonly.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate heterozygosity.
BLOCK_FILES=$(ls *bl*.anc_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#Next, generate some headers and info files.
#First, generate some headers and info files.
head -n1 <$(ls ${CALLING}_ann_individual_summary_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-27 > ind_headers.txt #Retrieve headers for files with individuals
cut -f2,5- ind_headers.txt > pop_headers.txt #Retrieve headers for files with populations
tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.anc_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirerror.anc_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_5xgenome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=27;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=27;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xgenome_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.boot_1.anc_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.boot_*[[:digit:]]*.anc_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$24 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootmean.anc_ann.txt

#sed 's/ 0.0/ 1.1/g' OR sed 's/\t0.0/\t1.1/g'
#^ use this to remove 0s

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterror.anc_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootmean.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterror.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirerror.anc_ann.txt))) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror.anc_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt


#Other files can also be relativised:
##Empirical errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror_ki_rel.anc_ann.txt
##Bootstrap errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterror_ki_rel.anc_ann.txt
##Bootstrap corrected errors divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected_ki_rel.anc_ann.txt

```

####Derived allele heterozygous counts (per site) 5x only.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate heterozygosity.
BLOCK_FILES=$(ls *bl*.anc_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#First, generate some headers and info files.
head -n1 <$(ls ${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-27 > ind_headers.txt #Retrieve headers for files with individuals
cut -f2,5- ind_headers.txt > pop_headers.txt #Retrieve headers for files with populations
tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_heterozygous_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.anc_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.anc_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirerror.anc_ann.txt


#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages (using 5x individuals only):
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_heterozygous_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_5xgenome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=16;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=16;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xgenome_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_heterozygous_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_heterozygous_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.boot_1.anc_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.boot_*[[:digit:]]*.anc_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",total[j","i]/nboot; printf "%.5f\n",total[j","NF]/nboot;}}' $(<${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootmean.anc_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; printf "%.5f\n",((sumsq[j","NF]-total[j","NF]*total[j","NF]/nboot)/(nboot-1))**0.5;}}' $(<${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterror.anc_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootmean.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterror.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirerror.anc_ann.txt))) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.anc_ann.txt

#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.anc_ann.txt. This file will be transposed to mimic the format of the population_average_heterozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):

grep -vE 'syn_pref|syn_unpref|UCNE_low' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt | awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' | grep 'callable_N_postfilter' > category_sizes.txt

#In order to relativise by the category size, divide both the population average and the total error by the size of the respective category:
##Population average:
cat pop_headers.txt <(paste pops.txt <(cat <(cut -f2- category_sizes.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_size_rel.anc_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(cut -f2- category_sizes.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_size_rel.anc_ann.txt

```

####Derived allele heterozygous counts (relative to Kirov) 5x only.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/$VAR


#All files used here were created in the previous section.

#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt

```

####Derived allele homozygous counts (relative to Kirov) 5x only.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_homozygous_counts/$VAR

#RUN THIS ONLY ONCE!! First, filter the per block individual counts to keep only the 5x individuals, which are the ones that we'll be using to estimate homozygosity.
BLOCK_FILES=$(ls *bl*.anc_ann.txt)
for file in ${BLOCK_FILES[@]}
  do
  awk '$3=="dataset" || $3=="5x"' ${file} > ${file/_individual_/_5xindividual_}
  done

#First, generate some headers and info files.
head -n1 <$(ls ${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-27 > ind_headers.txt #Retrieve headers for files with individuals
cut -f2,5- ind_headers.txt > pop_headers.txt #Retrieve headers for files with populations
tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_homozygous_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.anc_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.anc_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirerror.anc_ann.txt


#Next, generate several bootstrapped w-g counts by adding the counts of 2400 blocks pulled at random from the pool, and obtain the population averages (using 5x individuals only):
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_homozygous_'"$VAR"'_'"$TYPE"'./' ../../bootstrapped_genomes_coordinates/bootstrapped_genome_${boot}.list > bootstrapped_5xgenome_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=16;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=16;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xgenome_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_homozygous_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.5f\t",sum[p"."i]/N[p]); printf("%.5f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_homozygous_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.boot_1.anc_ann.txt > pops.txt #Retrieve column with population data


#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.boot_*[[:digit:]]*.anc_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",total[j","i]/nboot; printf "%.5f\n",total[j","NF]/nboot;}}' $(<${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootmean.anc_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; printf "%.5f\n",((sumsq[j","NF]-total[j","NF]*total[j","NF]/nboot)/(nboot-1))**0.5;}}' $(<${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterror.anc_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootmean.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterror.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirerror.anc_ann.txt))) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror.anc_ann.txt

#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.anc_ann.txt. This file will be transposed to mimic the format of the population_average_homozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):
grep -vE 'syn_pref|syn_unpref|UCNE_low' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/complete_category_sites_total_counts_percategoryfilter_corrected.lr_ann.txt | awk '{for (i=1; i<=NF; i++) {a[NR,i] = $i}} NF>p { p = NF } END {for(j=1; j<=p; j++) {str=a[1,j]; for(i=2; i<=NR; i++) {str=str"\t"a[i,j];} print str}}' | grep 'callable_N_postfilter' > category_sizes.txt

#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt

```

####Telomere derived allele counts 5xonly.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR

#Next, generate some headers and info files.
#tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.anc_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirerror.anc_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of the telomere blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_'"$VAR"'_'"$TYPE"'./' ../../tel_blocks/bootstrapped_tel_${boot}.list > bootstrapped_5xtel_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=27;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=27;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xtel_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_tel_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_tel_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

#cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_tel_${VAR}_${TYPE}.boot_1.anc_ann.txt > pops.txt #Retrieve column with population data

#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.boot_*[[:digit:]]*.anc_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootmean.anc_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterror.anc_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootmean.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterror.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirerror.anc_ann.txt))) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.totalerror.anc_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.totalerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt

```

####Centromere derived allele counts 5xonly.
```{bash}

N_BOOT=1000 #100 #1000
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR

#Next, generate some headers and info files.
#tail -n+2 <$(ls ${CALLING}_ann_5xindividual_summary_${VAR}_${TYPE}.bl*.anc_ann.txt | head -n1) | cut -f-4 > 5xids.txt #Retrieve first 4 columns with individual data

#Next, obtain the population average and sampling error (standard error) for the empirical data (5x only!!):
##Average:
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.anc_ann.txt

##Standard error (5x only!!): sqrt(variance(N)/N); we'll be using the N-1 version of the variance formula.
cat pop_headers.txt <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",(((sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1))/N[p])**0.5); printf("%.3f\n",(((sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1))/N[p])**0.5); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirerror.anc_ann.txt

#Next, generate several bootstrapped w-g counts by adding the counts of the centromere blocks pulled at random from the pool, and obtain the population averages:
for boot in $(seq 1 $N_BOOT) #100 (4mins) #1000
  do
  #First, edit the filenames of the input blocks so that they match the current variables:
  sed -e 's/^/'"$CALLING"'_ann_5xindividual_summary_'"$VAR"'_'"$TYPE"'./' ../../centr_blocks/bootstrapped_centr_${boot}.list > bootstrapped_5xcentr_${boot}.list
  #Next, generate the counts for each bootstrapped genome:
  echo "generating counts for bootstrapped genome number" $boot
  paste 5xids.txt <(awk '{for (i=5;i<=27;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=5;i<=27;i++) printf "%s\t",total[j","i]; print "";}}' $(cat bootstrapped_5xcentr_${boot}.list) | tail -n+2) > ${CALLING}_ann_5xindividual_summary_centr_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt
  #Next, obtain the population average for each bootstrap:
  echo "calculating population averages for bootstrapped genome number" $boot
  gawk '{N[$2]++; for (i=5;i<=NF;i++) {sum[$2"."i] += $i};} END {for (p in N) {printf "%s\t", p; for (i=5;i<NF;i++) printf("%.3f\t",sum[p"."i]/N[p]); printf("%.3f\n",sum[p"."NF]/N[p]);}}' ${CALLING}_ann_5xindividual_summary_centr_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.boot_${boot}.anc_ann.txt #(sum["ki."i]/N["ki"]) to relativise by Kirov
  done

#cut -f1 c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_centr_${VAR}_${TYPE}.boot_1.anc_ann.txt > pops.txt #Retrieve column with population data

#Next, obtain the population average of derived counts across bootstrapped genomes, but first store the correct files for the current $BOOT_N in an input list, so that only those (and not those from other bootstraps) are called:
ls ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.boot_*[[:digit:]]*.anc_ann.txt | awk -v nboot="$N_BOOT" -F"_|\\\\." '$25 <= nboot {print $0}' > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.Nboot_${N_BOOT}.list

cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) total[FNR","i]+=$i;} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.1f\t ",total[j","i]/nboot; print "";}}' $(<${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootmean.anc_ann.txt

#Next, obtain the bootstrap error of derived counts across bootstrapped genomes. The bootstrap error is the standard deviation of the N bootstraps, so it can be obtained as the sqrt(var(N)). We'll be using the N-1 version of the variance formula:
cat pop_headers.txt <(paste pops.txt <(awk -v nboot="$N_BOOT" '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END{for (j=1;j<=FNR;j++) {for (i=2;i<=NF;i++) printf "%.5f\t",((sumsq[j","i]-total[j","i]*total[j","i]/nboot)/(nboot-1))**0.5; print "";}}' $(<${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.Nboot_${N_BOOT}.list))) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterror.anc_ann.txt

#Since averages are different between the empirical data and the bootstrap, we need to correct the errors. 
##As a first step, the empirical means should be divided by the bootstrap means to obtain the correction factor for the bootstrap errors:
cat pop_headers.txt <(paste <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootmean.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i/$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt
##Then the errors can be multiplied by the correction factors in order to obtain the expected errors for the empirical means:
cat pop_headers.txt <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.bootcorrectionfactor.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterror.anc_ann.txt) | awk '{h=NF/2;for (i=2;i<=h;i++) $i=$i*$(i+h);NF=h};1' | tr " " "\t") > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt


#Per population total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the per population mean of the (corrected) bootstrap error, and ET is the per population empirical standard error (sampling error).
cat pop_headers.txt <(paste pops.txt <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=2;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirerror.anc_ann.txt))) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.totalerror.anc_ann.txt


#In order to relativise by the Kirov population average, divide both the population average and the total error by the Kirov empirical population averages:
##Empirical population averages divided by the empirical Kirov population average:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) | awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt
##Total error:
cat pop_headers.txt <(paste pops.txt <(cat <(grep 'ki' ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmean.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.totalerror.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=1; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt

```

###Species level (run population level first).
####Derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmeanN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmeanN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirvarN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirvarN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirvarweight.anc_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirvarweight.anc_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirvarweight.anc_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirvarN.anc_ann.txt) ${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt))) > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight_ll_rel.anc_ann.txt

```

####Derived allele heterozygous counts.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.anc_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmeanN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmeanN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_heterozygous_${VAR}_${TYPE}.anc_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirvarN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirvarN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirvarweight.anc_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirvarweight.anc_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirvarweight.anc_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirvarN.anc_ann.txt) ${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt))) > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt


#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/TCRLP_outerparsimony/complete_category_sites_total_counts_percategoryfilter_corrected.anc_ann.txt. This file will be transposed to mimic the format of the population_average_heterozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):

#In order to relativise by the category size, divide both the species average and the total error by the size of the respective category:
##Species average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight_size_rel.anc_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight_size_rel.anc_ann.txt

```

####Derived allele homozygous counts.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_homozygous_counts/$VAR

cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.anc_ann.txt)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmeanN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmeanN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.anc_ann.txt | cut -f-33)) > ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirvarN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirvarN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirvarweight.anc_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirvarweight.anc_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirvarweight.anc_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirvarN.anc_ann.txt) ${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt))) > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt


#Now relativise by the category size. 
##First we need to retrieve the category sizes. These were previously estimated as part of the script: "genetic_load_allpositions_final_pipeline.Rmd", and are stored in the following path: /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/TCRLP_outerparsimony/complete_category_sites_total_counts_percategoryfilter_corrected.anc_ann.txt. This file will be transposed to mimic the format of the population_average_homozygous files, and only the "callable_N_postfilter" row will be selected (i.e. the one with the estimated number of bp for each category after applying filters):

#In order to relativise by the category size, divide both the species average and the total error by the size of the respective category:
##Species average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight_size_rel.anc_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat category_sizes.txt <(tail -n+2 ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.8f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_homozygous_${VAR}_${TYPE}.totalerrorweight_size_rel.anc_ann.txt

```

####Telomere derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR

#cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
#echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmeanN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirmeanN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_tel_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirvarN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirvarN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirvarweight.anc_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirvarweight.anc_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirvarweight.anc_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.empirvarN.anc_ann.txt) ${CALLING}_ann_5xpopulation_average_tel_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt))) > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight_ll_rel.anc_ann.txt

```

####Centromere derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR

#cut -f1,5- ind_headers.txt > sp_headers.txt #Retrieve headers for files with species data
#echo -e "species\nll\nll\nlp\nlp\nll" > species_column.txt

#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size.
##First, for each population obtain its average mulitplied by the population size (the numerator of the weighted mean):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sum[p","i]/N[p])); printf("%.3f\n",N[p]*(sum[p","NF]/N[p])); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmeanN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirmeanN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt


#Weighted per species empirical variance (sampling variance): V = Sum(Vi*Ni) / Sum(Ni), where i refers to each population in the species, V is the sample variance, and N is the sample size.
##First, for each population obtain regular variance mulitplied by the population size (the numerator of the weighted variance):
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(gawk '$3=="5x" {N[$2]++; for (i=5;i<=NF;i++) {sum[$2","i] += $i; sumsq[$2","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=5;i<NF;i++) printf("%.3f\t",N[p]*(sumsq[p","i]-sum[p","i]*sum[p","i]/N[p])/(N[p]-1)); printf("%.3f\n",N[p]*(sumsq[p","NF]-sum[p","NF]*sum[p","NF]/N[p])/(N[p]-1)); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_centr_${VAR}_${TYPE}.anc_ann.txt | cut -f-27)) > ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirvarN.anc_ann.txt
##Then sum the numerator over populations and divide it by the sum of the populations' size (i.e. the species size) to obtain the weighted variance for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirvarN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirvarweight.anc_ann.txt


#Weighted per species empirical standard error (sampling error): ET(M) = (V/Sum(N))^0.5, where V is the sample variance, and N is the sample size.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirvarweight.anc_ann.txt | cut -f-2) <(awk '{for (i=3;i<=NF;i++) $i=sprintf("%.5f",($i/$2)**0.5);}1' <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirvarweight.anc_ann.txt) | tr " " "\t" | cut -f3-)) > ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt


#Weighted per species mean of the (corrected) bootstrap error: EB(M) = (Sum(EB^2)/P)^0.5 where EB is the (corrected) bootstrap error, and P the number of populations in the species.
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cut -f2 ../../../${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.empirvarN.anc_ann.txt) ${CALLING}_ann_5xpopulation_average_centr_${VAR}_${TYPE}.booterrorcorrected.anc_ann.txt | tail -n+2 | awk '{N[$1]++; for (i=2;i<=NF;i++) {total[$1","i]+=$i; sumsq[$1","i]+=$i*$i;}} END {for (p in N) {printf "%s\t%s\t", p,total[p","2]; for (i=4;i<NF;i++) printf("%.5f\t", (sumsq[p","i]/N[p])**0.5); printf("%.5f\n", (sumsq[p","NF]/N[p])**0.5);}}') > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt


#Weighted per species total error: Eglobal(M) = [EB^2(M) + ET^2(M)]^0.5 where EB is the weighted per species mean of the (corrected) bootstrap error, and ET is the weighted per species empirical standard error (sampling error).
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt | tail -n+2) <(awk '{for (i=2;i<=NF;i++) {total[FNR","i]+=$i; sumsq[FNR","i]+=$i*$i; }} END {for (j=1;j<=FNR;j++) {for (i=3;i<NF;i++) printf "%.8f\t",(sumsq[j","i])**0.5; printf "%.8f\n",(sumsq[j","NF])**0.5;}}' <(tail -n+2 ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.booterrorcorrectedweight.anc_ann.txt) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirerrorweight.anc_ann.txt))) > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
##Empirical population averages divided by the empirical Kirov population average:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt
##Total errors:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight_ll_rel.anc_ann.txt

```

####Xchr derived allele counts 5xonly.
```{bash}

#N_BOOT variable isn't used in this section. All files are inherited from the population level section, where N_BOOT is defined.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/$VAR


#Weighted per species empirical mean: M = Sum(Mi*Ni) / Sum(Ni), where i refers to each population in the species, and N is the sample size. However, since this is the Xchr, First, for each population obtain its X haploid average (derived counts divided by (the number of males + 2 times the number of females)) mulitplied by the population size (the numerator of the weighted mean). This will be achieved through multiple steps.
##First, obtain the number of males per population.
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N")) <(gawk '$3=="5x" && $5=="Male" {N[$2]++} END {for (p in N) {printf "%s\t%s\n", p, N[p]}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_Xchr_${VAR}_${TYPE}.anc_ann.txt | cut -f-5)) > ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.Nmales.anc_ann.txt
##Same for the females, but multiplied times 2.
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N")) <(gawk '$3=="5x" && $5=="Female" {N[$2]++} END {for (p in N) {printf "%s\t%s\n", p, 2*N[p]}}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_Xchr_${VAR}_${TYPE}.anc_ann.txt | cut -f-5)) > ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.Nfemales.anc_ann.txt
##Now, sum both datasets (= number of X chromosomes per population).
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N_Xchr")) <(paste <(tail -n+2 ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.Nmales.anc_ann.txt) <(tail -n+2 ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.Nfemales.anc_ann.txt) | awk '{printf "%s\t%s\n", $1, $2 + $4}') > ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.NXchr.anc_ann.txt
##Next, sum the individual counts across populations and divide them by the number of X chromosomes per population to obtain the proper population mean for the X chromosome.
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.NXchr.anc_ann.txt | tail -n+2) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i}} END {for (p in N) {printf "%s\t%s\t", p, N[p]; for (i=6;i<NF;i++) printf("%.3f\t",sum[p","i]); printf("%.3f\n",sum[p","NF]); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_Xchr_${VAR}_${TYPE}.anc_ann.txt | cut -f-28) | cut -f3-) | awk 'BEGIN {FS=OFS="\t"} {for (i=3;i<=NF;i++) $i=sprintf("%.5f",$i/$2)}1') > ../../../${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.empirmean.anc_ann.txt
##Now we can at last obtain for each population its average mulitplied by the Xchr population size (the numerator of the weighted mean), which is actually the same as the sum of all data in the population:
cat <(paste <(cut -f2 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(cut -f-2 ${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.NXchr.anc_ann.txt | tail -n+2) <(gawk '$3=="5x" {N[$2]++; for (i=6;i<=NF;i++) {sum[$2","i] += $i}} END {for (p in N) {for (i=6;i<NF;i++) printf("%.3f\t",sum[p","i]); printf("%.3f\n",sum[p","NF]); }}' <(tail -n+2 ../../../${CALLING}_ann_individual_summary_Xchr_${VAR}_${TYPE}.anc_ann.txt | cut -f-28))) > ../../../${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.empirmeanN.anc_ann.txt 
##Finally, sum the numerator over populations and divide it by the sum of the Xchr populations' size (i.e. the Xchr species size) to obtain the weighted mean for each species:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste species_column.txt <(cat pop_headers.txt <(tail -n+2 ../../../${CALLING}_ann_5xpopulation_average_correctedXchr_${VAR}_${TYPE}.empirmeanN.anc_ann.txt)) | tail -n+2 | awk '{N[$1]++; for (i=3;i<=NF;i++) {total[$1","i]+=$i; }} END {for (p in N) {printf "%s\t%s\t", p,total[p","3]; for (i=4;i<NF;i++) printf("%.5f\t", total[p","i]/total[p","3]); printf("%.5f\n", total[p","NF]/total[p","3]);}}') > ../../../${CALLING}_ann_5xspecies_average_correctedXchr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt


#In order to relativise by the Lynx lynx average, divide both the species average and the total error by the Eurasian species averages:
cat <(paste <(cut -f1 ind_headers.txt) <(echo "N") <(cut -f5- ind_headers.txt)) <(paste <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_correctedXchr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f1) <(cat <(grep -w 'll' ../../../${CALLING}_ann_5xspecies_average_correctedXchr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) <(tail -n+2 ../../../${CALLING}_ann_5xspecies_average_correctedXchr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt | cut -f2-) | column -t |
awk 'BEGIN {OFS = "\t"} NR == 1 {cols = split($0,m);next} NF == cols {for (i=2; i<=NF; i++) $i = sprintf ("%.5f", $i/m[i])}1')) > ${CALLING}_ann_5xspecies_average_correctedXchr_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt

```

#7: Download files:
##Rufus polarisation:
###Population level.
####Derived allele counts.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_population_average_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_population_average_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Derived allele counts 5xonly.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Derived allele heterozygous counts (size rel).
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_size_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_size_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Derived allele heterozygous counts (ki rel).
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Derived allele homozygous counts (ki rel).
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_homozygous_counts/${VAR}/${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror_ki_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

###Species level.
####Derived allele counts.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_species_average_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_species_average_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_species_average_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Derived allele counts 5xonly.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Derived allele heterozygous counts.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight_size_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight_size_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Telomere derived allele counts 5xonly.
#####Relative to Lynx lynx:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

#####Absolute:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Centromere derived allele counts 5xonly.
#####Relative to Lynx lynx:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

#####Absolute:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

####Xchr derived allele counts 5xonly.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_Xchr_${VAR}_${TYPE}.empirmeanweight_ll_rel.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

##TCRLP outerparsimony:
###Population level.
####Derived allele counts 5xonly.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Derived allele heterozygous counts (size rel).
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_size_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_size_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Derived allele heterozygous counts (ki rel).
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xpopulation_average_heterozygous_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Derived allele homozygous counts (ki rel).
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.empirmean_ki_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_homozygous_counts/${VAR}/${CALLING}_ann_5xpopulation_average_homozygous_${VAR}_${TYPE}.totalerror_ki_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_individual_summary_homozygous_${VAR}_${TYPE}.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

###Species level.
####Derived allele counts 5xonly.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(fixed) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_${VAR}_${TYPE}.totalerrorweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Derived allele heterozygous counts.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.empirmeanweight_size_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_heterozygous_counts/${VAR}/${CALLING}_ann_5xspecies_average_heterozygous_${VAR}_${TYPE}.totalerrorweight_size_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Telomere derived allele counts 5xonly.
#####Relative to Lynx lynx:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

#####Absolute:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_tel_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Centromere derived allele counts 5xonly.
#####Relative to Lynx lynx:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

#####Absolute:
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.empirmeanweight.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_centr_${VAR}_${TYPE}.totalerrorweight.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```

####Xchr derived allele counts 5xonly.
```{bash}

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/TCRLP_outerparsimony/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_5xspecies_average_correctedXchr_${VAR}_${TYPE}.empirmeanweight_ll_rel.anc_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony
unset SSHPASS

```


#8: Plot obsolete results.

##Population level.
###Derived allele counts.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="fixed" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_population_average_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_population_average_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","ratio"))
colnames(combined_tidy) <- c("population","ratio","mean","error")

combined_tidy$population = factor(combined_tidy$population,levels=c("ki","no","po","sm","do"))
levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","intronic","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
levels(combined_tidy$ratio) <- c("total","intergenic","introns","synonymous","synonymous_pref","synonymous_unpref","missense","missense_tolerated","missense_deleterious","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")
write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_relative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")

#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|nonsense|\\<UCNE\\>',ratio)), aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('synonymous_pref|synonymous_unpref|missense_tolerated|missense_deleterious|UCNE_mid|UCNE_high',ratio)), aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

ggplot_other <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE_low",], aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.text=element_text(colour="black",face="bold"),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other


if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Average genetic load relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to ki")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_relative2Kirov_bootstrapped.pdf"), width=41, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

```

###Derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")

#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.6,0.6,0.5,0.5),"max"=c(1.2,1.5,2,4),"breaks"=c(0.1,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5, size=1) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5, size=1) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_middle

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5, size=1) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

ggplot_other <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-11.9),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other


#Old:
if(type=="varssubs") {
  old_ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
} else if (type=="fixed") {
  old_ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

#Old:
ggsave(paste0(type,"_genetic_load_5xrelative2Kirov_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",old_ggplot_combined)

#Alt:
if(type=="varssubs") {
  alt_ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(13,0.1),layout_matrix=rbind(c(1,1),c(2,NA)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
} else if (type=="fixed") {
  alt_ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

#Alt:
ggsave(paste0(type,"_genetic_load_5xrelative2Kirov_bootstrapped.pdf"), width=40, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",alt_ggplot_combined)

#New:
if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_middle,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
} else if (type=="fixed") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

#New:
ggsave(paste0(type,"_genetic_load_5xrelative2Kirov_bootstrapped.pdf"), width=26, height=38, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

```

###Derived allele heterozygous counts (size rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.empirmean_size_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.totalerror_size_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_heterozygosity_5x_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%7.1e", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|synonymous',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up


average_ggplot_all_pi_downleft <- ggplot(data=filter(combined_tidy,grepl('missense|LoF|^UCNE$|moderate|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downleft


average_ggplot_all_pi_downright <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  scale_y_continuous(position="right",labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,2,0.5,-4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downright


old_average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_downleft,average_ggplot_all_pi_downright), set_panel_size,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,0.21,1),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average nucleotide heterozygosity")),rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))

average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_downleft), set_panel_size,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(1,1),layout_matrix=rbind(c(1,1),c(2,2)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average nucleotide heterozygosity")),rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))

ggsave(paste0(type,"_heterozygosity_population_5xpersite_bootstrapped.pdf"), width=30, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

```

###Derived allele heterozygous counts (ki rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)


averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high"))
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_heterozygosity_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%#.2f", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_middle


average_ggplot_all_pi_downleft <- ggplot(data=filter(combined_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downleft


average_ggplot_all_pi_downright <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2),position="right") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,2,0.5,-4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downright

#Old:
old_average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_downleft,average_ggplot_all_pi_downright), set_panel_size,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(2,0.53,0.54),layout_matrix=rbind(c(1,1,1),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average heterozygosity relative to KIR")),rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=2))

#Old:
ggsave(paste0(type,"_heterozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",old_average_ggplot_all_pi_combined)

#New:
average_ggplot_all_pi_combined <- grid.arrange(set_panel_size(average_ggplot_all_pi_up,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_ggplot_all_pi_middle,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_ggplot_all_pi_downleft,width=unit(4.5,"cm"),height=unit(10,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average heterozygosity relative to KIR")),rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))

#New:
ggsave(paste0(type,"_heterozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=26, height=38, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

```

###Derived allele homozygous counts (ki rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_homozygosity_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%#.2f", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_middle


average_ggplot_all_pi_downleft <- ggplot(data=filter(combined_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downleft


average_ggplot_all_pi_downright <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(population,mean,colour=population)) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4,1.6,1.8),limits=c(0.7,1.8),position="right") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,2,0.5,-4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downright

#Old:
average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_downleft,average_ggplot_all_pi_downright), set_panel_size,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(2,0.53,0.54),layout_matrix=rbind(c(1,1,1),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average derived homozygosity relative to KIR")),rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=2))

#Old:
ggsave(paste0(type,"_homozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

#New:
average_ggplot_all_pi_combined <- grid.arrange(set_panel_size(average_ggplot_all_pi_up,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_ggplot_all_pi_middle,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(average_ggplot_all_pi_downleft,width=unit(4.5,"cm"),height=unit(10,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average derived homozygosity relative to KIR")),rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))

#New:
ggsave(paste0(type,"_homozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=26, height=38, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

```

##Species level.
###Derived allele counts.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="fixed" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_species_average_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_species_average_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")

combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","intronic","synonymous","syn_pref","syn_unpref","missense","missense_tol","missense_del","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high"))
levels(combined_tidy$ratio) <- c("total","intergenic","introns","synonymous","synonymous_pref","synonymous_unpref","missense","missense_tolerated","missense_deleterious","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")
write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_species_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|nonsense|\\<UCNE\\>',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('synonymous_pref|synonymous_unpref|missense_tolerated|missense_deleterious|UCNE_mid|UCNE_high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

ggplot_other <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE_low",], aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.text=element_text(colour="black",face="bold"),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other


if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Average genetic load relative to Lynx lynx")~bolditalic("Lynx lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,1,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to")~bolditalic("Lynx lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_species_relative2ll_bootstrapped.pdf"), width=41, height=25, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)



#####Not relative to Lynx lynx####
wrap_ggplot <- ggplot(data=combined_tidy, aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio,scales="free",nrow = 2) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load per species",ifelse(type=="fixed","Derived fixation rate per species", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
  wrap_ggplot
ggsave(paste0(type,"_genetic_load_species_bootstrapped.pdf"), width=45, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",wrap_ggplot)

```

###Derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="fixed" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")

#combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
levels(combined_tidy$species) <- c("EL","IL")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.6,0.6,0.5,0.5),"max"=c(1.2,1.5,2,4),"breaks"=c(0.1,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_middle

#Add asterisk
middle_ann <- data.frame(species="IL",mean=1.15,lab="*",ratio=factor("LoF",levels=c("missense","LoF","UCNE")))
ggplot_middle <- ggplot_middle + geom_text(data=middle_ann,label="*",size=10, colour="black")


ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

#Add asterisk
down_ann <- data.frame(species="IL",mean=1.15,lab="*",ratio=factor("m. deleterious",levels=c("m. toelrated","m. deleterious","UCNE moderate","UCNE high")))
ggplot_down <- ggplot_down + geom_text(data=down_ann,label="*",size=10, colour="black")


ggplot_other <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-11.9),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other

#Old:
if(type=="varssubs") {
  old_ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
} else if (type=="fixed") {
  old_ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

#Old:
ggsave(paste0(type,"_genetic_load_5xspecies_relative2ll_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",old_ggplot_combined)

#New:
if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_middle,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
} else if (type=="fixed") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_middle,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_5xspecies_relative2ll_bootstrapped.pdf"), width=26, height=38, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

```

###Derived allele heterozygous counts.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_heterozygous_",type,"_SNP.empirmeanweight_size_rel.lr_ann.txt")) %>% select(.,species,contains("_H")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_heterozygous_",type,"_SNP.totalerrorweight_size_rel.lr_ann.txt")) %>% select(.,species,contains("_H")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","mistol","misdel","nonsense","UCNE","UCNE_low","UCNE_mid","UCNE_high")) 
combined_tidy
write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_heterozygosity_species_5xrelative2ll_bootstrapped.txt"), delim = "\t", na = "NA")

#Plots:
scaleFUN <- function(x) sprintf("%7.1e", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|synonymous|missense',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,0,0.5,1),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up


average_ggplot_all_pi_downleft <- ggplot(data=filter(combined_tidy,grepl('nonsense|\\<UCNE\\>|UCNE_mid|UCNE_high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,0,0.5,-2.8),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downleft


average_ggplot_all_pi_downright <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE_low",], aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  #ylab(paste0(type," population average ",plot_variable)) +
  scale_y_continuous(position="right",labels=scaleFUN) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      #axis.line=element_line(colour="black",size=1.5),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,2,0.5,-4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_downright


average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_downleft,average_ggplot_all_pi_downright), set_panel_size,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(7,0.2,1),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")))

ggsave(paste0(type,"_heterozygosity_species_5xpersite_bootstrapped.pdf"), width=32, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

```

```{bash}

#TOTAL:
bcftools view -Ou c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_aafilled.vcf | bcftools query -f '%REF\t%INFO/AA\n' | sed '/^.\{3\}./d' | wc -l #5324385

#REF=ANC:
bcftools view -Ou c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_aafilled.vcf | bcftools query -f '%REF\t%INFO/AA\n' | sed '/^.\{3\}./d' | awk '$1==$2' | wc -l #3604906

#Not polarisable:
bcftools view -Ou c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_aafilled.vcf | bcftools query -f '%REF\t%INFO/AA\n' | sed '/^.\{3\}./d' | awk '$2=="N"' | wc -l #86151

#REF=DER (substract the others):
#1633328

```

###Telomere derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")

combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
levels(combined_tidy$species) <- c("EL","IL")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_tel_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.2,0.2,0.5,0.5),"max"=c(1.7,1.7,2,4),"breaks"=c(0.2,0.2,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|LoF|^UCNE$',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

ggplot_other <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other


if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Average genetic load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_tel_5xspecies_relative2ll_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

```

###Centromere derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")

combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
levels(combined_tidy$species) <- c("EL","IL")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tolerated', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_centr_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.2,0.2,0.5,0.5),"max"=c(1.7,1.7,2,4),"breaks"=c(0.2,0.2,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|LoF|^UCNE$',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

ggplot_other <- ggplot(combined_tidy[combined_tidy$ratio=="UCNE low",], aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other


if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Average genetic load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_centr_5xspecies_relative2ll_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

```

###Xchr derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_Xchr_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
colnames(averages_tidy) <- c("species","ratio","mean")
averages_tidy$mean <- as.numeric(averages_tidy$mean)

averages_tidy$species = factor(averages$species,levels=c("ll","lp"))
levels(averages_tidy$species) <- c("EL","IL")
averages_tidy$ratio <- gsub('intronic', 'introns', averages_tidy$ratio)
averages_tidy$ratio <- gsub('coding', 'CDS', averages_tidy$ratio)
averages_tidy$ratio <- gsub('missense_tol', 'm. tolerated', averages_tidy$ratio)
averages_tidy$ratio <- gsub('missense_del', 'm. deleterious', averages_tidy$ratio)
averages_tidy$ratio <- gsub('nonsense', 'LoF', averages_tidy$ratio)
averages_tidy$ratio <- gsub('UCNE_low', 'UCNE low', averages_tidy$ratio)
averages_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', averages_tidy$ratio)
averages_tidy$ratio <- gsub('UCNE_high', 'UCNE high', averages_tidy$ratio)
#levels(averages_tidy$population)[levels(averages_tidy$population)=="sm"] <- "an"
averages_tidy$ratio = factor(averages_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 
write_delim(averages_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_Xchr_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.2,0.2,0.5,0.5),"max"=c(1.7,2.2,2,4),"breaks"=c(0.2,0.2,0.2,0.5))

ggplot_up <- ggplot(data=filter(averages_tidy,grepl('total|intergenic|introns|\\<synonymous\\>|\\<missense\\>|LoF|^UCNE$',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  #geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_down <- ggplot(data=filter(averages_tidy,grepl('tolerated|deleterious|moderate|high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  #geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

ggplot_other <- ggplot(averages_tidy[averages_tidy$ratio=="UCNE low",], aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_wrap(. ~ ratio) +
  geom_point() +
  #geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  #ylab("Average genetic load relative to ki") +
  scale_y_continuous(position="right",labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="other") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="other") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="other") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.text.y=element_text(size=12,colour="black"),
      axis.title.y=element_blank(),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,-1,0.5,-10),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_other


if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Average genetic load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
} else if (type=="fixed") {
    ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
}

ggsave(paste0(type,"_genetic_load_Xchr_5xspecies_relative2ll_bootstrapped.pdf"), width=41, height=26, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

```

###Telomere vs. centromere derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

tel_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.empirmeanweight.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
tel_averages$species = factor(tel_averages$species,levels=c("ll","lp"))

tel_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.totalerrorweight.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
tel_errors$species = factor(tel_averages$species,levels=c("ll","lp"))

tel_averages_tidy <- tel_averages %>% gather(ratio,value,-species,factor_key=T)
tel_averages_tidy

tel_errors_tidy <- tel_errors %>% gather(ratio,value,-species,factor_key=T)
tel_errors_tidy

tel_combined_tidy <- left_join(tel_averages_tidy,tel_errors_tidy,by=c("species","ratio"))
colnames(tel_combined_tidy) <- c("species","ratio","tel_mean","tel_error")

tel_combined_tidy$species = factor(tel_averages$species,levels=c("ll","lp"))
levels(tel_combined_tidy$species) <- c("EL","IL")
tel_combined_tidy$ratio <- gsub('intronic', 'introns', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('coding', 'CDS', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('missense_tol', 'm. tolerated', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('nonsense', 'LoF', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', tel_combined_tidy$ratio)
tel_combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', tel_combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
tel_combined_tidy$ratio = factor(tel_combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 

centr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.empirmeanweight.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
centr_averages$species = factor(centr_averages$species,levels=c("ll","lp"))

centr_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.totalerrorweight.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
centr_errors$species = factor(centr_averages$species,levels=c("ll","lp"))

centr_averages_tidy <- centr_averages %>% gather(ratio,value,-species,factor_key=T)
centr_averages_tidy

centr_errors_tidy <- centr_errors %>% gather(ratio,value,-species,factor_key=T)
centr_errors_tidy

centr_combined_tidy <- left_join(centr_averages_tidy,centr_errors_tidy,by=c("species","ratio"))
colnames(centr_combined_tidy) <- c("species","ratio","centr_mean","centr_error")

centr_combined_tidy$species = factor(centr_averages$species,levels=c("ll","lp"))
levels(centr_combined_tidy$species) <- c("EL","IL")
centr_combined_tidy$ratio <- gsub('intronic', 'introns', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('coding', 'CDS', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('missense_tol', 'm. tolerated', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('nonsense', 'LoF', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE moderate', centr_combined_tidy$ratio)
centr_combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', centr_combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
centr_combined_tidy$ratio = factor(centr_combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","synonymous","syn_pref","syn_unpref","missense","m. tolerated","m. deleterious","LoF","UCNE","UCNE low","UCNE moderate","UCNE high")) 

combined_tidy <- cbind(tel_combined_tidy,centr_combined_tidy[,c(3:4)])
write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_tel_vs_centr_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")

```

###Tel vs. cen vs. Xchr derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

tel_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="tel")
tel_averages$species = factor(tel_averages$species,levels=c("ll","lp"))

centr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="centr")
centr_averages$species = factor(centr_averages$species,levels=c("ll","lp"))

Xchr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_Xchr_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="Xchr")
Xchr_averages$species = factor(Xchr_averages$species,levels=c("ll","lp"))
print.data.frame(Xchr_averages)

all_averages <- rbind(tel_averages,centr_averages,Xchr_averages)
all_averages

tel_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="tel")
tel_errors$species = factor(tel_averages$species,levels=c("ll","lp"))

centr_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="centr")
centr_errors$species = factor(centr_averages$species,levels=c("ll","lp"))

all_errors <- rbind(tel_errors,centr_errors)
all_errors

all_averages_tidy <- all_averages %>% gather(ratio,value,-species,-region,factor_key=T) %>% filter(grepl("missense_del",ratio))
all_averages_tidy

all_errors_tidy <- all_errors %>% gather(ratio,value,-species,-region,factor_key=T) %>% filter(grepl("missense_del",ratio))
all_errors_tidy

all_combined_tidy <- full_join(all_averages_tidy,all_errors_tidy,by=c("species","ratio","region"))
colnames(all_combined_tidy) <- c("species","region","ratio","mean","error")

all_combined_tidy$species = factor(all_combined_tidy$species,levels=c("ll","lp"))
levels(all_combined_tidy$species) <- c("EL","IL")
all_combined_tidy$ratio <- gsub('missense_del', 'm. deleterious', all_combined_tidy$ratio)
all_combined_tidy$region <- gsub('tel', 'telomere', all_combined_tidy$region)
all_combined_tidy$region <- gsub('centr', 'centromere', all_combined_tidy$region)
all_combined_tidy$region <- gsub('Xchr', 'X chr.', all_combined_tidy$region)
all_combined_tidy$region = factor(all_combined_tidy$region,levels=c("telomere","centromere","X chr.")) 

#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.2,0.5,0.5),"max"=c(1.35,1.7,2,4),"breaks"=c(0.1,0.2,0.2,0.5))

ggplot_misdel <- ggplot(data=all_combined_tidy, aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ region) +
  geom_point() +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genomic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=30,hjust=1,size=12,colour="black"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(size=12,colour="black"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA,size=1.5),
      strip.background=element_rect(colour="black",size=1.5),
      strip.text=element_text(size=14),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_misdel

ggplot_misdel_final <- grid.arrange(set_panel_size(ggplot_misdel,width=unit(4.5,"cm"),height=unit(10,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=16,fontface="bold"),vjust=-2),left=textGrob(expression(bold("Average m. deleterious load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=16,fontface="bold"),vjust=3))
ggplot_misdel_final

ggsave(paste0(type,"_genetic_load_telcentrXchr_5xspecies_relative2ll_bootstrapped.pdf"), width=20, height=16, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_misdel_final)

```

#9: Plot manuscript versions.
##Rufus polarisation:
###Population level.
####Derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_genetic_load_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")

#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.6,0.6,0.5,0.5),"max"=c(1.2,1.5,2,4),"breaks"=c(0.1,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_middle

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

#New:
if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(1.65,"cm"),height=unit(4,"cm")), set_panel_size(ggplot_middle,width=unit(1.65,"cm"),height=unit(4,"cm")), set_panel_size(ggplot_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to KIR")), rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))
} else if (type=="fixed") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

#New:
ggsave(paste0(type,"_genetic_load_5xrelative2Kirov_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_combined)

#Stop
STOP

#Presentation version:
ggplot_pptx <- ggplot(data=filter(combined_tidy,grepl('intergenic|tol|del|LoF',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_pptx

ggplot_pptx_final <- grid.arrange(set_panel_size(ggplot_pptx,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=8,fontface="bold"),vjust=-2),left=textGrob(expression(bold("Average genomic load relative to KIR")),rot=90,gp=gpar(fontsize=8,fontface="bold"),vjust=0.5))

ggsave(paste0(type,"_genetic_load_5xrelative2Kirov_pptx.pdf"), width=8.2, height=7, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/thesis_presentation",ggplot_pptx_final)

```

####Derived allele heterozygous counts (size rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.empirmean_size_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.totalerror_size_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_heterozygosity_5x_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%7.1e", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_down <- ggplot(data=filter(combined_tidy,grepl('missense|LoF|^UCNE$|mod.|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_down

average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_down), set_panel_size,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average nucleotide heterozygosity")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=0.5))

ggsave(paste0(type,"_heterozygosity_population_5xpersite_bootstrapped.pdf"), width=10.3, height=11.6, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

#STOP
STOP

```

####Derived allele heterozygous counts (ki rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)


averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_heterozygosity_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%#.2f", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_middle

average_ggplot_all_pi_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_down

#New:
average_ggplot_all_pi_combined <- grid.arrange(set_panel_size(average_ggplot_all_pi_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average heterozygosity relative to KIR")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))

#New:
ggsave(paste0(type,"_heterozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

#STOP
STOP


#Presentation version:
average_ggplot_all_pi_pptx <- ggplot(data=filter(combined_tidy,grepl('intergenic|tol|del|LoF',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_pptx

average_ggplot_all_pi_pptx_final <- grid.arrange(set_panel_size(average_ggplot_all_pi_pptx,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=8,fontface="bold"),vjust=-2),left=textGrob(expression(bold("Average heterozygosity relative to KIR")),rot=90,gp=gpar(fontsize=8,fontface="bold"),vjust=0.5))

ggsave(paste0(type,"_heterozygosity_population_5xrelative2Kirov_pptx.pdf"), width=8.2, height=7, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/thesis_presentation",average_ggplot_all_pi_pptx_final)

```

####Derived allele homozygous counts (ki rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_",type,"_SNP.empirmean_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_",type,"_SNP.totalerror_ki_rel.lr_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/",type,"_homozygosity_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%#.2f", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_middle

average_ggplot_all_pi_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_down

#New:
average_ggplot_all_pi_combined <- grid.arrange(set_panel_size(average_ggplot_all_pi_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average derived homozygosity relative to KIR")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))

#New:
ggsave(paste0(type,"_homozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",average_ggplot_all_pi_combined)

#Stop
STOP


#Presentation version:
average_ggplot_all_pi_pptx <- ggplot(data=filter(combined_tidy,grepl('intergenic|tol|del|LoF',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.7,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_pptx


average_ggplot_all_pi_pptx_final <- grid.arrange(set_panel_size(average_ggplot_all_pi_pptx,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=8,fontface="bold"),vjust=-2),left=textGrob(expression(bold("Average derived homozygosity relative to KIR")),rot=90,gp=gpar(fontsize=8,fontface="bold"),vjust=0.5))

ggsave(paste0(type,"_homozygosity_population_5xrelative2Kirov_pptx.pdf"), width=8.2, height=7, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/thesis_presentation",average_ggplot_all_pi_pptx_final)


```

###Species level.
####Derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="fixed" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")

#combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
levels(combined_tidy$species) <- c("EL","IL")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0(wd_path,type,"_genetic_load_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.6,0.6,0.5,0.5),"max"=c(1.2,1.5,2,4),"breaks"=c(0.1,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_middle

#Add asterisk
if (type=="varssubs")
{
middle_ann <- data.frame(species="IL",mean=1.15,lab="*",ratio=factor("LoF",levels=c("missense","LoF","UCNE")))
ggplot_middle <- ggplot_middle + geom_text(data=middle_ann,label="*",size=6, colour="black")
}

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

#Add asterisk
if (type=="varssubs")
{
down_ann <- data.frame(species="IL",mean=1.15,lab="*",ratio=factor("m. del.",levels=c("m. tol.","m. del.","UCNE mod.","UCNE high")))
ggplot_down <- ggplot_down + geom_text(data=down_ann,label="*",size=6, colour="black")
}

#New:
if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average genomic load relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))
} else if (type=="fixed") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))
}

ggsave(paste0(type,"_genetic_load_5xspecies_relative2ll_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path=wd_path,ggplot_combined)

#Stop
STOP


#Presentation version:
ggplot_pptx <- ggplot(data=filter(combined_tidy,grepl('intergenic|tol|del|LoF',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_pptx


#New:
if(type=="varssubs") {
  ggplot_pptx_final <- grid.arrange(set_panel_size(ggplot_pptx,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=8,fontface="bold"),vjust=-2),left=textGrob(expression(bold("Average genomic load relative to Eurasian lynx")),rot=90,gp=gpar(fontsize=8,fontface="bold"),vjust=0.5))
} else if (type=="fixed") {
ggplot_pptx_final <- grid.arrange(set_panel_size(ggplot_pptx,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=8,fontface="bold"),vjust=-2),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")),rot=90,gp=gpar(fontsize=8,fontface="bold"),vjust=0.5))
}

ggsave(paste0(type,"_genetic_load_5xspecies_relative2ll_pptx.pdf"), width=8.2, height=7, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/thesis_presentation",ggplot_pptx_final)

```

####Tel vs. cen vs. Xchr derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/")

tel_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="tel")
tel_averages$species = factor(tel_averages$species,levels=c("ll","lp"))

centr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="centr")
centr_averages$species = factor(centr_averages$species,levels=c("ll","lp"))

Xchr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_Xchr_",type,"_SNP.empirmeanweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="Xchr")
Xchr_averages$species = factor(Xchr_averages$species,levels=c("ll","lp"))
print.data.frame(Xchr_averages)

all_averages <- rbind(tel_averages,centr_averages,Xchr_averages)
all_averages

tel_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="tel")
tel_errors$species = factor(tel_averages$species,levels=c("ll","lp"))

centr_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.totalerrorweight_ll_rel.lr_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="centr")
centr_errors$species = factor(centr_averages$species,levels=c("ll","lp"))

all_errors <- rbind(tel_errors,centr_errors)
all_errors

all_averages_tidy <- all_averages %>% gather(ratio,value,-species,-region,factor_key=T) %>% filter(grepl("missense_del",ratio))
all_averages_tidy

all_errors_tidy <- all_errors %>% gather(ratio,value,-species,-region,factor_key=T) %>% filter(grepl("missense_del",ratio))
all_errors_tidy

all_combined_tidy <- full_join(all_averages_tidy,all_errors_tidy,by=c("species","ratio","region"))
colnames(all_combined_tidy) <- c("species","region","ratio","mean","error")

all_combined_tidy$species = factor(all_combined_tidy$species,levels=c("ll","lp"))
levels(all_combined_tidy$species) <- c("EL","IL")
all_combined_tidy$ratio <- gsub('missense_del', 'm. del.', all_combined_tidy$ratio)
all_combined_tidy$region <- gsub('tel', 'telomere', all_combined_tidy$region)
all_combined_tidy$region <- gsub('centr', 'centromere', all_combined_tidy$region)
all_combined_tidy$region <- gsub('Xchr', 'X chr.', all_combined_tidy$region)
all_combined_tidy$region = factor(all_combined_tidy$region,levels=c("telomere","centromere","X chr.")) 

#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.2,0.5,0.5),"max"=c(1.35,1.7,2,4),"breaks"=c(0.1,0.2,0.2,0.5))

ggplot_misdel <- ggplot(data=all_combined_tidy, aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ region) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genomic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_misdel

ggplot_misdel_final <- grid.arrange(set_panel_size(ggplot_misdel,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average m. deleterious load\n   relative to Eurasian lynx")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=2))
ggplot_misdel_final

ggsave(paste0(type,"_genetic_load_telcentrXchr_5xspecies_relative2ll_bootstrapped.pdf"), width=7.4, height=6, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap",ggplot_misdel_final)

```

##TCRLP outerparsimony:
###Population level.
####Derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_",type,"_SNP.empirmean_ki_rel.anc_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_",type,"_SNP.totalerror_ki_rel.anc_ann.txt")) %>% select(.,population,contains("_A"),-contains("/")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/",type,"_genetic_load_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")

#Separate plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.6,0.6,0.5,0.5),"max"=c(1.2,1.5,2,4),"breaks"=c(0.1,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_middle

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to ki",ifelse(type=="fixed","Derived fixation rate relative to ki", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.4),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

#New:
if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(1.65,"cm"),height=unit(4,"cm")), set_panel_size(ggplot_middle,width=unit(1.65,"cm"),height=unit(4,"cm")), set_panel_size(ggplot_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average derived count relative to KIR")), rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))
} else if (type=="fixed") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(4.5,"cm"),height=unit(10,"cm")), set_panel_size(ggplot_down,width=unit(4.5,"cm"),height=unit(10,"cm")),set_panel_size(ggplot_other,width=unit(4.5,"cm"),height=unit(10,"cm")),widths=c(4,2.3,0.2),layout_matrix=rbind(c(1,1,NA),c(2,NA,3)),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=18,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to KIR")), rot=90,gp=gpar(fontsize=18,fontface="bold"),vjust=3))
}

#New:
ggsave(paste0(type,"_genetic_load_5xrelative2Kirov_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony",ggplot_combined)


#Stop
STOP

```

####Derived allele heterozygous counts (size rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.empirmean_size_rel.anc_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.totalerror_size_rel.anc_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/",type,"_heterozygosity_5x_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%7.1e", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_down <- ggplot(data=filter(combined_tidy,grepl('missense|LoF|^UCNE$|mod.|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_down

average_ggplot_all_pi_combined <- grid.arrange(grobs=lapply(list(average_ggplot_all_pi_up,average_ggplot_all_pi_down), set_panel_size,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average nucleotide heterozygosity")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=0.5))

ggsave(paste0(type,"_heterozygosity_population_5xpersite_bootstrapped.pdf"), width=10.3, height=11.6, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony",average_ggplot_all_pi_combined)

#STOP
STOP

```

####Derived allele heterozygous counts (ki rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.empirmean_ki_rel.anc_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_heterozygous_",type,"_SNP.totalerror_ki_rel.anc_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)


averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/",type,"_heterozygosity_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%#.2f", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_middle

average_ggplot_all_pi_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide heterozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.2,0.4,0.6,0.8,1.0,1.2),limits=c(0.1,1.2)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_down

#New:
average_ggplot_all_pi_combined <- grid.arrange(set_panel_size(average_ggplot_all_pi_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average heterozygosity relative to KIR")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))

#New:
ggsave(paste0(type,"_heterozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony",average_ggplot_all_pi_combined)

#STOP
STOP
```

####Derived allele homozygous counts (ki rel).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_",type,"_SNP.empirmean_ki_rel.anc_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
averages$population = factor(averages$population,levels=c("ki","no","po","sm","do"))
averages$species = factor(averages$species,levels=c("ll","lp"))
averages$size = factor(averages$size,levels=c("large","small"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xpopulation_average_homozygous_",type,"_SNP.totalerror_ki_rel.anc_ann.txt")) %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
errors$population = factor(errors$population,levels=c("ki","no","po","sm","do"))
errors$species = factor(errors$species,levels=c("ll","lp"))
errors$size = factor(errors$size,levels=c("large","small"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-population,-species,-size,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-population,-species,-size,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("population","species","size","ratio"))
colnames(combined_tidy) <- c("population","species","size","ratio","mean","error")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('mistol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('misdel', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
combined_tidy$population = factor(combined_tidy$population,levels=c("ki","po","no","sm","do"))
levels(combined_tidy$population) <- c("KIR","POL","NOR","AND","DON")
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/",type,"_homozygosity_5xrelative2Kirov_bootstrapped.txt"), delim = "\t", na = "NA")
combined_tidy

#Separate plots:
scaleFUN <- function(x) sprintf("%#.2f", x)
#type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.65,0.5,0.5),"max"=c(1.25,1.5,2,4),"breaks"=c(0.05,0.1,0.2,0.5))

average_ggplot_all_pi_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.65,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_up

average_ggplot_all_pi_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  #xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.65,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_middle

average_ggplot_all_pi_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(population,mean,colour=interaction(species,size),alpha=interaction(species,size))) +
  #facet_wrap(feature ~ population,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Population") +
  ylab("Average nucleotide homozygosity") +
  scale_y_continuous(labels=scaleFUN,breaks=c(0.8,1.0,1.2,1.4),limits=c(0.65,1.5)) +
  scale_colour_manual(values=c("steelblue4","indianred4","steelblue4","indianred4")) +
  scale_alpha_manual(values=c(1,1,0.4,0.4)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
average_ggplot_all_pi_down

#New:
average_ggplot_all_pi_combined <- grid.arrange(set_panel_size(average_ggplot_all_pi_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(average_ggplot_all_pi_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average derived homozygosity relative to KIR")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))

#New:
ggsave(paste0(type,"_homozygosity_population_5xrelative2Kirov_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony",average_ggplot_all_pi_combined)

#Stop
STOP

```
####Derived allele homozygous counts (absolute values).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)


type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")
data <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_summary_homozygous_",type,"_SNP.anc_ann.txt")) %>% filter(.,dataset=="5x") %>% select(.,population,contains("_H"),-contains("/")) %>% rename_at(vars(ends_with("_H")),funs(gsub("_H","",.))) %>% mutate(species=ifelse(population=="ki" | population=="po" | population=="no","ll","lp"),size=ifelse(population=="ki" | population=="sm","large","small"))
data$population = factor(data$population,levels=c("ki","po","no","sm","do"))
levels(data$population) <- c("KIR","POL","NOR","AND","DON")
data$species = factor(data$species,levels=c("ll","lp"))
levels(data$species) <- c("EL","IL")
data$size = factor(data$size,levels=c("large","small"))
print.data.frame(data)

data_tidy <- data %>% gather(ratio,value,-population,-species,-size,factor_key=T)
data_tidy$ratio <- gsub('intronic', 'introns', data_tidy$ratio)
data_tidy$ratio <- gsub('coding', 'CDS', data_tidy$ratio)
data_tidy$ratio <- gsub('synonymous', 'syn.', data_tidy$ratio)
data_tidy$ratio <- gsub('mistol', 'm. tol.', data_tidy$ratio)
data_tidy$ratio <- gsub('misdel', 'm. del.', data_tidy$ratio)
data_tidy$ratio <- gsub('nonsense', 'LoF', data_tidy$ratio)
data_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', data_tidy$ratio)
data_tidy$ratio <- gsub('UCNE_high', 'UCNE high', data_tidy$ratio)
data_tidy$ratio = factor(data_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
data_tidy

data_tidy_minmax <- data_tidy %>%
  group_by(ratio) %>%
  mutate(min = min(value)*0.98,max = max(value)*1.02) %>%
  ungroup()

scaleFUN <- function(x) sprintf("%#.2e", x)

derived_allele_homozygous_ggplot_up <-
ggplot(data=filter(data_tidy_minmax,grepl('total|intergenic|introns|syn\\.',ratio)), aes(population,value,colour=interaction(species,size),alpha=interaction(species,size))) +
  facet_wrap(. ~ ratio,scales="free_y",ncol=4) +
  geom_point(size=0.9,stroke=0,position = position_jitter(width=0.2,height=0)) +
  #geom_vline(xintercept=4.5,colour="black") +
  #ylab("N homozygous sites for the derived allele per individual") +
  scale_y_continuous(labels=scaleFUN) +
  geom_blank(aes(y = min)) + 
  geom_blank(aes(y = max)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
derived_allele_homozygous_ggplot_up

derived_allele_homozygous_ggplot_middle <- ggplot(data=filter(data_tidy_minmax,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(population,value,colour=interaction(species,size),alpha=interaction(species,size))) +
  facet_wrap(. ~ ratio,scales="free_y",ncol=4) +
  geom_point(size=0.9,stroke=0,position = position_jitter(width=0.2,height=0)) +
  #geom_vline(xintercept=4.5,colour="black") +
  #ylab("N homozygous sites for the derived allele per individual") +
  scale_y_continuous(labels=scaleFUN) +
  geom_blank(aes(y = min)) + 
  geom_blank(aes(y = max)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
derived_allele_homozygous_ggplot_middle

derived_allele_homozygous_ggplot_down <- ggplot(data=filter(data_tidy_minmax,grepl('tol|del|mod|high',ratio)), aes(population,value,colour=interaction(species,size),alpha=interaction(species,size))) +
  facet_wrap(. ~ ratio,scales="free_y",ncol=4) +
  geom_point(size=0.9,stroke=0,position = position_jitter(width=0.2,height=0)) +
  #geom_vline(xintercept=4.5,colour="black") +
  #ylab("N homozygous sites for the derived allele per individual") +
  scale_y_continuous(labels=scaleFUN) +
  geom_blank(aes(y = min)) + 
  geom_blank(aes(y = max)) +
  scale_colour_manual(values=c("steelblue3","indianred3","steelblue3","indianred3")) +
  scale_alpha_manual(values=c(1,1,0.5,0.5)) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      axis.title.y=element_blank(),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
derived_allele_homozygous_ggplot_down

#New:
derived_allele_homozygous_ggplot_combined <- grid.arrange(set_panel_size(derived_allele_homozygous_ggplot_up,width=unit(2,"cm"),height=unit(4,"cm")),set_panel_size(derived_allele_homozygous_ggplot_middle,width=unit(2,"cm"),height=unit(4,"cm")),set_panel_size(derived_allele_homozygous_ggplot_down,width=unit(2,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Population")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Number of derived homozygous sites")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))

#New:
ggsave(paste0(type,"_homozygosity_individual_5x.pdf"), width=14.5, height=17, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony",derived_allele_homozygous_ggplot_combined)

```

###Species level.
####Derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")

averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_",type,"_SNP.empirmeanweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.)))
averages$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(averages)

errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_",type,"_SNP.totalerrorweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) 
errors$species = factor(averages$species,levels=c("ll","lp"))
print.data.frame(errors)

averages_tidy <- averages %>% gather(ratio,value,-species,factor_key=T)
averages_tidy

errors_tidy <- errors %>% gather(ratio,value,-species,factor_key=T)
errors_tidy

combined_tidy <- left_join(averages_tidy,errors_tidy,by=c("species","ratio"))
colnames(combined_tidy) <- c("species","ratio","mean","error")

#combined_tidy$species = factor(averages$species,levels=c("ll","lp"))
levels(combined_tidy$species) <- c("EL","IL")
combined_tidy$ratio <- gsub('intronic', 'introns', combined_tidy$ratio)
combined_tidy$ratio <- gsub('coding', 'CDS', combined_tidy$ratio)
combined_tidy$ratio <- gsub('synonymous', 'syn.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_tol', 'm. tol.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('missense_del', 'm. del.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('nonsense', 'LoF', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_low', 'UCNE low', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_mid', 'UCNE mod.', combined_tidy$ratio)
combined_tidy$ratio <- gsub('UCNE_high', 'UCNE high', combined_tidy$ratio)
#levels(combined_tidy$population)[levels(combined_tidy$population)=="sm"] <- "an"
combined_tidy$ratio = factor(combined_tidy$ratio,levels=c("total","intergenic","introns","CDS","syn.","syn_pref","syn_unpref","missense","m. tol.","m. del.","LoF","UCNE","UCNE low","UCNE mod.","UCNE high")) 
#write_delim(combined_tidy,paste0(wd_path,type,"_genetic_load_5xspecies_bootstrapped.txt"), delim = "\t", na = "NA")


#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.6,0.6,0.5,0.5),"max"=c(1.2,1.5,2,4),"breaks"=c(0.1,0.1,0.2,0.5))

ggplot_up <- ggplot(data=filter(combined_tidy,grepl('total|intergenic|introns|syn\\.',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_up

ggplot_middle <- ggplot(data=filter(combined_tidy,grepl('\\<missense\\>|LoF|^UCNE$',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_middle

#Add asterisk
if (type=="varssubs")
{
middle_ann <- data.frame(species="IL",mean=1.15,lab="*",ratio=factor("LoF",levels=c("missense","LoF","UCNE")))
ggplot_middle <- ggplot_middle + geom_text(data=middle_ann,label="*",size=6, colour="black")
}

ggplot_down <- ggplot(data=filter(combined_tidy,grepl('tol|del|mod|high',ratio)), aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genetic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_down

#Add asterisk
if (type=="varssubs")
{
down_ann <- data.frame(species="IL",mean=1.15,lab="*",ratio=factor("m. del.",levels=c("m. tol.","m. del.","UCNE mod.","UCNE high")))
ggplot_down <- ggplot_down + geom_text(data=down_ann,label="*",size=6, colour="black")
}

#New:
if(type=="varssubs") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average derived count relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))
} else if (type=="fixed") {
  ggplot_combined <- grid.arrange(set_panel_size(ggplot_up,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_middle,width=unit(1.65,"cm"),height=unit(4,"cm")),set_panel_size(ggplot_down,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Derived fixation rate relative to Eurasian lynx")), rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=1))
}

ggsave(paste0(type,"_genetic_load_5xspecies_relative2ll_bootstrapped.pdf"), width=8.2, height=17, units="cm", device="pdf", path=wd_path,ggplot_combined)

#Stop
STOP

```

####Tel vs. cen vs. Xchr derived allele counts 5xonly.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)
library(grid)
library(gridExtra)
library(egg)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating
wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony/")

tel_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.empirmeanweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="tel")
tel_averages$species = factor(tel_averages$species,levels=c("ll","lp"))

centr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.empirmeanweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="centr")
centr_averages$species = factor(centr_averages$species,levels=c("ll","lp"))

Xchr_averages <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_correctedXchr_",type,"_SNP.empirmeanweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="Xchr")
Xchr_averages$species = factor(Xchr_averages$species,levels=c("ll","lp"))
print.data.frame(Xchr_averages)

all_averages <- rbind(tel_averages,centr_averages,Xchr_averages)
all_averages

tel_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_tel_",type,"_SNP.totalerrorweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="tel")
tel_errors$species = factor(tel_averages$species,levels=c("ll","lp"))

centr_errors <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_centr_",type,"_SNP.totalerrorweight_ll_rel.anc_ann.txt")) %>% select(.,species,contains("_A")) %>% rename_at(vars(ends_with("_A")),funs(gsub("_A","",.))) %>% mutate(region="centr")
centr_errors$species = factor(centr_averages$species,levels=c("ll","lp"))

all_errors <- rbind(tel_errors,centr_errors)
all_errors

all_averages_tidy <- all_averages %>% gather(ratio,value,-species,-region,factor_key=T) %>% filter(grepl("missense_del",ratio))
all_averages_tidy

all_errors_tidy <- all_errors %>% gather(ratio,value,-species,-region,factor_key=T) %>% filter(grepl("missense_del",ratio))
all_errors_tidy

all_combined_tidy <- full_join(all_averages_tidy,all_errors_tidy,by=c("species","ratio","region"))
colnames(all_combined_tidy) <- c("species","region","ratio","mean","error")

all_combined_tidy$species = factor(all_combined_tidy$species,levels=c("ll","lp"))
levels(all_combined_tidy$species) <- c("EL","IL")
all_combined_tidy$ratio <- gsub('missense_del', 'm. del.', all_combined_tidy$ratio)
all_combined_tidy$region <- gsub('tel', 'telomere', all_combined_tidy$region)
all_combined_tidy$region <- gsub('centr', 'centromere', all_combined_tidy$region)
all_combined_tidy$region <- gsub('Xchr', 'X chr.', all_combined_tidy$region)
all_combined_tidy$region = factor(all_combined_tidy$region,levels=c("telomere","centromere","X chr.")) 

#Plots:
twodecimalsFUN <- function(x) sprintf("%.2f", x)
type_range <- data.frame("var_type"=c("varssubs","varssubs","fixed","fixed"),"plot"=c("main","other","main","other"),"min"=c(0.65,0.2,0.5,0.5),"max"=c(1.35,1.7,2,4),"breaks"=c(0.1,0.2,0.2,0.5))

ggplot_misdel <- ggplot(data=all_combined_tidy, aes(species,mean,colour=species)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ region) +
  geom_point(size=0.5) +
  geom_errorbar(aes(ymin=mean-error, ymax=mean+error), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  xlab("Species") +
  ylab(ifelse(type=="varssubs","Average genomic load relative to Lynx lynx",ifelse(type=="fixed","Derived fixation rate relative to Lynx lynx", "Check code"))) +
  scale_y_continuous(labels=twodecimalsFUN, breaks=seq(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F), filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F), by = filter(type_range,var_type==type & plot=="main") %>% select(breaks) %>% unlist(.,use.names=F)), limits=c(filter(type_range,var_type==type & plot=="main") %>% select(min) %>% unlist(.,use.names=F),filter(type_range,var_type==type & plot=="main") %>% select(max) %>% unlist(.,use.names=F))) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  scale_colour_manual(values=c("steelblue3","indianred3")) +
  theme_bw() +
  theme(text=element_text(size=9,face="bold"),
      rect=element_rect(size=1),
      axis.line=element_blank(),
      axis.title=element_text(size=16),
      axis.text.x=element_text(angle=45,hjust=1,colour="black",face="bold"),
      axis.title.x=element_blank(),
      axis.title.y=element_blank(),
      axis.text.y=element_text(colour="black",face="bold"),
      #axis.title.y=element_text(margin=unit(c(0,0.5,0,0),"cm")),
      panel.background=element_blank(),
      panel.border=element_rect(colour="black",fill=NA),
      panel.spacing.x=unit(0.075,"cm"),
      strip.background=element_rect(colour="black"),
      strip.text=element_text(size=8),
      #panel.grid=element_blank(),
      #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
      #plot.margin=unit(c(0.5,1,0.5,0.2),"cm"),
      #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
      legend.background=element_rect(linetype="solid", colour="black", size=.5),
      #legend.justification=c(0,0),
      legend.key=element_rect(colour="white"),
      legend.key.size=unit(0.5,"cm"),
      legend.position="none",
      legend.title=element_blank()
  )
ggplot_misdel

ggplot_misdel_final <- grid.arrange(set_panel_size(ggplot_misdel,width=unit(1.65,"cm"),height=unit(4,"cm")),bottom=textGrob(expression(bold("Species")),gp=gpar(fontsize=10,fontface="bold")),left=textGrob(expression(bold("Average m. deleterious derived\n count relative to Eurasian lynx")),rot=90,gp=gpar(fontsize=10,fontface="bold"),vjust=2))
ggplot_misdel_final

ggsave(paste0(type,"_genetic_load_telcentrXchr_5xspecies_relative2ll_bootstrapped.pdf"), width=7.4, height=6, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/TCRLP_outerparsimony",ggplot_misdel_final)

```

#Others:
##Download other files.
```{bash}
#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions #segregating #fixed #private_segregating
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_population_average_${VAR}_${TYPE}.empirerror.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_5xpopulation_average_${VAR}_${TYPE}.empirerror.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/bootstrap/derived_allele_counts/${VAR}/${CALLING}_ann_population_average_${VAR}_${TYPE}.booterrorcorrected.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/
unset SSHPASS

```

##Remove _V columns:
```{r}

c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_varssubs_SNP.empirmeanweight.lr_ann
c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_5xspecies_average_varssubs_SNP.totalerrorweight.lr_ann

file <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/bootstrap/nm3/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm3nm3_origcov_ann_5xspecies_average_varssubs_SNP.totalerrorweight.lr_ann"
read_tsv(paste0(file,".txt")) %>% select(.,species,contains("_A")) %>% write_tsv(paste0(file,"As.txt"))
                                                                                       
```

