---
title: "GO_KO_enrichment"
author: "Dani"
date: "18 de febrero de 2019"
output: html_document
---

#0: Define paths.

```{r Define paths, eval=FALSE, engine='bash'}

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/BAM_nm_filtered #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

```

#1: TopGO approach (obsolete).
##Obtain gene list of interest.
###Example

```{r Obtain gene list of interest, eval=FALSE, engine='bash'}

#Extract genes with NS variants from an example population:
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/annotation
FILE=(c_lp_do_nm2_samecov_perpop.lr_ann)
echo "${FILE}"
FILE_PATH=$(find . -name "${FILE}.vcf")
echo "${FILE_PATH}"
grep 'missense_variant' $FILE_PATH | cut -f8 | cut -d'|' -f4 > ${FILE_PATH/.vcf/.NSYN.genlist}
grep 'synonymous_variant' $FILE_PATH | cut -f8 | cut -d'|' -f4 >  ${FILE_PATH/.vcf/.SYN.genlist}

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/annotation//c_lp_do_nm2_samecov_perpop/c_lp_do_nm2_samecov_perpop.lr_ann.*.genlist /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/input

```

##Perform GO terms enrichment.
###Retrieve GO terms annotation from the Iberian lynx project.

```{r Perform GO terms enrichment}

library(topGO)
library(dplyr)
library(plyr)
library(tidyr)

# Notas de María:
# Este es el script donde realizo el GO term enrichment analysis y el pathway enrichment analysis. 
# 
# Lo primero que hago es pedirle a Godoy una lista donde cada gen tenga asociado su GO-term. Él me pasa una carpeta donde hay varios archivos.
# 
# GO2LYNPC.txt
# GO2LYPA.LYPA23C.APPRIS.fatigo.genes.txt
# GO2LYPA.LYPA23C.APPRIS.fatigo.txt
# GO_enrichment.docx
# genes2transcript2go.list.txt
# lp23.allgenes.fatigo.txt
# lp23.allgenes.goterms
# lp23.allgenes.id
# 
# En el archivo docx, se explican los pasos que siguió Godoy para hacer el enrichment. Lo que deduzco es que las que se llaman lp23* son listas antiguas, y despues están las GO2LYPA que sólo contienen las isoformas principales. 
# 
# Los distintos archivos son:
# 
# GO2LYNPC.txt --> Tiene el nombre del gen, el GO_term, el GO_name y el GO_type; parece que incluye todas las isoformas.
# GO2LYPA.LYPA23C.APPRIS.fatigo.genes.txt --> Creo que es el que tengo que usar. Nombre de gen, GO_ACC, GO_name y el GO_type.
# GO2LYPA.LYPA23C.APPRIS.fatigo.txt --> Igual pero con proteínas.
# GO_enrichment.docx --> Archivo docx donde se explica el procedimiento.
# genes2transcript2go.list.txt --> Creo que es la carpeta que tiene el nombre de gen, isoforma principal y proteína y despues GO term y sus características.
# 
# Los que comienzan por lp23 son antiguos sin la isoforma principal.

wd <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/Lypa23c_GO_terms_annotation/"
tabla_all_genes_GO_term_complete_info <- read_tsv(paste0(wd, "GO2LYPA.LYPA23C.APPRIS.fatigo.genes.txt"),col_names=T) %>% dplyr::select(GENE_NAME, GO_ACC) %>% mutate(GO_ACC=as.character(GO_ACC))
#Convert  data frame in a list of list based on a Key value
#geneID2GO <- readMappings(file = system.file("examples/geneid2go.map", package = "topGO"))
list_of_list_all_genes_GO_term <- dlply(tabla_all_genes_GO_term_complete_info, .(GENE_NAME), function(x)x$GO_ACC)
list_of_list_all_genes_GO_term_chr <- lapply(list_of_list_all_genes_GO_term, function(x) as.character(unlist(x)))
str(head(list_of_list_all_genes_GO_term_chr))
#Guardo los nombres de los genes en cuestión. 
geneNames <- names(list_of_list_all_genes_GO_term_chr)


#Obtain list of GO terms and the genes associated to each of them:
tabla_all_genes_GO_term_complete_info_modif <- read_tsv(paste0(wd, "GO2LYPA.LYPA23C.APPRIS.fatigo.genes.txt"),col_names=T) %>% dplyr::select (-"GO_TYPE") %>% dplyr::group_by(GO_ACC,GO_NAME) %>% dplyr::summarise(Genes=paste(GENE_NAME, collapse=' ')) %>% as.data.frame()

write.table(tabla_all_genes_GO_term_complete_info_modif, paste0(wd, "GO2LYPA.LYPA23C.APPRIS.fatigo.genes_modif.NAMES.gmt"), quote = F, row.names = F, col.names = F, sep = "\t" )

```

###Perform GO terms enrichment analysis for each population and feature of interest.

```{r Perform GO terms enrichment}

library(topGO)
library(dplyr)
library(plyr)
library(tidyr)
library(readr)

wd <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/Lypa23c_GO_terms_annotation/"
wd_input <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/input/"
wd_output <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/output/"

pop_genlists <- list.files(wd_input, pattern="c_lp_do_nm2_samecov_perpop.lr_ann.*")
ontologies <- c("BP","MF","CC")
for (genlist in pop_genlists) {
  description <- paste(strsplit(genlist,"\\.")[[1]][c(1,3)],collapse="_")
  print(description)
  #Guardo la tabla de genes de interés en formato vector.
  genes_duplicate_list <- unlist(read.table(paste0(wd_input,genlist),stringsAsFactors=F),use.names=F)
  #Hago un vector con 0 y 1 con los genes que tengo que filtrar.
  geneList <- factor(as.integer(geneNames %in% genes_duplicate_list))
  #Asigno esa lista a cada gen (pongo header)
  names(geneList) <- as.factor(geneNames)
  #str(geneList)
  for (ontology in ontologies) {
    print(ontology)
    GOdata <- new("topGOdata", ontology = ontology, description = description, allGenes = geneList, annot = annFUN.gene2GO, gene2GO = list_of_list_all_genes_GO_term_chr)
    description(GOdata)
    termStat(GOdata)
    test.stat <- new("weight01Count", testStatistic = GOFisherTest, name = "Fisher test")
    resultFisher <- getSigGroups(GOdata, test.stat)
    head(score(resultFisher))
    hist(score(resultFisher))
    geneData(resultFisher)
    allRes <- GenTable(GOdata, weight01=resultFisher, orderBy="weight01", ranksOf="weight01", topNodes=100)
    write_tsv(allRes,paste0(wd_output,description,".",ontology,".go_res"))
  }
}


```

##Perform pathway (KO terms) enrichment analysis for each population and feature of interest.
###Retrieve KO terms annotation from the Iberian lynx project.

```{r Perform pathway (KO terms) enrichment analysis for each population and feature of interest, eval=FALSE, engine='bash'}

#Voy a seleccionar los KO (que son los nombres de proteinas comunes a todas las especies)
cd /GRUPOS/grupolince/Lyp_annotation_Apr14_final

grep "ko_group" LYPA23C.all.fix.nr.gff3 > all_mRNA
#Perform a sanity check:
#all ko groups
grep -o '\bko_group=\w*' all_mRNA | wc -l #10213
#all parent_id
grep -o '\bParent=\w*' all_mRNA | wc -l #10213
#Igual! Perfecto, los podemos juntar
paste <(grep -o '\bParent=\w*' all_mRNA) <(grep -o '\bko_group=\w*' all_mRNA) | sed 's/Parent=//g' | sed 's/ko_group=//g' > genes_to_KO.txt
rm all_mRNA

#From outside the server:
scp dkleinman@genomics-b.ebd.csic.es:/GRUPOS/grupolince/Lyp_annotation_Apr14_final/genes_to_KO.txt /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/Lypa23c_GO_terms_annotation/

```

###Perform GO terms enrichment analysis for each population and feature of interest.

```{r Perform pathway (KO terms) enrichment analysis for each population and feature of interest}

library(topGO)
library(dplyr)
library(plyr)
library(tidyr)
library(readr)

wd <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/Lypa23c_GO_terms_annotation/"
wd_input <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/input/"
wd_output <- "/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/output/"

#Archivo de equivalencias KO con genes
genes_to_KO.list <- read.table(paste0(wd,"genes_to_KO.txt"),header=F,stringsAsFactors=F)

pop_genlists <- list.files(wd_input, pattern="c_lp_do_nm2_samecov_perpop.lr_ann.*")
for (genlist in pop_genlists) {
  description <- paste(strsplit(genlist,"\\.")[[1]][c(1,3)],collapse="_")
  print(description)
  #Guardo la tabla de genes de interés en formato vector.
  genes_duplicate_df <- read.table(paste0(wd_input,genlist),stringsAsFactors=F)
  #Intersecto la tabla de genes de interés y la de KO terms. 
  genes_of_interest_KO <- inner_join(genes_to_KO.list, genes_duplicate_df, by=c("V1"="V1")) %>% distinct(V1,V2, .keep_all=TRUE)
  write.table(genes_of_interest_KO, paste0(wd,"genes_of_interes_KO_ratio_above_1_CDS.txt"), row.names = F, quote = F)
  write_tsv(as.data.frame(genes_of_interest_KO$V2),paste0(wd_output,description,".ko_res"),col_names=F)
  }

#Take the output file to https://reactome.org/PathwayBrowser/#TOOL=AT and perform the analysis checking both boxes (project to human, include interactome). Then download and rename the pdf with the report.

```

#2: Gowinda approach.
##Getting started.
```{bash}

#Gowinda programme is installed in genomics_a: /opt/Gowinda/Gowinda-1.12.jar

#Make a working directory.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/goterms
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/goterms

#I'll be following María's script: "4.Enrichment_analysis.Rmd" (in the historical_ancient_diversity repository) which in turn is based on the following wiki: 
https://sourceforge.net/p/gowinda/wiki/Main/

#To run Gowinda we need the following:
#1) a file containing the annotation of the genome in .gtf
#2) a gene set file, containing for every gene set (e.g.: GO category) a list of the associated gene IDs
#3) a file containing the total set of SNPs
#4) a file containing the candidate SNPs

```

##Prepare input files:
###1: Genome annotation file.
```{bash}

#The gtf already exists:
/GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.all.fix.nr.gtf

#María created it using a python script gff2gtf.py that transformed the .gff3 to .gtf. The python code can be found in her enrichment analysis script (referenced above).

#To obtain the list of gene names from here:
cut -f9 /GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.all.fix.nr.gtf | cut -d'"' -f2 | sort | uniq #21243 genes

```

###2: Gene sets and gene IDs file.
```{bash}

#The file with GO terms and their respective genes was created in the topGO section. It's the following file:
/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/Lypa23c_GO_terms_annotation/GO2LYPA.LYPA23C.APPRIS.fatigo.genes_modif.NAMES.gmt
 
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
scp /Users/dani/ownCloud/backup/g-w_analysis/genetic_load/GO_analysis/Lypa23c_GO_terms_annotation/GO2LYPA.LYPA23C.APPRIS.fatigo.genes_modif.NAMES.gmt dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/goterms/

#To obtain the list of gene names from here:
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/goterms
cut -f3 GO2LYPA.LYPA23C.APPRIS.fatigo.genes_modif.NAMES.gmt | tr ' ' '\n' | sort | uniq | wc -l #18513 genes

```

###3: Total SNP file.
```{bash}

#This file must contain all the SNPs used for the GWAS, in a simple tab-delimited file format. Gowinda also ignores entries starting with a '#' thus a '.vcf' file may also be provided.
/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_polarized_filteredall_varssubs_SNP.lr_ann.vcf

```

###4: Candidate SNP file.
```{bash}

#The same applies as for the total SNP file, except that the candidate SNPs must be a subset of the total SNPs. The candidate file will be the vcf for each population and/or category.

```

##Run the programme:
###Output format explanation:
```{bash}

#Results file:
#column 1: the GO term
#column 2: on the average this number of genes are found per simulation for the given GO category. In --mode gene every gene is only counted once whereas in --mode snp a single gene may be counted several times dependent on the SNP
#column 3: using the candidate SNPs this number of genes was found for the given GO category. In --mode gene every gene is only counted once whereas in --mode snp a single gene may be counted several times dependent on the SNP
#column 4: p-value (uncorrected for multiple testing)
#column 5: FDR (p-value after adjustment for multiple testing)
#column 6: the number of genes (uniq) found for the given GO category
#column 7: the number of genes that could at most be found for the given GO category, i.e.: genes of the given GO category that have an corresponding entry in the annotation file and contain at least one SNP
#column 8: total number of genes for the given GO category in the GO association file
#column 9: description of the given GO term
#column 10: comma separated list of the gene_ids found for the given GO category

```

###Per population GO enrichment analysis:
```{bash}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/goterms/
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_polarized_perpop_cds_GO_enrichment_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

N_POPS=$(awk -F"_" '{print (NF-2)/3}' <<< $CALLING)
POPS=$(echo $CALLING | fold -w8 | cut -c1-7 | head -n$N_POPS | sort | uniq)
NM_COV=$(echo ${CALLING} | rev | cut -d'_' -f1,2 | rev)

#Define universal variables:
ANNOTATION_FILE=/GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.all.fix.nr.gtf
GENE_SET_FILE=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/goterms/GO2LYPA.LYPA23C.APPRIS.fatigo.genes_modif.NAMES.gmt
SNP_TOTAL_FILE=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_polarized_filteredall_${VAR}_${TYPE}.lr_ann.vcf

for POP in ${POPS[@]}
  do
  echo "preparing input file for population" $POP
  #Define population-dependant variables:
  SPECIES=$(echo $POP | cut -c1-4)
  cat ./../${POP}_${NM_COV}_perpop/${SPECIES}_*_${POP}_${NM_COV}_perpop_${VAR}_${TYPE}.lr_ann.vcf | grep -v '#' | cut -f1,2 | sort -k1,1 -k2,2n | uniq > ${SPECIES}_${POP}_${NM_COV}_perpop_${VAR}_${TYPE}.lr_ann.txt
  SNP_CANDIDATE_FILE=${SPECIES}_${POP}_${NM_COV}_perpop_${VAR}_${TYPE}.lr_ann.txt
  OUTPUT_FILE=$(echo ${SNP_CANDIDATE_FILE/.txt/.gowinda})
  #Launch the programme:
  echo "starting GO enrichment analysis for population" $POP
  java -Xmx4g -jar /opt/Gowinda/Gowinda-1.12.jar \
  --annotation-file $ANNOTATION_FILE \
  --gene-set-file $GENE_SET_FILE \
  --snp-file $SNP_TOTAL_FILE \
  --candidate-snp-file $SNP_CANDIDATE_FILE \
  --output-file $OUTPUT_FILE \
  --mode gene \
  --gene-definition cds \
  --simulations 100000 \
  --min-genes 1 \
  --min-significance 1 \
  --detailed-log \
  --threads 10
  done

```

