---
title: "ll_phylogeography_TreeMix"
author: "Dani"
date: "8 de noviembre de 2017"
output: html_document
---

#0a: Define paths.

```{r Define paths, eval=FALSE, engine='bash'}

V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani #VCFs path
O_PATH=/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

```

#0b: Produce a gVCF and VCF data for the outgroup (the lynx rufus sample).

```{r Produce a gVCF and VCF data for the outgroup (the lynx rufus sample), eval=FALSE, engine='bash'}

#I first did all these with the subsampled rufus BAM, but eventually I substituted them with the original one.

#This was only used to produce the gVCF file for the original (not subsampled) Macrogen Lynx rufus sample, and was actually performed as part of the substitutions_lynx_genus_VCFs script:
cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/MACROGEN_samples_25x
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
    -T HaplotypeCaller \
    -R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
    -I c_lr_zz_0001_recal_round-1_25x.bam \
    --emitRefConfidence GVCF \
    -o /home/GRUPOS/grupolince/lynx_genomes_5x/gVCFs/c_lr_zz_0001_recal_round-1_25x.g.vcf.gz
    
#Next, build a VCF for the rufus sample:
cd /home/GRUPOS/grupolince/lynx_genomes_5x/gVCFs/macrogen_samples_25x
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
    -T GenotypeGVCFs \
    -R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
    -V c_lr_zz_0001_recal_round-1_25x.g.vcf.gz \
    -o /home/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/c_lr.vcf

#Drop all multiallelic SNPs as well as all INDELs from the rufus VCFs:
cd $V_PATH
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
    -T SelectVariants \
    -selectType SNP \
    -restrictAllelesTo BIALLELIC \
    -R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
    -V c_lr.vcf \
    -o c_lr_SNPs.vcf

#Tag SNPs that should be filtered:
cd $V_PATH
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
    -T VariantFiltration \
    --filterName "snpsfilter" \
    --filterExpression "QD<2.0 || FS>60.0 || MQ<40.0 || MQRankSum<-12.5 || ReadPosRankSum<-8.0 || SOR>3.0" \
    -R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
    -V c_lr_SNPs.vcf \
    -o c_lr_SNPs_tagged.vcf

#Filter previously tagged SNPs:
cd $V_PATH
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
    -T SelectVariants \
    -select 'vc.isNotFiltered()' \
    -R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
    -V c_lr_SNPs_tagged.vcf \
    -o c_lr_SNPs_tagged_filtered.vcf
    
grep -v '#' c_lr_SNPs_tagged_filtered.vcf | wc -l #16685199

```

#1a: Build whole-genome TreeMix input. Combine per population allele counts for all SNPs in a single gzipped file matching the required format for TreeMix inputs.

```{r Build whole-genome TreeMix input, eval=FALSE, engine='bash'}

#First create the allelic counts file for Eurasian lynx (without the root):
cd $V_PATH
shopt -s extglob #the extglob shell option gives you more powerful pattern matching in the command line.
POPLIST=($(ls ll*[^pv]_perpop_standard_filter.vcf | cut -d "_" -f1,2 | sort | uniq)) #selects all populations except for País Vasco
cd $O_PATH
rm SNP_names_treemix_input.txt
SNP_LIST="$(ls "$V_PATH"/ll*_perpop_standard_filter.vcf | head -1)" #any population will be useful to retrieve the SNP list because I didn't trim the subVCFs so all populations have all the SNPs (from their species)
grep -v '#' $SNP_LIST | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2}' > SNP_names_treemix_input.txt
unset POPNAMES #removes the variable so that it starts all over again inside the loop
POPNAMES="SNP"
#echo "pop_name,total_SNPs,homoz_ref,heteroz,homoz_alt,unaccounted,unaccounted_cat" > c_VCF_raw.stats 
for pop in "${POPLIST[@]}"
  do
  echo "${pop}"
  NAME="${pop}"
  POPNAMES=$POPNAMES" "$NAME
  echo $POPNAMES
  SAMPLE_NUMBER="$(/opt/bcftools-1.6/bcftools query -l "$V_PATH/${pop}"_perpop_standard_filter.vcf | wc -l)"
  echo $SAMPLE_NUMBER
  TOTAL_SNPS="$(grep -v '#' "$V_PATH/${pop}"_perpop_standard_filter.vcf | wc -l)" #11412781
  echo $TOTAL_SNPS
  #done
  grep -v '#' "$V_PATH/${pop}"_perpop_standard_filter.vcf | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | cut -d$'=' -f1,2,3 --output-delimiter=' ' | awk '{print $1" "$2" "$6-$4","$4}' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3}' > temp_SNP_counts.txt
  #LANG=en_EN join -j 1 <(LANG=en_EN sort -k1,1 -k2,2n SNP_names_treemix_input.txt | awk '{print $1"_"$2}') <(LANG=en_EN sort -k1,1 -k2,2n temp_SNP_counts.txt | awk '{print $1"_"$2" "$3}') > intermediate_file.borrar.txt
  LANG=en_EN join -j 1 SNP_names_treemix_input.txt temp_SNP_counts.txt > intermediate_file.borrar.txt
  mv intermediate_file.borrar.txt SNP_names_treemix_input.txt
  done
rm temp_SNP_counts.txt
LONGNAMES=$(echo $POPNAMES | sed -e "s/ll_ba/Balkans/g; s/ll_cr/Carpathians/g; s/ll_ka/Khentii/g; s/ll_ki/Kirov/g; s/ll_la/Latvia/g; s/ll_no/Norway/g; s/ll_og/Ömnögovi/g; s/ll_po/Bialowieza/g; s/ll_to/Töv/g; s/ll_tu/Tuva/g; s/ll_ur/Urals/g; s/ll_vl/Vladivostok/g; s/ll_ya/Yakutia/g; s/lp_sm/Sierra Morena/g; s/lp_do/Doñana/g; s/lr_zz/Rufus/g") #swaps the short names with the full names
echo $LONGNAMES
sed -i -e '1i'"$LONGNAMES"'\' SNP_names_treemix_input.txt #adds the headers
#sed -i "1s/.*/$LONGNAMES/" SNP_names_treemix_input.txt #substitutes old headers

grep -v ' 0,0 ' SNP_names_treemix_input.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input.txt.gz #removes rows with any missing data, and also the SNP names
shopt -u extglob #disable extglob

#Get a version of the TreeMix input without Balkans:
cut -d$' ' -f2 --complement SNP_names_treemix_input.txt > SNP_names_treemix_input_wout_ba.txt #removes the Balkans column
grep -v ' 0,0 ' SNP_names_treemix_input_wout_ba.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_wout_ba.txt.gz #removes rows with any missing data, and also the SNP names

#Get a version of the TreeMix input that includes Lynx rufus as the outgroup:
cd $O_PATH
#LANG=en_EN sort -k1,1 -k2,2n SNP_names_treemix_input.txt
grep -v '#' $V_PATH/c_lr_SNPs.vcf | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | cut -d$'=' -f1,2,3 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$6-$4","$4}' > c_lr_SNP_counts.txt #gets the allele counts for all variable positions within the rufus individual
sed -i -e '1i'"SNP Rufus"'\' c_lr_SNP_counts.txt #adds column names; #16987505 lines in c_lr_SNP_counts.txt
LANG=en_EN join -a1 -e 2,0 -o auto -j 1 <(LANG=en_EN sort -k1 SNP_names_treemix_input.txt | tail -n +2) <(LANG=en_EN sort -k1 c_lr_SNP_counts.txt | tail -n +2) | cut -d$'_' -f1,2 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9" "$10" "$11" "$12" "$13" "$14" "$15" "$16}' > intermediate_file.borrar.txt #joins the eurasian lynx allelic counts with the rufus allelic counts, assigning "2,0" for all missing positions from Rufus
LANG=en_EN join -a2 -e 'NA' -o auto -j 1 <(LANG=en_EN sort -k1 SNP_names_treemix_input.txt | tail -n +2) <(LANG=en_EN sort -k1 c_lr_SNP_counts.txt | tail -n +2) | cut -d$'_' -f1,2 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9" "$10" "$11" "$12" "$13" "$14" "$15" "$16}' > intermediate_file_2.borrar.txt #joins the eurasian lynx allelic counts with the rufus allelic counts, assigning "NA" for all missing positions from Eurasian
LANG=en_EN join -a1 -a2 -j 1 <(LANG=en_EN sort -k1 intermediate_file.borrar.txt) <(LANG=en_EN sort -k1 intermediate_file_2.borrar.txt) | cut -d$'_' -f1,2 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9" "$10" "$11" "$12" "$13" "$14" "$15" "$16}' > intermediate_file_joined.borrar.txt #joins both previous allelic count files in a single one with all variable positions
cp intermediate_file_joined.borrar.txt intermediate_file_joined_2.borrar.txt #duplicates the joined intermediate file
sed -i 's/NA NA NA NA NA NA NA NA NA NA NA NA NA/6,0 12,0 8,0 26,0 12,0 16,0 4,0 16,0 4,0 12,0 12,0 16,0 16,0/g' intermediate_file_joined_2.borrar.txt #replaces all rows with NA in the Eurasian fields with max allelic counts for the reference allele
mv intermediate_file_joined_2.borrar.txt SNP_names_treemix_input_rooted.txt #final name
sed -i -e '1i'"$LONGNAMES Rufus"'\' SNP_names_treemix_input_rooted.txt #adds headers with population names; #26974289 lines in SNP_names_treemix_input_rooted.txt
grep -v ' 0,0 ' SNP_names_treemix_input_rooted.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_rooted.txt.gz #removes rows with any missing data, and also the SNP names
rm *.borrar.txt #removes all intermediate files

#Get a version of the rufus-rooted TreeMix input without Balkans:
cut -d$' ' -f2 --complement SNP_names_treemix_input_rooted.txt > SNP_names_treemix_input_rooted_wout_ba.txt #removes the Balkans column
grep -v ' 0,0 ' SNP_names_treemix_input_rooted_wout_ba.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_rooted_wout_ba.txt.gz #removes rows with any missing data, and also the SNP names

#Get a version of the rufus-rooted TreeMix input without Balkans and with Khentii, Töv & Ömnögovi grouped together as Mongolia:
cut -d$' ' -f1,4,8,10  SNP_names_treemix_input_rooted.txt | cut -d$',' -f1,2,3,4 --output-delimiter=' ' | cut -d$' ' -f1,2,3,4,5,6,7 | awk '{print $1" "$2+$4+$6","$3+$5+$7}' > SNP_names_threepop_input_ka_om_to.txt #sums allele counts for the three Mongolian populations
sed -i '1s/.*/SNP Mongolia/' SNP_names_threepop_input_ka_om_to.txt #adds correct column names
awk -v f2=SNP_names_treemix_input_rooted.txt '{ c = $2; getline < f2; print $0, c; }' SNP_names_threepop_input_ka_om_to.txt | cut -d$' ' -f2,4,8,10 --complement > SNP_names_treemix_input_rooted_wout_ba_w_mo.txt #copies the Mongolia column to a new TreeMix input where Balkans and the three Mongolian populations have been removed
grep -v ' 0,0 ' SNP_names_treemix_input_rooted_wout_ba_w_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_rooted_wout_ba_w_mo.txt.gz #removes rows with any missing data, and also the SNP names

```

#1b: Build intergenic TreeMix input. Combine per population allele counts for intergenic SNPs in a single gzipped file matching the required format for TreeMix inputs.

```{r Build intergenic TreeMix input, eval=FALSE, engine='bash'}

#Generate VCF with only intergenic positions for Lynx rufus:
cd $V_PATH
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar $GATK \
  -T SelectVariants \
  -R $REF \
  -V c_lr_SNPs_tagged_filtered.vcf \
  -o $V_PATH/intergenic_plus_1000/c_lr_SNPs_tagged_filtered_intergenic.vcf \
  -L /GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.intergenic.PLUS1000.bed
  
grep -v '#' c_lr_SNPs_tagged_filtered_intergenic.vcf | wc -l #10492014

#Generate VCF with only intergenic positions for all Eurasian lynx populations:
cd $V_PATH
shopt -s extglob #the extglob shell option gives you more powerful pattern matching in the command line.
POP_VCFs=($(ls ll*[^pv]_perpop_standard_filter.vcf)) #selects all populational VCFs except for País Vasco
for pop in "${POP_VCFs[@]}"
  do
  echo "${pop}"
  java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar $GATK \
    -T SelectVariants \
    -R $REF \
    -V "${pop}" \
    -o $V_PATH/intergenic_plus_1000/${pop/.vcf/_intergenic.vcf} \
    -L /GRUPOS/grupolince/Lyp_annotation_Apr14_final/LYPA23C.intergenic.PLUS1000.bed
  done
shopt -u extglob #disable extglob


#Create the allelic counts file for Eurasian lynx (without the root):
cd $V_PATH
shopt -s extglob #the extglob shell option gives you more powerful pattern matching in the command line.
POPLIST=($(ls ll*[^pv]_perpop_standard_filter.vcf | cut -d "_" -f1,2 | sort | uniq)) #selects all populations except for País Vasco
cd $O_PATH/intergenic_plus_1000
rm SNP_names_treemix_input_intergenic.txt
SNP_LIST="$(ls "$V_PATH"/intergenic_plus_1000/ll*_perpop_standard_filter_intergenic.vcf | head -1)" #any population will be useful to retrieve the SNP list because I didn't trim the subVCFs so all populations have all the SNPs (from their species)
grep -v '#' $SNP_LIST | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2}' > SNP_names_treemix_input_intergenic.txt
unset POPNAMES #removes the variable so that it starts all over again inside the loop
POPNAMES="SNP"
#echo "pop_name,total_SNPs,homoz_ref,heteroz,homoz_alt,unaccounted,unaccounted_cat" > c_VCF_raw.stats 
for pop in "${POPLIST[@]}"
  do
  echo "${pop}"
  NAME="${pop}"
  POPNAMES=$POPNAMES" "$NAME
  echo $POPNAMES
  SAMPLE_NUMBER="$(/opt/bcftools-1.6/bcftools query -l "$V_PATH"/intergenic_plus_1000/"${pop}"_perpop_standard_filter_intergenic.vcf | wc -l)"
  echo $SAMPLE_NUMBER
  TOTAL_SNPS="$(grep -v '#' "$V_PATH"/intergenic_plus_1000/"${pop}"_perpop_standard_filter_intergenic.vcf | wc -l)" #7252375
  echo $TOTAL_SNPS
  #done
  grep -v '#' $V_PATH/intergenic_plus_1000/${pop}_perpop_standard_filter_intergenic.vcf | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | cut -d$'=' -f1,2,3 --output-delimiter=' ' | awk '{print $1" "$2" "$6-$4","$4}' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3}' > temp_SNP_counts.txt
  #LANG=en_EN join -j 1 <(LANG=en_EN sort -k1,1 -k2,2n SNP_names_treemix_input.txt | awk '{print $1"_"$2}') <(LANG=en_EN sort -k1,1 -k2,2n temp_SNP_counts.txt | awk '{print $1"_"$2" "$3}') > intermediate_file.borrar.txt
  LANG=en_EN join -j 1 SNP_names_treemix_input_intergenic.txt temp_SNP_counts.txt > intermediate_file.borrar.txt
  mv intermediate_file.borrar.txt SNP_names_treemix_input_intergenic.txt
  done
rm temp_SNP_counts.txt
LONGNAMES=$(echo $POPNAMES | sed -e "s/ll_ba/Balkans/g; s/ll_cr/Carpathians/g; s/ll_ka/Khentii/g; s/ll_ki/Kirov/g; s/ll_la/Latvia/g; s/ll_no/Norway/g; s/ll_og/Ömnögovi/g; s/ll_po/Bialowieza/g; s/ll_to/Töv/g; s/ll_tu/Tuva/g; s/ll_ur/Urals/g; s/ll_vl/Vladivostok/g; s/ll_ya/Yakutia/g; s/lp_sm/Sierra Morena/g; s/lp_do/Doñana/g; s/lr_zz/Rufus/g") #swaps the short names with the full names
echo $LONGNAMES
sed -i -e '1i'"$LONGNAMES"'\' SNP_names_treemix_input_intergenic.txt #adds the headers
#sed -i "1s/.*/$LONGNAMES/" SNP_names_treemix_input.txt #substitutes old headers

grep -v ' 0,0 ' SNP_names_treemix_input_intergenic.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_intergenic.txt.gz #removes rows with any missing data, and also the SNP names
shopt -u extglob #disable extglob

#Get a version of the TreeMix input that includes Lynx rufus as the outgroup:
cd $O_PATH/intergenic_plus_1000
grep -v '#' $V_PATH/intergenic_plus_1000/c_lr_SNPs_tagged_filtered_intergenic.vcf | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | cut -d$'=' -f1,2,3 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$6-$4","$4}' > c_lr_SNP_counts_intergenic.txt #gets the allele counts for all variable positions within the rufus individual
sed -i -e '1i'"SNP Rufus"'\' c_lr_SNP_counts_intergenic.txt #adds column names; #10492015 lines in c_lr_SNP_counts_intergenic.txt
LANG=en_EN join -a1 -e 2,0 -o auto -j 1 <(LANG=en_EN sort -k1 SNP_names_treemix_input_intergenic.txt | tail -n +2) <(LANG=en_EN sort -k1 c_lr_SNP_counts_intergenic.txt | tail -n +2) | cut -d$'_' -f1,2 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9" "$10" "$11" "$12" "$13" "$14" "$15" "$16}' > intermediate_file.borrar.txt #joins the eurasian lynx allelic counts with the rufus allelic counts, assigning "2,0" for all missing positions from Rufus
LANG=en_EN join -a2 -e 'NA' -o auto -j 1 <(LANG=en_EN sort -k1 SNP_names_treemix_input_intergenic.txt | tail -n +2) <(LANG=en_EN sort -k1 c_lr_SNP_counts_intergenic.txt | tail -n +2) | cut -d$'_' -f1,2 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9" "$10" "$11" "$12" "$13" "$14" "$15" "$16}' > intermediate_file_2.borrar.txt #joins the eurasian lynx allelic counts with the rufus allelic counts, assigning "NA" for all missing positions from Eurasian
LANG=en_EN join -a1 -a2 -j 1 <(LANG=en_EN sort -k1 intermediate_file.borrar.txt) <(LANG=en_EN sort -k1 intermediate_file_2.borrar.txt) | cut -d$'_' -f1,2 --output-delimiter=' ' | LANG=en_EN sort -k1,1 -k2,2n | awk '{print $1"_"$2" "$3" "$4" "$5" "$6" "$7" "$8" "$9" "$10" "$11" "$12" "$13" "$14" "$15" "$16}' > intermediate_file_joined.borrar.txt #joins both previous allelic count files in a single one with all variable positions
cp intermediate_file_joined.borrar.txt intermediate_file_joined_2.borrar.txt #duplicates the joined intermediate file
sed -i 's/NA NA NA NA NA NA NA NA NA NA NA NA NA/6,0 12,0 8,0 26,0 12,0 16,0 4,0 16,0 4,0 12,0 12,0 16,0 16,0/g' intermediate_file_joined_2.borrar.txt #replaces all rows with NA in the Eurasian fields with max allelic counts for the reference allele
mv intermediate_file_joined_2.borrar.txt SNP_names_treemix_input_intergenic_rooted.txt #final name
sed -i -e '1i'"$LONGNAMES Rufus"'\' SNP_names_treemix_input_intergenic_rooted.txt #adds headers with population names; #16842636 lines in SNP_names_treemix_input_intergenic_rooted.txt
grep -v ' 0,0 ' SNP_names_treemix_input_intergenic_rooted.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_intergenic_rooted.txt.gz #removes rows with any missing data, and also the SNP names
rm *.borrar.txt #removes all intermediate files

#Get a version of the rufus-rooted TreeMix input without Balkans:
cut -d$' ' -f2 --complement SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_treemix_input_intergenic_rooted_wout_ba.txt #removes the Balkans column
grep -v ' 0,0 ' SNP_names_treemix_input_intergenic_rooted_wout_ba.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_intergenic_rooted_wout_ba.txt.gz #removes rows with any missing data, and also the SNP names

#Get a version of the rufus-rooted TreeMix input without Balkans and with Khentii, Töv & Ömnögovi grouped together as Mongolia:
cut -d$' ' -f1,4,8,10  SNP_names_treemix_input_intergenic_rooted.txt | cut -d$',' -f1,2,3,4 --output-delimiter=' ' | cut -d$' ' -f1,2,3,4,5,6,7 | awk '{print $1" "$2+$4+$6","$3+$5+$7}' > SNP_names_threepop_input_intergenic_ka_om_to.txt #sums allele counts for the three Mongolian populations
sed -i '1s/.*/SNP Mongolia/' SNP_names_threepop_input_intergenic_ka_om_to.txt #adds correct column names
awk -v f2=SNP_names_treemix_input_intergenic_rooted.txt '{ c = $2; getline < f2; print $0, c; }' SNP_names_threepop_input_intergenic_ka_om_to.txt | cut -d$' ' -f2,4,8,10 --complement > SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt #copies the Mongolia column to a new TreeMix input where Balkans and the three Mongolian populations have been removed
grep -v ' 0,0 ' SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_intergenic_rooted_wout_ba_w_mo.txt.gz #removes rows with any missing data, and also the SNP names

```

#2: Run TreeMix. Run TreeMix under different scenarios.

```{r Run TreeMix, eval=FALSE, engine='bash'}

V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani #VCFs path
O_PATH=/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path

#Basic command to build a ML tree
cd $O_PATH
$T_PATH/treemix -i treemix_input.txt.gz -o eurasian_lynx_out_stem

#Basic command to build a ML tree without Balkans
cd $O_PATH
$T_PATH/treemix -i treemix_input_wout_ba.txt.gz -o eurasian_lynx_wout_ba_out_stem

#Basic command to build a rooted ML tree, followed by a loop that runs TreeMix with different number of migration events
cd $O_PATH
$T_PATH/treemix -i treemix_input_rooted.txt.gz -root Rufus -k 100 -o eurasian_lynx_rooted_out_stem
END=6
for i in $(seq 1 $END)
  do echo "migration events number = " $i
  $T_PATH/treemix -i treemix_input_rooted.txt.gz -root Rufus -k 100 -m ${i} -o eurasian_lynx_rooted_out_stem_"${i}"_migration
  done

#Basic command to build a rooted ML tree without Balkans, followed by a loop that runs TreeMix with different number of migration events
cd $O_PATH
$T_PATH/treemix -i treemix_input_rooted_wout_ba.txt.gz -root Rufus -k 100 -o eurasian_lynx_rooted_wout_ba_out_stem #replace this with a run with m=0 inside the loop
END=6
for i in $(seq 0 $END)
  do echo "migration events number = " $i
  $T_PATH/treemix -i treemix_input_rooted_wout_ba.txt.gz -root Rufus -k 100 -m ${i} -o eurasian_lynx_rooted_wout_ba_out_stem_"${i}"_migration
  done

#Basic command to build a rooted ML tree without Balkans and with Mongolia, followed by a loop that runs TreeMix with different number of migration events
cd $O_PATH
#$T_PATH/treemix -i treemix_input_rooted_wout_ba_w_mo.txt.gz -root Rufus -k 100 -o eurasian_lynx_rooted_wout_ba_w_mo_out_stem #replace this with a run with m=0 inside the loop
END=6
for i in $(seq 0 $END)
  do echo "migration events number = " $i
  $T_PATH/treemix -i treemix_input_rooted_wout_ba_w_mo.txt.gz -root Rufus -k 100 -m ${i} -o eurasian_lynx_rooted_wout_ba_w_mo_out_stem_"${i}"_migration
  done

#Basic command to build a rooted ML tree without Balkans and with Mongolia, followed by a loop that runs TreeMix with different number of migration events
cd $O_PATH/intergenic_plus_1000
#$T_PATH/treemix -i treemix_input_intergenic_rooted_wout_ba_w_mo.txt.gz -root Rufus -k 100 -se -o eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem #replace this with a run with m=0 inside the loop
END=6
for i in $(seq 0 $END)
  do echo "migration events number = " $i
  $T_PATH/treemix -i treemix_input_intergenic_rooted_wout_ba_w_mo.txt.gz -root Rufus -k 100 -m ${i} -se -o eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_"${i}"_migration
  done

#Code to run additional TreeMix runs with M=2. I sent 9 nine additional runs (from run_2 to run_10):
zcat treemix_input_intergenic_rooted_wout_ba_w_mo_bis.txt.gz | sed 's/Rufus/Bobcat/g' | gzip > treemix_input_intergenic_rooted_wout_ba_w_mo.txt.gz #First, use this line to replace 'Rufus' with 'Bobcat'

#Then, run 9 independent TreeMix threads as follows (replacing run_2 with subsequent numbers):
screen -S treemix_intergenic_wout_ba_w_mo_migration_2_run_10.log
script treemix_intergenic_wout_ba_w_mo_migration_2_run_10.log
O_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path
cd $O_PATH/intergenic_plus_1000
$T_PATH/treemix -i treemix_input_intergenic_rooted_wout_ba_w_mo.txt.gz -root Bobcat -k 100 -m 2 -se -o eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_2_migration_run_10

#scp dkleinman@genomics-b.ebd.csic.es:/opt/treemix-1.12/src/plotting_funcs.R /Users/Dani/ownCloud/backup/annotation/TreeMix

#Copy files for all_positions trees:
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_out_stem* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/unrooted/all_pops
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_wout_ba_out_stem* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/unrooted/wout_ba
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_rooted_out_stem.* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/all_pops
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_rooted_out_stem*migration* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/all/migration
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_rooted_wout_ba_out_stem.* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/wout_ba
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_rooted_wout_ba_out_stem*migration* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/wout_ba/migration
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_rooted_wout_ba_w_mo_out_stem.* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/wout_ba_w_mo
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/eurasian_lynx_rooted_wout_ba_w_mo_out_stem*migration* /Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/wout_ba_w_mo/migration

#Copy files for intergenic positions trees:
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/intergenic_plus_1000/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem*migration* /Users/Dani/ownCloud/backup/annotation/TreeMix/intergenic/rooted/wout_ba_w_mo/
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/intergenic_plus_1000/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_2_migration_run* /Users/Dani/ownCloud/backup/annotation/TreeMix/intergenic/rooted/wout_ba_w_mo/independent_m_2_runs


#Basic command to build a rooted ML tree, followed by a loop that runs TreeMix with different number of migration events, with a bigger block size
cd $O_PATH/test_block_size
screen -S treemix_block_size_test
script treemix_block_size_test.log

#Basic command to build a rooted ML tree without Balkans and with Mongolia, followed by a loop that runs TreeMix with different number of migration events
cd $O_PATH/test_block_size
END=6
for i in $(seq 0 $END)
  do echo "migration events number = " $i
  $T_PATH/treemix -i ./../intergenic_plus_1000/treemix_input_intergenic_rooted_wout_ba_w_mo.txt.gz -root Bobcat -k 1000 -m ${i} -se -o eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_"${i}"_migration
  done

```

#3a: Plot TreeMix trees and residuals.

```{r Plot TreeMix trees and residuals}

library(readr)
library(ggtree)

local_repo <- file.path("/Users/Dani/ownCloud/backup/annotation/TreeMix/")
#poporder <- c("Balkans","Carpathians","Norway","Bialowieza","Latvia","Kirov","Urals","Tuva","Ömnögovi","Töv","Khentii","Mongolia","Yakutia","Vladivostok","Rufus")

source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"toy/ll_toy_basic_tree.pdf"),width=20,height=10)
toy_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/toy/toy_out_stem")
toy_out_stem
dev.off()

toy_wout_ba_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/toy/toy_wout_ba_out_stem")
toy_wout_ba_out_stem

#all Eurasian lynx populations
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"eurasian_lynx_out_stem.pdf"),width=20,height=10)
eurasian_lynx_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/unrooted/all_pops/eurasian_lynx_out_stem")
eurasian_lynx_out_stem
dev.off()
plot_resid(eurasian_lynx_out_stem, poporder)

#all Eurasian lynx populations except for Balkans
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"eurasian_lynx_wout_ba_out_stem.pdf"),width=20,height=10)
eurasian_lynx_wout_ba_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/unrooted/wout_ba/eurasian_lynx_wout_ba_out_stem")
eurasian_lynx_wout_ba_out_stem
dev.off()

#all Eurasian lynx populations, rooted with Lynx rufus, with 0-6 migration events
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"rooted/all_pops/eurasian_lynx_rooted_out_stem_tree.pdf"),width=20,height=10)
eurasian_lynx_rooted_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/all_pops/eurasian_lynx_rooted_out_stem")
eurasian_lynx_rooted_out_stem
dev.off()
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"all_positions/rooted/all_pops/eurasian_lynx_rooted_out_stem_residuals.pdf"),width=20,height=10)
eurasian_lynx_rooted_out_stem <- plot_resid("/Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/all_pops/eurasian_lynx_rooted_out_stem",paste0(local_repo,"all_positions/rooted/all_pops/migration/pop_order"))
eurasian_lynx_rooted_out_stem
dev.off()

for (i in 1:6) {
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"all_positions/rooted/all_pops/migration/eurasian_lynx_rooted_out_stem_",i,"_migration_tree.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_tree(paste0(local_repo,"all_positions/rooted/all_pops/migration/eurasian_lynx_rooted_out_stem_",i,"_migration"))
  eurasian_lynx_rooted_migration
  dev.off()
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"all_positions/rooted/all_pops/migration/eurasian_lynx_rooted_out_stem_",i,"_migration_residuals.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_resid(paste0(local_repo,"all_positions/rooted/all_pops/migration/eurasian_lynx_rooted_out_stem_",i,"_migration"),paste0(local_repo,"all_positions/rooted/all_pops/migration/pop_order"))
  eurasian_lynx_rooted_migration
  dev.off()
}

# #all Eurasian lynx populations except for Balkans, rooted with Lynx rufus
# source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
# pdf(paste0(local_repo,"eurasian_lynx_rooted_wout_ba_out_stem.pdf"),width=20,height=10)
# eurasian_lynx_rooted_wout_ba_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/eurasian_lynx_rooted_wout_ba_out_stem")
# eurasian_lynx_rooted_wout_ba_out_stem
# dev.off()

#all Eurasian lynx populations except for Balkans, with Mongolia grouped, rooted with Lynx rufus, with 0-6 migration events
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/eurasian_lynx_rooted_wout_ba_w_mo_out_stem_tree.pdf"),width=20,height=10)
eurasian_lynx_rooted_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/wout_ba_w_mo/eurasian_lynx_rooted_wout_ba_w_mo_out_stem")
eurasian_lynx_rooted_out_stem
dev.off()
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/eurasian_lynx_rooted_wout_ba_w_mo_out_stem_residuals.pdf"),width=20,height=10)
eurasian_lynx_rooted_out_stem <- plot_resid("/Users/Dani/ownCloud/backup/annotation/TreeMix/all_positions/rooted/wout_ba_w_mo/eurasian_lynx_rooted_wout_ba_w_mo_out_stem",paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/migration/pop_order"))
eurasian_lynx_rooted_out_stem
dev.off()

for (i in 1:6) {
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/migration/eurasian_lynx_rooted_wout_ba_w_mo_out_stem_",i,"_migration_tree.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_tree(paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/migration/eurasian_lynx_rooted_wout_ba_w_mo_out_stem_",i,"_migration"))
  eurasian_lynx_rooted_migration
  dev.off()
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/migration/eurasian_lynx_rooted_wout_ba_w_mo_out_stem_",i,"_migration_residuals.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_resid(paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/migration/eurasian_lynx_rooted_wout_ba_w_mo_out_stem_",i,"_migration"),paste0(local_repo,"all_positions/rooted/wout_ba_w_mo/migration/pop_order"))
  eurasian_lynx_rooted_migration
  dev.off()
}


#intergenic for all Eurasian lynx populations except for Balkans, with Mongolia grouped, rooted with Lynx rufus, with 0-6 migration events
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_tree.pdf"),width=20,height=10)
eurasian_lynx_rooted_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem")
eurasian_lynx_rooted_out_stem
dev.off()
source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_residuals.pdf"),width=20,height=10)
eurasian_lynx_rooted_out_stem <- plot_resid("/Users/Dani/ownCloud/backup/annotation/TreeMix/intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem",paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/migration/pop_order"))
eurasian_lynx_rooted_out_stem
dev.off()

for (i in 0:6) {
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration_tree.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_tree(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration"))
  eurasian_lynx_rooted_migration
  dev.off()
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration_residuals.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_resid(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration"),paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/pop_order"))
  eurasian_lynx_rooted_migration
  dev.off()
}

for (i in 2:10) {
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/independent_m_2_runs/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_2_migration_run_",i,"_tree.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_tree(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/independent_m_2_runs/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_2_migration_run_",i))
  eurasian_lynx_rooted_migration
  dev.off()
  source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
  pdf(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/independent_m_2_runs/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_2_migration_run_",i,"_residuals.pdf"),width=20,height=10)
  eurasian_lynx_rooted_migration <- plot_resid(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/independent_m_2_runs/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_2_migration_run_",i),paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/independent_m_2_runs/pop_order"))
  eurasian_lynx_rooted_migration
  dev.off()
}


#Visualize trees with ggtree (NOT).

import_tree <- read.tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_3_migration.treeout.gz")

basic_ggtree <- ggplot(import_tree) + geom_tree() + theme_tree() + theme_tree2() + geom_tiplab(size=3)
basic_ggtree

group_df <- data_frame("Pop"=c("Rufus","Carpathians","Bialowieza","Latvia","Urals","Kirov","Norway","Mongolia","Tuva","Yakutia","Vladivostok"),"Group"=c("O","W","W","W","W","W","W","E","E","E","E"))

group_info <- split(import_tree$tip.label,unlist(group_df[match(import_tree$tip.label,unlist(group_df[,1],use.names=F)),2],use.names = F))
import_tree <- groupOTU(import_tree,group_info)

grouped_tree <- ggplot(import_tree, aes(colour=group)) + geom_tree() + theme_tree() + theme_tree2() + geom_tiplab(size=3) + scale_color_manual(values=c("black", "red","black","blue"))
grouped_tree

```

#3a: Plot TreeMix variances.

```{r Plot TreeMix variances}

library(readr)
library(dplyr)
library(ggplot2)

local_repo <- file.path("/Users/Dani/ownCloud/backup/annotation/TreeMix/")
#poporder <- c("Balkans","Carpathians","Norway","Bialowieza","Latvia","Kirov","Urals","Tuva","Ömnögovi","Töv","Khentii","Mongolia","Yakutia","Vladivostok","Rufus")

fraction_variance <- data_frame("m"=numeric(0),"variance"=double(0))
for (i in 0:6) {
args <- commandArgs(TRUE)
calcVarExplain <- function(stem){
  #Calculate standard error of covariance matrix
  se = read.table(gzfile(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration.covse.gz")), as.is = T, head = T, quote = "", comment.char = "")
  seBar = apply(se, 1, mean)
  seBar = mean(seBar)
  #Read in two covariance matrices
  WHat <- read.table(gzfile(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration.cov.gz")), as.is = T, head = T, quote = "", comment.char = "")
  W <- read.table(gzfile(paste0(local_repo,"intergenic/rooted/wout_ba_w_mo/eurasian_lynx_intergenic_rooted_wout_ba_w_mo_out_stem_",i,"_migration.modelcov.gz")), as.is = T, head = T, quote = "", comment.char = "")
  #Set row and column names and order properly
  names(WHat) = rownames(WHat)
  names(W) = rownames(W)
  WHat = WHat[order(names(WHat)), order(names(WHat))]
  W = W[order(names(W)), order(names(W))]
  #Calculate residuals
  R = WHat - W
  #Set residual lower matrix triangle to NA
  R[lower.tri(R, diag = TRUE)] <- NA
  #Calculate mean residual across upper matrix triangle
  RBar = mean(as.matrix(R), na.rm = TRUE)
  #Set data-estimated W(hat) matrix lower matrix triangle to NA
  WHat[lower.tri(WHat, diag = TRUE)] <- NA
  #Calculate mean W(hat)
  WHatBar = mean(as.matrix(WHat), na.rm = TRUE)
  #Calculate numerator of equation 30
  fNum = sum((R - RBar)^2, na.rm = TRUE)
  #Calculate denominator of equation 30
  fDen = sum((WHat - WHatBar)^2, na.rm = TRUE)
  #Calculate f in equation 30
  f = 1 - (fNum/fDen)
  #Return output
  outList = list("StdErr" = seBar, "VarExplain" = f)
  return(outList)
}
output <- calcVarExplain(args[1])
fraction_variance[i+1,] <- c(i,output$VarExplain)
}
fraction_variance

fraction_variance_ggplot <- ggplot(data=fraction_variance, aes(x=m,y=variance)) + #[-c(1:3),] to plot from 3 to 6
  geom_point() +
  #ggtitle("Number of private SNPs per population with different filterings") +
  ylab("Fraction of variance") +
  xlab("m (Number of migration events)") +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=2),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm")#,
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        #legend.background=element_rect(linetype="solid", colour="black", size=1),
        #legend.justification=c(0,0),
        #legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.14,0.8),
        #legend.title=element_blank()
       )
fraction_variance_ggplot
ggsave("/intergenic/rooted/wout_ba_w_mo/fraction_of_variance.pdf", width=20, height=10, units="cm", device="pdf", path=local_repo)

```

#4a: Whole-genome ThreePop. Run ThreePop admixture tests.

```{r Whole-genome ThreePop, eval=FALSE, engine='bash'}

V_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani #VCFs path
O_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path


#Run threepop test to evaluate admixture between populations (without Balkans). I eventually abort it because it will take MONTHS.
cd $O_PATH
$T_PATH/threepop -i treemix_input_rooted_wout_ba.txt.gz -k 100


#####ThreePop with Khentii in the East:

#Get a ThreePop input with only Urals, Tuva & Khentii:
cut -d$' ' -f1,4,11,12  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ka_tu_ur.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ka_tu_ur.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ka_tu_ur.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Khentii:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ka_tu_ur.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Khentii:
cut -d$' ' -f1,4,5,11  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ka_ki_tu.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ka_ki_tu.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ka_ki_tu.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Khentii:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ka_ki_tu.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Khentii:
cut -d$' ' -f1,3,4,11  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ca_ka_tu.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ca_ka_tu.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ca_ka_tu.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Khentii:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ca_ka_tu.txt.gz -k 100


#####ThreePop with Yakutia in the East:

#Get a ThreePop input with only Urals, Tuva & Yakutia:
cut -d$' ' -f1,11,12,14  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_tu_ur_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_tu_ur_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_tu_ur_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Yakutia:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_tu_ur_ya.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Yakutia:
cut -d$' ' -f1,5,11,14  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ki_tu_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ki_tu_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ki_tu_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Yakutia:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ki_tu_ya.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Yakutia:
cut -d$' ' -f1,3,11,14  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ca_tu_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ca_tu_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ca_tu_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Yakutia:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ca_tu_ya.txt.gz -k 100


#####ThreePop with Vladivostok in the East:

#Get a ThreePop input with only Urals, Tuva & Vladivostok:
cut -d$' ' -f1,11,12,13  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_tu_ur_vl.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_tu_ur_vl.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_tu_ur_vl.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Vladivostok:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_tu_ur_vl.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Vladivostok:
cut -d$' ' -f1,5,11,13  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ki_tu_vl.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ki_tu_vl.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ki_tu_vl.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Vladivostok:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ki_tu_vl.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Vladivostok:
cut -d$' ' -f1,3,11,13  SNP_names_treemix_input_rooted.txt > SNP_names_threepop_input_ca_tu_vl.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ca_tu_vl.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ca_tu_vl.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Vladivostok:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ca_tu_vl.txt.gz -k 100


#####ThreePop with Mongolia (grouped) in the East:

#Get a ThreePop input with only Urals, Tuva & Mongolia:
cut -d$' ' -f1,11,12,16  SNP_names_treemix_input_rooted_w_mo.txt > SNP_names_threepop_input_tu_ur_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_tu_ur_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_tu_ur_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Mongolia:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_tu_ur_mo.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Mongolia:
cut -d$' ' -f1,5,11,16  SNP_names_treemix_input_rooted_w_mo.txt > SNP_names_threepop_input_ki_tu_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ki_tu_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ki_tu_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Mongolia:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ki_tu_mo.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Mongolia:
cut -d$' ' -f1,3,11,16  SNP_names_treemix_input_rooted_w_mo.txt > SNP_names_threepop_input_ca_tu_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_ca_tu_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_ca_tu_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Mongolia:
cd $O_PATH
$T_PATH/threepop -i SNP_names_threepop_input_ca_tu_mo.txt.gz -k 100

```

#4b: Intergenic ThreePop for Tuva. Run ThreePop admixture tests with Tuva as admixed.

```{r Intergenic ThreePop, eval=FALSE, engine='bash'}

V_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani #VCFs path
O_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path


#####ThreePop with Yakutia in the East:

#Get a ThreePop input with only Urals, Tuva & Yakutia:
cut -d$' ' -f1,11,12,14  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_tu_ur_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_tu_ur_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_tu_ur_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Yakutia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_tu_ur_ya.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Yakutia:
cut -d$' ' -f1,5,11,14  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ki_tu_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ki_tu_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ki_tu_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Yakutia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ki_tu_ya.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Yakutia:
cut -d$' ' -f1,3,11,14  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ca_tu_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ca_tu_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ca_tu_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Yakutia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ca_tu_ya.txt.gz -k 100


#####ThreePop with Vladivostok in the East:

#Get a ThreePop input with only Urals, Tuva & Vladivostok:
cut -d$' ' -f1,11,12,13  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_tu_ur_vl.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_tu_ur_vl.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_tu_ur_vl.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Vladivostok:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_tu_ur_vl.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Vladivostok:
cut -d$' ' -f1,5,11,13  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ki_tu_vl.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ki_tu_vl.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ki_tu_vl.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Vladivostok:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ki_tu_vl.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Vladivostok:
cut -d$' ' -f1,3,11,13  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ca_tu_vl.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ca_tu_vl.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ca_tu_vl.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Vladivostok:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ca_tu_vl.txt.gz -k 100


#####ThreePop with Mongolia (grouped) in the East:

#Get a ThreePop input with only Urals, Tuva & Mongolia:
cut -d$' ' -f1,7,8,12  SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt > SNP_names_threepop_input_intergenic_tu_ur_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_tu_ur_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_tu_ur_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Tuva and Mongolia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_tu_ur_mo.txt.gz -k 100


#Get a ThreePop input with only Kirov, Tuva & Mongolia:
cut -d$' ' -f1,3,7,12  SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt > SNP_names_threepop_input_intergenic_ki_tu_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ki_tu_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ki_tu_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Tuva and Mongolia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ki_tu_mo.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Tuva & Mongolia:
cut -d$' ' -f1,2,7,12  SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt > SNP_names_threepop_input_intergenic_ca_tu_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ca_tu_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ca_tu_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Tuva and Mongolia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ca_tu_mo.txt.gz -k 100

```

#4c: Intergenic ThreePop for Yakutia. Run ThreePop admixture tests with Yakutia as admixed.

```{r Intergenic ThreePop, eval=FALSE, engine='bash'}

V_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani #VCFs path
O_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path


#####ThreePop with Tuva in the East:

#Get a ThreePop input with only Urals, Yakutia & Tuva: already done for Tuva before! Check SNP_names_threepop_input_intergenic_tu_ur_ya.txt.gz


#Get a ThreePop input with only Kirov, Yakutia & Tuva: already done for Tuva before! Check SNP_names_threepop_input_intergenic_ki_tu_ya.txt.gz


#Get a ThreePop input with only Carpathians, Yakutia & Tuva: already done for Tuva before! Check SNP_names_threepop_input_intergenic_ca_tu_ya.txt.gz


#####ThreePop with Vladivostok in the East:

#Get a ThreePop input with only Urals, Yakutia & Vladivostok:
cut -d$' ' -f1,12,13,14  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ur_vl_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ur_vl_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ur_vl_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Yakutia and Vladivostok:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ur_vl_ya.txt.gz -k 100


#Get a ThreePop input with only Kirov, Yakutia & Vladivostok:
cut -d$' ' -f1,5,13,14  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ki_vl_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ki_vl_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ki_vl_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Yakutia and Vladivostok:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ki_vl_ya.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Yakutia & Vladivostok:
cut -d$' ' -f1,3,13,14  SNP_names_treemix_input_intergenic_rooted.txt > SNP_names_threepop_input_intergenic_ca_vl_ya.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ca_vl_ya.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ca_vl_ya.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Yakutia and Vladivostok:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ca_vl_ya.txt.gz -k 100


#####ThreePop with Mongolia (grouped) in the East:

#Get a ThreePop input with only Urals, Yakutia & Mongolia:
cut -d$' ' -f1,8,10,12  SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt > SNP_names_threepop_input_intergenic_ur_ya_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ur_ya_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ur_ya_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Urals, Yakutia and Mongolia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ur_ya_mo.txt.gz -k 100


#Get a ThreePop input with only Kirov, Yakutia & Mongolia:
cut -d$' ' -f1,3,10,12  SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt > SNP_names_threepop_input_intergenic_ki_ya_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ki_ya_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ki_ya_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Kirov, Yakutia and Mongolia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ki_ya_mo.txt.gz -k 100


#Get a ThreePop input with only Carpathians, Yakutia & Mongolia:
cut -d$' ' -f1,2,10,12  SNP_names_treemix_input_intergenic_rooted_wout_ba_w_mo.txt > SNP_names_threepop_input_intergenic_ca_ya_mo.txt #keeps only columns for the three desired populations
grep -v ' 0,0 ' SNP_names_threepop_input_intergenic_ca_ya_mo.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > SNP_names_threepop_input_intergenic_ca_ya_mo.txt.gz #removes rows with any missing data, and also the SNP names

#Run threepop test to evaluate admixture between Carpathians, Yakutia and Mongolia:
cd $O_PATH/intergenic_plus_1000
$T_PATH/threepop -i SNP_names_threepop_input_intergenic_ca_ya_mo.txt.gz -k 100

```
