---
title: "rvis_database"
author: "Dani"
date: "8 de julio de 2019"
output: html_document
---

#1: Prepare RVIS database.
##Prepare human RVIS database.
```{bash}

mkdir -p /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis

#Download the database with the RVIS scores:
curl http://genic-intolerance.org/data/RVIS_Unpublished_ExACv2_March2017.txt --output rvis_database.txt

#Extract gene names from the RVIS database:
tail -n +2 rvis_database.txt | cut -f1 > rvis_gene_names.txt

#Since genes in the database are registered with their gene name, but our own Lynx pardinus database uses the UniProt ID codes, we need to convert them first. The UniProt website has a tool for this. First, download rvis_gene_names.txt to local. Next, upload it to https://www.uniprot.org/uploadlists/, select "from gene names to UniProtKB", insert Homo sapiens as the species, and submit the search. Once it finishes (17505 out of 17518 Gene name identifiers were successfully mapped to 17564 UniProtKB IDs), filter in only the reviewed results, and select the desired columns (in mycase I included my input list as the 6th column). Then download the file as rvis_gene_names_uniprot_codes_dirty.txt and upload it to the working folder inside the server.

#Separate comma-separated entries into new lines to clean the gene_names_uniprot_codes_dirty.txt file:
tail -n +2 rvis_gene_names_uniprot_codes_dirty.txt | awk -F"\t" '{printf ("%s\t%s\n", $6,$1)}' | grep ',' > rvis_gene_names_uniprot_codes_dirty_rows.txt
rm rvis_gene_names_uniprot_codes_clean.txt
while read row;
  do
  #echo ${row}
  GENE_NAMES=$(echo "$row" | cut -f1 | tr , '\n')
  #echo $GENE_NAMES
  UNIPROT=$(echo "$row" | cut -f2)
  #echo $UNIPROT
  for GENE in ${GENE_NAMES[@]}
    do
    #echo $GENE
    echo -e "$GENE\t$UNIPROT" >> rvis_gene_names_uniprot_codes_clean.txt
    done
  done < rvis_gene_names_uniprot_codes_dirty_rows.txt
tail -n +2 rvis_gene_names_uniprot_codes_dirty.txt | awk -F"\t" '{printf ("%s\t%s\n", $6,$1)}' | grep -v ',' >> rvis_gene_names_uniprot_codes_clean.txt

#Join the RVIS database and the UniProt codes file:
join -t $'\t' <(sort -k 1b,1 rvis_gene_names_uniprot_codes_clean.txt) <(tail -n +2 rvis_database.txt | sort -k 1b,1) > rvis_database_uniprot_joined.txt

```

##Join it with the lynx genes.
###Reactome pathway genes:
```{bash}

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis
#Extract the Lynx pardinus genes (change LYPA23B for LYPA23C) and their UniProt codes from the proper file that was generated as part of the Genome Project:
tail -n +2 /backup/grupolince/Lynx_annotation/Protein\ annotation/Funtional_annotation_140305/Reactome_pathway_annotations.txt | grep "Homo sapiens" | awk -F"\t|," '{printf ("%s\t%s\t%s\n", $1,$3,$5)}' | awk -F"\t|]" '{printf ("%s\t%s\n", $1,$3)}' | awk 'BEGIN{FS=OFS=";"} {gsub("B", "C", $1)} 1' | sort -k 2,2 | uniq > lypa_reactome_gene_names_uniprot_codes.txt

#Join the Lynx pardinus and the RVIS databases, and sort genes from most deleterious (most negative score in column 5) to least:
join -1 2 -2 2 -t $'\t' <(sort -k 2b,2 lypa_reactome_gene_names_uniprot_codes.txt) <(tail -n +2 rvis_database_uniprot_joined.txt | sort -k 2b,2) | awk 'BEGIN{FS=OFS="\t"} {$2 = substr($2, 1, length($2)-2)}1' | sort -n -k 5,5 > lypa_reactome_rvis_database_uniprot_joined.txt

```

###Orhtologous genes:
```{bash}

#The lynx-human orthologs were already compiled before. I obtained the UniProt codes in the script "all_genes_database.Rmd", section: "Obtain UniProt codes for human-lynx orthologous genes database".
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis

awk -F"\t| " '{printf ("%s\t%s\n", $2,$3)}' /GRUPOS/grupolince/ortologous/HUMAN.1to1_and_multi_orthologs.joined_codes.txt | sort -k 2,2 | uniq | awk 'BEGIN{FS=OFS="\t"} {gsub("LYPA23A", "LYPA23C", $1)} 1' > lypa_orthologs_gene_names_uniprot_codes.txt

#Join the Lynx pardinus and the RVIS databases, and sort genes from most deleterious (most negative score in column 5) to least:
join -1 2 -2 2 -t $'\t' <(sort -k 2b,2 lypa_orthologs_gene_names_uniprot_codes.txt) <(tail -n +2 rvis_database_uniprot_joined.txt | sort -k 2b,2) | awk 'BEGIN{FS=OFS="\t"} {$2 = substr($2, 1, length($2)-2)}1' | sort -n -k 5,5 > lypa_orthologs_rvis_database_uniprot_joined.txt

```

###Combine both:
```{bash}

cat lypa_*_rvis_database_uniprot_joined.txt | sort -k2,2 | uniq > lypa_rvis_database_uniprot_joined.txt

```

#2: Explore RVIS scores.

```{bash}

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis

#awk -v FS="|" 'NR==FNR{rows[$1]++;next}(substr($NF,1,length($NF)-1) in rows)' /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.vcf <(head -n 100 lypa_rvis_database_uniprot_joined.txt | cut -f 2) | less -S

#Count number of deleterious variants within low tolerance (deleterious) or high tolerance (tolerant) genes, and calculate the average AF:

screen -S del.af
DEL_GENES=$(awk '$6 < 3' lypa_rvis_database_uniprot_joined.txt | cut -f 2) #255
COUNTER_MIS=0
COUNTER_LOF=0
rm c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.del.af
for gen in ${DEL_GENES[@]}
  do
  echo ${gen}
  GREP="\|${gen}\|"
  GREP_MIS=$(grep -E "missense_variant.*${GREP}" /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.vcf)
  N_MIS=$(echo "$GREP_MIS" | wc -l)
  echo $N_MIS
  echo "$GREP_MIS" | awk -F"\t|;|=" '{printf ("%s\n", $13)}' >> c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.del.af
  N_LOF=$(grep -E "\|HIGH\|.*${GREP}" /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.vcf | wc -l)
  COUNTER_MIS=$((COUNTER_MIS+N_MIS))
  COUNTER_LOF=$((COUNTER_LOF+N_LOF))
  done
echo "total of MIS is" $COUNTER_MIS #555
SUM_AF_MIS=$(awk NF c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.del.af | paste -sd+ | bc)
AVG_AF_MIS=$(echo "scale=4; $SUM_AF_MIS/$COUNTER_MIS" | bc)
echo "average AF of MIS is" $AVG_AF_MIS #0.1994
echo "total of LOF is" $COUNTER_LOF #16


screen -S tol.af
TOL_GENES=$(awk '$6 > 94' lypa_rvis_database_uniprot_joined.txt | cut -f 2) #429
COUNTER_MIS=0
COUNTER_LOF=0
rm c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.tol.af
for gen in ${TOL_GENES[@]}
  do
  echo ${gen}
  GREP="\|${gen}\|"
  GREP_MIS=$(grep -E "missense_variant.*${GREP}" /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.vcf)
  N_MIS=$(echo "$GREP_MIS" | wc -l)
  echo $N_MIS
  echo "$GREP_MIS" | awk -F"\t|;|=" '{printf ("%s\n", $13)}' >> c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.tol.af
  N_LOF=$(grep -E "\|HIGH\|.*${GREP}" /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.vcf | wc -l)
  COUNTER_MIS=$((COUNTER_MIS+N_MIS))
  COUNTER_LOF=$((COUNTER_LOF+N_LOF))
  done
echo "total of MIS is" $COUNTER_MIS #1651
SUM_AF_MIS=$(awk NF c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.tol.af | paste -sd+ | bc)
AVG_AF_MIS=$(echo "scale=4; $SUM_AF_MIS/$COUNTER_MIS" | bc)
echo "average AF of MIS is" $AVG_AF_MIS #0.2788
echo "total of LOF is" $COUNTER_LOF #26



#?
DEL_GENES=$(awk '$6 < 3' lypa_rvis_database_uniprot_joined.txt | cut -f 2) #91
for gen in ${DEL_GENES[@]}
  do
  echo ${gen}
  GREP="\|${gen}\|"
  grep -E "\|HIGH\|.*${GREP}" /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_polarized_filteredall_varssubs_SNP.lr_ann.vcf
  done

```

#3: Obtain BED file for the most deleterious genes.
```{bash}

cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis

awk -F"\t" '$5 <= -2' lypa_rvis_database_uniprot_joined.txt | sort -rk 5 | cut -f2 > deleterious_genes_list.txt

```

#4: Subset deleterious genes from the individual VCF files.
```{bash}

#This code will only work if the whole-genome individual VCFs have been generated before.
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "individual_deleterious_genes_${VAR}_${CALLING}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

DEL_GENES_FILE=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/rvis/deleterious_genes_list.txt

cd $V_PATH/$CALLING/annotation
INDLIST=($(ls `find . -name *"_individual_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  grep '#' ${i} > ${i/_individual_/_individual_deleterious_genes_} 
  grep -f $DEL_GENES_FILE ${i} >> ${i/_individual_/_individual_deleterious_genes_}
  done

```

#5: Get counts (of variants, substitutions or vars+subs).
```{r Get annotation statistics, eval=FALSE, engine='bash'}

CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
cd /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation
screen -S "${CALLING}-${VAR}-${TYPE}"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)
script "${CALLING}_ann_individual_deleterious_genes_summary_${VAR}_${TYPE}.lr_ann.log"
CALLING=$(echo ${STY#*.} | cut -d'-' -f1)
VAR=$(echo ${STY#*.} | cut -d'-' -f2)
TYPE=$(echo ${STY#*.} | cut -d'-' -f3)

S_PATH=/opt/snpEff #software path
C_PATH=/home/dkleinman/datos/snpEff #config file path
O_PATH=/home/dkleinman/datos/snpEff #output path
I_PATH=/home/GRUPOS/grupolince/immunocapture/prueba_highdiv #immunocapture path
V_PATH=/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs #VCFs path
G_PATH=/GRUPOS/grupolince/lynx_genomes_5x/gVCFs #gVCFs path
B_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final #BAM files path
REF=/home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa #path to reference genome
GATK=/opt/GATK-3.7/GenomeAnalysisTK.jar #GATK software path
BCF=/opt/bcftools-1.6/bcftools #BCFtools software path

cd $V_PATH/$CALLING/annotation
rm ${CALLING}"_ann_individual_deleterious_genes_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
echo -e "species\tpopulation\tdataset\tsample\ttotal_V\ttotal_A\tintergenic_V\tintergenic_A\tintronic_V\tintronic_A\tcoding_V\tsynonymous_V\tsynonymous_A\tsyn_pref_V\tsyn_pref_A\tsyn_unpref_V\tsyn_unpref_A\tmissense_V\tmissense_A\tmissense_tol_V\tmissense_tol_A\tmissense_del_V\tmissense_del_A\tnonsense_V\tnonsense_A\tUCNE_V\tUCNE_A\tUCNE_low_V\tUCNE_low_A\tUCNE_mid_V\tUCNE_mid_A\tUCNE_high_V\tUCNE_high_A\tmissense/synonymous_V\tmissense/synonymous_A\tsynonymous/intronic_V\tmissense/intronic_V" > ${CALLING}"_ann_individual_deleterious_genes_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
INDLIST=($(ls `find . -name *"_individual_deleterious_genes_"${VAR}"_"${TYPE}".lr_ann.vcf" -print`))
for i in "${INDLIST[@]}"
  do
  echo "${i}"
  ind=$(echo "${i}" | awk -F'[/]' '{print $3}' | cut -c1-12)
  echo "${ind}"
  SPECIES=$(echo "${ind}" | cut -c3-4)
  POPULATION=$(echo "${ind}" | cut -c6-7)
  DATASET=$(if [ $ind = "c_lp_sm_0221" ]; then echo "REF"; elif [ $ind = "c_ll_ki_0090" ]; then echo "MG"; elif [ $ind = "h_ll_pv_0223" ]; then echo "LD"; elif grep -Fxq $ind /GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final/c_lp_5x_samples || [ $SPECIES = "ll" ]; then echo "5x"; else echo "GP"; fi)
  SAMPLE=$(echo "${ind}" | cut -c9-12)
  TOTAL_V=$(grep -v '#' ${i} | wc -l)
  TOTAL_A=$(grep -v '#' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTERGENIC_V=$(grep 'intergenic' ${i} | wc -l)
  INTERGENIC_A=$(grep 'intergenic' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  INTRONIC_V=$(grep 'intron_variant' ${i} | wc -l)
  INTRONIC_A=$(grep 'intron_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  CODING_V=$(grep 'CDS' ${i} | wc -l)
  SYNONYMOUS_V=$(grep 'synonymous_variant' ${i} | wc -l)
  SYNONYMOUS_A=$(grep 'synonymous_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_0to1_lr.bed > ${VAR}_delgenes_syn_pref.temp.borrar
  SYN_PREF_V=$(wc -l < ${VAR}_delgenes_syn_pref.temp.borrar)
  SYN_PREF_A=$(cut -f8 ${VAR}_delgenes_syn_pref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/syn_pref/synonymous_variants_complete_info_1to0_lr.bed > ${VAR}_delgenes_syn_unpref.temp.borrar
  SYN_UNPREF_V=$(wc -l < ${VAR}_delgenes_syn_unpref.temp.borrar)
  SYN_UNPREF_A=$(cut -f8 ${VAR}_delgenes_syn_unpref.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  MISSENSE_V=$(grep 'missense_variant' ${i} | wc -l)
  MISSENSE_A=$(grep 'missense_variant' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_tolerated.txt > ${VAR}_delgenes_mis_tol.temp.borrar
  MISSENSE_TOL_V=$(wc -l < ${VAR}_delgenes_mis_tol.temp.borrar)
  MISSENSE_TOL_A=$(cut -f8 ${VAR}_delgenes_mis_tol.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/provean/missense_variants_provean_scores_deleterious.txt > ${VAR}_delgenes_mis_del.temp.borrar
  MISSENSE_DEL_V=$(wc -l < ${VAR}_delgenes_mis_del.temp.borrar)
  MISSENSE_DEL_A=$(cut -f8 ${VAR}_delgenes_mis_del.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc)
  NONSENSE_V=$(grep '|HIGH|' ${i} | wc -l)
  if [ $NONSENSE_V == 0 ]; then NONSENSE_A=0; else NONSENSE_A=$(grep '|HIGH|' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
  UCNE_V=$(grep 'UCNE' ${i} | wc -l)
  if [ $UCNE_V == 0 ]; then UCNE_A=0; else UCNE_A=$(grep 'UCNE' ${i} | cut -f8 | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_lt2.gerp.bed > ${VAR}_delgenes_ucne_low.temp.borrar
  UCNE_LOW_V=$(wc -l < ${VAR}_delgenes_ucne_low.temp.borrar)
  if [ $UCNE_LOW_V == 0 ]; then UCNE_LOW_A=0; else UCNE_LOW_A=$(cut -f8 ${VAR}_delgenes_ucne_low.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt2lt5.gerp.bed > ${VAR}_delgenes_ucne_mid.temp.borrar
  UCNE_MID_V=$(wc -l < ${VAR}_delgenes_ucne_mid.temp.borrar)
  if [ $UCNE_MID_V == 0 ]; then UCNE_MID_A=0; else UCNE_MID_A=$(cut -f8 ${VAR}_delgenes_ucne_mid.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
  bedtools intersect -a ${i} -b /GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov/annotation/ucne_database/gerp_analysis/c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov.UCNE.derived_gt5.gerp.bed > ${VAR}_delgenes_ucne_high.temp.borrar
  UCNE_HIGH_V=$(wc -l < ${VAR}_delgenes_ucne_high.temp.borrar)
  if [ $UCNE_HIGH_V == 0 ]; then UCNE_HIGH_A=0; else UCNE_HIGH_A=$(cut -f8 ${VAR}_delgenes_ucne_high.temp.borrar | cut -d';' -f2 | cut -d'=' -f2 | paste -sd+ | bc); fi
  MISSENSE_SYNONYMOUS_V=$(echo "scale=4; $MISSENSE_V/$SYNONYMOUS_V" | bc)
  MISSENSE_SYNONYMOUS_A=$(echo "scale=4; $MISSENSE_A/$SYNONYMOUS_A" | bc)
  SYNONYMOUS_INTRONIC_V=$(echo "scale=4; $SYNONYMOUS_V/$INTRONIC_V" | bc)
  MISSENSE_INTRONIC_V=$(echo "scale=4; $MISSENSE_V/$INTRONIC_V" | bc)
  echo -e "$SPECIES\t$POPULATION\t$DATASET\t$SAMPLE\t$TOTAL_V\t$TOTAL_A\t$INTERGENIC_V\t$INTERGENIC_A\t$INTRONIC_V\t$INTRONIC_A\t$CODING_V\t$SYNONYMOUS_V\t$SYNONYMOUS_A\t$SYN_PREF_V\t$SYN_PREF_A\t$SYN_UNPREF_V\t$SYN_UNPREF_A\t$MISSENSE_V\t$MISSENSE_A\t$MISSENSE_TOL_V\t$MISSENSE_TOL_A\t$MISSENSE_DEL_V\t$MISSENSE_DEL_A\t$NONSENSE_V\t$NONSENSE_A\t$UCNE_V\t$UCNE_A\t$UCNE_LOW_V\t$UCNE_LOW_A\t$UCNE_MID_V\t$UCNE_MID_A\t$UCNE_HIGH_V\t$UCNE_HIGH_A\t$MISSENSE_SYNONYMOUS_V\t$MISSENSE_SYNONYMOUS_A\t$SYNONYMOUS_INTRONIC_V\t$MISSENSE_INTRONIC_V" >> ${CALLING}"_ann_individual_deleterious_genes_summary_"${VAR}"_"${TYPE}".lr_ann.txt"
  done

#From outside the server:
export SSHPASS=$(cat /Users/dani/Documents/genomics_pass.txt)
CALLING=(c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov)
VAR=(varssubs) #varssubs #variants #substitutions
TYPE=(SNP) #write down SNP or INDEL
sshpass -e scp dkleinman@genomics-a.ebd.csic.es:/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/nmVCFs/$CALLING/annotation/${CALLING}_ann_individual_deleterious_genes_summary_${VAR}_${TYPE}.lr_ann.txt /Users/Dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/deleterious_genes/
unset SSHPASS

```

#6: Visualise relativised derived allele counts.
#####Relative to introns (population average version).
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants #substitutions #segregating #fixed #private_segregating

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/deleterious_genes/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_deleterious_genes_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,synpref=syn_pref_A/intronic_A,synunpref=syn_unpref_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A,UCNElow=UCNE_low_A/intronic_A,UCNEmid=UCNE_mid_A/intronic_A,UCNEhigh=UCNE_high_A/intronic_A) %>% select(c(1:4,38:48)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

#Obtain per population averages and standard errors:
se <- function(x) sqrt(var(x)/length(x)) #first define the standard error function

average_relativised_variants_and_subst_wg_alleleR <- data_frame("species"=character(0),"population"=character(0),"ratio"=character(0),"avg_ki_relative_value"=character(0),"se_ki_relative_value"=character(0)) #next, create the empty dataframe

for (pop in unique(relativised_variants_and_subst_wg_alleleR$population)) { #then loop over each population and feature to get the (relativised) mean and standard error, and feed the dataframe
  print(pop)
  species <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(species) %>% unlist(.,use.names=F) %>% unique()
  for (r in unique(relativised_variants_and_subst_wg_alleleR$ratio)) {
    print(r)
    pop_mean <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% mean()
    #print(paste0(pop," feature ",r," average is ",pop_mean))
    pop_se <- filter(relativised_variants_and_subst_wg_alleleR,ratio==r & population==pop) %>% select(ki_relative_value) %>% unlist(.,use.names=F) %>% se()
    #print(paste0(pop," feature ",r," std error is ",pop_se))
    row_data <- cbind(species,pop,r,pop_mean,pop_se)
    colnames(row_data) <- c("species","population","ratio","avg_ki_relative_value","se_ki_relative_value")
    average_relativised_variants_and_subst_wg_alleleR <- rbind(average_relativised_variants_and_subst_wg_alleleR,row_data,stringsAsFactors=F)
  }
}
average_relativised_variants_and_subst_wg_alleleR$population = factor(average_relativised_variants_and_subst_wg_alleleR$population,levels=c("ki","no","po","sm","do"))
levels(average_relativised_variants_and_subst_wg_alleleR$population)[levels(average_relativised_variants_and_subst_wg_alleleR$population)=="sm"] <- "an"
average_relativised_variants_and_subst_wg_alleleR$ratio = factor(average_relativised_variants_and_subst_wg_alleleR$ratio,levels=c("synonymous","synpref","synunpref","missense","mistol","misdel","nonsense","UCNE","UCNElow","UCNEmid","UCNEhigh"))
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synpref"] <- "synonymous_pref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="synunpref"] <- "synonymous_unpref"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="mistol"] <- "missense_tolerated"
levels(average_relativised_variants_and_subst_wg_alleleR$ratio)[levels(average_relativised_variants_and_subst_wg_alleleR$ratio)=="misdel"] <- "missense_deleterious"
average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$avg_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value <- as.numeric(average_relativised_variants_and_subst_wg_alleleR$se_ki_relative_value)
average_relativised_variants_and_subst_wg_alleleR
write_csv(average_relativised_variants_and_subst_wg_alleleR,paste0("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/deleterious_genes/",type,"_derived_allele_allele_ratio_relative2introns_mean.csv"))


#Separate plots:
average_relativised_derived_allele_allele_ratio_ggplot1 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "synonymous" | ratio == "synonymous_pref" | ratio == "synonymous_unpref" | ratio == "missense" | ratio == "missense_tolerated" | ratio == "missense_deleterious"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0.7, 1.2, by = 0.1)) +
  ylim(0.7,1.2) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot1
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_part1_mean.pdf"), width=35, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/deleterious_genes/")

average_relativised_derived_allele_allele_ratio_ggplot2 <- ggplot(data=filter(average_relativised_variants_and_subst_wg_alleleR,ratio == "nonsense" | ratio == "UCNE" | ratio == "UCNElow" | ratio == "UCNEmid" | ratio == "UCNEhigh"), aes(population,avg_ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(. ~ ratio) +
  geom_point() +
  geom_errorbar(aes(ymin=avg_ki_relative_value-se_ki_relative_value, ymax=avg_ki_relative_value+se_ki_relative_value), position=position_dodge(), width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("population mean ratio") +
  scale_y_continuous(breaks = seq(0, 14, by = 2)) +
  ylim(0,14) +
  #ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=16,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=20),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
average_relativised_derived_allele_allele_ratio_ggplot2
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_part2_mean.pdf"), width=30, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/deleterious_genes/")


average_relativised_derived_allele_allele_ratio_ggplotcombined <- grid.arrange(average_relativised_derived_allele_allele_ratio_ggplot1, average_relativised_derived_allele_allele_ratio_ggplot2, nrow = 2, layout_matrix = rbind(c(rep(1,20)), c(rep(2,18),rep(NA,3))))
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns_all.pdf"), width=40, height=20, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/deleterious_genes/", average_relativised_derived_allele_allele_ratio_ggplotcombined)

```

##Relative to introns.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_deleterious_genes_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/intronic_A,missense=missense_A/intronic_A,mistol=missense_tol_A/intronic_A,misdel=missense_del_A/intronic_A,nonsense=nonsense_A/intronic_A,UCNE=UCNE_A/intronic_A) %>% select(c(1:4,28:33)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=relativised_variants_and_subst_wg_alleleR, aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .,scales="fixed") +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 1.2, by = 0.2)) +
  ylim(0.5,1.2) +
  ggtitle(paste0("ratio of ",type," relative to intronic and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_derived_allele_allele_ratio_relative2introns.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```

##Relative to synonymous.
```{r Plot variant count results}

library(readr)
library(dplyr)
library(ggplot2)
library(tidyr)

type="varssubs" #varssubs #variants

wd_path <- ("/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")
variants_and_subst_wg <- read_tsv(paste0(wd_path,"c_lp_sm_c_lp_do_c_ll_ki_c_ll_no_c_ll_po_nm2nm3_origcov_ann_individual_deleterious_genes_summary_",type,"_SNP.lr_ann.txt"))

variants_and_subst_wg$dataset <- as.factor(variants_and_subst_wg$dataset)
variants_and_subst_wg$dataset = factor(variants_and_subst_wg$dataset,levels=c("REF","GP","5x","MG")) #Reorder factor levels to: REF, GP, 5x, MG
variants_and_subst_wg$population = factor(variants_and_subst_wg$population,levels=c("ki","no","po","sm","do"))
print.data.frame(variants_and_subst_wg)

variants_and_subst_wg_alleleR <- variants_and_subst_wg %>% mutate(synonymous=synonymous_A/synonymous_A,missense=missense_A/synonymous_A,mistol=missense_tol_A/synonymous_A,misdel=missense_del_A/synonymous_A,nonsense=nonsense_A/synonymous_A,UCNE=UCNE_A/synonymous_A) %>% select(c(1:4,28:33)) %>% gather(ratio,value,-species,-population,-dataset,-sample,factor_key=T)
variants_and_subst_wg_alleleR

r_average_vector <- c()
for (r in unique(variants_and_subst_wg_alleleR$ratio)) {
  print(r)
  r_average <- filter(variants_and_subst_wg_alleleR,r==ratio & population=="ki") %>% select(value) %>% unlist(.,use.names=F) %>% mean()
  r_average_vector <- c(r_average_vector,rep(r_average,nrow(filter(variants_and_subst_wg_alleleR,r==ratio))))
}
print(r_average_vector)

relativised_variants_and_subst_wg_alleleR <- mutate(variants_and_subst_wg_alleleR, ki_relative_value=value/r_average_vector)

relativised_derived_allele_allele_ratio_ggplot <- ggplot(data=relativised_variants_and_subst_wg_alleleR, aes(population,ki_relative_value,colour=population)) +
  #facet_wrap(feature ~ species,nrow=6,ncol=2,scales="free") +
  facet_grid(ratio ~ .) +
  geom_boxplot(width=0.5) +
  #ggtitle("Proportion of reads at different NM") +
  ylab("ratio") +
  scale_y_continuous(breaks = seq(0.5, 2.3, by = 0.2)) +
  ylim(0.5,2.3) +
  ggtitle(paste0("ratio of ",type," relative to synonymous and Kirov")) +
  theme_bw() +
  theme(text=element_text(size=12,face="bold"),
        rect=element_rect(size=1),
        axis.line=element_line(colour="black"),
        axis.title=element_text(size=16),
        #axis.text.x=element_text(angle=45, hjust=1, size=24,colour="black"),
        #axis.text.y=element_text(size=24,colour="black",margin=margin(t=0.5,unit="cm")),
        #axis.title.y=element_text(size=30,margin=margin(r=0.5,unit="cm")),
        panel.background=element_blank(),
        panel.border=element_rect(colour="black"),
        #panel.grid=element_blank(),
        #panel.grid.major=element_line(colour="grey", linetype="dashed", size=0.4),
        plot.margin=unit(c(0.5,1,0.5,0.5),"cm"),
        #plot.title=element_text(size=36, face="bold", margin=margin(b=0.5, unit="cm")),
        legend.background=element_rect(linetype="solid", colour="black", size=.5),
        #legend.justification=c(0,0),
        legend.key=element_rect(colour="white"),
        #legend.key.size=unit(1.3,"cm"),
        #legend.position=c(0.07,0.84),
        legend.title=element_blank()
  )
  relativised_derived_allele_allele_ratio_ggplot
ggsave(paste0(type,"_deleterious_genes_derived_allele_allele_ratio_relative2synonymous.pdf"), width=15, height=15, units="cm", device="pdf", path="/Users/dani/ownCloud/backup/g-w_analysis/genetic_load/snpeff_summary_ratios/")

```
