---
title: "TreeMix_script"
author: "Dani"
date: "8/11/2017"
output: html_document
---

#0a: Define paths.

```{r Define paths, eval=FALSE, engine='bash'}

V_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani #VCFs path
O_PATH=/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix #TreeMix analyses output path
T_PATH=/opt/treemix-1.12/src #TreeMix software path

```

#0b: Produce a gVCF and allelic counts data for the outgroup (the lynx rufus sample).

```{r Define paths, eval=FALSE, engine='bash'}

cd /home/GRUPOS/grupolince/lynx_genomes_5x/BAM_files_final
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -Xms16g -Xmx32g -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
    -T HaplotypeCaller \
    -R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
    -I c_lr_zz_0001_recal_round-1.bam \
    --emitRefConfidence GVCF \
    -o /home/GRUPOS/grupolince/lynx_genomes_5x/gVCFs/c_lr_zz_0001_recal_round.g.vcf.gz

```

#1: Build TreeMix input. Combine per population allele counts for all SNPs in a single gzipped file matching the required format for TreeMix inputs.

```{r Remove monomorphic SNPs, eval=FALSE, engine='bash'}

cd $V_PATH
shopt -s extglob #the extglob shell option gives you more powerful pattern matching in the command line.
POPLIST=($(ls c_ll*_perpop.vcf | cut -d "_" -f1,2,3 | sort | uniq))
cd $O_PATH
rm SNP_names_treemix_input.txt
SNP_LIST="$(ls "$V_PATH"/c_ll*_perpop.vcf | head -1)" #any population will be useful to retrieve the SNP list because I didn't trim the subVCFs so all populations have all the SNPs (from their species)
grep -v '#' $SNP_LIST | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | awk '{print $1"_"$2}' > SNP_names_treemix_input.txt
unset POPNAMES #removes the variable so that it starts all over again inside the loop
POPNAMES="SNP"
#echo "pop_name,total_SNPs,homoz_ref,heteroz,homoz_alt,unaccounted,unaccounted_cat" > c_VCF_raw.stats 
for pop in "${POPLIST[@]}"
  do
  echo "${pop}"
  NAME="${pop}"
  POPNAMES=$POPNAMES" "$NAME
  echo $POPNAMES
  SAMPLE_NUMBER="$(/opt/bcftools-1.6/bcftools query -l "$V_PATH/${pop}"_perpop.vcf | wc -l)"
  echo $SAMPLE_NUMBER
  TOTAL_SNPS="$(grep -v '#' "$V_PATH/${pop}"_perpop.vcf | wc -l)"
  echo $TOTAL_SNPS
  #done
  grep -v '#' "$V_PATH/${pop}"_perpop.vcf | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | cut -d$'=' -f1,2,3 --output-delimiter=' ' | awk '{print $1"_"$2" "$6-$4","$4}' > temp_SNP_counts.txt
  LANG=en_EN join -j 1 <(LANG=en_EN sort -k1 SNP_names_treemix_input.txt) <(LANG=en_EN sort -k1 temp_SNP_counts.txt) > intermediate_file.borrar.txt
  mv intermediate_file.borrar.txt SNP_names_treemix_input.txt
  done
rm temp_SNP_counts.txt
sed -i -e '1i'"$POPNAMES"'\' 
#adds the headers
grep -v ' 0,0 ' SNP_names_treemix_input.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input.txt.gz #removes rows with any missing data, and also the SNP names
shopt -u extglob #disable extglob


  #sed -i "1 s/$/ $NAME/" toy_treemix_input.txt
  #paste -d' ' toy_treemix_input.txt temp_SNP_counts.txt > toy_treemix_input.borrar.txt
  #mv toy_treemix_input.borrar.txt toy_treemix_input.txt


#Get a version of the TreeMix input without Balkans:
cut -d$' ' -f2 --complement SNP_names_treemix_input.txt > SNP_names_treemix_input_wout_ba.txt #removes the Balkans column
grep -v ' 0,0 ' SNP_names_treemix_input_wout_ba.txt | grep -v ' 0,0$' | cut -d$' ' -f 2- | gzip > treemix_input_wout_ba.txt.gz #removes rows with any missing data, and also the SNP names


#Get a version that includes Lynx rufus (first test it with pardinus) as the outgroup:
cd $O_PATH
SNP_LIST="$(ls "$V_PATH"/c_ll*_perpop.vcf | head -1)"
grep -v '#' $SNP_LIST | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | awk '{printf ("%1s\t%s\n",$1,$2)}' > SNP_names_only.txt
/opt/bcftools-1.6/bcftools filter -R SNP_names_only.txt -Ov -o $O_PATH/c_lp_do_0007_subset_positions.vcf /home/GRUPOS/grupolince/lynx_genomes_5x/gVCFs/c_lp_do_0007_recal_round-1.g.vcf.gz

cd $O_PATH
grep -v '#' $O_PATH/c_lp_do_0007_subset_positions.vcf | cut -d$'\t' -f1,2,8 | cut -d$';' -f1,3 --output-delimiter=' ' | cut -d$'=' -f1,2,3 --output-delimiter=' ' | awk '{print $1"_"$2" "$6-$4","$4}' > c_lp_do_0007_SNP_counts.txt

grep -v '#' c_lp_do_0007_subset_positions.vcf | cut -d$' ' -f1,2,10 | cut -d$':' -f1 | awk '{print $1"_"$2" "$3}' | NUEVA VARIABLE que para 0,0 sea 2,0; para 0,1 sea 1,1; para 1,1 sea 0,2 > new_test.txt



Coger directamente esos!!! E intersectar con el input de TreeMix.


#Build a VCF for the rufus sample
cd /home/GRUPOS/grupolince/lynx_genomes_5x/gVCFs
java -XX:MaxMetaspaceSize=1g -XX:+UseG1GC -XX:+UseStringDeduplication -jar /opt/GATK-3.7/GenomeAnalysisTK.jar \
-T GenotypeGVCFs \
-R /home/GRUPOS/grupolince/reference_genomes/lynx_pardinus_genome/lp23.fa \
$(for var in c_lr_zz_0001_recal_round.g.vcf.gz; do echo -V ${var}" ";done) \
-o /home/GRUPOS/grupolince/lynx_genomes_5x/VCFs_Dani/c_lr.vcf


```

#2: Run TreeMix. Test the waters with different TreeMix runs.

```{r Run TreeMix, eval=FALSE, engine='bash'}

#Basic command to build a ML tree
cd $O_PATH
$T_PATH/treemix -i treemix_input.txt.gz -o toy_out_stem

#Basic command to build a ML tree without Balkans
cd $O_PATH
$T_PATH/treemix -i treemix_input_wout_ba.txt.gz -o toy_wout_ba_out_stem


scp dkleinman@genomics-b.ebd.csic.es:/opt/treemix-1.12/src/plotting_funcs.R /Users/Dani/ownCloud/backup/annotation/TreeMix
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/toy_out_stem* /Users/Dani/ownCloud/backup/annotation/TreeMix
scp dkleinman@genomics-b.ebd.csic.es:/home/GRUPOS/grupolince/lynx_genomes_5x/TreeMix/toy_wout_ba_out_stem* /Users/Dani/ownCloud/backup/annotation/TreeMix

```

#3: Plot TreeMix.

```{r Plot TreeMix}

local_repo <- file.path("/Users/Dani/ownCloud/backup/annotation/TreeMix/")

source("/Users/Dani/ownCloud/backup/annotation/TreeMix/plotting_funcs.R")
pdf(paste0(local_repo,"ll_toy_basic_tree.pdf"),width=20,height=10)
toy_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/toy_out_stem")
toy_out_stem
dev.off()

png("CrabMigrationAllSNPSmig3.png",width=2400,height=2000,res = 300)
plot_tree("crabmigout3")
dev.off()

toy_wout_ba_out_stem <- plot_tree("/Users/Dani/ownCloud/backup/annotation/TreeMix/toy_wout_ba_out_stem")
toy_wout_ba_out_stem

```
